{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "77334dcd",
   "metadata": {},
   "source": [
    "# Inference of MicroRNA-Messenger RNA Interactions in TCGA-BRCA\n",
    "TCGA: The Cancer Genome Atlas  \n",
    "BRCA: Breast Invasive Carcinoma"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5221ad8c",
   "metadata": {},
   "source": [
    "# Importing Libraries and Configurations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "001781d4",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import sys\n",
    "\n",
    "import pandas as pd\n",
    "from joblib import Parallel, delayed\n",
    "from scipy.stats import false_discovery_control, spearmanr\n",
    "\n",
    "# Add project root to Python's path\n",
    "sys.path.append(os.path.abspath(os.path.join('..', '..')))\n",
    "\n",
    "from config import (\n",
    "    AGGREGATED_READS_FILES,\n",
    "    BRCA_INTERIM_FILES_DIRS,\n",
    "    BRCA_PROCESSED_FILES_DIRS,\n",
    "    BRCA_PROCESSED_FILES_PATHS,\n",
    "    INTERACTION_INFERENCE_PARAMETERS,\n",
    "    MIRWALK_MIR_MAPPING_FILE_PATH,\n",
    "    MIRWALK_PROCESSED_DATA_DIR,\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9069dd41",
   "metadata": {},
   "source": [
    "# Function"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "871bf352",
   "metadata": {},
   "outputs": [],
   "source": [
    "def prepare_data_for_spearman(processed_dir_path):\n",
    "    \"\"\"\n",
    "    Prepare aggregated expression data of a group for Spearman correlation analysis.\n",
    "\n",
    "    Parameters:\n",
    "    -----------\n",
    "    processed_dir_path : str\n",
    "        Path to the directory containing processed expression data files.\n",
    "\n",
    "    Returns:\n",
    "    --------\n",
    "    dict of pd.DataFrame\n",
    "        A dictionary with keys:\n",
    "        - 'mir' : DataFrame with aggregated microRNA normalized read counts.\n",
    "        - 'rna' : DataFrame with aggregated messenger RNA normalized read counts.\n",
    "        Both DataFrames have case IDs as indices and molecule identifiers as columns.\n",
    "    \"\"\"\n",
    "    # Create a DataFrame for the files metadata\n",
    "    df_files_metadata = pd.read_csv(BRCA_PROCESSED_FILES_PATHS['files'])\n",
    "\n",
    "    # Initialize a dictionary for Spearman-ready DataFrames\n",
    "    aggregated_reads = dict()\n",
    "\n",
    "    # Prepare the aggregated reads files\n",
    "    for experimental_strategy in ['mir', 'rna']:\n",
    "        # Define the experimental strategy related parameters\n",
    "        drop_column = ['is_expressed']\n",
    "        if experimental_strategy == 'mir':\n",
    "            index_column = 'accession_id'\n",
    "            file_name = AGGREGATED_READS_FILES['mir-normalized']\n",
    "        else:\n",
    "            index_column = 'gene_name'\n",
    "            drop_column = drop_column + ['gene_id']\n",
    "            file_name = AGGREGATED_READS_FILES['rna-normalized']\n",
    "        \n",
    "        # Get the expressed molecules and set the primary key as index\n",
    "        file_path = os.path.join(processed_dir_path, file_name)\n",
    "        df_agg_reads = pd.read_csv(file_path) \\\n",
    "            .query('is_expressed == 1') \\\n",
    "            .drop(columns=drop_column) \\\n",
    "            .set_index(keys=index_column)\n",
    "        \n",
    "        # Map the file ID to the associated case ID\n",
    "        file_ids = list(df_agg_reads.columns)\n",
    "        df_files_mapping = pd.DataFrame(file_ids, columns=['file_id'])\n",
    "        df_files_mapping = df_files_mapping \\\n",
    "            .merge(\n",
    "                right=df_files_metadata,\n",
    "                left_on='file_id',\n",
    "                right_on='file_id',\n",
    "                how='left',\n",
    "            ) \\\n",
    "            [['file_id', 'case_id']]\n",
    "        \n",
    "        # Transpose the DataFrame and set the case ID as index\n",
    "        df_agg_reads = df_agg_reads \\\n",
    "            .transpose() \\\n",
    "            .reset_index() \\\n",
    "            .rename(columns={'index': 'file_id'}) \\\n",
    "            .merge(\n",
    "                right=df_files_mapping,\n",
    "                left_on='file_id',\n",
    "                right_on='file_id',\n",
    "                how='inner',\n",
    "            ) \\\n",
    "            .drop(columns=['file_id']) \\\n",
    "            .set_index(keys='case_id') \\\n",
    "            .sort_index(ascending=True)\n",
    "        \n",
    "        # Store the prepared DataFrame in the dictionary\n",
    "        aggregated_reads[experimental_strategy] = df_agg_reads.copy()\n",
    "    \n",
    "    return aggregated_reads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "5ca411e7",
   "metadata": {},
   "outputs": [],
   "source": [
    "def compute_single_pair_spearman(ser_mir_expression, ser_rna_expression):\n",
    "    \"\"\"\n",
    "    Calculate the Spearman correlation coefficient between the expression profile \n",
    "    of a single microRNA and messenger RNA pair.\n",
    "    \n",
    "    Parameters:\n",
    "    -----------\n",
    "    ser_mir_expression : pd.Series\n",
    "        Expression values of a single microRNA across samples.\n",
    "    ser_rna_expression : pd.Series\n",
    "        Expression values of a single messenger RNA across the same samples.\n",
    "\n",
    "    Returns:\n",
    "    --------\n",
    "    dict\n",
    "        A dictionary containing:\n",
    "        - 'accession_id': microRNA accession ID.\n",
    "        - 'gene_name': gene name of the messenger RNA.\n",
    "        - 'correlation': Spearman correlation coefficient.\n",
    "        - 'pvalue': p-value for the hypothesis test that correlation < 0.\n",
    "    \"\"\"\n",
    "    # Calculate the Spearman correlation coefficient for the pair\n",
    "    axis = INTERACTION_INFERENCE_PARAMETERS['axis']\n",
    "    alternative = INTERACTION_INFERENCE_PARAMETERS['alternative']\n",
    "    correlation, pvalue = spearmanr(\n",
    "        a=ser_mir_expression, # Samples of the expressed microRNA\n",
    "        b=ser_rna_expression, # Samples of the expressed messenger RNAs\n",
    "        axis=axis, # Each row is a observation, while the columns are variables\n",
    "        alternative=alternative, # The correlation is negative in miRNA-mRNA inferred_interactions\n",
    "    )\n",
    "    \n",
    "    # Create a dictionary to represent the results\n",
    "    results = {\n",
    "        'accession_id': ser_mir_expression.name,\n",
    "        'gene_name': ser_rna_expression.name,\n",
    "        'correlation': correlation,\n",
    "        'pvalue': pvalue,\n",
    "    }\n",
    "    \n",
    "    return results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "865d69f3",
   "metadata": {},
   "outputs": [],
   "source": [
    "def infer_mir_rna_interactions(df_mirwalk_interactions, group):\n",
    "    \"\"\"\n",
    "    Infer significant microRNA (miRNA) and messenger RNA (mRNA) interactions based on \n",
    "    expression data and validated miRWalk interactions for a specific group of samples.\n",
    "\n",
    "    Parameters:\n",
    "    -----------\n",
    "    df_mirwalk_interactions : pd.DataFrame\n",
    "        A DataFrame containing known miRNA-mRNA interactions from the miRWalk database.\n",
    "        Expected to contain columns ['accession_id', 'gene_name'] at minimum.\n",
    "    \n",
    "    group : str\n",
    "        Name of the group used to locate interim and processed data directories.\n",
    "\n",
    "    Returns:\n",
    "    --------\n",
    "    pd.DataFrame\n",
    "        A DataFrame of statistically significant miRNA-mRNA interactions, including\n",
    "        Spearman correlation coefficients, p-values, FDR-adjusted q-values, and\n",
    "        miRWalk evidence.\n",
    "    \"\"\"\n",
    "    # Define the group interim and processed directories path\n",
    "    dir_base_name = (group.lower()).replace(' ', '-')\n",
    "    interim_dir_path = BRCA_INTERIM_FILES_DIRS[dir_base_name]\n",
    "    processed_dir_path = BRCA_PROCESSED_FILES_DIRS[dir_base_name]\n",
    "    \n",
    "    # Prepare the expression data of the expressed molecules\n",
    "    aggregated_reads = prepare_data_for_spearman(processed_dir_path)\n",
    "    df_mir_expression = aggregated_reads['mir']\n",
    "    df_rna_expression = aggregated_reads['rna']\n",
    "    \n",
    "    # Define all potential inferred_interactions between expressed microRNAs and messenger RNAs\n",
    "    df_expressed_mirs = pd.DataFrame(\n",
    "        data=list(df_mir_expression.columns), columns=['accession_id']\n",
    "    )\n",
    "    df_expressed_rnas = pd.DataFrame(\n",
    "        data=list(df_rna_expression.columns), columns=['gene_name']\n",
    "    )\n",
    "    df_potential_inferred_interactions = df_expressed_mirs \\\n",
    "        .merge(right=df_expressed_rnas, how='cross')\n",
    "    \n",
    "    # Select inferred_interactions with miRWalk retrieved evidence\n",
    "    primary_key = ['accession_id', 'gene_name']\n",
    "    df_inferred_interactions = df_potential_inferred_interactions \\\n",
    "        .merge(\n",
    "            right=df_mirwalk_interactions,\n",
    "            left_on=primary_key,\n",
    "            right_on=primary_key,\n",
    "            how='inner',\n",
    "        )\n",
    "    group_inferred_interactions = zip(\n",
    "        df_inferred_interactions['accession_id'], df_inferred_interactions['gene_name'],\n",
    "    )\n",
    "    \n",
    "    # Compute in parallel the Spearman correlation coefficient for each pair\n",
    "    results = Parallel(n_jobs=-1, prefer='processes')(\n",
    "        delayed(compute_single_pair_spearman)\n",
    "        (df_mir_expression[mir], df_rna_expression[rna])\n",
    "        for mir, rna in group_inferred_interactions\n",
    "    )\n",
    "    \n",
    "    # Create a DataFrame for the computing results\n",
    "    df_results = pd.DataFrame(results)\n",
    "    \n",
    "    # Adjust the p-values to control the false discovery fate (FDR)\n",
    "    axis = INTERACTION_INFERENCE_PARAMETERS['axis']\n",
    "    method = INTERACTION_INFERENCE_PARAMETERS['fdr-method']\n",
    "    qvalues = false_discovery_control(\n",
    "        ps=df_results['pvalue'], # The p-values to adjust\n",
    "        axis=axis, # The axis along which to perform the adjustment\n",
    "        method=method, # FDR control procedure\n",
    "    )\n",
    "    df_results['qvalue'] = qvalues\n",
    "    \n",
    "    # Add miRWalk data to results\n",
    "    df_results = df_mirwalk_interactions \\\n",
    "        .merge(\n",
    "            right=df_results,\n",
    "            left_on=primary_key,\n",
    "            right_on=primary_key,\n",
    "            how='inner',\n",
    "        )\n",
    "    \n",
    "    # Store the DataFrame of inferred interactions\n",
    "    file_name = INTERACTION_INFERENCE_PARAMETERS['file-name']\n",
    "    df_results.to_csv(os.path.join(interim_dir_path, file_name), index=False)\n",
    "    \n",
    "    return df_results"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "39372508",
   "metadata": {},
   "source": [
    "# miRWalk Interactions of Interest"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "01b3d654",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Initialize a DataFrame for the inferred interactions of interest from miRWalk\n",
    "mirwalk_columns = ['mirna_name', 'gene_name', 'mirtarbase']\n",
    "df_mirwalk_interactions = pd.DataFrame(columns=mirwalk_columns)\n",
    "\n",
    "# List the interaction files downloaded from miRWalk\n",
    "files = [f for f in os.listdir(MIRWALK_PROCESSED_DATA_DIR) if f.startswith('MIMAT')]\n",
    "\n",
    "# Iterate over each inferred_interactions file from miRWalk\n",
    "for file in files:\n",
    "    # Create a DataFrame for the inferred interactions of interest of the microRNA\n",
    "    file_path = os.path.join(MIRWALK_PROCESSED_DATA_DIR, file)    \n",
    "    df_mir_inferred_interactions = pd.read_csv(file_path, low_memory=False) \\\n",
    "        .query('is_interaction_of_interest == 1') \\\n",
    "        [mirwalk_columns] \\\n",
    "        .drop_duplicates()\n",
    "    \n",
    "    # Concatenate the inferred interactions of interest to the others\n",
    "    df_mirwalk_interactions = pd.concat(\n",
    "        [df_mirwalk_interactions, df_mir_inferred_interactions], ignore_index=True\n",
    "    )\n",
    "\n",
    "# Add the microRNA accession IDs to the DataFrame\n",
    "df_mir_mapping = pd.read_csv(MIRWALK_MIR_MAPPING_FILE_PATH)\n",
    "df_mirwalk_interactions = df_mir_mapping \\\n",
    "    .merge(\n",
    "        right=df_mirwalk_interactions,\n",
    "        left_on='mirna_name',\n",
    "        right_on='mirna_name',\n",
    "        how='inner',\n",
    "    )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "86955188",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>accession_id</th>\n",
       "      <th>mirna_name</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>mirtarbase</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>PKIA</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>GJC1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>PBX3</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>ITSN1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>SRGAP1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41848</th>\n",
       "      <td>MIMAT0026483</td>\n",
       "      <td>hsa-miR-370-5p</td>\n",
       "      <td>STUM</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41849</th>\n",
       "      <td>MIMAT0026483</td>\n",
       "      <td>hsa-miR-370-5p</td>\n",
       "      <td>CRISPLD2</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41850</th>\n",
       "      <td>MIMAT0026483</td>\n",
       "      <td>hsa-miR-370-5p</td>\n",
       "      <td>ADAM19</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41851</th>\n",
       "      <td>MIMAT0026557</td>\n",
       "      <td>hsa-miR-412-5p</td>\n",
       "      <td>FAM53C</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41852</th>\n",
       "      <td>MIMAT0026557</td>\n",
       "      <td>hsa-miR-412-5p</td>\n",
       "      <td>FICD</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>41853 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       accession_id      mirna_name gene_name mirtarbase\n",
       "0      MIMAT0000062   hsa-let-7a-5p      PKIA        NaN\n",
       "1      MIMAT0000062   hsa-let-7a-5p      GJC1        NaN\n",
       "2      MIMAT0000062   hsa-let-7a-5p      PBX3        NaN\n",
       "3      MIMAT0000062   hsa-let-7a-5p     ITSN1        NaN\n",
       "4      MIMAT0000062   hsa-let-7a-5p    SRGAP1        NaN\n",
       "...             ...             ...       ...        ...\n",
       "41848  MIMAT0026483  hsa-miR-370-5p      STUM        NaN\n",
       "41849  MIMAT0026483  hsa-miR-370-5p  CRISPLD2        NaN\n",
       "41850  MIMAT0026483  hsa-miR-370-5p    ADAM19        NaN\n",
       "41851  MIMAT0026557  hsa-miR-412-5p    FAM53C        NaN\n",
       "41852  MIMAT0026557  hsa-miR-412-5p      FICD        NaN\n",
       "\n",
       "[41853 rows x 4 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the DataFrame of inferred_interactions of interest from miRWalk\n",
    "df_mirwalk_interactions"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "adaa1b0b",
   "metadata": {},
   "source": [
    "# Interaction Inference"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f9a5d2ac",
   "metadata": {},
   "source": [
    "## Basal-like"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "6927253e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Infer the interactions for Basal-like tumor tissue\n",
    "inferred_interactions = infer_mir_rna_interactions(df_mirwalk_interactions, 'Basal-like')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "288e3888",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>accession_id</th>\n",
       "      <th>mirna_name</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>mirtarbase</th>\n",
       "      <th>correlation</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>qvalue</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>PKIA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.078607</td>\n",
       "      <td>0.234621</td>\n",
       "      <td>0.745015</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>GJC1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.018153</td>\n",
       "      <td>0.566268</td>\n",
       "      <td>0.978192</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>PBX3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.143526</td>\n",
       "      <td>0.907620</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>ITSN1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.033079</td>\n",
       "      <td>0.380501</td>\n",
       "      <td>0.873011</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>SRGAP1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.047532</td>\n",
       "      <td>0.669012</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41164</th>\n",
       "      <td>MIMAT0026483</td>\n",
       "      <td>hsa-miR-370-5p</td>\n",
       "      <td>STUM</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.092500</td>\n",
       "      <td>0.802930</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41165</th>\n",
       "      <td>MIMAT0026483</td>\n",
       "      <td>hsa-miR-370-5p</td>\n",
       "      <td>CRISPLD2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.430324</td>\n",
       "      <td>0.999984</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41166</th>\n",
       "      <td>MIMAT0026483</td>\n",
       "      <td>hsa-miR-370-5p</td>\n",
       "      <td>ADAM19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.221807</td>\n",
       "      <td>0.980527</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41167</th>\n",
       "      <td>MIMAT0026557</td>\n",
       "      <td>hsa-miR-412-5p</td>\n",
       "      <td>FAM53C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.016884</td>\n",
       "      <td>0.438326</td>\n",
       "      <td>0.911847</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41168</th>\n",
       "      <td>MIMAT0026557</td>\n",
       "      <td>hsa-miR-412-5p</td>\n",
       "      <td>FICD</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.110946</td>\n",
       "      <td>0.846853</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>41169 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       accession_id      mirna_name gene_name mirtarbase  correlation  \\\n",
       "0      MIMAT0000062   hsa-let-7a-5p      PKIA        NaN    -0.078607   \n",
       "1      MIMAT0000062   hsa-let-7a-5p      GJC1        NaN     0.018153   \n",
       "2      MIMAT0000062   hsa-let-7a-5p      PBX3        NaN     0.143526   \n",
       "3      MIMAT0000062   hsa-let-7a-5p     ITSN1        NaN    -0.033079   \n",
       "4      MIMAT0000062   hsa-let-7a-5p    SRGAP1        NaN     0.047532   \n",
       "...             ...             ...       ...        ...          ...   \n",
       "41164  MIMAT0026483  hsa-miR-370-5p      STUM        NaN     0.092500   \n",
       "41165  MIMAT0026483  hsa-miR-370-5p  CRISPLD2        NaN     0.430324   \n",
       "41166  MIMAT0026483  hsa-miR-370-5p    ADAM19        NaN     0.221807   \n",
       "41167  MIMAT0026557  hsa-miR-412-5p    FAM53C        NaN    -0.016884   \n",
       "41168  MIMAT0026557  hsa-miR-412-5p      FICD        NaN     0.110946   \n",
       "\n",
       "         pvalue    qvalue  \n",
       "0      0.234621  0.745015  \n",
       "1      0.566268  0.978192  \n",
       "2      0.907620  1.000000  \n",
       "3      0.380501  0.873011  \n",
       "4      0.669012  1.000000  \n",
       "...         ...       ...  \n",
       "41164  0.802930  1.000000  \n",
       "41165  0.999984  1.000000  \n",
       "41166  0.980527  1.000000  \n",
       "41167  0.438326  0.911847  \n",
       "41168  0.846853  1.000000  \n",
       "\n",
       "[41169 rows x 7 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the inferred interactions associated with this group\n",
    "inferred_interactions"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a3a473b5",
   "metadata": {},
   "source": [
    "## HER2-enriched"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "bc3d7400",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Infer the interactions for HER2-enriched tumor tissue\n",
    "inferred_interactions = infer_mir_rna_interactions(df_mirwalk_interactions, 'HER2-enriched')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "3f25d0b1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>accession_id</th>\n",
       "      <th>mirna_name</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>mirtarbase</th>\n",
       "      <th>correlation</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>qvalue</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>PKIA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.013944</td>\n",
       "      <td>0.459379</td>\n",
       "      <td>0.871133</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>GJC1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.248189</td>\n",
       "      <td>0.967434</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>PBX3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.003896</td>\n",
       "      <td>0.511368</td>\n",
       "      <td>0.894937</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>ITSN1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.102802</td>\n",
       "      <td>0.774561</td>\n",
       "      <td>0.982053</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>SRGAP1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.067943</td>\n",
       "      <td>0.309403</td>\n",
       "      <td>0.789843</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41164</th>\n",
       "      <td>MIMAT0026483</td>\n",
       "      <td>hsa-miR-370-5p</td>\n",
       "      <td>STUM</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.378153</td>\n",
       "      <td>0.997971</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41165</th>\n",
       "      <td>MIMAT0026483</td>\n",
       "      <td>hsa-miR-370-5p</td>\n",
       "      <td>CRISPLD2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.378153</td>\n",
       "      <td>0.997971</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41166</th>\n",
       "      <td>MIMAT0026483</td>\n",
       "      <td>hsa-miR-370-5p</td>\n",
       "      <td>ADAM19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.262629</td>\n",
       "      <td>0.974737</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41167</th>\n",
       "      <td>MIMAT0026557</td>\n",
       "      <td>hsa-miR-412-5p</td>\n",
       "      <td>FAM53C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.236705</td>\n",
       "      <td>0.960494</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41168</th>\n",
       "      <td>MIMAT0026557</td>\n",
       "      <td>hsa-miR-412-5p</td>\n",
       "      <td>FICD</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.221394</td>\n",
       "      <td>0.949478</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>41169 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       accession_id      mirna_name gene_name mirtarbase  correlation  \\\n",
       "0      MIMAT0000062   hsa-let-7a-5p      PKIA        NaN    -0.013944   \n",
       "1      MIMAT0000062   hsa-let-7a-5p      GJC1        NaN     0.248189   \n",
       "2      MIMAT0000062   hsa-let-7a-5p      PBX3        NaN     0.003896   \n",
       "3      MIMAT0000062   hsa-let-7a-5p     ITSN1        NaN     0.102802   \n",
       "4      MIMAT0000062   hsa-let-7a-5p    SRGAP1        NaN    -0.067943   \n",
       "...             ...             ...       ...        ...          ...   \n",
       "41164  MIMAT0026483  hsa-miR-370-5p      STUM        NaN     0.378153   \n",
       "41165  MIMAT0026483  hsa-miR-370-5p  CRISPLD2        NaN     0.378153   \n",
       "41166  MIMAT0026483  hsa-miR-370-5p    ADAM19        NaN     0.262629   \n",
       "41167  MIMAT0026557  hsa-miR-412-5p    FAM53C        NaN     0.236705   \n",
       "41168  MIMAT0026557  hsa-miR-412-5p      FICD        NaN     0.221394   \n",
       "\n",
       "         pvalue    qvalue  \n",
       "0      0.459379  0.871133  \n",
       "1      0.967434  1.000000  \n",
       "2      0.511368  0.894937  \n",
       "3      0.774561  0.982053  \n",
       "4      0.309403  0.789843  \n",
       "...         ...       ...  \n",
       "41164  0.997971  1.000000  \n",
       "41165  0.997971  1.000000  \n",
       "41166  0.974737  1.000000  \n",
       "41167  0.960494  1.000000  \n",
       "41168  0.949478  1.000000  \n",
       "\n",
       "[41169 rows x 7 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the inferred interactions associated with this group\n",
    "inferred_interactions"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "474a2945",
   "metadata": {},
   "source": [
    "## Luminal A"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "9206a496",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Infer the interactions for Luminal A tumor tissue\n",
    "inferred_interactions = infer_mir_rna_interactions(df_mirwalk_interactions, 'Luminal A')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "0622aa4c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>accession_id</th>\n",
       "      <th>mirna_name</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>mirtarbase</th>\n",
       "      <th>correlation</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>qvalue</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>PKIA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.010777</td>\n",
       "      <td>0.436428</td>\n",
       "      <td>0.841128</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>GJC1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.043499</td>\n",
       "      <td>0.259064</td>\n",
       "      <td>0.648037</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>PBX3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.059124</td>\n",
       "      <td>0.810222</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>ITSN1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.061249</td>\n",
       "      <td>0.818681</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>SRGAP1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.035360</td>\n",
       "      <td>0.299708</td>\n",
       "      <td>0.697952</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41164</th>\n",
       "      <td>MIMAT0026483</td>\n",
       "      <td>hsa-miR-370-5p</td>\n",
       "      <td>STUM</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.143357</td>\n",
       "      <td>0.983815</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41165</th>\n",
       "      <td>MIMAT0026483</td>\n",
       "      <td>hsa-miR-370-5p</td>\n",
       "      <td>CRISPLD2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.456634</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41166</th>\n",
       "      <td>MIMAT0026483</td>\n",
       "      <td>hsa-miR-370-5p</td>\n",
       "      <td>ADAM19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.252987</td>\n",
       "      <td>0.999933</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41167</th>\n",
       "      <td>MIMAT0026557</td>\n",
       "      <td>hsa-miR-412-5p</td>\n",
       "      <td>FAM53C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.017501</td>\n",
       "      <td>0.602526</td>\n",
       "      <td>0.970287</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41168</th>\n",
       "      <td>MIMAT0026557</td>\n",
       "      <td>hsa-miR-412-5p</td>\n",
       "      <td>FICD</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.029306</td>\n",
       "      <td>0.331685</td>\n",
       "      <td>0.734977</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>41169 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       accession_id      mirna_name gene_name mirtarbase  correlation  \\\n",
       "0      MIMAT0000062   hsa-let-7a-5p      PKIA        NaN    -0.010777   \n",
       "1      MIMAT0000062   hsa-let-7a-5p      GJC1        NaN    -0.043499   \n",
       "2      MIMAT0000062   hsa-let-7a-5p      PBX3        NaN     0.059124   \n",
       "3      MIMAT0000062   hsa-let-7a-5p     ITSN1        NaN     0.061249   \n",
       "4      MIMAT0000062   hsa-let-7a-5p    SRGAP1        NaN    -0.035360   \n",
       "...             ...             ...       ...        ...          ...   \n",
       "41164  MIMAT0026483  hsa-miR-370-5p      STUM        NaN     0.143357   \n",
       "41165  MIMAT0026483  hsa-miR-370-5p  CRISPLD2        NaN     0.456634   \n",
       "41166  MIMAT0026483  hsa-miR-370-5p    ADAM19        NaN     0.252987   \n",
       "41167  MIMAT0026557  hsa-miR-412-5p    FAM53C        NaN     0.017501   \n",
       "41168  MIMAT0026557  hsa-miR-412-5p      FICD        NaN    -0.029306   \n",
       "\n",
       "         pvalue    qvalue  \n",
       "0      0.436428  0.841128  \n",
       "1      0.259064  0.648037  \n",
       "2      0.810222  1.000000  \n",
       "3      0.818681  1.000000  \n",
       "4      0.299708  0.697952  \n",
       "...         ...       ...  \n",
       "41164  0.983815  1.000000  \n",
       "41165  1.000000  1.000000  \n",
       "41166  0.999933  1.000000  \n",
       "41167  0.602526  0.970287  \n",
       "41168  0.331685  0.734977  \n",
       "\n",
       "[41169 rows x 7 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the inferred interactions associated with this group\n",
    "inferred_interactions"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "36ce0bae",
   "metadata": {},
   "source": [
    "## Luminal B"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "e4384539",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Infer the interactions for Luminal B tumor tissue\n",
    "inferred_interactions = infer_mir_rna_interactions(df_mirwalk_interactions, 'Luminal B')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "0e683555",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>accession_id</th>\n",
       "      <th>mirna_name</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>mirtarbase</th>\n",
       "      <th>correlation</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>qvalue</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>PKIA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.162456</td>\n",
       "      <td>0.038132</td>\n",
       "      <td>0.262916</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>GJC1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.096166</td>\n",
       "      <td>0.851953</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>PBX3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.064838</td>\n",
       "      <td>0.759150</td>\n",
       "      <td>0.995618</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>ITSN1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.149260</td>\n",
       "      <td>0.948140</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>SRGAP1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.173825</td>\n",
       "      <td>0.971198</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41164</th>\n",
       "      <td>MIMAT0026483</td>\n",
       "      <td>hsa-miR-370-5p</td>\n",
       "      <td>STUM</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.076256</td>\n",
       "      <td>0.796110</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41165</th>\n",
       "      <td>MIMAT0026483</td>\n",
       "      <td>hsa-miR-370-5p</td>\n",
       "      <td>CRISPLD2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.317266</td>\n",
       "      <td>0.999793</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41166</th>\n",
       "      <td>MIMAT0026483</td>\n",
       "      <td>hsa-miR-370-5p</td>\n",
       "      <td>ADAM19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.141708</td>\n",
       "      <td>0.938692</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41167</th>\n",
       "      <td>MIMAT0026557</td>\n",
       "      <td>hsa-miR-412-5p</td>\n",
       "      <td>FAM53C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.070633</td>\n",
       "      <td>0.221659</td>\n",
       "      <td>0.598784</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41168</th>\n",
       "      <td>MIMAT0026557</td>\n",
       "      <td>hsa-miR-412-5p</td>\n",
       "      <td>FICD</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.016142</td>\n",
       "      <td>0.430545</td>\n",
       "      <td>0.799941</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>41169 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       accession_id      mirna_name gene_name mirtarbase  correlation  \\\n",
       "0      MIMAT0000062   hsa-let-7a-5p      PKIA        NaN    -0.162456   \n",
       "1      MIMAT0000062   hsa-let-7a-5p      GJC1        NaN     0.096166   \n",
       "2      MIMAT0000062   hsa-let-7a-5p      PBX3        NaN     0.064838   \n",
       "3      MIMAT0000062   hsa-let-7a-5p     ITSN1        NaN     0.149260   \n",
       "4      MIMAT0000062   hsa-let-7a-5p    SRGAP1        NaN     0.173825   \n",
       "...             ...             ...       ...        ...          ...   \n",
       "41164  MIMAT0026483  hsa-miR-370-5p      STUM        NaN     0.076256   \n",
       "41165  MIMAT0026483  hsa-miR-370-5p  CRISPLD2        NaN     0.317266   \n",
       "41166  MIMAT0026483  hsa-miR-370-5p    ADAM19        NaN     0.141708   \n",
       "41167  MIMAT0026557  hsa-miR-412-5p    FAM53C        NaN    -0.070633   \n",
       "41168  MIMAT0026557  hsa-miR-412-5p      FICD        NaN    -0.016142   \n",
       "\n",
       "         pvalue    qvalue  \n",
       "0      0.038132  0.262916  \n",
       "1      0.851953  1.000000  \n",
       "2      0.759150  0.995618  \n",
       "3      0.948140  1.000000  \n",
       "4      0.971198  1.000000  \n",
       "...         ...       ...  \n",
       "41164  0.796110  1.000000  \n",
       "41165  0.999793  1.000000  \n",
       "41166  0.938692  1.000000  \n",
       "41167  0.221659  0.598784  \n",
       "41168  0.430545  0.799941  \n",
       "\n",
       "[41169 rows x 7 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the inferred interactions associated with this group\n",
    "inferred_interactions"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a6f9f340",
   "metadata": {},
   "source": [
    "## Paired Normal"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "63f609cf",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Infer the interactions for Paired Normal tissue\n",
    "inferred_interactions = infer_mir_rna_interactions(df_mirwalk_interactions, 'Paired Normal')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "5acc63a5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>accession_id</th>\n",
       "      <th>mirna_name</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>mirtarbase</th>\n",
       "      <th>correlation</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>qvalue</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>PKIA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.237936</td>\n",
       "      <td>0.961289</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>GJC1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.015379</td>\n",
       "      <td>0.455214</td>\n",
       "      <td>0.898533</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>PBX3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.007314</td>\n",
       "      <td>0.521332</td>\n",
       "      <td>0.969059</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>ITSN1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.077102</td>\n",
       "      <td>0.286103</td>\n",
       "      <td>0.674410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>SRGAP1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.162748</td>\n",
       "      <td>0.884626</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41164</th>\n",
       "      <td>MIMAT0026483</td>\n",
       "      <td>hsa-miR-370-5p</td>\n",
       "      <td>STUM</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.198770</td>\n",
       "      <td>0.929030</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41165</th>\n",
       "      <td>MIMAT0026483</td>\n",
       "      <td>hsa-miR-370-5p</td>\n",
       "      <td>CRISPLD2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.176897</td>\n",
       "      <td>0.903923</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41166</th>\n",
       "      <td>MIMAT0026483</td>\n",
       "      <td>hsa-miR-370-5p</td>\n",
       "      <td>ADAM19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.176555</td>\n",
       "      <td>0.903485</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41167</th>\n",
       "      <td>MIMAT0026557</td>\n",
       "      <td>hsa-miR-412-5p</td>\n",
       "      <td>FAM53C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.143814</td>\n",
       "      <td>0.854845</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41168</th>\n",
       "      <td>MIMAT0026557</td>\n",
       "      <td>hsa-miR-412-5p</td>\n",
       "      <td>FICD</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.260560</td>\n",
       "      <td>0.973781</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>41169 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       accession_id      mirna_name gene_name mirtarbase  correlation  \\\n",
       "0      MIMAT0000062   hsa-let-7a-5p      PKIA        NaN     0.237936   \n",
       "1      MIMAT0000062   hsa-let-7a-5p      GJC1        NaN    -0.015379   \n",
       "2      MIMAT0000062   hsa-let-7a-5p      PBX3        NaN     0.007314   \n",
       "3      MIMAT0000062   hsa-let-7a-5p     ITSN1        NaN    -0.077102   \n",
       "4      MIMAT0000062   hsa-let-7a-5p    SRGAP1        NaN     0.162748   \n",
       "...             ...             ...       ...        ...          ...   \n",
       "41164  MIMAT0026483  hsa-miR-370-5p      STUM        NaN     0.198770   \n",
       "41165  MIMAT0026483  hsa-miR-370-5p  CRISPLD2        NaN     0.176897   \n",
       "41166  MIMAT0026483  hsa-miR-370-5p    ADAM19        NaN     0.176555   \n",
       "41167  MIMAT0026557  hsa-miR-412-5p    FAM53C        NaN     0.143814   \n",
       "41168  MIMAT0026557  hsa-miR-412-5p      FICD        NaN     0.260560   \n",
       "\n",
       "         pvalue    qvalue  \n",
       "0      0.961289  1.000000  \n",
       "1      0.455214  0.898533  \n",
       "2      0.521332  0.969059  \n",
       "3      0.286103  0.674410  \n",
       "4      0.884626  1.000000  \n",
       "...         ...       ...  \n",
       "41164  0.929030  1.000000  \n",
       "41165  0.903923  1.000000  \n",
       "41166  0.903485  1.000000  \n",
       "41167  0.854845  1.000000  \n",
       "41168  0.973781  1.000000  \n",
       "\n",
       "[41169 rows x 7 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the inferred interactions associated with this group\n",
    "inferred_interactions"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "venv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
