{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "77334dcd",
   "metadata": {},
   "source": [
    "# Inference of MicroRNA-Messenger RNA Interactions in TCGA-BRCA\n",
    "TCGA: The Cancer Genome Atlas  \n",
    "BRCA: Breast Invasive Carcinoma"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5221ad8c",
   "metadata": {},
   "source": [
    "# Importing Libraries and Configurations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "001781d4",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import sys\n",
    "\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "from joblib import Parallel, delayed\n",
    "from scipy.stats import spearmanr\n",
    "\n",
    "# Add project root to Python's path\n",
    "sys.path.append(os.path.abspath(os.path.join('..', '..')))\n",
    "\n",
    "from config import (\n",
    "    AGGREGATED_READS_FILES,\n",
    "    BRCA_INTERIM_FILES_DIRS,\n",
    "    BRCA_PROCESSED_FILES_DIRS,\n",
    "    BRCA_PROCESSED_FILES_PATHS,\n",
    "    INTERACTION_INFERENCE_PARAMETERS,\n",
    "    MIRWALK_MIR_MAPPING_FILE_PATH,\n",
    "    MIRWALK_PROCESSED_DATA_DIR,\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9069dd41",
   "metadata": {},
   "source": [
    "# Function"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "871bf352",
   "metadata": {},
   "outputs": [],
   "source": [
    "def preparing_data_for_spearman(processed_dir_path):\n",
    "    \"\"\"\n",
    "    Prepare normalized expression data of a cohort for Spearman correlation analysis.\n",
    "\n",
    "    Parameters\n",
    "    ----------\n",
    "    processed_dir_path : str\n",
    "        Path to the directory containing processed expression data files.\n",
    "\n",
    "    Returns\n",
    "    -------\n",
    "    dict of pd.DataFrame\n",
    "        A dictionary with keys:\n",
    "        - 'mir' : DataFrame with aggregated microRNA read counts.\n",
    "        - 'rna' : DataFrame with aggregated messenger RNA read counts.\n",
    "        Both DataFrames have case IDs as indices and molecule identifiers as columns.\n",
    "    \"\"\"\n",
    "    # Create a DataFrame for the files metadata\n",
    "    df_files_metadata = pd.read_csv(BRCA_PROCESSED_FILES_PATHS['files'])\n",
    "\n",
    "    # Initialize a dictionary for the prepared DataFrames\n",
    "    aggregated_norm_reads = dict()\n",
    "\n",
    "    # Prepare the aggregated normalized reads files\n",
    "    for experimental_strategy in ['mir', 'rna']:\n",
    "        # Define the experimental strategy related parameters\n",
    "        drop_column = ['is_expressed']\n",
    "        if experimental_strategy == 'mir':\n",
    "            file_name = AGGREGATED_READS_FILES['mir-norm']\n",
    "            index_column = 'accession_id'\n",
    "        else:\n",
    "            file_name = AGGREGATED_READS_FILES['rna-norm']\n",
    "            index_column = 'gene_name'\n",
    "            drop_column = drop_column + ['gene_id']\n",
    "        \n",
    "        # Get the expressed molecules and set the primary key as index\n",
    "        file_path = os.path.join(processed_dir_path, file_name)\n",
    "        df_aggregated_norm_reads = pd.read_csv(file_path) \\\n",
    "            .query('is_expressed == 1') \\\n",
    "            .drop(columns=drop_column) \\\n",
    "            .set_index(keys=index_column)\n",
    "        \n",
    "        # Map the file ID to the associated case ID\n",
    "        related_files = list(df_aggregated_norm_reads.columns)\n",
    "        df_files_mapping = pd.DataFrame(related_files, columns=['file_id'])\n",
    "        df_files_mapping = df_files_mapping \\\n",
    "            .merge(\n",
    "                right=df_files_metadata,\n",
    "                left_on='file_id',\n",
    "                right_on='file_id',\n",
    "                how='left',\n",
    "            ) \\\n",
    "            [['file_id', 'case_id']]\n",
    "        \n",
    "        # Transpose the DataFrame and set the case ID as index\n",
    "        df_aggregated_norm_reads = df_aggregated_norm_reads \\\n",
    "            .transpose() \\\n",
    "            .reset_index() \\\n",
    "            .rename(columns={'index': 'file_id'}) \\\n",
    "            .merge(\n",
    "                right=df_files_mapping,\n",
    "                left_on='file_id',\n",
    "                right_on='file_id',\n",
    "                how='inner',\n",
    "            ) \\\n",
    "            .drop(columns=['file_id']) \\\n",
    "            .set_index(keys='case_id') \\\n",
    "            .sort_index(ascending=True)\n",
    "        \n",
    "        # Store the prepared DataFrame in the dictionary\n",
    "        aggregated_norm_reads[experimental_strategy] = \\\n",
    "            df_aggregated_norm_reads.copy()\n",
    "    \n",
    "    return aggregated_norm_reads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "5cb6f8d9",
   "metadata": {},
   "outputs": [],
   "source": [
    "def computing_single_pair_spearman(df_mir_expression, df_rna_expression):\n",
    "    \"\"\"\n",
    "    Calculate the Spearman correlation coefficient between the expression profile \n",
    "    of a single microRNA and messenger RNA pair.\n",
    "    \n",
    "    Parameters\n",
    "    ----------\n",
    "    df_mir_expression : pd.Series\n",
    "        Expression values of a single microRNA across samples.\n",
    "    df_rna_expression : pd.Series\n",
    "        Expression values of a single messenger RNA across the same samples.\n",
    "\n",
    "    Returns\n",
    "    -------\n",
    "    dict\n",
    "        A dictionary containing:\n",
    "        - 'accession_id': microRNA accession ID.\n",
    "        - 'gene_name': gene name of the messenger RNA.\n",
    "        - 'correlation': Spearman correlation coefficient.\n",
    "        - 'pvalue': p-value for the hypothesis test that correlation < 0.\n",
    "    \"\"\"\n",
    "    # Calculate the Spearman correlation coefficient for the pair\n",
    "    correlation, pvalue = spearmanr(\n",
    "        a=df_mir_expression, # Samples of the expressed microRNA\n",
    "        b=df_rna_expression, # Samples of the expressed messenger RNAs\n",
    "        axis=0, # Each row is a observation, while the columns are variables\n",
    "        nan_policy='propagate', # Return NaN when input contains it\n",
    "        alternative='less', # The correlation is negative in miRNA-mRNA interactions\n",
    "    )\n",
    "    \n",
    "    # Create a dictionary to represent the result\n",
    "    result = {\n",
    "        'accession_id': df_mir_expression.name,\n",
    "        'gene_name': df_rna_expression.name,\n",
    "        'correlation': correlation,\n",
    "        'pvalue': pvalue,\n",
    "    }\n",
    "    \n",
    "    return result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "865d69f3",
   "metadata": {},
   "outputs": [],
   "source": [
    "def computing_parallel_spearman(cohort):\n",
    "    \"\"\"\n",
    "    Calculate Spearman correlation coefficients between all pairs of expressed microRNAs and \n",
    "    messenger RNAs for a given cohort using parallel processing.\n",
    "\n",
    "    Parameter\n",
    "    ---------\n",
    "    cohort : str\n",
    "        Name of the cohort used to locate interim and processed data directories.\n",
    "\n",
    "    Returns\n",
    "    -------\n",
    "    pd.DataFrame\n",
    "        A DataFrame containing the Spearman correlation results with one row per microRNA-\n",
    "        messenger RNA pair and their corresponding correlation values.\n",
    "    \"\"\"\n",
    "    # Define the cohort interim and processed directories path\n",
    "    dir_base_name = (cohort.lower()).replace(' ', '-')\n",
    "    interim_dir_path = BRCA_INTERIM_FILES_DIRS[dir_base_name]\n",
    "    processed_dir_path = BRCA_PROCESSED_FILES_DIRS[dir_base_name]\n",
    "    \n",
    "    # Prepare the expression data of the expressed molecules\n",
    "    aggregated_norm_reads = preparing_data_for_spearman(processed_dir_path)\n",
    "    df_mir_expression = aggregated_norm_reads['mir']\n",
    "    df_rna_expression = aggregated_norm_reads['rna']\n",
    "    \n",
    "    # Get the sets of expressed microRNAs and messenger RNAs\n",
    "    expressed_mirs = list(df_mir_expression.columns)\n",
    "    expressed_rnas = list(df_rna_expression.columns)\n",
    "    \n",
    "    # Compute in parallel the Spearman correlation coefficient for each pair\n",
    "    results = Parallel(n_jobs=-1, prefer='processes')(\n",
    "        delayed(computing_single_pair_spearman)\n",
    "        (df_mir_expression[mir], df_rna_expression[rna])\n",
    "        for mir in expressed_mirs\n",
    "        for rna in expressed_rnas\n",
    "    )\n",
    "    \n",
    "    # Create a DataFrame for the computing results\n",
    "    df_results = pd.DataFrame(results)\n",
    "    \n",
    "    # Store the DataFrame of Spearman correlation analysis\n",
    "    df_results.to_csv(\n",
    "        os.path.join(interim_dir_path, INTERACTION_INFERENCE_PARAMETERS['interim-file']),\n",
    "        index=False\n",
    "    )\n",
    "    \n",
    "    return df_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "29962316",
   "metadata": {},
   "outputs": [],
   "source": [
    "def inferring_interactions(cohort, df_mirwalk_interactions):\n",
    "    \"\"\"\n",
    "    Infer microRNA-messenger RNA interactions of interest present in the cohort based on \n",
    "    expression correlation and known microRNA-target interactions.\n",
    "\n",
    "    Parameters\n",
    "    ----------\n",
    "    cohort : str\n",
    "        Name of the cohort. Used to resolve file paths.\n",
    "    df_mirwalk_interactions : pd.DataFrame\n",
    "        DataFrame containing microRNA-target interactions of interest from miRWalk. \n",
    "\n",
    "    Returns\n",
    "    -------\n",
    "    pd.DataFrame\n",
    "        DataFrame with inferred interactions.\n",
    "    \"\"\"\n",
    "    # Define the cohort interim and processed directories path\n",
    "    dir_base_name = (cohort.lower()).replace(' ', '-')\n",
    "    interim_dir_path = BRCA_INTERIM_FILES_DIRS[dir_base_name]\n",
    "    processed_dir_path = BRCA_PROCESSED_FILES_DIRS[dir_base_name]\n",
    "    \n",
    "    # Create a DataFrame for the Spearman correlation analysis results\n",
    "    df_spearman_pairs = pd.read_csv(\n",
    "        os.path.join(interim_dir_path, INTERACTION_INFERENCE_PARAMETERS['interim-file'])\n",
    "    )\n",
    "    \n",
    "    # Merge Spearman results with miRWalk interactions of interest\n",
    "    df_interactions = df_spearman_pairs \\\n",
    "        .merge(\n",
    "            right=df_mirwalk_interactions,\n",
    "            left_on=['accession_id', 'gene_name'],\n",
    "            right_on=['accession_id', 'gene_name'],\n",
    "            how='left',\n",
    "        ) \\\n",
    "        [[\n",
    "            'accession_id',\n",
    "            'mirna_name',\n",
    "            'gene_name',\n",
    "            'correlation',\n",
    "            'pvalue',\n",
    "        ]]\n",
    "        \n",
    "    # Flag interactions of interest, i.e. miRWalk interactions of interest that are \n",
    "    # statistically relevant in the cohort expression data\n",
    "    correlation = INTERACTION_INFERENCE_PARAMETERS['correlation']\n",
    "    pvalue = INTERACTION_INFERENCE_PARAMETERS['pvalue']\n",
    "    df_interactions['is_interaction_of_interest'] = np.where(\n",
    "        ((df_interactions['correlation'] <= correlation)\n",
    "         & (df_interactions['pvalue'] <= pvalue)\n",
    "         & (df_interactions['mirna_name'].notna())), 1, 0\n",
    "    )\n",
    "    \n",
    "    # Store the DataFrame of flagged inferred interactions\n",
    "    df_interactions.to_csv(\n",
    "        os.path.join(processed_dir_path, INTERACTION_INFERENCE_PARAMETERS['processed-file']),\n",
    "        index=False\n",
    "    )\n",
    "    \n",
    "    return df_interactions"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "adaa1b0b",
   "metadata": {},
   "source": [
    "# Spearman Correlation Analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f9a5d2ac",
   "metadata": {},
   "source": [
    "## Basal-like"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "6927253e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Compute the Spearman correlation coefficients for Basal-like tumor tissue\n",
    "df_results = computing_parallel_spearman('Basal-like')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "088780f6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>accession_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>correlation</th>\n",
       "      <th>pvalue</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>TSPAN6</td>\n",
       "      <td>-0.099821</td>\n",
       "      <td>0.178811</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>DPM1</td>\n",
       "      <td>0.066141</td>\n",
       "      <td>0.728626</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>SCYL3</td>\n",
       "      <td>0.176059</td>\n",
       "      <td>0.948574</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>C1orf112</td>\n",
       "      <td>-0.026828</td>\n",
       "      <td>0.402586</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>FGR</td>\n",
       "      <td>-0.114274</td>\n",
       "      <td>0.145960</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3569745</th>\n",
       "      <td>MIMAT0026480</td>\n",
       "      <td>AC055839.2</td>\n",
       "      <td>0.209047</td>\n",
       "      <td>0.974003</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3569746</th>\n",
       "      <td>MIMAT0026480</td>\n",
       "      <td>NOTCH2NLC</td>\n",
       "      <td>0.138569</td>\n",
       "      <td>0.899723</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3569747</th>\n",
       "      <td>MIMAT0026480</td>\n",
       "      <td>AP003071.5</td>\n",
       "      <td>0.066159</td>\n",
       "      <td>0.728682</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3569748</th>\n",
       "      <td>MIMAT0026480</td>\n",
       "      <td>DUS4L-BCAP29</td>\n",
       "      <td>0.158554</td>\n",
       "      <td>0.928785</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3569749</th>\n",
       "      <td>MIMAT0026480</td>\n",
       "      <td>AP006621.6</td>\n",
       "      <td>-0.040935</td>\n",
       "      <td>0.353291</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3569750 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         accession_id     gene_name  correlation    pvalue\n",
       "0        MIMAT0000062        TSPAN6    -0.099821  0.178811\n",
       "1        MIMAT0000062          DPM1     0.066141  0.728626\n",
       "2        MIMAT0000062         SCYL3     0.176059  0.948574\n",
       "3        MIMAT0000062      C1orf112    -0.026828  0.402586\n",
       "4        MIMAT0000062           FGR    -0.114274  0.145960\n",
       "...               ...           ...          ...       ...\n",
       "3569745  MIMAT0026480    AC055839.2     0.209047  0.974003\n",
       "3569746  MIMAT0026480     NOTCH2NLC     0.138569  0.899723\n",
       "3569747  MIMAT0026480    AP003071.5     0.066159  0.728682\n",
       "3569748  MIMAT0026480  DUS4L-BCAP29     0.158554  0.928785\n",
       "3569749  MIMAT0026480    AP006621.6    -0.040935  0.353291\n",
       "\n",
       "[3569750 rows x 4 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the DataFrame of Spearman results for this cohort\n",
    "df_results"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a3a473b5",
   "metadata": {},
   "source": [
    "## HER2-enriched"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "bc3d7400",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Compute the Spearman correlation coefficients for HER2-enriched tumor tissue\n",
    "df_results = computing_parallel_spearman('HER2-enriched')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "3f25d0b1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>accession_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>correlation</th>\n",
       "      <th>pvalue</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>TSPAN6</td>\n",
       "      <td>0.055024</td>\n",
       "      <td>0.656445</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>DPM1</td>\n",
       "      <td>0.019959</td>\n",
       "      <td>0.558041</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>SCYL3</td>\n",
       "      <td>0.136295</td>\n",
       "      <td>0.841736</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>C1orf112</td>\n",
       "      <td>0.128640</td>\n",
       "      <td>0.827638</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>FGR</td>\n",
       "      <td>-0.126316</td>\n",
       "      <td>0.176791</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3615177</th>\n",
       "      <td>MIMAT0000281</td>\n",
       "      <td>AC055839.2</td>\n",
       "      <td>-0.151128</td>\n",
       "      <td>0.133106</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3615178</th>\n",
       "      <td>MIMAT0000281</td>\n",
       "      <td>NOTCH2NLC</td>\n",
       "      <td>-0.140260</td>\n",
       "      <td>0.151261</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3615179</th>\n",
       "      <td>MIMAT0000281</td>\n",
       "      <td>AP003071.5</td>\n",
       "      <td>0.016473</td>\n",
       "      <td>0.547957</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3615180</th>\n",
       "      <td>MIMAT0000281</td>\n",
       "      <td>DUS4L-BCAP29</td>\n",
       "      <td>-0.037662</td>\n",
       "      <td>0.391435</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3615181</th>\n",
       "      <td>MIMAT0000281</td>\n",
       "      <td>AP006621.6</td>\n",
       "      <td>0.053725</td>\n",
       "      <td>0.652936</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3615182 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         accession_id     gene_name  correlation    pvalue\n",
       "0        MIMAT0000062        TSPAN6     0.055024  0.656445\n",
       "1        MIMAT0000062          DPM1     0.019959  0.558041\n",
       "2        MIMAT0000062         SCYL3     0.136295  0.841736\n",
       "3        MIMAT0000062      C1orf112     0.128640  0.827638\n",
       "4        MIMAT0000062           FGR    -0.126316  0.176791\n",
       "...               ...           ...          ...       ...\n",
       "3615177  MIMAT0000281    AC055839.2    -0.151128  0.133106\n",
       "3615178  MIMAT0000281     NOTCH2NLC    -0.140260  0.151261\n",
       "3615179  MIMAT0000281    AP003071.5     0.016473  0.547957\n",
       "3615180  MIMAT0000281  DUS4L-BCAP29    -0.037662  0.391435\n",
       "3615181  MIMAT0000281    AP006621.6     0.053725  0.652936\n",
       "\n",
       "[3615182 rows x 4 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the DataFrame of Spearman results for this cohort\n",
    "df_results"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "474a2945",
   "metadata": {},
   "source": [
    "## Luminal A"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "9206a496",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Compute the Spearman correlation coefficients for Luminal A tumor tissue\n",
    "df_results = computing_parallel_spearman('Luminal A')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "0622aa4c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>accession_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>correlation</th>\n",
       "      <th>pvalue</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>TSPAN6</td>\n",
       "      <td>0.004186</td>\n",
       "      <td>0.524780</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>DPM1</td>\n",
       "      <td>-0.006052</td>\n",
       "      <td>0.464194</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>SCYL3</td>\n",
       "      <td>0.191154</td>\n",
       "      <td>0.997915</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>C1orf112</td>\n",
       "      <td>0.085533</td>\n",
       "      <td>0.898390</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>FGR</td>\n",
       "      <td>0.039186</td>\n",
       "      <td>0.719751</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3523599</th>\n",
       "      <td>MIMAT0026480</td>\n",
       "      <td>AC055839.2</td>\n",
       "      <td>0.034058</td>\n",
       "      <td>0.693528</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3523600</th>\n",
       "      <td>MIMAT0026480</td>\n",
       "      <td>NOTCH2NLC</td>\n",
       "      <td>-0.010078</td>\n",
       "      <td>0.440517</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3523601</th>\n",
       "      <td>MIMAT0026480</td>\n",
       "      <td>AP003071.5</td>\n",
       "      <td>0.032441</td>\n",
       "      <td>0.685047</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3523602</th>\n",
       "      <td>MIMAT0026480</td>\n",
       "      <td>DUS4L-BCAP29</td>\n",
       "      <td>0.168949</td>\n",
       "      <td>0.994247</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3523603</th>\n",
       "      <td>MIMAT0026480</td>\n",
       "      <td>AP006621.6</td>\n",
       "      <td>0.114674</td>\n",
       "      <td>0.956226</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3523604 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         accession_id     gene_name  correlation    pvalue\n",
       "0        MIMAT0000062        TSPAN6     0.004186  0.524780\n",
       "1        MIMAT0000062          DPM1    -0.006052  0.464194\n",
       "2        MIMAT0000062         SCYL3     0.191154  0.997915\n",
       "3        MIMAT0000062      C1orf112     0.085533  0.898390\n",
       "4        MIMAT0000062           FGR     0.039186  0.719751\n",
       "...               ...           ...          ...       ...\n",
       "3523599  MIMAT0026480    AC055839.2     0.034058  0.693528\n",
       "3523600  MIMAT0026480     NOTCH2NLC    -0.010078  0.440517\n",
       "3523601  MIMAT0026480    AP003071.5     0.032441  0.685047\n",
       "3523602  MIMAT0026480  DUS4L-BCAP29     0.168949  0.994247\n",
       "3523603  MIMAT0026480    AP006621.6     0.114674  0.956226\n",
       "\n",
       "[3523604 rows x 4 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the DataFrame of Spearman results for this cohort\n",
    "df_results"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "36ce0bae",
   "metadata": {},
   "source": [
    "## Luminal B"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "e4384539",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Compute the Spearman correlation coefficients for Luminal B tumor tissue\n",
    "df_results = computing_parallel_spearman('Luminal B')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "0e683555",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>accession_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>correlation</th>\n",
       "      <th>pvalue</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>TSPAN6</td>\n",
       "      <td>0.121078</td>\n",
       "      <td>0.906133</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>DPM1</td>\n",
       "      <td>0.004202</td>\n",
       "      <td>0.518164</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>SCYL3</td>\n",
       "      <td>0.225023</td>\n",
       "      <td>0.993263</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>C1orf112</td>\n",
       "      <td>0.062984</td>\n",
       "      <td>0.752825</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>FGR</td>\n",
       "      <td>-0.086221</td>\n",
       "      <td>0.174542</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3462111</th>\n",
       "      <td>MIMAT0026480</td>\n",
       "      <td>AC055839.2</td>\n",
       "      <td>0.137787</td>\n",
       "      <td>0.933290</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3462112</th>\n",
       "      <td>MIMAT0026480</td>\n",
       "      <td>NOTCH2NLC</td>\n",
       "      <td>0.183395</td>\n",
       "      <td>0.977519</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3462113</th>\n",
       "      <td>MIMAT0026480</td>\n",
       "      <td>AP003071.5</td>\n",
       "      <td>-0.085193</td>\n",
       "      <td>0.177441</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3462114</th>\n",
       "      <td>MIMAT0026480</td>\n",
       "      <td>DUS4L-BCAP29</td>\n",
       "      <td>0.206896</td>\n",
       "      <td>0.988314</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3462115</th>\n",
       "      <td>MIMAT0026480</td>\n",
       "      <td>AP006621.6</td>\n",
       "      <td>0.062129</td>\n",
       "      <td>0.749883</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3462116 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         accession_id     gene_name  correlation    pvalue\n",
       "0        MIMAT0000062        TSPAN6     0.121078  0.906133\n",
       "1        MIMAT0000062          DPM1     0.004202  0.518164\n",
       "2        MIMAT0000062         SCYL3     0.225023  0.993263\n",
       "3        MIMAT0000062      C1orf112     0.062984  0.752825\n",
       "4        MIMAT0000062           FGR    -0.086221  0.174542\n",
       "...               ...           ...          ...       ...\n",
       "3462111  MIMAT0026480    AC055839.2     0.137787  0.933290\n",
       "3462112  MIMAT0026480     NOTCH2NLC     0.183395  0.977519\n",
       "3462113  MIMAT0026480    AP003071.5    -0.085193  0.177441\n",
       "3462114  MIMAT0026480  DUS4L-BCAP29     0.206896  0.988314\n",
       "3462115  MIMAT0026480    AP006621.6     0.062129  0.749883\n",
       "\n",
       "[3462116 rows x 4 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the DataFrame of Spearman results for this cohort\n",
    "df_results"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a6f9f340",
   "metadata": {},
   "source": [
    "## Paired Normal"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "63f609cf",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Compute the Spearman correlation coefficients for Paired Normal tissue\n",
    "df_results = computing_parallel_spearman('Paired Normal')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "5acc63a5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>accession_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>correlation</th>\n",
       "      <th>pvalue</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>TSPAN6</td>\n",
       "      <td>0.180451</td>\n",
       "      <td>0.908384</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>TNMD</td>\n",
       "      <td>0.108749</td>\n",
       "      <td>0.787512</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>DPM1</td>\n",
       "      <td>0.226521</td>\n",
       "      <td>0.953404</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>SCYL3</td>\n",
       "      <td>0.260150</td>\n",
       "      <td>0.973588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>C1orf112</td>\n",
       "      <td>0.339508</td>\n",
       "      <td>0.994764</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4140823</th>\n",
       "      <td>MIMAT0026738</td>\n",
       "      <td>AC022415.2</td>\n",
       "      <td>-0.105641</td>\n",
       "      <td>0.219204</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4140824</th>\n",
       "      <td>MIMAT0026738</td>\n",
       "      <td>AC055839.2</td>\n",
       "      <td>0.072522</td>\n",
       "      <td>0.702348</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4140825</th>\n",
       "      <td>MIMAT0026738</td>\n",
       "      <td>NOTCH2NLC</td>\n",
       "      <td>-0.018045</td>\n",
       "      <td>0.447491</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4140826</th>\n",
       "      <td>MIMAT0026738</td>\n",
       "      <td>AP003071.5</td>\n",
       "      <td>-0.117567</td>\n",
       "      <td>0.194084</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4140827</th>\n",
       "      <td>MIMAT0026738</td>\n",
       "      <td>DUS4L-BCAP29</td>\n",
       "      <td>-0.139166</td>\n",
       "      <td>0.153172</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4140828 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         accession_id     gene_name  correlation    pvalue\n",
       "0        MIMAT0000062        TSPAN6     0.180451  0.908384\n",
       "1        MIMAT0000062          TNMD     0.108749  0.787512\n",
       "2        MIMAT0000062          DPM1     0.226521  0.953404\n",
       "3        MIMAT0000062         SCYL3     0.260150  0.973588\n",
       "4        MIMAT0000062      C1orf112     0.339508  0.994764\n",
       "...               ...           ...          ...       ...\n",
       "4140823  MIMAT0026738    AC022415.2    -0.105641  0.219204\n",
       "4140824  MIMAT0026738    AC055839.2     0.072522  0.702348\n",
       "4140825  MIMAT0026738     NOTCH2NLC    -0.018045  0.447491\n",
       "4140826  MIMAT0026738    AP003071.5    -0.117567  0.194084\n",
       "4140827  MIMAT0026738  DUS4L-BCAP29    -0.139166  0.153172\n",
       "\n",
       "[4140828 rows x 4 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the DataFrame of Spearman results for this cohort\n",
    "df_results"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1042d63e",
   "metadata": {},
   "source": [
    "# miRWalk Interactions of Interest"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "4aa367b8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Initialize a DataFrame for the interactions of interest from miRWalk\n",
    "df_mirwalk_interactions = pd.DataFrame(columns=['mirna_name', 'gene_name'])\n",
    "\n",
    "# List the interaction files downloaded from miRWalk\n",
    "files = [f for f in os.listdir(MIRWALK_PROCESSED_DATA_DIR) if f.startswith('MIMAT')]\n",
    "\n",
    "# Iterate over each interactions file from miRWalk\n",
    "for file in files:\n",
    "    # Create a DataFrame for the interactions of interest of this microRNA\n",
    "    file_path = os.path.join(MIRWALK_PROCESSED_DATA_DIR, file)    \n",
    "    df_mir_interactions = pd.read_csv(file_path, low_memory=False) \\\n",
    "        .query('is_interaction_of_interest == 1') \\\n",
    "        [['mirna_name', 'gene_name']] \\\n",
    "        .drop_duplicates()\n",
    "    \n",
    "    # Concatenate the interactions of interest to the others\n",
    "    df_mirwalk_interactions = pd.concat(\n",
    "        [df_mirwalk_interactions, df_mir_interactions], ignore_index=True\n",
    "    )\n",
    "\n",
    "# Add the microRNA accession IDs to the DataFrame\n",
    "df_mir_mapping = pd.read_csv(MIRWALK_MIR_MAPPING_FILE_PATH)\n",
    "df_mirwalk_interactions = df_mir_mapping \\\n",
    "    .merge(\n",
    "        right=df_mirwalk_interactions,\n",
    "        left_on='mirna_name',\n",
    "        right_on='mirna_name',\n",
    "        how='inner',\n",
    "    )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "35698fbe",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>accession_id</th>\n",
       "      <th>mirna_name</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MIMAT0003322</td>\n",
       "      <td>hsa-miR-652-3p</td>\n",
       "      <td>KCNN3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MIMAT0003322</td>\n",
       "      <td>hsa-miR-652-3p</td>\n",
       "      <td>HSD3B7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MIMAT0003322</td>\n",
       "      <td>hsa-miR-652-3p</td>\n",
       "      <td>UBE2I</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MIMAT0003322</td>\n",
       "      <td>hsa-miR-652-3p</td>\n",
       "      <td>SLC35C2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MIMAT0003322</td>\n",
       "      <td>hsa-miR-652-3p</td>\n",
       "      <td>GGCX</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51125</th>\n",
       "      <td>MIMAT0000736</td>\n",
       "      <td>hsa-miR-381-3p</td>\n",
       "      <td>SLC6A17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51126</th>\n",
       "      <td>MIMAT0000736</td>\n",
       "      <td>hsa-miR-381-3p</td>\n",
       "      <td>XKR7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51127</th>\n",
       "      <td>MIMAT0000736</td>\n",
       "      <td>hsa-miR-381-3p</td>\n",
       "      <td>RFFL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51128</th>\n",
       "      <td>MIMAT0000736</td>\n",
       "      <td>hsa-miR-381-3p</td>\n",
       "      <td>DNAJB14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51129</th>\n",
       "      <td>MIMAT0000736</td>\n",
       "      <td>hsa-miR-381-3p</td>\n",
       "      <td>NRF1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>51130 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       accession_id      mirna_name gene_name\n",
       "0      MIMAT0003322  hsa-miR-652-3p     KCNN3\n",
       "1      MIMAT0003322  hsa-miR-652-3p    HSD3B7\n",
       "2      MIMAT0003322  hsa-miR-652-3p     UBE2I\n",
       "3      MIMAT0003322  hsa-miR-652-3p   SLC35C2\n",
       "4      MIMAT0003322  hsa-miR-652-3p      GGCX\n",
       "...             ...             ...       ...\n",
       "51125  MIMAT0000736  hsa-miR-381-3p   SLC6A17\n",
       "51126  MIMAT0000736  hsa-miR-381-3p      XKR7\n",
       "51127  MIMAT0000736  hsa-miR-381-3p      RFFL\n",
       "51128  MIMAT0000736  hsa-miR-381-3p   DNAJB14\n",
       "51129  MIMAT0000736  hsa-miR-381-3p      NRF1\n",
       "\n",
       "[51130 rows x 3 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the DataFrame of interactions of interest from miRWalk\n",
    "df_mirwalk_interactions"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "63b7408c",
   "metadata": {},
   "source": [
    "# Interaction Inference\n",
    "According to Anticorrelation Analysis [`statistic`] and miRWalk Data [`biology`]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "46a96a26",
   "metadata": {},
   "source": [
    "## Basal-like"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fa97947d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Infer the interactions of interest present in Basal-like tumor tissue\n",
    "df_interactions = inferring_interactions('Basal-like', df_mirwalk_interactions)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "651e59f1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>accession_id</th>\n",
       "      <th>mirna_name</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>correlation</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>is_interaction_of_interest</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TSPAN6</td>\n",
       "      <td>-0.099821</td>\n",
       "      <td>0.178811</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DPM1</td>\n",
       "      <td>0.066141</td>\n",
       "      <td>0.728626</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>SCYL3</td>\n",
       "      <td>0.176059</td>\n",
       "      <td>0.948574</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C1orf112</td>\n",
       "      <td>-0.026828</td>\n",
       "      <td>0.402586</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FGR</td>\n",
       "      <td>-0.114274</td>\n",
       "      <td>0.145960</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3569745</th>\n",
       "      <td>MIMAT0026480</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AC055839.2</td>\n",
       "      <td>0.209047</td>\n",
       "      <td>0.974003</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3569746</th>\n",
       "      <td>MIMAT0026480</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NOTCH2NLC</td>\n",
       "      <td>0.138569</td>\n",
       "      <td>0.899723</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3569747</th>\n",
       "      <td>MIMAT0026480</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AP003071.5</td>\n",
       "      <td>0.066159</td>\n",
       "      <td>0.728682</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3569748</th>\n",
       "      <td>MIMAT0026480</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DUS4L-BCAP29</td>\n",
       "      <td>0.158554</td>\n",
       "      <td>0.928785</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3569749</th>\n",
       "      <td>MIMAT0026480</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AP006621.6</td>\n",
       "      <td>-0.040935</td>\n",
       "      <td>0.353291</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3569750 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         accession_id     mirna_name     gene_name  correlation    pvalue  \\\n",
       "0        MIMAT0000062            NaN        TSPAN6    -0.099821  0.178811   \n",
       "1        MIMAT0000062            NaN          DPM1     0.066141  0.728626   \n",
       "2        MIMAT0000062  hsa-let-7a-5p         SCYL3     0.176059  0.948574   \n",
       "3        MIMAT0000062            NaN      C1orf112    -0.026828  0.402586   \n",
       "4        MIMAT0000062            NaN           FGR    -0.114274  0.145960   \n",
       "...               ...            ...           ...          ...       ...   \n",
       "3569745  MIMAT0026480            NaN    AC055839.2     0.209047  0.974003   \n",
       "3569746  MIMAT0026480            NaN     NOTCH2NLC     0.138569  0.899723   \n",
       "3569747  MIMAT0026480            NaN    AP003071.5     0.066159  0.728682   \n",
       "3569748  MIMAT0026480            NaN  DUS4L-BCAP29     0.158554  0.928785   \n",
       "3569749  MIMAT0026480            NaN    AP006621.6    -0.040935  0.353291   \n",
       "\n",
       "         is_interaction_of_interest  \n",
       "0                                 0  \n",
       "1                                 0  \n",
       "2                                 0  \n",
       "3                                 0  \n",
       "4                                 0  \n",
       "...                             ...  \n",
       "3569745                           0  \n",
       "3569746                           0  \n",
       "3569747                           0  \n",
       "3569748                           0  \n",
       "3569749                           0  \n",
       "\n",
       "[3569750 rows x 6 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the DataFrame of inferred interactions for this cohort\n",
    "df_interactions"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "144895c2",
   "metadata": {},
   "source": [
    "## HER2-enriched"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "b5e06728",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Infer the interactions of interest present in HER2-enriched tumor tissue\n",
    "df_interactions = inferring_interactions('HER2-enriched', df_mirwalk_interactions)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1f34fb9c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>accession_id</th>\n",
       "      <th>mirna_name</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>correlation</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>is_interaction_of_interest</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TSPAN6</td>\n",
       "      <td>0.055024</td>\n",
       "      <td>0.656445</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DPM1</td>\n",
       "      <td>0.019959</td>\n",
       "      <td>0.558041</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>SCYL3</td>\n",
       "      <td>0.136295</td>\n",
       "      <td>0.841736</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C1orf112</td>\n",
       "      <td>0.128640</td>\n",
       "      <td>0.827638</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FGR</td>\n",
       "      <td>-0.126316</td>\n",
       "      <td>0.176791</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3615177</th>\n",
       "      <td>MIMAT0000281</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AC055839.2</td>\n",
       "      <td>-0.151128</td>\n",
       "      <td>0.133106</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3615178</th>\n",
       "      <td>MIMAT0000281</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NOTCH2NLC</td>\n",
       "      <td>-0.140260</td>\n",
       "      <td>0.151261</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3615179</th>\n",
       "      <td>MIMAT0000281</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AP003071.5</td>\n",
       "      <td>0.016473</td>\n",
       "      <td>0.547957</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3615180</th>\n",
       "      <td>MIMAT0000281</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DUS4L-BCAP29</td>\n",
       "      <td>-0.037662</td>\n",
       "      <td>0.391435</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3615181</th>\n",
       "      <td>MIMAT0000281</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AP006621.6</td>\n",
       "      <td>0.053725</td>\n",
       "      <td>0.652936</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3615182 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         accession_id     mirna_name     gene_name  correlation    pvalue  \\\n",
       "0        MIMAT0000062            NaN        TSPAN6     0.055024  0.656445   \n",
       "1        MIMAT0000062            NaN          DPM1     0.019959  0.558041   \n",
       "2        MIMAT0000062  hsa-let-7a-5p         SCYL3     0.136295  0.841736   \n",
       "3        MIMAT0000062            NaN      C1orf112     0.128640  0.827638   \n",
       "4        MIMAT0000062            NaN           FGR    -0.126316  0.176791   \n",
       "...               ...            ...           ...          ...       ...   \n",
       "3615177  MIMAT0000281            NaN    AC055839.2    -0.151128  0.133106   \n",
       "3615178  MIMAT0000281            NaN     NOTCH2NLC    -0.140260  0.151261   \n",
       "3615179  MIMAT0000281            NaN    AP003071.5     0.016473  0.547957   \n",
       "3615180  MIMAT0000281            NaN  DUS4L-BCAP29    -0.037662  0.391435   \n",
       "3615181  MIMAT0000281            NaN    AP006621.6     0.053725  0.652936   \n",
       "\n",
       "         is_interaction_of_interest  \n",
       "0                                 0  \n",
       "1                                 0  \n",
       "2                                 0  \n",
       "3                                 0  \n",
       "4                                 0  \n",
       "...                             ...  \n",
       "3615177                           0  \n",
       "3615178                           0  \n",
       "3615179                           0  \n",
       "3615180                           0  \n",
       "3615181                           0  \n",
       "\n",
       "[3615182 rows x 6 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the DataFrame of inferred interactions for this cohort\n",
    "df_interactions"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4163aa9c",
   "metadata": {},
   "source": [
    "## Luminal A"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e2bcaf1d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Infer the interactions of interest present in Luminal A tumor tissue\n",
    "df_interactions = inferring_interactions('Luminal A', df_mirwalk_interactions)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "97271637",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>accession_id</th>\n",
       "      <th>mirna_name</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>correlation</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>is_interaction_of_interest</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TSPAN6</td>\n",
       "      <td>0.004186</td>\n",
       "      <td>0.524780</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DPM1</td>\n",
       "      <td>-0.006052</td>\n",
       "      <td>0.464194</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>SCYL3</td>\n",
       "      <td>0.191154</td>\n",
       "      <td>0.997915</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C1orf112</td>\n",
       "      <td>0.085533</td>\n",
       "      <td>0.898390</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FGR</td>\n",
       "      <td>0.039186</td>\n",
       "      <td>0.719751</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3523599</th>\n",
       "      <td>MIMAT0026480</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AC055839.2</td>\n",
       "      <td>0.034058</td>\n",
       "      <td>0.693528</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3523600</th>\n",
       "      <td>MIMAT0026480</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NOTCH2NLC</td>\n",
       "      <td>-0.010078</td>\n",
       "      <td>0.440517</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3523601</th>\n",
       "      <td>MIMAT0026480</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AP003071.5</td>\n",
       "      <td>0.032441</td>\n",
       "      <td>0.685047</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3523602</th>\n",
       "      <td>MIMAT0026480</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DUS4L-BCAP29</td>\n",
       "      <td>0.168949</td>\n",
       "      <td>0.994247</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3523603</th>\n",
       "      <td>MIMAT0026480</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AP006621.6</td>\n",
       "      <td>0.114674</td>\n",
       "      <td>0.956226</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3523604 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         accession_id     mirna_name     gene_name  correlation    pvalue  \\\n",
       "0        MIMAT0000062            NaN        TSPAN6     0.004186  0.524780   \n",
       "1        MIMAT0000062            NaN          DPM1    -0.006052  0.464194   \n",
       "2        MIMAT0000062  hsa-let-7a-5p         SCYL3     0.191154  0.997915   \n",
       "3        MIMAT0000062            NaN      C1orf112     0.085533  0.898390   \n",
       "4        MIMAT0000062            NaN           FGR     0.039186  0.719751   \n",
       "...               ...            ...           ...          ...       ...   \n",
       "3523599  MIMAT0026480            NaN    AC055839.2     0.034058  0.693528   \n",
       "3523600  MIMAT0026480            NaN     NOTCH2NLC    -0.010078  0.440517   \n",
       "3523601  MIMAT0026480            NaN    AP003071.5     0.032441  0.685047   \n",
       "3523602  MIMAT0026480            NaN  DUS4L-BCAP29     0.168949  0.994247   \n",
       "3523603  MIMAT0026480            NaN    AP006621.6     0.114674  0.956226   \n",
       "\n",
       "         is_interaction_of_interest  \n",
       "0                                 0  \n",
       "1                                 0  \n",
       "2                                 0  \n",
       "3                                 0  \n",
       "4                                 0  \n",
       "...                             ...  \n",
       "3523599                           0  \n",
       "3523600                           0  \n",
       "3523601                           0  \n",
       "3523602                           0  \n",
       "3523603                           0  \n",
       "\n",
       "[3523604 rows x 6 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the DataFrame of inferred interactions for this cohort\n",
    "df_interactions"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "71054d06",
   "metadata": {},
   "source": [
    "## Luminal B"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "2b1d6df7",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Infer the interactions of interest present in Luminal B tumor tissue\n",
    "df_interactions = inferring_interactions('Luminal B', df_mirwalk_interactions)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "66164dff",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>accession_id</th>\n",
       "      <th>mirna_name</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>correlation</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>is_interaction_of_interest</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TSPAN6</td>\n",
       "      <td>0.121078</td>\n",
       "      <td>0.906133</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DPM1</td>\n",
       "      <td>0.004202</td>\n",
       "      <td>0.518164</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>SCYL3</td>\n",
       "      <td>0.225023</td>\n",
       "      <td>0.993263</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C1orf112</td>\n",
       "      <td>0.062984</td>\n",
       "      <td>0.752825</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FGR</td>\n",
       "      <td>-0.086221</td>\n",
       "      <td>0.174542</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3462111</th>\n",
       "      <td>MIMAT0026480</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AC055839.2</td>\n",
       "      <td>0.137787</td>\n",
       "      <td>0.933290</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3462112</th>\n",
       "      <td>MIMAT0026480</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NOTCH2NLC</td>\n",
       "      <td>0.183395</td>\n",
       "      <td>0.977519</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3462113</th>\n",
       "      <td>MIMAT0026480</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AP003071.5</td>\n",
       "      <td>-0.085193</td>\n",
       "      <td>0.177441</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3462114</th>\n",
       "      <td>MIMAT0026480</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DUS4L-BCAP29</td>\n",
       "      <td>0.206896</td>\n",
       "      <td>0.988314</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3462115</th>\n",
       "      <td>MIMAT0026480</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AP006621.6</td>\n",
       "      <td>0.062129</td>\n",
       "      <td>0.749883</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3462116 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         accession_id     mirna_name     gene_name  correlation    pvalue  \\\n",
       "0        MIMAT0000062            NaN        TSPAN6     0.121078  0.906133   \n",
       "1        MIMAT0000062            NaN          DPM1     0.004202  0.518164   \n",
       "2        MIMAT0000062  hsa-let-7a-5p         SCYL3     0.225023  0.993263   \n",
       "3        MIMAT0000062            NaN      C1orf112     0.062984  0.752825   \n",
       "4        MIMAT0000062            NaN           FGR    -0.086221  0.174542   \n",
       "...               ...            ...           ...          ...       ...   \n",
       "3462111  MIMAT0026480            NaN    AC055839.2     0.137787  0.933290   \n",
       "3462112  MIMAT0026480            NaN     NOTCH2NLC     0.183395  0.977519   \n",
       "3462113  MIMAT0026480            NaN    AP003071.5    -0.085193  0.177441   \n",
       "3462114  MIMAT0026480            NaN  DUS4L-BCAP29     0.206896  0.988314   \n",
       "3462115  MIMAT0026480            NaN    AP006621.6     0.062129  0.749883   \n",
       "\n",
       "         is_interaction_of_interest  \n",
       "0                                 0  \n",
       "1                                 0  \n",
       "2                                 0  \n",
       "3                                 0  \n",
       "4                                 0  \n",
       "...                             ...  \n",
       "3462111                           0  \n",
       "3462112                           0  \n",
       "3462113                           0  \n",
       "3462114                           0  \n",
       "3462115                           0  \n",
       "\n",
       "[3462116 rows x 6 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the DataFrame of inferred interactions for this cohort\n",
    "df_interactions"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "99db44f5",
   "metadata": {},
   "source": [
    "## Paired Normal"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "179cee9d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Infer the interactions of interest present in Paired Normal tissue\n",
    "df_interactions = inferring_interactions('Paired Normal', df_mirwalk_interactions)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "12e67699",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>accession_id</th>\n",
       "      <th>mirna_name</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>correlation</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>is_interaction_of_interest</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TSPAN6</td>\n",
       "      <td>0.180451</td>\n",
       "      <td>0.908384</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TNMD</td>\n",
       "      <td>0.108749</td>\n",
       "      <td>0.787512</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DPM1</td>\n",
       "      <td>0.226521</td>\n",
       "      <td>0.953404</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>SCYL3</td>\n",
       "      <td>0.260150</td>\n",
       "      <td>0.973588</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C1orf112</td>\n",
       "      <td>0.339508</td>\n",
       "      <td>0.994764</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4140823</th>\n",
       "      <td>MIMAT0026738</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AC022415.2</td>\n",
       "      <td>-0.105641</td>\n",
       "      <td>0.219204</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4140824</th>\n",
       "      <td>MIMAT0026738</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AC055839.2</td>\n",
       "      <td>0.072522</td>\n",
       "      <td>0.702348</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4140825</th>\n",
       "      <td>MIMAT0026738</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NOTCH2NLC</td>\n",
       "      <td>-0.018045</td>\n",
       "      <td>0.447491</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4140826</th>\n",
       "      <td>MIMAT0026738</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AP003071.5</td>\n",
       "      <td>-0.117567</td>\n",
       "      <td>0.194084</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4140827</th>\n",
       "      <td>MIMAT0026738</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DUS4L-BCAP29</td>\n",
       "      <td>-0.139166</td>\n",
       "      <td>0.153172</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4140828 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         accession_id     mirna_name     gene_name  correlation    pvalue  \\\n",
       "0        MIMAT0000062            NaN        TSPAN6     0.180451  0.908384   \n",
       "1        MIMAT0000062            NaN          TNMD     0.108749  0.787512   \n",
       "2        MIMAT0000062            NaN          DPM1     0.226521  0.953404   \n",
       "3        MIMAT0000062  hsa-let-7a-5p         SCYL3     0.260150  0.973588   \n",
       "4        MIMAT0000062            NaN      C1orf112     0.339508  0.994764   \n",
       "...               ...            ...           ...          ...       ...   \n",
       "4140823  MIMAT0026738            NaN    AC022415.2    -0.105641  0.219204   \n",
       "4140824  MIMAT0026738            NaN    AC055839.2     0.072522  0.702348   \n",
       "4140825  MIMAT0026738            NaN     NOTCH2NLC    -0.018045  0.447491   \n",
       "4140826  MIMAT0026738            NaN    AP003071.5    -0.117567  0.194084   \n",
       "4140827  MIMAT0026738            NaN  DUS4L-BCAP29    -0.139166  0.153172   \n",
       "\n",
       "         is_interaction_of_interest  \n",
       "0                                 0  \n",
       "1                                 0  \n",
       "2                                 0  \n",
       "3                                 0  \n",
       "4                                 0  \n",
       "...                             ...  \n",
       "4140823                           0  \n",
       "4140824                           0  \n",
       "4140825                           0  \n",
       "4140826                           0  \n",
       "4140827                           0  \n",
       "\n",
       "[4140828 rows x 6 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the DataFrame of inferred interactions for this cohort\n",
    "df_interactions"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
