{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "77334dcd",
   "metadata": {},
   "source": [
    "# Inference of MicroRNA-Messenger RNA Interactions in TCGA-BRCA\n",
    "TCGA: The Cancer Genome Atlas  \n",
    "BRCA: Breast Invasive Carcinoma"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5221ad8c",
   "metadata": {},
   "source": [
    "# Importing Libraries and Configurations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "001781d4",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import sys\n",
    "\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "from joblib import Parallel, delayed\n",
    "from scipy.stats import false_discovery_control, spearmanr\n",
    "\n",
    "# Add project root to Python's path\n",
    "sys.path.append(os.path.abspath(os.path.join('..', '..')))\n",
    "\n",
    "from config import (\n",
    "    AGGREGATED_READS_FILES,\n",
    "    BRCA_PROCESSED_FILES_DIRS,\n",
    "    BRCA_PROCESSED_FILES_PATHS,\n",
    "    INTERACTION_INFERENCE_PARAMETERS,\n",
    "    MIRWALK_MIR_MAPPING_FILE_PATH,\n",
    "    MIRWALK_PROCESSED_DATA_DIR,\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9069dd41",
   "metadata": {},
   "source": [
    "# Function"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "871bf352",
   "metadata": {},
   "outputs": [],
   "source": [
    "def prepare_data_for_spearman(processed_dir_path):\n",
    "    \"\"\"\n",
    "    Prepare aggregated expression data of a group for Spearman correlation analysis.\n",
    "\n",
    "    Parameters\n",
    "    ----------\n",
    "    processed_dir_path : str\n",
    "        Path to the directory containing processed expression data files.\n",
    "\n",
    "    Returns\n",
    "    -------\n",
    "    dict of pd.DataFrame\n",
    "        A dictionary with keys:\n",
    "        - 'mir' : DataFrame with aggregated microRNA normalized read counts.\n",
    "        - 'rna' : DataFrame with aggregated messenger RNA normalized read counts.\n",
    "        Both DataFrames have case IDs as indices and molecule identifiers as columns.\n",
    "    \"\"\"\n",
    "    # Create a DataFrame for the files metadata\n",
    "    df_files_metadata = pd.read_csv(BRCA_PROCESSED_FILES_PATHS['files'])\n",
    "\n",
    "    # Initialize a dictionary for Spearman-ready DataFrames\n",
    "    aggregated_reads = dict()\n",
    "\n",
    "    # Prepare the aggregated reads files\n",
    "    for experimental_strategy in ['mir', 'rna']:\n",
    "        # Define the experimental strategy related parameters\n",
    "        drop_column = ['is_expressed']\n",
    "        if experimental_strategy == 'mir':\n",
    "            index_column = 'accession_id'\n",
    "            file_name = AGGREGATED_READS_FILES['mir-normalized']\n",
    "        else:\n",
    "            index_column = 'gene_name'\n",
    "            drop_column = drop_column + ['gene_id']\n",
    "            file_name = AGGREGATED_READS_FILES['rna-normalized']\n",
    "        \n",
    "        # Get the expressed molecules and set the primary key as index\n",
    "        file_path = os.path.join(processed_dir_path, file_name)\n",
    "        df_agg_reads = pd.read_csv(file_path) \\\n",
    "            .query('is_expressed == 1') \\\n",
    "            .drop(columns=drop_column) \\\n",
    "            .set_index(keys=index_column)\n",
    "        \n",
    "        # Map the file ID to the associated case ID\n",
    "        file_ids = list(df_agg_reads.columns)\n",
    "        df_files_mapping = pd.DataFrame(file_ids, columns=['file_id'])\n",
    "        df_files_mapping = df_files_mapping \\\n",
    "            .merge(\n",
    "                right=df_files_metadata,\n",
    "                left_on='file_id',\n",
    "                right_on='file_id',\n",
    "                how='left',\n",
    "            ) \\\n",
    "            [['file_id', 'case_id']]\n",
    "        \n",
    "        # Transpose the DataFrame and set the case ID as index\n",
    "        df_agg_reads = df_agg_reads \\\n",
    "            .transpose() \\\n",
    "            .reset_index() \\\n",
    "            .rename(columns={'index': 'file_id'}) \\\n",
    "            .merge(\n",
    "                right=df_files_mapping,\n",
    "                left_on='file_id',\n",
    "                right_on='file_id',\n",
    "                how='inner',\n",
    "            ) \\\n",
    "            .drop(columns=['file_id']) \\\n",
    "            .set_index(keys='case_id') \\\n",
    "            .sort_index(ascending=True)\n",
    "        \n",
    "        # Store the prepared DataFrame in the dictionary\n",
    "        aggregated_reads[experimental_strategy] = df_agg_reads.copy()\n",
    "    \n",
    "    return aggregated_reads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "5ca411e7",
   "metadata": {},
   "outputs": [],
   "source": [
    "def compute_single_pair_spearman(ser_mir_expression, ser_rna_expression):\n",
    "    \"\"\"\n",
    "    Calculate the Spearman correlation coefficient between the expression profile \n",
    "    of a single microRNA and messenger RNA pair.\n",
    "    \n",
    "    Parameters\n",
    "    ----------\n",
    "    ser_mir_expression : pd.Series\n",
    "        Expression values of a single microRNA across samples.\n",
    "    ser_rna_expression : pd.Series\n",
    "        Expression values of a single messenger RNA across the same samples.\n",
    "\n",
    "    Returns\n",
    "    -------\n",
    "    dict\n",
    "        A dictionary containing:\n",
    "        - 'accession_id': microRNA accession ID.\n",
    "        - 'gene_name': gene name of the messenger RNA.\n",
    "        - 'correlation': Spearman correlation coefficient.\n",
    "        - 'pvalue': p-value for the hypothesis test that correlation < 0.\n",
    "    \"\"\"\n",
    "    # Calculate the Spearman correlation coefficient for the pair\n",
    "    axis = INTERACTION_INFERENCE_PARAMETERS['axis']\n",
    "    alternative = INTERACTION_INFERENCE_PARAMETERS['alternative']\n",
    "    correlation, pvalue = spearmanr(\n",
    "        a=ser_mir_expression, # Samples of the expressed microRNA\n",
    "        b=ser_rna_expression, # Samples of the expressed messenger RNAs\n",
    "        axis=axis, # Each row is a observation, while the columns are variables\n",
    "        alternative=alternative, # The correlation is negative in miRNA-mRNA inferred_interactions\n",
    "    )\n",
    "    \n",
    "    # Create a dictionary to represent the results\n",
    "    results = {\n",
    "        'accession_id': ser_mir_expression.name,\n",
    "        'gene_name': ser_rna_expression.name,\n",
    "        'correlation': correlation,\n",
    "        'pvalue': pvalue,\n",
    "    }\n",
    "    \n",
    "    return results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "865d69f3",
   "metadata": {},
   "outputs": [],
   "source": [
    "def infer_mir_rna_interactions(df_mirwalk_interactions, group):\n",
    "    \"\"\"\n",
    "    Calculate Spearman correlation coefficients between all pairs of expressed microRNAs and \n",
    "    messenger RNAs for a given group using parallel processing.\n",
    "\n",
    "    Parameter\n",
    "    ---------\n",
    "    group : str\n",
    "        Name of the group used to locate interim and processed data directories.\n",
    "\n",
    "    Returns\n",
    "    -------\n",
    "    pd.DataFrame\n",
    "        A DataFrame containing the Spearman correlation results with one row per microRNA-\n",
    "        messenger RNA pair and their corresponding correlation values.\n",
    "    \"\"\"\n",
    "    # Define the group interim and processed directories path\n",
    "    dir_base_name = (group.lower()).replace(' ', '-')\n",
    "    processed_dir_path = BRCA_PROCESSED_FILES_DIRS[dir_base_name]\n",
    "    \n",
    "    # Prepare the expression data of the expressed molecules\n",
    "    aggregated_reads = prepare_data_for_spearman(processed_dir_path)\n",
    "    df_mir_expression = aggregated_reads['mir']\n",
    "    df_rna_expression = aggregated_reads['rna']\n",
    "    \n",
    "    # Define all potential inferred_interactions between expressed microRNAs and messenger RNAs\n",
    "    df_expressed_mirs = pd.DataFrame(\n",
    "        data=list(df_mir_expression.columns), columns=['accession_id']\n",
    "    )\n",
    "    df_expressed_rnas = pd.DataFrame(\n",
    "        data=list(df_rna_expression.columns), columns=['gene_name']\n",
    "    )\n",
    "    df_potential_inferred_interactions = df_expressed_mirs \\\n",
    "        .merge(right=df_expressed_rnas, how='cross')\n",
    "    \n",
    "    # Select inferred_interactions with miRWalk retrieved evidence\n",
    "    primary_key = ['accession_id', 'gene_name']\n",
    "    df_inferred_interactions = df_potential_inferred_interactions \\\n",
    "        .merge(\n",
    "            right=df_mirwalk_interactions,\n",
    "            left_on=primary_key,\n",
    "            right_on=primary_key,\n",
    "            how='inner',\n",
    "        )\n",
    "    group_inferred_interactions = zip(\n",
    "        df_inferred_interactions['accession_id'], df_inferred_interactions['gene_name'],\n",
    "    )\n",
    "    \n",
    "    # Compute in parallel the Spearman correlation coefficient for each pair\n",
    "    results = Parallel(n_jobs=-1, prefer='processes')(\n",
    "        delayed(compute_single_pair_spearman)\n",
    "        (df_mir_expression[mir], df_rna_expression[rna])\n",
    "        for mir, rna in group_inferred_interactions\n",
    "    )\n",
    "    \n",
    "    # Create a DataFrame for the computing results\n",
    "    df_results = pd.DataFrame(results)\n",
    "    \n",
    "    # Adjust the p-values to control the false discovery fate (FDR)\n",
    "    axis = INTERACTION_INFERENCE_PARAMETERS['axis']\n",
    "    method = INTERACTION_INFERENCE_PARAMETERS['fdr-method']\n",
    "    qvalues = false_discovery_control(\n",
    "        ps=df_results['pvalue'], # The p-values to adjust\n",
    "        axis=axis, # The axis along which to perform the adjustment\n",
    "        method=method, # FDR control procedure\n",
    "    )\n",
    "    df_results['qvalue'] = qvalues\n",
    "    \n",
    "    # Add miRWalk data to results\n",
    "    df_results = df_mirwalk_interactions \\\n",
    "        .merge(\n",
    "            right=df_results,\n",
    "            left_on=primary_key,\n",
    "            right_on=primary_key,\n",
    "            how='inner',\n",
    "        )\n",
    "        \n",
    "    # Flag the inferred interactions of interest, i.e. those with correlation and \n",
    "    # q-value values below the thresholds\n",
    "    min_correlation = INTERACTION_INFERENCE_PARAMETERS['min_correlation']\n",
    "    min_qvalue = INTERACTION_INFERENCE_PARAMETERS['min_qvalue']\n",
    "    df_results['is_interaction_of_interest'] = np.where(\n",
    "        ((df_results['correlation'] < min_correlation)\n",
    "         & (df_results['qvalue'] < min_qvalue)), 1, 0\n",
    "    )\n",
    "    \n",
    "    # Store the DataFrame of inferred interactions\n",
    "    file_name = INTERACTION_INFERENCE_PARAMETERS['file']\n",
    "    df_results.to_csv(os.path.join(processed_dir_path, file_name), index=False)\n",
    "    \n",
    "    return df_results"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "39372508",
   "metadata": {},
   "source": [
    "# miRWalk Interactions of Interest"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "01b3d654",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Initialize a DataFrame for the inferred interactions of interest from miRWalk\n",
    "mirwalk_columns = ['mirna_name', 'gene_name', 'mirtarbase']\n",
    "df_mirwalk_interactions = pd.DataFrame(columns=mirwalk_columns)\n",
    "\n",
    "# List the interaction files downloaded from miRWalk\n",
    "files = [f for f in os.listdir(MIRWALK_PROCESSED_DATA_DIR) if f.startswith('MIMAT')]\n",
    "\n",
    "# Iterate over each inferred_interactions file from miRWalk\n",
    "for file in files:\n",
    "    # Create a DataFrame for the inferred interactions of interest of the microRNA\n",
    "    file_path = os.path.join(MIRWALK_PROCESSED_DATA_DIR, file)    \n",
    "    df_mir_inferred_interactions = pd.read_csv(file_path, low_memory=False) \\\n",
    "        .query('is_interaction_of_interest == 1') \\\n",
    "        [mirwalk_columns] \\\n",
    "        .drop_duplicates()\n",
    "    \n",
    "    # Concatenate the inferred interactions of interest to the others\n",
    "    df_mirwalk_interactions = pd.concat(\n",
    "        [df_mirwalk_interactions, df_mir_inferred_interactions], ignore_index=True\n",
    "    )\n",
    "\n",
    "# Add the microRNA accession IDs to the DataFrame\n",
    "df_mir_mapping = pd.read_csv(MIRWALK_MIR_MAPPING_FILE_PATH)\n",
    "df_mirwalk_interactions = df_mir_mapping \\\n",
    "    .merge(\n",
    "        right=df_mirwalk_interactions,\n",
    "        left_on='mirna_name',\n",
    "        right_on='mirna_name',\n",
    "        how='inner',\n",
    "    )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "86955188",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>accession_id</th>\n",
       "      <th>mirna_name</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>mirtarbase</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>PKIA</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>GJC1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>PBX3</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>ITSN1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>SRGAP1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41848</th>\n",
       "      <td>MIMAT0026483</td>\n",
       "      <td>hsa-miR-370-5p</td>\n",
       "      <td>STUM</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41849</th>\n",
       "      <td>MIMAT0026483</td>\n",
       "      <td>hsa-miR-370-5p</td>\n",
       "      <td>CRISPLD2</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41850</th>\n",
       "      <td>MIMAT0026483</td>\n",
       "      <td>hsa-miR-370-5p</td>\n",
       "      <td>ADAM19</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41851</th>\n",
       "      <td>MIMAT0026557</td>\n",
       "      <td>hsa-miR-412-5p</td>\n",
       "      <td>FAM53C</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41852</th>\n",
       "      <td>MIMAT0026557</td>\n",
       "      <td>hsa-miR-412-5p</td>\n",
       "      <td>FICD</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>41853 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       accession_id      mirna_name gene_name mirtarbase\n",
       "0      MIMAT0000062   hsa-let-7a-5p      PKIA        NaN\n",
       "1      MIMAT0000062   hsa-let-7a-5p      GJC1        NaN\n",
       "2      MIMAT0000062   hsa-let-7a-5p      PBX3        NaN\n",
       "3      MIMAT0000062   hsa-let-7a-5p     ITSN1        NaN\n",
       "4      MIMAT0000062   hsa-let-7a-5p    SRGAP1        NaN\n",
       "...             ...             ...       ...        ...\n",
       "41848  MIMAT0026483  hsa-miR-370-5p      STUM        NaN\n",
       "41849  MIMAT0026483  hsa-miR-370-5p  CRISPLD2        NaN\n",
       "41850  MIMAT0026483  hsa-miR-370-5p    ADAM19        NaN\n",
       "41851  MIMAT0026557  hsa-miR-412-5p    FAM53C        NaN\n",
       "41852  MIMAT0026557  hsa-miR-412-5p      FICD        NaN\n",
       "\n",
       "[41853 rows x 4 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the DataFrame of inferred_interactions of interest from miRWalk\n",
    "df_mirwalk_interactions"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "adaa1b0b",
   "metadata": {},
   "source": [
    "# Interaction Inference"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f9a5d2ac",
   "metadata": {},
   "source": [
    "## Basal-like"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "6927253e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Infer the interactions for Basal-like tumor tissue\n",
    "inferred_interactions = infer_mir_rna_interactions(df_mirwalk_interactions, 'Basal-like')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "288e3888",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>accession_id</th>\n",
       "      <th>mirna_name</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>mirtarbase</th>\n",
       "      <th>correlation</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>qvalue</th>\n",
       "      <th>is_interaction_of_interest</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>PKIA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.078607</td>\n",
       "      <td>0.234621</td>\n",
       "      <td>0.745015</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>GJC1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.018153</td>\n",
       "      <td>0.566268</td>\n",
       "      <td>0.978192</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>PBX3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.143526</td>\n",
       "      <td>0.907620</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>ITSN1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.033079</td>\n",
       "      <td>0.380501</td>\n",
       "      <td>0.873011</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>SRGAP1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.047532</td>\n",
       "      <td>0.669012</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41164</th>\n",
       "      <td>MIMAT0026483</td>\n",
       "      <td>hsa-miR-370-5p</td>\n",
       "      <td>STUM</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.092500</td>\n",
       "      <td>0.802930</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41165</th>\n",
       "      <td>MIMAT0026483</td>\n",
       "      <td>hsa-miR-370-5p</td>\n",
       "      <td>CRISPLD2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.430324</td>\n",
       "      <td>0.999984</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41166</th>\n",
       "      <td>MIMAT0026483</td>\n",
       "      <td>hsa-miR-370-5p</td>\n",
       "      <td>ADAM19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.221807</td>\n",
       "      <td>0.980527</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41167</th>\n",
       "      <td>MIMAT0026557</td>\n",
       "      <td>hsa-miR-412-5p</td>\n",
       "      <td>FAM53C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.016884</td>\n",
       "      <td>0.438326</td>\n",
       "      <td>0.911847</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41168</th>\n",
       "      <td>MIMAT0026557</td>\n",
       "      <td>hsa-miR-412-5p</td>\n",
       "      <td>FICD</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.110946</td>\n",
       "      <td>0.846853</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>41169 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       accession_id      mirna_name gene_name mirtarbase  correlation  \\\n",
       "0      MIMAT0000062   hsa-let-7a-5p      PKIA        NaN    -0.078607   \n",
       "1      MIMAT0000062   hsa-let-7a-5p      GJC1        NaN     0.018153   \n",
       "2      MIMAT0000062   hsa-let-7a-5p      PBX3        NaN     0.143526   \n",
       "3      MIMAT0000062   hsa-let-7a-5p     ITSN1        NaN    -0.033079   \n",
       "4      MIMAT0000062   hsa-let-7a-5p    SRGAP1        NaN     0.047532   \n",
       "...             ...             ...       ...        ...          ...   \n",
       "41164  MIMAT0026483  hsa-miR-370-5p      STUM        NaN     0.092500   \n",
       "41165  MIMAT0026483  hsa-miR-370-5p  CRISPLD2        NaN     0.430324   \n",
       "41166  MIMAT0026483  hsa-miR-370-5p    ADAM19        NaN     0.221807   \n",
       "41167  MIMAT0026557  hsa-miR-412-5p    FAM53C        NaN    -0.016884   \n",
       "41168  MIMAT0026557  hsa-miR-412-5p      FICD        NaN     0.110946   \n",
       "\n",
       "         pvalue    qvalue  is_interaction_of_interest  \n",
       "0      0.234621  0.745015                           0  \n",
       "1      0.566268  0.978192                           0  \n",
       "2      0.907620  1.000000                           0  \n",
       "3      0.380501  0.873011                           0  \n",
       "4      0.669012  1.000000                           0  \n",
       "...         ...       ...                         ...  \n",
       "41164  0.802930  1.000000                           0  \n",
       "41165  0.999984  1.000000                           0  \n",
       "41166  0.980527  1.000000                           0  \n",
       "41167  0.438326  0.911847                           0  \n",
       "41168  0.846853  1.000000                           0  \n",
       "\n",
       "[41169 rows x 8 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the inferred interactions associated with this group\n",
    "inferred_interactions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "db41e12e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>accession_id</th>\n",
       "      <th>mirna_name</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>mirtarbase</th>\n",
       "      <th>correlation</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>qvalue</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MIMAT0000064</td>\n",
       "      <td>hsa-let-7c-5p</td>\n",
       "      <td>IGF1R</td>\n",
       "      <td>MIRT053087</td>\n",
       "      <td>-0.388423</td>\n",
       "      <td>0.000101</td>\n",
       "      <td>0.028263</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MIMAT0000064</td>\n",
       "      <td>hsa-let-7c-5p</td>\n",
       "      <td>EIF2S2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.416928</td>\n",
       "      <td>0.000029</td>\n",
       "      <td>0.017575</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MIMAT0000065</td>\n",
       "      <td>hsa-let-7d-5p</td>\n",
       "      <td>ITSN1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.378399</td>\n",
       "      <td>0.000151</td>\n",
       "      <td>0.033543</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MIMAT0000066</td>\n",
       "      <td>hsa-let-7e-5p</td>\n",
       "      <td>CD86</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.367774</td>\n",
       "      <td>0.000229</td>\n",
       "      <td>0.039746</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MIMAT0000066</td>\n",
       "      <td>hsa-let-7e-5p</td>\n",
       "      <td>CD200R1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.372494</td>\n",
       "      <td>0.000191</td>\n",
       "      <td>0.037333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>291</th>\n",
       "      <td>MIMAT0002820</td>\n",
       "      <td>hsa-miR-497-5p</td>\n",
       "      <td>HMGA1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.392998</td>\n",
       "      <td>0.000083</td>\n",
       "      <td>0.026621</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>292</th>\n",
       "      <td>MIMAT0002820</td>\n",
       "      <td>hsa-miR-497-5p</td>\n",
       "      <td>EIF3M</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.401217</td>\n",
       "      <td>0.000059</td>\n",
       "      <td>0.023656</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>293</th>\n",
       "      <td>MIMAT0003301</td>\n",
       "      <td>hsa-miR-33b-5p</td>\n",
       "      <td>TARDBP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.400115</td>\n",
       "      <td>0.000062</td>\n",
       "      <td>0.023656</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>294</th>\n",
       "      <td>MIMAT0004693</td>\n",
       "      <td>hsa-miR-330-5p</td>\n",
       "      <td>NGFR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.375501</td>\n",
       "      <td>0.000169</td>\n",
       "      <td>0.035921</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>295</th>\n",
       "      <td>MIMAT0016847</td>\n",
       "      <td>hsa-miR-378c</td>\n",
       "      <td>HSPA12A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.361158</td>\n",
       "      <td>0.000294</td>\n",
       "      <td>0.045228</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>296 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     accession_id      mirna_name gene_name  mirtarbase  correlation  \\\n",
       "0    MIMAT0000064   hsa-let-7c-5p     IGF1R  MIRT053087    -0.388423   \n",
       "1    MIMAT0000064   hsa-let-7c-5p    EIF2S2         NaN    -0.416928   \n",
       "2    MIMAT0000065   hsa-let-7d-5p     ITSN1         NaN    -0.378399   \n",
       "3    MIMAT0000066   hsa-let-7e-5p      CD86         NaN    -0.367774   \n",
       "4    MIMAT0000066   hsa-let-7e-5p   CD200R1         NaN    -0.372494   \n",
       "..            ...             ...       ...         ...          ...   \n",
       "291  MIMAT0002820  hsa-miR-497-5p     HMGA1         NaN    -0.392998   \n",
       "292  MIMAT0002820  hsa-miR-497-5p     EIF3M         NaN    -0.401217   \n",
       "293  MIMAT0003301  hsa-miR-33b-5p    TARDBP         NaN    -0.400115   \n",
       "294  MIMAT0004693  hsa-miR-330-5p      NGFR         NaN    -0.375501   \n",
       "295  MIMAT0016847    hsa-miR-378c   HSPA12A         NaN    -0.361158   \n",
       "\n",
       "       pvalue    qvalue  \n",
       "0    0.000101  0.028263  \n",
       "1    0.000029  0.017575  \n",
       "2    0.000151  0.033543  \n",
       "3    0.000229  0.039746  \n",
       "4    0.000191  0.037333  \n",
       "..        ...       ...  \n",
       "291  0.000083  0.026621  \n",
       "292  0.000059  0.023656  \n",
       "293  0.000062  0.023656  \n",
       "294  0.000169  0.035921  \n",
       "295  0.000294  0.045228  \n",
       "\n",
       "[296 rows x 7 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the inferred interactions of interest associated with this group\n",
    "inferred_interactions \\\n",
    "    .query('is_interaction_of_interest == 1') \\\n",
    "    .drop(columns=['is_interaction_of_interest']) \\\n",
    "    .reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a3a473b5",
   "metadata": {},
   "source": [
    "## HER2-enriched"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "bc3d7400",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Infer the interactions for HER2-enriched tumor tissue\n",
    "inferred_interactions = infer_mir_rna_interactions(df_mirwalk_interactions, 'HER2-enriched')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "3f25d0b1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>accession_id</th>\n",
       "      <th>mirna_name</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>mirtarbase</th>\n",
       "      <th>correlation</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>qvalue</th>\n",
       "      <th>is_interaction_of_interest</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>PKIA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.013944</td>\n",
       "      <td>0.459379</td>\n",
       "      <td>0.871133</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>GJC1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.248189</td>\n",
       "      <td>0.967434</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>PBX3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.003896</td>\n",
       "      <td>0.511368</td>\n",
       "      <td>0.894937</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>ITSN1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.102802</td>\n",
       "      <td>0.774561</td>\n",
       "      <td>0.982053</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>SRGAP1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.067943</td>\n",
       "      <td>0.309403</td>\n",
       "      <td>0.789843</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41164</th>\n",
       "      <td>MIMAT0026483</td>\n",
       "      <td>hsa-miR-370-5p</td>\n",
       "      <td>STUM</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.378153</td>\n",
       "      <td>0.997971</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41165</th>\n",
       "      <td>MIMAT0026483</td>\n",
       "      <td>hsa-miR-370-5p</td>\n",
       "      <td>CRISPLD2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.378153</td>\n",
       "      <td>0.997971</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41166</th>\n",
       "      <td>MIMAT0026483</td>\n",
       "      <td>hsa-miR-370-5p</td>\n",
       "      <td>ADAM19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.262629</td>\n",
       "      <td>0.974737</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41167</th>\n",
       "      <td>MIMAT0026557</td>\n",
       "      <td>hsa-miR-412-5p</td>\n",
       "      <td>FAM53C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.236705</td>\n",
       "      <td>0.960494</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41168</th>\n",
       "      <td>MIMAT0026557</td>\n",
       "      <td>hsa-miR-412-5p</td>\n",
       "      <td>FICD</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.221394</td>\n",
       "      <td>0.949478</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>41169 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       accession_id      mirna_name gene_name mirtarbase  correlation  \\\n",
       "0      MIMAT0000062   hsa-let-7a-5p      PKIA        NaN    -0.013944   \n",
       "1      MIMAT0000062   hsa-let-7a-5p      GJC1        NaN     0.248189   \n",
       "2      MIMAT0000062   hsa-let-7a-5p      PBX3        NaN     0.003896   \n",
       "3      MIMAT0000062   hsa-let-7a-5p     ITSN1        NaN     0.102802   \n",
       "4      MIMAT0000062   hsa-let-7a-5p    SRGAP1        NaN    -0.067943   \n",
       "...             ...             ...       ...        ...          ...   \n",
       "41164  MIMAT0026483  hsa-miR-370-5p      STUM        NaN     0.378153   \n",
       "41165  MIMAT0026483  hsa-miR-370-5p  CRISPLD2        NaN     0.378153   \n",
       "41166  MIMAT0026483  hsa-miR-370-5p    ADAM19        NaN     0.262629   \n",
       "41167  MIMAT0026557  hsa-miR-412-5p    FAM53C        NaN     0.236705   \n",
       "41168  MIMAT0026557  hsa-miR-412-5p      FICD        NaN     0.221394   \n",
       "\n",
       "         pvalue    qvalue  is_interaction_of_interest  \n",
       "0      0.459379  0.871133                           0  \n",
       "1      0.967434  1.000000                           0  \n",
       "2      0.511368  0.894937                           0  \n",
       "3      0.774561  0.982053                           0  \n",
       "4      0.309403  0.789843                           0  \n",
       "...         ...       ...                         ...  \n",
       "41164  0.997971  1.000000                           0  \n",
       "41165  0.997971  1.000000                           0  \n",
       "41166  0.974737  1.000000                           0  \n",
       "41167  0.960494  1.000000                           0  \n",
       "41168  0.949478  1.000000                           0  \n",
       "\n",
       "[41169 rows x 8 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the inferred interactions associated with this group\n",
    "inferred_interactions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "2348e5fe",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>accession_id</th>\n",
       "      <th>mirna_name</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>mirtarbase</th>\n",
       "      <th>correlation</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>qvalue</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MIMAT0000065</td>\n",
       "      <td>hsa-let-7d-5p</td>\n",
       "      <td>HIF1AN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.539303</td>\n",
       "      <td>8.986564e-06</td>\n",
       "      <td>0.041561</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MIMAT0000065</td>\n",
       "      <td>hsa-let-7d-5p</td>\n",
       "      <td>MEF2D</td>\n",
       "      <td>MIRT735654</td>\n",
       "      <td>-0.562406</td>\n",
       "      <td>3.225376e-06</td>\n",
       "      <td>0.032134</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MIMAT0000065</td>\n",
       "      <td>hsa-let-7d-5p</td>\n",
       "      <td>SMARCC1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.546275</td>\n",
       "      <td>6.649086e-06</td>\n",
       "      <td>0.041507</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MIMAT0000065</td>\n",
       "      <td>hsa-let-7d-5p</td>\n",
       "      <td>PRKAR2A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.536637</td>\n",
       "      <td>1.006593e-05</td>\n",
       "      <td>0.041561</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MIMAT0000072</td>\n",
       "      <td>hsa-miR-18a-5p</td>\n",
       "      <td>ZBTB20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.558715</td>\n",
       "      <td>3.818749e-06</td>\n",
       "      <td>0.032134</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MIMAT0000244</td>\n",
       "      <td>hsa-miR-30c-5p</td>\n",
       "      <td>SNAI1</td>\n",
       "      <td>MIRT006762</td>\n",
       "      <td>-0.529050</td>\n",
       "      <td>1.382919e-05</td>\n",
       "      <td>0.043795</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>MIMAT0000259</td>\n",
       "      <td>hsa-miR-182-5p</td>\n",
       "      <td>GPHN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.531852</td>\n",
       "      <td>1.230922e-05</td>\n",
       "      <td>0.043795</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>MIMAT0000432</td>\n",
       "      <td>hsa-miR-141-3p</td>\n",
       "      <td>IGF2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.587491</td>\n",
       "      <td>9.681125e-07</td>\n",
       "      <td>0.032134</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>MIMAT0000646</td>\n",
       "      <td>hsa-miR-155-5p</td>\n",
       "      <td>VAV3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.536569</td>\n",
       "      <td>1.009512e-05</td>\n",
       "      <td>0.041561</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>MIMAT0000680</td>\n",
       "      <td>hsa-miR-106b-5p</td>\n",
       "      <td>PLXDC2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.558237</td>\n",
       "      <td>3.902691e-06</td>\n",
       "      <td>0.032134</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>MIMAT0000680</td>\n",
       "      <td>hsa-miR-106b-5p</td>\n",
       "      <td>SALL1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.544908</td>\n",
       "      <td>7.057413e-06</td>\n",
       "      <td>0.041507</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>MIMAT0000732</td>\n",
       "      <td>hsa-miR-378a-3p</td>\n",
       "      <td>ACP6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.561791</td>\n",
       "      <td>3.317913e-06</td>\n",
       "      <td>0.032134</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>MIMAT0005792</td>\n",
       "      <td>hsa-miR-320b</td>\n",
       "      <td>MYEF2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.529323</td>\n",
       "      <td>1.367360e-05</td>\n",
       "      <td>0.043795</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    accession_id       mirna_name gene_name  mirtarbase  correlation  \\\n",
       "0   MIMAT0000065    hsa-let-7d-5p    HIF1AN         NaN    -0.539303   \n",
       "1   MIMAT0000065    hsa-let-7d-5p     MEF2D  MIRT735654    -0.562406   \n",
       "2   MIMAT0000065    hsa-let-7d-5p   SMARCC1         NaN    -0.546275   \n",
       "3   MIMAT0000065    hsa-let-7d-5p   PRKAR2A         NaN    -0.536637   \n",
       "4   MIMAT0000072   hsa-miR-18a-5p    ZBTB20         NaN    -0.558715   \n",
       "5   MIMAT0000244   hsa-miR-30c-5p     SNAI1  MIRT006762    -0.529050   \n",
       "6   MIMAT0000259   hsa-miR-182-5p      GPHN         NaN    -0.531852   \n",
       "7   MIMAT0000432   hsa-miR-141-3p      IGF2         NaN    -0.587491   \n",
       "8   MIMAT0000646   hsa-miR-155-5p      VAV3         NaN    -0.536569   \n",
       "9   MIMAT0000680  hsa-miR-106b-5p    PLXDC2         NaN    -0.558237   \n",
       "10  MIMAT0000680  hsa-miR-106b-5p     SALL1         NaN    -0.544908   \n",
       "11  MIMAT0000732  hsa-miR-378a-3p      ACP6         NaN    -0.561791   \n",
       "12  MIMAT0005792     hsa-miR-320b     MYEF2         NaN    -0.529323   \n",
       "\n",
       "          pvalue    qvalue  \n",
       "0   8.986564e-06  0.041561  \n",
       "1   3.225376e-06  0.032134  \n",
       "2   6.649086e-06  0.041507  \n",
       "3   1.006593e-05  0.041561  \n",
       "4   3.818749e-06  0.032134  \n",
       "5   1.382919e-05  0.043795  \n",
       "6   1.230922e-05  0.043795  \n",
       "7   9.681125e-07  0.032134  \n",
       "8   1.009512e-05  0.041561  \n",
       "9   3.902691e-06  0.032134  \n",
       "10  7.057413e-06  0.041507  \n",
       "11  3.317913e-06  0.032134  \n",
       "12  1.367360e-05  0.043795  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the inferred interactions of interest associated with this group\n",
    "inferred_interactions \\\n",
    "    .query('is_interaction_of_interest == 1') \\\n",
    "    .drop(columns=['is_interaction_of_interest']) \\\n",
    "    .reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "474a2945",
   "metadata": {},
   "source": [
    "## Luminal A"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "9206a496",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Infer the interactions for Luminal A tumor tissue\n",
    "inferred_interactions = infer_mir_rna_interactions(df_mirwalk_interactions, 'Luminal A')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "0622aa4c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>accession_id</th>\n",
       "      <th>mirna_name</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>mirtarbase</th>\n",
       "      <th>correlation</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>qvalue</th>\n",
       "      <th>is_interaction_of_interest</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>PKIA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.010777</td>\n",
       "      <td>0.436428</td>\n",
       "      <td>0.841128</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>GJC1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.043499</td>\n",
       "      <td>0.259064</td>\n",
       "      <td>0.648037</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>PBX3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.059124</td>\n",
       "      <td>0.810222</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>ITSN1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.061249</td>\n",
       "      <td>0.818681</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>SRGAP1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.035360</td>\n",
       "      <td>0.299708</td>\n",
       "      <td>0.697952</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41164</th>\n",
       "      <td>MIMAT0026483</td>\n",
       "      <td>hsa-miR-370-5p</td>\n",
       "      <td>STUM</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.143357</td>\n",
       "      <td>0.983815</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41165</th>\n",
       "      <td>MIMAT0026483</td>\n",
       "      <td>hsa-miR-370-5p</td>\n",
       "      <td>CRISPLD2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.456634</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41166</th>\n",
       "      <td>MIMAT0026483</td>\n",
       "      <td>hsa-miR-370-5p</td>\n",
       "      <td>ADAM19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.252987</td>\n",
       "      <td>0.999933</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41167</th>\n",
       "      <td>MIMAT0026557</td>\n",
       "      <td>hsa-miR-412-5p</td>\n",
       "      <td>FAM53C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.017501</td>\n",
       "      <td>0.602526</td>\n",
       "      <td>0.970287</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41168</th>\n",
       "      <td>MIMAT0026557</td>\n",
       "      <td>hsa-miR-412-5p</td>\n",
       "      <td>FICD</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.029306</td>\n",
       "      <td>0.331685</td>\n",
       "      <td>0.734977</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>41169 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       accession_id      mirna_name gene_name mirtarbase  correlation  \\\n",
       "0      MIMAT0000062   hsa-let-7a-5p      PKIA        NaN    -0.010777   \n",
       "1      MIMAT0000062   hsa-let-7a-5p      GJC1        NaN    -0.043499   \n",
       "2      MIMAT0000062   hsa-let-7a-5p      PBX3        NaN     0.059124   \n",
       "3      MIMAT0000062   hsa-let-7a-5p     ITSN1        NaN     0.061249   \n",
       "4      MIMAT0000062   hsa-let-7a-5p    SRGAP1        NaN    -0.035360   \n",
       "...             ...             ...       ...        ...          ...   \n",
       "41164  MIMAT0026483  hsa-miR-370-5p      STUM        NaN     0.143357   \n",
       "41165  MIMAT0026483  hsa-miR-370-5p  CRISPLD2        NaN     0.456634   \n",
       "41166  MIMAT0026483  hsa-miR-370-5p    ADAM19        NaN     0.252987   \n",
       "41167  MIMAT0026557  hsa-miR-412-5p    FAM53C        NaN     0.017501   \n",
       "41168  MIMAT0026557  hsa-miR-412-5p      FICD        NaN    -0.029306   \n",
       "\n",
       "         pvalue    qvalue  is_interaction_of_interest  \n",
       "0      0.436428  0.841128                           0  \n",
       "1      0.259064  0.648037                           0  \n",
       "2      0.810222  1.000000                           0  \n",
       "3      0.818681  1.000000                           0  \n",
       "4      0.299708  0.697952                           0  \n",
       "...         ...       ...                         ...  \n",
       "41164  0.983815  1.000000                           0  \n",
       "41165  1.000000  1.000000                           0  \n",
       "41166  0.999933  1.000000                           0  \n",
       "41167  0.602526  0.970287                           0  \n",
       "41168  0.331685  0.734977                           0  \n",
       "\n",
       "[41169 rows x 8 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the inferred interactions associated with this group\n",
    "inferred_interactions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "eafc3e23",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>accession_id</th>\n",
       "      <th>mirna_name</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>mirtarbase</th>\n",
       "      <th>correlation</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>qvalue</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MIMAT0000064</td>\n",
       "      <td>hsa-let-7c-5p</td>\n",
       "      <td>PARD6B</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.305317</td>\n",
       "      <td>1.699354e-06</td>\n",
       "      <td>0.000311</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MIMAT0000069</td>\n",
       "      <td>hsa-miR-16-5p</td>\n",
       "      <td>TLE4</td>\n",
       "      <td>MIRT031643</td>\n",
       "      <td>-0.315052</td>\n",
       "      <td>7.889337e-07</td>\n",
       "      <td>0.000169</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MIMAT0000069</td>\n",
       "      <td>hsa-miR-16-5p</td>\n",
       "      <td>PLSCR4</td>\n",
       "      <td>MIRT031585</td>\n",
       "      <td>-0.338303</td>\n",
       "      <td>1.127362e-07</td>\n",
       "      <td>0.000046</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MIMAT0000069</td>\n",
       "      <td>hsa-miR-16-5p</td>\n",
       "      <td>ETV1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.325088</td>\n",
       "      <td>3.474575e-07</td>\n",
       "      <td>0.000097</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MIMAT0000069</td>\n",
       "      <td>hsa-miR-16-5p</td>\n",
       "      <td>ACVR2A</td>\n",
       "      <td>MIRT000536</td>\n",
       "      <td>-0.312958</td>\n",
       "      <td>9.326604e-07</td>\n",
       "      <td>0.000193</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>254</th>\n",
       "      <td>MIMAT0003301</td>\n",
       "      <td>hsa-miR-33b-5p</td>\n",
       "      <td>CROT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.302597</td>\n",
       "      <td>2.095452e-06</td>\n",
       "      <td>0.000356</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>255</th>\n",
       "      <td>MIMAT0003301</td>\n",
       "      <td>hsa-miR-33b-5p</td>\n",
       "      <td>SLC16A7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.323011</td>\n",
       "      <td>4.127512e-07</td>\n",
       "      <td>0.000109</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>256</th>\n",
       "      <td>MIMAT0003393</td>\n",
       "      <td>hsa-miR-425-5p</td>\n",
       "      <td>PPP2CB</td>\n",
       "      <td>MIRT016657</td>\n",
       "      <td>-0.310217</td>\n",
       "      <td>1.158869e-06</td>\n",
       "      <td>0.000233</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>257</th>\n",
       "      <td>MIMAT0004945</td>\n",
       "      <td>hsa-miR-744-5p</td>\n",
       "      <td>HPSE2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.378697</td>\n",
       "      <td>2.572259e-09</td>\n",
       "      <td>0.000003</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>258</th>\n",
       "      <td>MIMAT0022834</td>\n",
       "      <td>hsa-miR-365b-3p</td>\n",
       "      <td>MEF2D</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.302939</td>\n",
       "      <td>2.041148e-06</td>\n",
       "      <td>0.000351</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>259 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     accession_id       mirna_name gene_name  mirtarbase  correlation  \\\n",
       "0    MIMAT0000064    hsa-let-7c-5p    PARD6B         NaN    -0.305317   \n",
       "1    MIMAT0000069    hsa-miR-16-5p      TLE4  MIRT031643    -0.315052   \n",
       "2    MIMAT0000069    hsa-miR-16-5p    PLSCR4  MIRT031585    -0.338303   \n",
       "3    MIMAT0000069    hsa-miR-16-5p      ETV1         NaN    -0.325088   \n",
       "4    MIMAT0000069    hsa-miR-16-5p    ACVR2A  MIRT000536    -0.312958   \n",
       "..            ...              ...       ...         ...          ...   \n",
       "254  MIMAT0003301   hsa-miR-33b-5p      CROT         NaN    -0.302597   \n",
       "255  MIMAT0003301   hsa-miR-33b-5p   SLC16A7         NaN    -0.323011   \n",
       "256  MIMAT0003393   hsa-miR-425-5p    PPP2CB  MIRT016657    -0.310217   \n",
       "257  MIMAT0004945   hsa-miR-744-5p     HPSE2         NaN    -0.378697   \n",
       "258  MIMAT0022834  hsa-miR-365b-3p     MEF2D         NaN    -0.302939   \n",
       "\n",
       "           pvalue    qvalue  \n",
       "0    1.699354e-06  0.000311  \n",
       "1    7.889337e-07  0.000169  \n",
       "2    1.127362e-07  0.000046  \n",
       "3    3.474575e-07  0.000097  \n",
       "4    9.326604e-07  0.000193  \n",
       "..            ...       ...  \n",
       "254  2.095452e-06  0.000356  \n",
       "255  4.127512e-07  0.000109  \n",
       "256  1.158869e-06  0.000233  \n",
       "257  2.572259e-09  0.000003  \n",
       "258  2.041148e-06  0.000351  \n",
       "\n",
       "[259 rows x 7 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the inferred interactions of interest associated with this group\n",
    "inferred_interactions \\\n",
    "    .query('is_interaction_of_interest == 1') \\\n",
    "    .drop(columns=['is_interaction_of_interest']) \\\n",
    "    .reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "36ce0bae",
   "metadata": {},
   "source": [
    "## Luminal B"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "e4384539",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Infer the interactions for Luminal B tumor tissue\n",
    "inferred_interactions = infer_mir_rna_interactions(df_mirwalk_interactions, 'Luminal B')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "0e683555",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>accession_id</th>\n",
       "      <th>mirna_name</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>mirtarbase</th>\n",
       "      <th>correlation</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>qvalue</th>\n",
       "      <th>is_interaction_of_interest</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>PKIA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.162456</td>\n",
       "      <td>0.038132</td>\n",
       "      <td>0.262916</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>GJC1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.096166</td>\n",
       "      <td>0.851953</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>PBX3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.064838</td>\n",
       "      <td>0.759150</td>\n",
       "      <td>0.995618</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>ITSN1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.149260</td>\n",
       "      <td>0.948140</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>SRGAP1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.173825</td>\n",
       "      <td>0.971198</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41164</th>\n",
       "      <td>MIMAT0026483</td>\n",
       "      <td>hsa-miR-370-5p</td>\n",
       "      <td>STUM</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.076256</td>\n",
       "      <td>0.796110</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41165</th>\n",
       "      <td>MIMAT0026483</td>\n",
       "      <td>hsa-miR-370-5p</td>\n",
       "      <td>CRISPLD2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.317266</td>\n",
       "      <td>0.999793</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41166</th>\n",
       "      <td>MIMAT0026483</td>\n",
       "      <td>hsa-miR-370-5p</td>\n",
       "      <td>ADAM19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.141708</td>\n",
       "      <td>0.938692</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41167</th>\n",
       "      <td>MIMAT0026557</td>\n",
       "      <td>hsa-miR-412-5p</td>\n",
       "      <td>FAM53C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.070633</td>\n",
       "      <td>0.221659</td>\n",
       "      <td>0.598784</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41168</th>\n",
       "      <td>MIMAT0026557</td>\n",
       "      <td>hsa-miR-412-5p</td>\n",
       "      <td>FICD</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.016142</td>\n",
       "      <td>0.430545</td>\n",
       "      <td>0.799941</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>41169 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       accession_id      mirna_name gene_name mirtarbase  correlation  \\\n",
       "0      MIMAT0000062   hsa-let-7a-5p      PKIA        NaN    -0.162456   \n",
       "1      MIMAT0000062   hsa-let-7a-5p      GJC1        NaN     0.096166   \n",
       "2      MIMAT0000062   hsa-let-7a-5p      PBX3        NaN     0.064838   \n",
       "3      MIMAT0000062   hsa-let-7a-5p     ITSN1        NaN     0.149260   \n",
       "4      MIMAT0000062   hsa-let-7a-5p    SRGAP1        NaN     0.173825   \n",
       "...             ...             ...       ...        ...          ...   \n",
       "41164  MIMAT0026483  hsa-miR-370-5p      STUM        NaN     0.076256   \n",
       "41165  MIMAT0026483  hsa-miR-370-5p  CRISPLD2        NaN     0.317266   \n",
       "41166  MIMAT0026483  hsa-miR-370-5p    ADAM19        NaN     0.141708   \n",
       "41167  MIMAT0026557  hsa-miR-412-5p    FAM53C        NaN    -0.070633   \n",
       "41168  MIMAT0026557  hsa-miR-412-5p      FICD        NaN    -0.016142   \n",
       "\n",
       "         pvalue    qvalue  is_interaction_of_interest  \n",
       "0      0.038132  0.262916                           0  \n",
       "1      0.851953  1.000000                           0  \n",
       "2      0.759150  0.995618                           0  \n",
       "3      0.948140  1.000000                           0  \n",
       "4      0.971198  1.000000                           0  \n",
       "...         ...       ...                         ...  \n",
       "41164  0.796110  1.000000                           0  \n",
       "41165  0.999793  1.000000                           0  \n",
       "41166  0.938692  1.000000                           0  \n",
       "41167  0.221659  0.598784                           0  \n",
       "41168  0.430545  0.799941                           0  \n",
       "\n",
       "[41169 rows x 8 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the inferred interactions associated with this group\n",
    "inferred_interactions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "eb8b24e1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>accession_id</th>\n",
       "      <th>mirna_name</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>mirtarbase</th>\n",
       "      <th>correlation</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>qvalue</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MIMAT0000063</td>\n",
       "      <td>hsa-let-7b-5p</td>\n",
       "      <td>SENP5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.305959</td>\n",
       "      <td>0.000339</td>\n",
       "      <td>0.027519</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MIMAT0000063</td>\n",
       "      <td>hsa-let-7b-5p</td>\n",
       "      <td>PEG10</td>\n",
       "      <td>MIRT561775</td>\n",
       "      <td>-0.351302</td>\n",
       "      <td>0.000042</td>\n",
       "      <td>0.010108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MIMAT0000065</td>\n",
       "      <td>hsa-let-7d-5p</td>\n",
       "      <td>PDE12</td>\n",
       "      <td>MIRT120920</td>\n",
       "      <td>-0.301715</td>\n",
       "      <td>0.000406</td>\n",
       "      <td>0.029633</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MIMAT0000065</td>\n",
       "      <td>hsa-let-7d-5p</td>\n",
       "      <td>CALU</td>\n",
       "      <td>MIRT123316</td>\n",
       "      <td>-0.300611</td>\n",
       "      <td>0.000425</td>\n",
       "      <td>0.030313</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MIMAT0000065</td>\n",
       "      <td>hsa-let-7d-5p</td>\n",
       "      <td>CSRNP3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.316057</td>\n",
       "      <td>0.000219</td>\n",
       "      <td>0.022328</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>579</th>\n",
       "      <td>MIMAT0016847</td>\n",
       "      <td>hsa-miR-378c</td>\n",
       "      <td>TOLLIP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.317707</td>\n",
       "      <td>0.000203</td>\n",
       "      <td>0.021596</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>580</th>\n",
       "      <td>MIMAT0016847</td>\n",
       "      <td>hsa-miR-378c</td>\n",
       "      <td>MEF2D</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.330541</td>\n",
       "      <td>0.000113</td>\n",
       "      <td>0.016482</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>581</th>\n",
       "      <td>MIMAT0016847</td>\n",
       "      <td>hsa-miR-378c</td>\n",
       "      <td>ABI2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.312810</td>\n",
       "      <td>0.000252</td>\n",
       "      <td>0.024002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>582</th>\n",
       "      <td>MIMAT0018443</td>\n",
       "      <td>hsa-miR-374c-5p</td>\n",
       "      <td>ATXN1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.304957</td>\n",
       "      <td>0.000354</td>\n",
       "      <td>0.027953</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>583</th>\n",
       "      <td>MIMAT0018443</td>\n",
       "      <td>hsa-miR-374c-5p</td>\n",
       "      <td>PRRC2C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.339867</td>\n",
       "      <td>0.000073</td>\n",
       "      <td>0.013476</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>584 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     accession_id       mirna_name gene_name  mirtarbase  correlation  \\\n",
       "0    MIMAT0000063    hsa-let-7b-5p     SENP5         NaN    -0.305959   \n",
       "1    MIMAT0000063    hsa-let-7b-5p     PEG10  MIRT561775    -0.351302   \n",
       "2    MIMAT0000065    hsa-let-7d-5p     PDE12  MIRT120920    -0.301715   \n",
       "3    MIMAT0000065    hsa-let-7d-5p      CALU  MIRT123316    -0.300611   \n",
       "4    MIMAT0000065    hsa-let-7d-5p    CSRNP3         NaN    -0.316057   \n",
       "..            ...              ...       ...         ...          ...   \n",
       "579  MIMAT0016847     hsa-miR-378c    TOLLIP         NaN    -0.317707   \n",
       "580  MIMAT0016847     hsa-miR-378c     MEF2D         NaN    -0.330541   \n",
       "581  MIMAT0016847     hsa-miR-378c      ABI2         NaN    -0.312810   \n",
       "582  MIMAT0018443  hsa-miR-374c-5p     ATXN1         NaN    -0.304957   \n",
       "583  MIMAT0018443  hsa-miR-374c-5p    PRRC2C         NaN    -0.339867   \n",
       "\n",
       "       pvalue    qvalue  \n",
       "0    0.000339  0.027519  \n",
       "1    0.000042  0.010108  \n",
       "2    0.000406  0.029633  \n",
       "3    0.000425  0.030313  \n",
       "4    0.000219  0.022328  \n",
       "..        ...       ...  \n",
       "579  0.000203  0.021596  \n",
       "580  0.000113  0.016482  \n",
       "581  0.000252  0.024002  \n",
       "582  0.000354  0.027953  \n",
       "583  0.000073  0.013476  \n",
       "\n",
       "[584 rows x 7 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the inferred interactions of interest associated with this group\n",
    "inferred_interactions \\\n",
    "    .query('is_interaction_of_interest == 1') \\\n",
    "    .drop(columns=['is_interaction_of_interest']) \\\n",
    "    .reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a6f9f340",
   "metadata": {},
   "source": [
    "## Paired Normal"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "63f609cf",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Infer the interactions for Paired Normal tissue\n",
    "inferred_interactions = infer_mir_rna_interactions(df_mirwalk_interactions, 'Paired Normal')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "5acc63a5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>accession_id</th>\n",
       "      <th>mirna_name</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>mirtarbase</th>\n",
       "      <th>correlation</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>qvalue</th>\n",
       "      <th>is_interaction_of_interest</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>PKIA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.237936</td>\n",
       "      <td>0.961289</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>GJC1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.015379</td>\n",
       "      <td>0.455214</td>\n",
       "      <td>0.898533</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>PBX3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.007314</td>\n",
       "      <td>0.521332</td>\n",
       "      <td>0.969059</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>ITSN1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.077102</td>\n",
       "      <td>0.286103</td>\n",
       "      <td>0.674410</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>SRGAP1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.162748</td>\n",
       "      <td>0.884626</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41164</th>\n",
       "      <td>MIMAT0026483</td>\n",
       "      <td>hsa-miR-370-5p</td>\n",
       "      <td>STUM</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.198770</td>\n",
       "      <td>0.929030</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41165</th>\n",
       "      <td>MIMAT0026483</td>\n",
       "      <td>hsa-miR-370-5p</td>\n",
       "      <td>CRISPLD2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.176897</td>\n",
       "      <td>0.903923</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41166</th>\n",
       "      <td>MIMAT0026483</td>\n",
       "      <td>hsa-miR-370-5p</td>\n",
       "      <td>ADAM19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.176555</td>\n",
       "      <td>0.903485</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41167</th>\n",
       "      <td>MIMAT0026557</td>\n",
       "      <td>hsa-miR-412-5p</td>\n",
       "      <td>FAM53C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.143814</td>\n",
       "      <td>0.854845</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41168</th>\n",
       "      <td>MIMAT0026557</td>\n",
       "      <td>hsa-miR-412-5p</td>\n",
       "      <td>FICD</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.260560</td>\n",
       "      <td>0.973781</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>41169 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       accession_id      mirna_name gene_name mirtarbase  correlation  \\\n",
       "0      MIMAT0000062   hsa-let-7a-5p      PKIA        NaN     0.237936   \n",
       "1      MIMAT0000062   hsa-let-7a-5p      GJC1        NaN    -0.015379   \n",
       "2      MIMAT0000062   hsa-let-7a-5p      PBX3        NaN     0.007314   \n",
       "3      MIMAT0000062   hsa-let-7a-5p     ITSN1        NaN    -0.077102   \n",
       "4      MIMAT0000062   hsa-let-7a-5p    SRGAP1        NaN     0.162748   \n",
       "...             ...             ...       ...        ...          ...   \n",
       "41164  MIMAT0026483  hsa-miR-370-5p      STUM        NaN     0.198770   \n",
       "41165  MIMAT0026483  hsa-miR-370-5p  CRISPLD2        NaN     0.176897   \n",
       "41166  MIMAT0026483  hsa-miR-370-5p    ADAM19        NaN     0.176555   \n",
       "41167  MIMAT0026557  hsa-miR-412-5p    FAM53C        NaN     0.143814   \n",
       "41168  MIMAT0026557  hsa-miR-412-5p      FICD        NaN     0.260560   \n",
       "\n",
       "         pvalue    qvalue  is_interaction_of_interest  \n",
       "0      0.961289  1.000000                           0  \n",
       "1      0.455214  0.898533                           0  \n",
       "2      0.521332  0.969059                           0  \n",
       "3      0.286103  0.674410                           0  \n",
       "4      0.884626  1.000000                           0  \n",
       "...         ...       ...                         ...  \n",
       "41164  0.929030  1.000000                           0  \n",
       "41165  0.903923  1.000000                           0  \n",
       "41166  0.903485  1.000000                           0  \n",
       "41167  0.854845  1.000000                           0  \n",
       "41168  0.973781  1.000000                           0  \n",
       "\n",
       "[41169 rows x 8 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the inferred interactions associated with this group\n",
    "inferred_interactions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "c0215783",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>accession_id</th>\n",
       "      <th>mirna_name</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>mirtarbase</th>\n",
       "      <th>correlation</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>qvalue</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>GIPC1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.427751</td>\n",
       "      <td>5.038145e-04</td>\n",
       "      <td>0.008435</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>MMP11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.339986</td>\n",
       "      <td>5.178013e-03</td>\n",
       "      <td>0.044181</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MIMAT0000063</td>\n",
       "      <td>hsa-let-7b-5p</td>\n",
       "      <td>GJC1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.598291</td>\n",
       "      <td>5.586415e-07</td>\n",
       "      <td>0.000054</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MIMAT0000063</td>\n",
       "      <td>hsa-let-7b-5p</td>\n",
       "      <td>MEF2C</td>\n",
       "      <td>MIRT052262</td>\n",
       "      <td>-0.340807</td>\n",
       "      <td>5.080635e-03</td>\n",
       "      <td>0.043603</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MIMAT0000063</td>\n",
       "      <td>hsa-let-7b-5p</td>\n",
       "      <td>SOX6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.426589</td>\n",
       "      <td>5.218179e-04</td>\n",
       "      <td>0.008635</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5067</th>\n",
       "      <td>MIMAT0026483</td>\n",
       "      <td>hsa-miR-370-5p</td>\n",
       "      <td>KCNB1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.355776</td>\n",
       "      <td>3.561907e-03</td>\n",
       "      <td>0.033726</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5068</th>\n",
       "      <td>MIMAT0026483</td>\n",
       "      <td>hsa-miR-370-5p</td>\n",
       "      <td>ANTXR2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.337184</td>\n",
       "      <td>5.522999e-03</td>\n",
       "      <td>0.046262</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5069</th>\n",
       "      <td>MIMAT0026483</td>\n",
       "      <td>hsa-miR-370-5p</td>\n",
       "      <td>PIP4P2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.334586</td>\n",
       "      <td>5.860281e-03</td>\n",
       "      <td>0.048118</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5070</th>\n",
       "      <td>MIMAT0026483</td>\n",
       "      <td>hsa-miR-370-5p</td>\n",
       "      <td>ATP1A2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.339986</td>\n",
       "      <td>5.178013e-03</td>\n",
       "      <td>0.044181</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5071</th>\n",
       "      <td>MIMAT0026483</td>\n",
       "      <td>hsa-miR-370-5p</td>\n",
       "      <td>ST6GALNAC3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.344771</td>\n",
       "      <td>4.631989e-03</td>\n",
       "      <td>0.040948</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5072 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      accession_id      mirna_name   gene_name  mirtarbase  correlation  \\\n",
       "0     MIMAT0000062   hsa-let-7a-5p       GIPC1         NaN    -0.427751   \n",
       "1     MIMAT0000062   hsa-let-7a-5p       MMP11         NaN    -0.339986   \n",
       "2     MIMAT0000063   hsa-let-7b-5p        GJC1         NaN    -0.598291   \n",
       "3     MIMAT0000063   hsa-let-7b-5p       MEF2C  MIRT052262    -0.340807   \n",
       "4     MIMAT0000063   hsa-let-7b-5p        SOX6         NaN    -0.426589   \n",
       "...            ...             ...         ...         ...          ...   \n",
       "5067  MIMAT0026483  hsa-miR-370-5p       KCNB1         NaN    -0.355776   \n",
       "5068  MIMAT0026483  hsa-miR-370-5p      ANTXR2         NaN    -0.337184   \n",
       "5069  MIMAT0026483  hsa-miR-370-5p      PIP4P2         NaN    -0.334586   \n",
       "5070  MIMAT0026483  hsa-miR-370-5p      ATP1A2         NaN    -0.339986   \n",
       "5071  MIMAT0026483  hsa-miR-370-5p  ST6GALNAC3         NaN    -0.344771   \n",
       "\n",
       "            pvalue    qvalue  \n",
       "0     5.038145e-04  0.008435  \n",
       "1     5.178013e-03  0.044181  \n",
       "2     5.586415e-07  0.000054  \n",
       "3     5.080635e-03  0.043603  \n",
       "4     5.218179e-04  0.008635  \n",
       "...            ...       ...  \n",
       "5067  3.561907e-03  0.033726  \n",
       "5068  5.522999e-03  0.046262  \n",
       "5069  5.860281e-03  0.048118  \n",
       "5070  5.178013e-03  0.044181  \n",
       "5071  4.631989e-03  0.040948  \n",
       "\n",
       "[5072 rows x 7 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the inferred interactions of interest associated with this group\n",
    "inferred_interactions \\\n",
    "    .query('is_interaction_of_interest == 1') \\\n",
    "    .drop(columns=['is_interaction_of_interest']) \\\n",
    "    .reset_index(drop=True)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "venv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
