{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# **Preprocessing Metadata from TCGA-BRCA Project**\n",
    "TCGA: The Cancer Genome Atlas  \n",
    "BRCA: Breast Invasive Carcinoma"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Importing Libraries and Configurations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "import os\n",
    "import sys\n",
    "\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import requests\n",
    "\n",
    "# Add project root to Python's path\n",
    "sys.path.append(os.path.abspath(os.path.join('..', '..')))\n",
    "\n",
    "from config import (\n",
    "    BRCA_INTERIM_FILE_PATHS,\n",
    "    BRCA_PAPER_FILE_PATH,\n",
    "    BRCA_PREPROCESSING_PARAMETERS,\n",
    "    GDC_API_ENDPOINTS,\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Function"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "def gdc_api_request(endpoint, fields, filters={}, size=1):\n",
    "    \"\"\"\n",
    "    Send a request to a Genomic Data Commons (GDC) API endpoint with the specified parameters.\n",
    "\n",
    "    Parameters:\n",
    "    -----------\n",
    "    endpoint : str\n",
    "        The URL of the endpoint to send the request to.\n",
    "    fields : list of str\n",
    "        A list of fields to include in the response.\n",
    "    filters : dict, optional\n",
    "        A dictionary of filters to apply to the request. Default is an empty dictionary.\n",
    "    size : int, optional\n",
    "        The number of results to retrieve. Default is 1.\n",
    "\n",
    "    Returns:\n",
    "    --------\n",
    "    list\n",
    "        A list of hits (data items) retrieved from the endpoint response.\n",
    "    \"\"\"\n",
    "    # Parameters used in the endpoint request\n",
    "    params = {\n",
    "        'fields': ','.join(fields),\n",
    "        'filters': filters,\n",
    "        'size': str(size)\n",
    "    }\n",
    "    \n",
    "    # Request the objects of interest to the endpoint\n",
    "    response = requests.post(\n",
    "        url=endpoint,\n",
    "        headers={'Content-Type': 'application/json'},\n",
    "        json=params\n",
    "    )\n",
    "    response.raise_for_status()\n",
    "\n",
    "    # Convert the response content to a JSON\n",
    "    json_response = json.loads(response.content.decode('utf-8'))\n",
    "\n",
    "    return json_response['data']['hits']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# TCGA-BRCA Paper Metadata\n",
    "The Cancer Genome Atlas Network. Comprehensive molecular portraits of human breast tumours. Nature 490, 61–70 (2012). https://doi.org/10.1038/nature11412."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Loading and Flagging Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create the DataFrame of 'Supplementary Table 1' from the TCGA-BRCA paper\n",
    "df_paper_cases = pd.read_excel(\n",
    "    BRCA_PAPER_FILE_PATH, sheet_name='SuppTable1', skiprows=1\n",
    ")\n",
    "\n",
    "# Standardize DataFrame column names\n",
    "columns = dict()\n",
    "for name in list(df_paper_cases.columns):\n",
    "    columns[name] = name.replace(' ', '_').lower()\n",
    "df_paper_cases = df_paper_cases.rename(columns=columns)\n",
    "\n",
    "# Flag the cases of interest, i.e., those classified into specific molecular\n",
    "# subtypes of breast cancer by the PAM50 signature\n",
    "molecular_subtypes = BRCA_PREPROCESSING_PARAMETERS['molecular_subtypes']\n",
    "df_paper_cases['is_case_of_interest'] = np.where(\n",
    "    df_paper_cases['pam50_mrna'].isin(molecular_subtypes), 1, 0\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Flagged Paper Cases DataFrame"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Store the DataFrame of flagged cases in a CSV file\n",
    "df_paper_cases.to_csv(BRCA_INTERIM_FILE_PATHS['paper'], index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>complete_tcga_id</th>\n",
       "      <th>gender</th>\n",
       "      <th>age_at_initial_pathologic_diagnosis</th>\n",
       "      <th>er_status</th>\n",
       "      <th>pr_status</th>\n",
       "      <th>her2_final_status</th>\n",
       "      <th>tumor</th>\n",
       "      <th>tumor--t1_coded</th>\n",
       "      <th>node</th>\n",
       "      <th>node-coded</th>\n",
       "      <th>...</th>\n",
       "      <th>sigclust_unsupervised_mrna</th>\n",
       "      <th>sigclust_intrinsic_mrna</th>\n",
       "      <th>mirna_clusters</th>\n",
       "      <th>methylation_clusters</th>\n",
       "      <th>rppa_clusters</th>\n",
       "      <th>cn_clusters</th>\n",
       "      <th>integrated_clusters_(with_pam50)</th>\n",
       "      <th>integrated_clusters_(no_exp)</th>\n",
       "      <th>integrated_clusters_(unsup_exp)</th>\n",
       "      <th>is_case_of_interest</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TCGA-A2-A0T2</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>66.0</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>T3</td>\n",
       "      <td>T_Other</td>\n",
       "      <td>N3</td>\n",
       "      <td>Positive</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-13.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Basal</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TCGA-A2-A04P</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>36.0</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>T2</td>\n",
       "      <td>T_Other</td>\n",
       "      <td>N3</td>\n",
       "      <td>Positive</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-13.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Basal</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TCGA-A1-A0SK</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>54.0</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>T2</td>\n",
       "      <td>T_Other</td>\n",
       "      <td>N0</td>\n",
       "      <td>Negative</td>\n",
       "      <td>...</td>\n",
       "      <td>-6.0</td>\n",
       "      <td>-13.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Basal</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TCGA-A2-A0CM</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>40.0</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>T2</td>\n",
       "      <td>T_Other</td>\n",
       "      <td>N0</td>\n",
       "      <td>Negative</td>\n",
       "      <td>...</td>\n",
       "      <td>-12.0</td>\n",
       "      <td>-13.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>Basal</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TCGA-AR-A1AR</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>50.0</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>T1</td>\n",
       "      <td>T1</td>\n",
       "      <td>N2</td>\n",
       "      <td>Positive</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-13.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>820</th>\n",
       "      <td>TCGA-AC-A2FF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>821</th>\n",
       "      <td>TCGA-AC-A2FB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>822</th>\n",
       "      <td>TCGA-AC-A2FG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>823</th>\n",
       "      <td>TCGA-GI-A2C8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>824</th>\n",
       "      <td>TCGA-E9-A295</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>825 rows × 31 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    complete_tcga_id  gender  age_at_initial_pathologic_diagnosis er_status  \\\n",
       "0       TCGA-A2-A0T2  FEMALE                                 66.0  Negative   \n",
       "1       TCGA-A2-A04P  FEMALE                                 36.0  Negative   \n",
       "2       TCGA-A1-A0SK  FEMALE                                 54.0  Negative   \n",
       "3       TCGA-A2-A0CM  FEMALE                                 40.0  Negative   \n",
       "4       TCGA-AR-A1AR  FEMALE                                 50.0  Negative   \n",
       "..               ...     ...                                  ...       ...   \n",
       "820     TCGA-AC-A2FF     NaN                                  NaN       NaN   \n",
       "821     TCGA-AC-A2FB     NaN                                  NaN       NaN   \n",
       "822     TCGA-AC-A2FG     NaN                                  NaN       NaN   \n",
       "823     TCGA-GI-A2C8     NaN                                  NaN       NaN   \n",
       "824     TCGA-E9-A295     NaN                                  NaN       NaN   \n",
       "\n",
       "    pr_status her2_final_status tumor tumor--t1_coded node node-coded  ...  \\\n",
       "0    Negative          Negative    T3         T_Other   N3   Positive  ...   \n",
       "1    Negative          Negative    T2         T_Other   N3   Positive  ...   \n",
       "2    Negative          Negative    T2         T_Other   N0   Negative  ...   \n",
       "3    Negative          Negative    T2         T_Other   N0   Negative  ...   \n",
       "4    Negative          Negative    T1              T1   N2   Positive  ...   \n",
       "..        ...               ...   ...             ...  ...        ...  ...   \n",
       "820       NaN               NaN   NaN             NaN  NaN        NaN  ...   \n",
       "821       NaN               NaN   NaN             NaN  NaN        NaN  ...   \n",
       "822       NaN               NaN   NaN             NaN  NaN        NaN  ...   \n",
       "823       NaN               NaN   NaN             NaN  NaN        NaN  ...   \n",
       "824       NaN               NaN   NaN             NaN  NaN        NaN  ...   \n",
       "\n",
       "    sigclust_unsupervised_mrna sigclust_intrinsic_mrna mirna_clusters  \\\n",
       "0                          0.0                   -13.0            3.0   \n",
       "1                          0.0                   -13.0            5.0   \n",
       "2                         -6.0                   -13.0            5.0   \n",
       "3                        -12.0                   -13.0            4.0   \n",
       "4                          0.0                   -13.0            5.0   \n",
       "..                         ...                     ...            ...   \n",
       "820                        NaN                     NaN            NaN   \n",
       "821                        NaN                     NaN            NaN   \n",
       "822                        NaN                     NaN            NaN   \n",
       "823                        NaN                     NaN            NaN   \n",
       "824                        NaN                     NaN            NaN   \n",
       "\n",
       "    methylation_clusters rppa_clusters cn_clusters  \\\n",
       "0                    5.0         Basal         3.0   \n",
       "1                    5.0         Basal         1.0   \n",
       "2                    5.0         Basal         1.0   \n",
       "3                    4.0         Basal         4.0   \n",
       "4                    5.0           NaN         1.0   \n",
       "..                   ...           ...         ...   \n",
       "820                  1.0           NaN         1.0   \n",
       "821                  1.0           NaN         3.0   \n",
       "822                  1.0           NaN         2.0   \n",
       "823                  3.0           NaN         3.0   \n",
       "824                  4.0           NaN         3.0   \n",
       "\n",
       "    integrated_clusters_(with_pam50)  integrated_clusters_(no_exp)  \\\n",
       "0                                2.0                           2.0   \n",
       "1                                2.0                           2.0   \n",
       "2                                2.0                           2.0   \n",
       "3                                2.0                           1.0   \n",
       "4                                NaN                           NaN   \n",
       "..                               ...                           ...   \n",
       "820                              NaN                           NaN   \n",
       "821                              NaN                           NaN   \n",
       "822                              NaN                           NaN   \n",
       "823                              NaN                           NaN   \n",
       "824                              NaN                           NaN   \n",
       "\n",
       "     integrated_clusters_(unsup_exp)  is_case_of_interest  \n",
       "0                                2.0                    1  \n",
       "1                                2.0                    1  \n",
       "2                                2.0                    1  \n",
       "3                                1.0                    1  \n",
       "4                                NaN                    1  \n",
       "..                               ...                  ...  \n",
       "820                              NaN                    0  \n",
       "821                              NaN                    0  \n",
       "822                              NaN                    0  \n",
       "823                              NaN                    0  \n",
       "824                              NaN                    0  \n",
       "\n",
       "[825 rows x 31 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the DataFrame of flagged cases from the TCGA-BRCA paper\n",
    "df_paper_cases"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# TCGA-BRCA Metadata from GDC API\n",
    "GDC: Genomic Data Commons"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Status"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{\n",
      "    \"commit\": \"4bb408881e6dc67eca93ff9fd913629a8f2d11c2\",\n",
      "    \"data_release\": \"Data Release 42.0 - January 30, 2025\",\n",
      "    \"data_release_version\": {\n",
      "        \"major\": 42,\n",
      "        \"minor\": 0,\n",
      "        \"release_date\": \"2025-01-30\"\n",
      "    },\n",
      "    \"status\": \"OK\",\n",
      "    \"tag\": \"7.8.5\",\n",
      "    \"version\": 1\n",
      "}\n"
     ]
    }
   ],
   "source": [
    "# Request the API version and status to the endpoint\n",
    "response = requests.get(GDC_API_ENDPOINTS['status'])\n",
    "\n",
    "# Print the response content\n",
    "try:\n",
    "    parsed = json.loads(response.content.decode('utf-8'))\n",
    "    print(json.dumps(parsed, indent=4))\n",
    "except json.JSONDecodeError:\n",
    "    print('Response is not valid JSON:')\n",
    "    print(response.content.decode('utf-8'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Projects"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Endpoint Response"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Fields available in the endpoint\n",
    "fields = [\n",
    "    'disease_type',\n",
    "    'name',\n",
    "    'primary_site',\n",
    "    'summary.case_count',\n",
    "    'summary.data_categories.case_count',\n",
    "    'summary.data_categories.data_category',\n",
    "    'summary.data_categories.file_count',\n",
    "    'summary.experimental_strategies.case_count',\n",
    "    'summary.experimental_strategies.experimental_strategy',\n",
    "    'summary.experimental_strategies.file_count',\n",
    "    'summary.file_count',\n",
    "]\n",
    "\n",
    "# Filter to be used in the projects endpoint request\n",
    "filters = {\n",
    "    'op': 'in',\n",
    "    'content': {\n",
    "        'field': 'projects.project_id',\n",
    "        'value': BRCA_PREPROCESSING_PARAMETERS['project_ids']\n",
    "    }\n",
    "}\n",
    "\n",
    "# Get the response from the projects endpoint request\n",
    "projects_response = gdc_api_request(\n",
    "    endpoint=GDC_API_ENDPOINTS['projects'],\n",
    "    fields=fields,\n",
    "    filters=filters,\n",
    "    size=len(BRCA_PREPROCESSING_PARAMETERS['project_ids'])\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Project DataFrame"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create the TCGA-BRCA project metadata DataFrame\n",
    "df_project = pd.json_normalize(projects_response) \\\n",
    "    .rename(columns={\n",
    "        'id': 'project_id',\n",
    "        'name': 'project_name',\n",
    "        'summary.case_count': 'case_count',\n",
    "        'summary.data_categories': 'data_categories',\n",
    "        'summary.experimental_strategies': 'experimental_strategies',\n",
    "        'summary.file_count': 'file_count',\n",
    "    }) \\\n",
    "    [[\n",
    "        'project_id',\n",
    "        'project_name',\n",
    "        'primary_site',\n",
    "        'disease_type',\n",
    "        'experimental_strategies',\n",
    "        'data_categories',\n",
    "        'case_count',\n",
    "        'file_count',\n",
    "    ]]\n",
    "\n",
    "# Store the project metadata DataFrame in a CSV file\n",
    "df_project.to_csv(BRCA_INTERIM_FILE_PATHS['project'], index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>project_id</th>\n",
       "      <th>project_name</th>\n",
       "      <th>primary_site</th>\n",
       "      <th>disease_type</th>\n",
       "      <th>experimental_strategies</th>\n",
       "      <th>data_categories</th>\n",
       "      <th>case_count</th>\n",
       "      <th>file_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TCGA-BRCA</td>\n",
       "      <td>Breast Invasive Carcinoma</td>\n",
       "      <td>[Breast]</td>\n",
       "      <td>[Epithelial Neoplasms, NOS, Adnexal and Skin Appendage Neoplasms, Squamous Cell Neoplasms, Adenomas and Adenocarcinomas, Complex Epithelial Neoplasms, Fibroepithelial Neoplasms, Cystic, Mucinous and Serous Neoplasms, Basal Cell Neoplasms, Ductal and Lobular Neoplasms]</td>\n",
       "      <td>[{'file_count': 11079, 'case_count': 1095, 'experimental_strategy': 'RNA-Seq'}, {'file_count': 17049, 'case_count': 1072, 'experimental_strategy': 'WXS'}, {'file_count': 3621, 'case_count': 1079, 'experimental_strategy': 'miRNA-Seq'}, {'file_count': 10572, 'case_count': 952, 'experimental_strategy': 'WGS'}, {'file_count': 75, 'case_count': 74, 'experimental_strategy': 'ATAC-Seq'}, {'file_count': 14329, 'case_count': 1098, 'experimental_strategy': 'Genotyping Array'}, {'file_count': 3714, 'case_count': 1097, 'experimental_strategy': 'Methylation Array'}, {'file_count': 919, 'case_count': 881, 'experimental_strategy': 'Reverse Phase Protein Array'}, {'file_count': 1133, 'case_count': 1062, 'experimental_strategy': 'Diagnostic Slide'}, {'file_count': 1978, 'case_count': 1093, 'experimental_strategy': 'Tissue Slide'}]</td>\n",
       "      <td>[{'file_count': 19753, 'case_count': 1098, 'data_category': 'Simple Nucleotide Variation'}, {'file_count': 9282, 'case_count': 1098, 'data_category': 'Sequencing Reads'}, {'file_count': 5316, 'case_count': 1098, 'data_category': 'Biospecimen'}, {'file_count': 2288, 'case_count': 1098, 'data_category': 'Clinical'}, {'file_count': 14346, 'case_count': 1098, 'data_category': 'Copy Number Variation'}, {'file_count': 4876, 'case_count': 1097, 'data_category': 'Transcriptome Profiling'}, {'file_count': 3714, 'case_count': 1097, 'data_category': 'DNA Methylation'}, {'file_count': 919, 'case_count': 881, 'data_category': 'Proteome Profiling'}, {'file_count': 2696, 'case_count': 784, 'data_category': 'Somatic Structural Variation'}, {'file_count': 5772, 'case_count': 1098, 'data_category': 'Structural Variation'}]</td>\n",
       "      <td>1098</td>\n",
       "      <td>68962</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  project_id               project_name primary_site  \\\n",
       "0  TCGA-BRCA  Breast Invasive Carcinoma     [Breast]   \n",
       "\n",
       "                                                                                                                                                                                                                                                                   disease_type  \\\n",
       "0  [Epithelial Neoplasms, NOS, Adnexal and Skin Appendage Neoplasms, Squamous Cell Neoplasms, Adenomas and Adenocarcinomas, Complex Epithelial Neoplasms, Fibroepithelial Neoplasms, Cystic, Mucinous and Serous Neoplasms, Basal Cell Neoplasms, Ductal and Lobular Neoplasms]   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     experimental_strategies  \\\n",
       "0  [{'file_count': 11079, 'case_count': 1095, 'experimental_strategy': 'RNA-Seq'}, {'file_count': 17049, 'case_count': 1072, 'experimental_strategy': 'WXS'}, {'file_count': 3621, 'case_count': 1079, 'experimental_strategy': 'miRNA-Seq'}, {'file_count': 10572, 'case_count': 952, 'experimental_strategy': 'WGS'}, {'file_count': 75, 'case_count': 74, 'experimental_strategy': 'ATAC-Seq'}, {'file_count': 14329, 'case_count': 1098, 'experimental_strategy': 'Genotyping Array'}, {'file_count': 3714, 'case_count': 1097, 'experimental_strategy': 'Methylation Array'}, {'file_count': 919, 'case_count': 881, 'experimental_strategy': 'Reverse Phase Protein Array'}, {'file_count': 1133, 'case_count': 1062, 'experimental_strategy': 'Diagnostic Slide'}, {'file_count': 1978, 'case_count': 1093, 'experimental_strategy': 'Tissue Slide'}]   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    data_categories  \\\n",
       "0  [{'file_count': 19753, 'case_count': 1098, 'data_category': 'Simple Nucleotide Variation'}, {'file_count': 9282, 'case_count': 1098, 'data_category': 'Sequencing Reads'}, {'file_count': 5316, 'case_count': 1098, 'data_category': 'Biospecimen'}, {'file_count': 2288, 'case_count': 1098, 'data_category': 'Clinical'}, {'file_count': 14346, 'case_count': 1098, 'data_category': 'Copy Number Variation'}, {'file_count': 4876, 'case_count': 1097, 'data_category': 'Transcriptome Profiling'}, {'file_count': 3714, 'case_count': 1097, 'data_category': 'DNA Methylation'}, {'file_count': 919, 'case_count': 881, 'data_category': 'Proteome Profiling'}, {'file_count': 2696, 'case_count': 784, 'data_category': 'Somatic Structural Variation'}, {'file_count': 5772, 'case_count': 1098, 'data_category': 'Structural Variation'}]   \n",
       "\n",
       "   case_count  file_count  \n",
       "0        1098       68962  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the TCGA-BRCA project metadata DataFrame\n",
    "pd.set_option('display.max_colwidth', 900)\n",
    "df_project"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Cases"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Endpoint Response"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Fields of interest for the cases endpoint request\n",
    "fields = ['disease_type', 'submitter_id',]\n",
    "\n",
    "# Filter to be used in the cases endpoint request\n",
    "filters = {\n",
    "    'op': 'in',\n",
    "    'content': {\n",
    "        'field': 'project.project_id',\n",
    "        'value': BRCA_PREPROCESSING_PARAMETERS['project_ids']\n",
    "    }\n",
    "}\n",
    "\n",
    "# Get the response from the cases endpoint request\n",
    "cases_response = gdc_api_request(\n",
    "    endpoint=GDC_API_ENDPOINTS['cases'],\n",
    "    fields=fields,\n",
    "    filters=filters,\n",
    "    size=sum(df_project['case_count'])\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Cases DataFrame"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create the TCGA-BRCA cases metadata DataFrame\n",
    "df_api_cases = pd.json_normalize(cases_response) \\\n",
    "    .rename(columns={'id': 'case_id'}) \\\n",
    "    [['case_id', 'submitter_id', 'disease_type']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>case_id</th>\n",
       "      <th>submitter_id</th>\n",
       "      <th>disease_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2d29a4ac-98e7-4663-9dd6-5681bc32ac2e</td>\n",
       "      <td>TCGA-A2-A3KC</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>de531604-bd1a-49df-96c2-6de3ae703f1d</td>\n",
       "      <td>TCGA-D8-A1XG</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>de646f6a-6d50-4be2-b3cd-f17554b234df</td>\n",
       "      <td>TCGA-AR-A24L</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>de75d0b9-0f47-4732-8df5-05c350cfcd32</td>\n",
       "      <td>TCGA-AN-A0FY</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>deba32e4-0e68-4711-941b-3b63bd965afb</td>\n",
       "      <td>TCGA-A2-A3XY</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1093</th>\n",
       "      <td>35bd694d-1dd2-466f-ab27-03320614b40e</td>\n",
       "      <td>TCGA-OL-A66I</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1094</th>\n",
       "      <td>2c90d5e3-ac8c-4c86-a06f-42ccc064eb49</td>\n",
       "      <td>TCGA-D8-A1XZ</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1095</th>\n",
       "      <td>360d1d6d-cf3e-4c78-b36f-ef5e3160aa6b</td>\n",
       "      <td>TCGA-BH-A0AU</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1096</th>\n",
       "      <td>2cf68894-168b-458b-af4f-53cad72989a8</td>\n",
       "      <td>TCGA-A7-A13F</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1097</th>\n",
       "      <td>36d77496-99f8-4911-ba7c-89fae32662aa</td>\n",
       "      <td>TCGA-A2-A0T5</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1098 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   case_id  submitter_id  \\\n",
       "0     2d29a4ac-98e7-4663-9dd6-5681bc32ac2e  TCGA-A2-A3KC   \n",
       "1     de531604-bd1a-49df-96c2-6de3ae703f1d  TCGA-D8-A1XG   \n",
       "2     de646f6a-6d50-4be2-b3cd-f17554b234df  TCGA-AR-A24L   \n",
       "3     de75d0b9-0f47-4732-8df5-05c350cfcd32  TCGA-AN-A0FY   \n",
       "4     deba32e4-0e68-4711-941b-3b63bd965afb  TCGA-A2-A3XY   \n",
       "...                                    ...           ...   \n",
       "1093  35bd694d-1dd2-466f-ab27-03320614b40e  TCGA-OL-A66I   \n",
       "1094  2c90d5e3-ac8c-4c86-a06f-42ccc064eb49  TCGA-D8-A1XZ   \n",
       "1095  360d1d6d-cf3e-4c78-b36f-ef5e3160aa6b  TCGA-BH-A0AU   \n",
       "1096  2cf68894-168b-458b-af4f-53cad72989a8  TCGA-A7-A13F   \n",
       "1097  36d77496-99f8-4911-ba7c-89fae32662aa  TCGA-A2-A0T5   \n",
       "\n",
       "                      disease_type  \n",
       "0     Ductal and Lobular Neoplasms  \n",
       "1     Ductal and Lobular Neoplasms  \n",
       "2     Ductal and Lobular Neoplasms  \n",
       "3     Ductal and Lobular Neoplasms  \n",
       "4     Ductal and Lobular Neoplasms  \n",
       "...                            ...  \n",
       "1093  Ductal and Lobular Neoplasms  \n",
       "1094  Ductal and Lobular Neoplasms  \n",
       "1095  Ductal and Lobular Neoplasms  \n",
       "1096  Ductal and Lobular Neoplasms  \n",
       "1097  Ductal and Lobular Neoplasms  \n",
       "\n",
       "[1098 rows x 3 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the TCGA-BRCA cases metadata DataFrame\n",
    "df_api_cases"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Files"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Endpoint Response"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Fields of interest for the files endpoint request\n",
    "fields = [\n",
    "    'cases.case_id',\n",
    "    'cases.samples.sample_type',\n",
    "    'cases.samples.tissue_type',\n",
    "    'data_format',\n",
    "    'data_type',\n",
    "    'experimental_strategy',\n",
    "]\n",
    "\n",
    "# Filter to be used in the files endpoint request\n",
    "filters = {\n",
    "    'op': 'in',\n",
    "    'content': {\n",
    "        'field': 'cases.project.project_id',\n",
    "        'value': BRCA_PREPROCESSING_PARAMETERS['project_ids']\n",
    "    }\n",
    "}\n",
    "\n",
    "# Get the response from the files endpoint request\n",
    "files_response = gdc_api_request(\n",
    "    endpoint=GDC_API_ENDPOINTS['files'],\n",
    "    fields=fields,\n",
    "    filters=filters,\n",
    "    size=sum(df_project['file_count'])\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Files DataFrame"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create the TCGA-BRCA files metadata DataFrame\n",
    "df_files = pd.json_normalize(files_response) \\\n",
    "    .rename(columns={'id': 'file_id'})\n",
    "\n",
    "# Explode lists with dictionaries containing cases metadata\n",
    "df_files = df_files.explode('cases')\n",
    "\n",
    "# Explode dictionary contents with cases metadata into columns\n",
    "df_files = pd.concat(\n",
    "    objs=[\n",
    "        df_files.reset_index(drop=True),\n",
    "        pd.json_normalize(df_files['cases'])\n",
    "    ],\n",
    "    axis='columns'\n",
    ")\n",
    "\n",
    "# Reorganize the DataFrame columns\n",
    "df_files = df_files \\\n",
    "    .drop(columns=['cases']) \\\n",
    "    [[\n",
    "        'file_id',\n",
    "        'case_id',\n",
    "        'experimental_strategy',\n",
    "        'data_type',\n",
    "        'data_format',\n",
    "        'samples',\n",
    "    ]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>file_id</th>\n",
       "      <th>case_id</th>\n",
       "      <th>experimental_strategy</th>\n",
       "      <th>data_type</th>\n",
       "      <th>data_format</th>\n",
       "      <th>samples</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>27016c1c-f696-4a54-8c4a-6f9ba8a1c9e3</td>\n",
       "      <td>3b1f7e48-216d-47de-ba79-db350f52bff0</td>\n",
       "      <td>WXS</td>\n",
       "      <td>Annotated Somatic Mutation</td>\n",
       "      <td>VCF</td>\n",
       "      <td>[{'sample_type': 'Blood Derived Normal', 'tissue_type': 'Normal'}, {'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>776f51f8-6f23-42d5-a96b-d7c934dcaa12</td>\n",
       "      <td>3b1f7e48-216d-47de-ba79-db350f52bff0</td>\n",
       "      <td>WXS</td>\n",
       "      <td>Aligned Reads</td>\n",
       "      <td>BAM</td>\n",
       "      <td>[{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>b1680590-4e97-4a3d-b41b-d9e84290de17</td>\n",
       "      <td>3b1f7e48-216d-47de-ba79-db350f52bff0</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>Aligned Reads</td>\n",
       "      <td>BAM</td>\n",
       "      <td>[{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>65cab127-616b-45c7-98b4-fdf8e01881fb</td>\n",
       "      <td>3b1f7e48-216d-47de-ba79-db350f52bff0</td>\n",
       "      <td>WGS</td>\n",
       "      <td>Aligned Reads</td>\n",
       "      <td>BAM</td>\n",
       "      <td>[{'sample_type': 'Blood Derived Normal', 'tissue_type': 'Normal'}]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>a0409565-1baf-46f4-bae9-85a1dbe34160</td>\n",
       "      <td>3b1f7e48-216d-47de-ba79-db350f52bff0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Biospecimen Supplement</td>\n",
       "      <td>BCR XML</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89800</th>\n",
       "      <td>cc6ca81e-5bb7-4c87-9602-ec5919b443a8</td>\n",
       "      <td>00807dae-9f4a-4fd1-aac2-82eb11bf2afb</td>\n",
       "      <td>WGS</td>\n",
       "      <td>Aligned Reads</td>\n",
       "      <td>BAM</td>\n",
       "      <td>[{'sample_type': 'Blood Derived Normal', 'tissue_type': 'Normal'}]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89801</th>\n",
       "      <td>82e26986-3016-4302-8929-e8a10078bd1a</td>\n",
       "      <td>0130d616-885e-4a6c-9d03-2f17dd692a05</td>\n",
       "      <td>WGS</td>\n",
       "      <td>Structural Rearrangement</td>\n",
       "      <td>VCF</td>\n",
       "      <td>[{'sample_type': 'Blood Derived Normal', 'tissue_type': 'Normal'}, {'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89802</th>\n",
       "      <td>6f4842e1-5b24-45ec-b03e-b81b63050111</td>\n",
       "      <td>0130d616-885e-4a6c-9d03-2f17dd692a05</td>\n",
       "      <td>Methylation Array</td>\n",
       "      <td>Masked Intensities</td>\n",
       "      <td>IDAT</td>\n",
       "      <td>[{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89803</th>\n",
       "      <td>c88d52bf-0d3e-4e13-87c6-c33b29417bc6</td>\n",
       "      <td>00a2d166-78c9-4687-a195-3d6315c27574</td>\n",
       "      <td>WGS</td>\n",
       "      <td>Annotated Somatic Mutation</td>\n",
       "      <td>VCF</td>\n",
       "      <td>[{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}, {'sample_type': 'Blood Derived Normal', 'tissue_type': 'Normal'}]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89804</th>\n",
       "      <td>f0107961-a5e1-46ec-b232-a47f0049bc5f</td>\n",
       "      <td>125b0a9a-6cbe-4207-9178-e7ff03487e87</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>Gene Expression Quantification</td>\n",
       "      <td>TSV</td>\n",
       "      <td>[{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>89805 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                    file_id  \\\n",
       "0      27016c1c-f696-4a54-8c4a-6f9ba8a1c9e3   \n",
       "1      776f51f8-6f23-42d5-a96b-d7c934dcaa12   \n",
       "2      b1680590-4e97-4a3d-b41b-d9e84290de17   \n",
       "3      65cab127-616b-45c7-98b4-fdf8e01881fb   \n",
       "4      a0409565-1baf-46f4-bae9-85a1dbe34160   \n",
       "...                                     ...   \n",
       "89800  cc6ca81e-5bb7-4c87-9602-ec5919b443a8   \n",
       "89801  82e26986-3016-4302-8929-e8a10078bd1a   \n",
       "89802  6f4842e1-5b24-45ec-b03e-b81b63050111   \n",
       "89803  c88d52bf-0d3e-4e13-87c6-c33b29417bc6   \n",
       "89804  f0107961-a5e1-46ec-b232-a47f0049bc5f   \n",
       "\n",
       "                                    case_id experimental_strategy  \\\n",
       "0      3b1f7e48-216d-47de-ba79-db350f52bff0                   WXS   \n",
       "1      3b1f7e48-216d-47de-ba79-db350f52bff0                   WXS   \n",
       "2      3b1f7e48-216d-47de-ba79-db350f52bff0               RNA-Seq   \n",
       "3      3b1f7e48-216d-47de-ba79-db350f52bff0                   WGS   \n",
       "4      3b1f7e48-216d-47de-ba79-db350f52bff0                   NaN   \n",
       "...                                     ...                   ...   \n",
       "89800  00807dae-9f4a-4fd1-aac2-82eb11bf2afb                   WGS   \n",
       "89801  0130d616-885e-4a6c-9d03-2f17dd692a05                   WGS   \n",
       "89802  0130d616-885e-4a6c-9d03-2f17dd692a05     Methylation Array   \n",
       "89803  00a2d166-78c9-4687-a195-3d6315c27574                   WGS   \n",
       "89804  125b0a9a-6cbe-4207-9178-e7ff03487e87               RNA-Seq   \n",
       "\n",
       "                            data_type data_format  \\\n",
       "0          Annotated Somatic Mutation         VCF   \n",
       "1                       Aligned Reads         BAM   \n",
       "2                       Aligned Reads         BAM   \n",
       "3                       Aligned Reads         BAM   \n",
       "4              Biospecimen Supplement     BCR XML   \n",
       "...                               ...         ...   \n",
       "89800                   Aligned Reads         BAM   \n",
       "89801        Structural Rearrangement         VCF   \n",
       "89802              Masked Intensities        IDAT   \n",
       "89803      Annotated Somatic Mutation         VCF   \n",
       "89804  Gene Expression Quantification         TSV   \n",
       "\n",
       "                                                                                                                            samples  \n",
       "0      [{'sample_type': 'Blood Derived Normal', 'tissue_type': 'Normal'}, {'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]  \n",
       "1                                                                        [{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]  \n",
       "2                                                                        [{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]  \n",
       "3                                                                [{'sample_type': 'Blood Derived Normal', 'tissue_type': 'Normal'}]  \n",
       "4                                                                                                                               NaN  \n",
       "...                                                                                                                             ...  \n",
       "89800                                                            [{'sample_type': 'Blood Derived Normal', 'tissue_type': 'Normal'}]  \n",
       "89801  [{'sample_type': 'Blood Derived Normal', 'tissue_type': 'Normal'}, {'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]  \n",
       "89802                                                                    [{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]  \n",
       "89803  [{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}, {'sample_type': 'Blood Derived Normal', 'tissue_type': 'Normal'}]  \n",
       "89804                                                                    [{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]  \n",
       "\n",
       "[89805 rows x 6 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the TCGA-BRCA files metadata DataFrame\n",
    "pd.set_option('display.max_colwidth', 300)\n",
    "df_files"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Cases and Files of Interest"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Potential Cases of Interest"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Filter the cases of interest from the paper\n",
    "df_paper_cases_of_interest = df_paper_cases \\\n",
    "    .query('is_case_of_interest == 1') \\\n",
    "    [['complete_tcga_id', 'pam50_mrna']]\n",
    "\n",
    "# Retrieve the molecular subtype classification of the cases\n",
    "df_cases = df_api_cases \\\n",
    "    .merge(\n",
    "        right=df_paper_cases_of_interest,\n",
    "        left_on='submitter_id',\n",
    "        right_on='complete_tcga_id',\n",
    "        how='left'\n",
    "    ) \\\n",
    "    [['case_id', 'submitter_id', 'disease_type', 'pam50_mrna']]\n",
    "    \n",
    "# Flag the cases of interest, i.e., those classified into a molecular\n",
    "# subtype of breast cancer and a specific type of disease\n",
    "disease_types = BRCA_PREPROCESSING_PARAMETERS['disease_types']\n",
    "df_cases['is_case_of_interest'] = np.where(\n",
    "    ((df_cases['disease_type'].isin(disease_types)) \n",
    "     & (df_cases['pam50_mrna'].notna())), 1, 0\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>case_id</th>\n",
       "      <th>submitter_id</th>\n",
       "      <th>disease_type</th>\n",
       "      <th>pam50_mrna</th>\n",
       "      <th>is_case_of_interest</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2d29a4ac-98e7-4663-9dd6-5681bc32ac2e</td>\n",
       "      <td>TCGA-A2-A3KC</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>de531604-bd1a-49df-96c2-6de3ae703f1d</td>\n",
       "      <td>TCGA-D8-A1XG</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>de646f6a-6d50-4be2-b3cd-f17554b234df</td>\n",
       "      <td>TCGA-AR-A24L</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>de75d0b9-0f47-4732-8df5-05c350cfcd32</td>\n",
       "      <td>TCGA-AN-A0FY</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal B</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>deba32e4-0e68-4711-941b-3b63bd965afb</td>\n",
       "      <td>TCGA-A2-A3XY</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1093</th>\n",
       "      <td>35bd694d-1dd2-466f-ab27-03320614b40e</td>\n",
       "      <td>TCGA-OL-A66I</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1094</th>\n",
       "      <td>2c90d5e3-ac8c-4c86-a06f-42ccc064eb49</td>\n",
       "      <td>TCGA-D8-A1XZ</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1095</th>\n",
       "      <td>360d1d6d-cf3e-4c78-b36f-ef5e3160aa6b</td>\n",
       "      <td>TCGA-BH-A0AU</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal B</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1096</th>\n",
       "      <td>2cf68894-168b-458b-af4f-53cad72989a8</td>\n",
       "      <td>TCGA-A7-A13F</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal B</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1097</th>\n",
       "      <td>36d77496-99f8-4911-ba7c-89fae32662aa</td>\n",
       "      <td>TCGA-A2-A0T5</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1098 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   case_id  submitter_id  \\\n",
       "0     2d29a4ac-98e7-4663-9dd6-5681bc32ac2e  TCGA-A2-A3KC   \n",
       "1     de531604-bd1a-49df-96c2-6de3ae703f1d  TCGA-D8-A1XG   \n",
       "2     de646f6a-6d50-4be2-b3cd-f17554b234df  TCGA-AR-A24L   \n",
       "3     de75d0b9-0f47-4732-8df5-05c350cfcd32  TCGA-AN-A0FY   \n",
       "4     deba32e4-0e68-4711-941b-3b63bd965afb  TCGA-A2-A3XY   \n",
       "...                                    ...           ...   \n",
       "1093  35bd694d-1dd2-466f-ab27-03320614b40e  TCGA-OL-A66I   \n",
       "1094  2c90d5e3-ac8c-4c86-a06f-42ccc064eb49  TCGA-D8-A1XZ   \n",
       "1095  360d1d6d-cf3e-4c78-b36f-ef5e3160aa6b  TCGA-BH-A0AU   \n",
       "1096  2cf68894-168b-458b-af4f-53cad72989a8  TCGA-A7-A13F   \n",
       "1097  36d77496-99f8-4911-ba7c-89fae32662aa  TCGA-A2-A0T5   \n",
       "\n",
       "                      disease_type pam50_mrna  is_case_of_interest  \n",
       "0     Ductal and Lobular Neoplasms        NaN                    0  \n",
       "1     Ductal and Lobular Neoplasms        NaN                    0  \n",
       "2     Ductal and Lobular Neoplasms        NaN                    0  \n",
       "3     Ductal and Lobular Neoplasms  Luminal B                    1  \n",
       "4     Ductal and Lobular Neoplasms        NaN                    0  \n",
       "...                            ...        ...                  ...  \n",
       "1093  Ductal and Lobular Neoplasms        NaN                    0  \n",
       "1094  Ductal and Lobular Neoplasms        NaN                    0  \n",
       "1095  Ductal and Lobular Neoplasms  Luminal B                    1  \n",
       "1096  Ductal and Lobular Neoplasms  Luminal B                    1  \n",
       "1097  Ductal and Lobular Neoplasms  Luminal A                    1  \n",
       "\n",
       "[1098 rows x 5 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the DataFrame of flagged cases\n",
    "df_cases"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Potential Files of Interest"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Retrieve the flag of interest of the cases\n",
    "df_files = df_files \\\n",
    "    .merge(\n",
    "        right=df_cases,\n",
    "        left_on='case_id',\n",
    "        right_on='case_id',\n",
    "        how='inner'\n",
    "    ) \\\n",
    "    .drop(columns=['submitter_id', 'disease_type', 'pam50_mrna'])\n",
    "\n",
    "# Filter the files based on data type and case\n",
    "data_types = BRCA_PREPROCESSING_PARAMETERS['data_types']\n",
    "df_files_of_interest = df_files \\\n",
    "    .query('(data_type == @data_types) and (is_case_of_interest == 1)')\n",
    "\n",
    "# Explode the lists with dictionaries containing samples metadata\n",
    "df_files_of_interest = df_files_of_interest.explode('samples')\n",
    "df_files_of_interest = pd.concat(\n",
    "    objs=[\n",
    "        df_files_of_interest.reset_index(drop=True),\n",
    "        pd.json_normalize(df_files_of_interest['samples'])\n",
    "    ],\n",
    "    axis='columns'\n",
    ")\n",
    "\n",
    "# Rearrange the DataFrame columns\n",
    "df_files_of_interest = df_files_of_interest \\\n",
    "    .drop(columns=[\n",
    "        'experimental_strategy',\n",
    "        'data_type',\n",
    "        'data_format',\n",
    "        'samples',\n",
    "        'is_case_of_interest',\n",
    "    ])\n",
    "    \n",
    "# Flag the files of interest, i.e., those related to a case\n",
    "# of interest, specific data type, and specific sample type\n",
    "sample_types = BRCA_PREPROCESSING_PARAMETERS['sample_types']\n",
    "df_files_of_interest['is_file_of_interest'] = np.where(\n",
    "    df_files_of_interest['sample_type'].isin(sample_types), 1, 0\n",
    ")\n",
    "\n",
    "# Extend the flag column to all files\n",
    "df_files = df_files \\\n",
    "    .merge(\n",
    "        right=df_files_of_interest,\n",
    "        left_on=['file_id', 'case_id'],\n",
    "        right_on=['file_id', 'case_id'],\n",
    "        how='left'\n",
    "    ) \\\n",
    "    .drop(columns=['is_case_of_interest']) \\\n",
    "    .fillna(value={'is_file_of_interest': 0}) \\\n",
    "    .astype({'is_file_of_interest': 'int'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>file_id</th>\n",
       "      <th>case_id</th>\n",
       "      <th>experimental_strategy</th>\n",
       "      <th>data_type</th>\n",
       "      <th>data_format</th>\n",
       "      <th>samples</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>tissue_type</th>\n",
       "      <th>is_file_of_interest</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>27016c1c-f696-4a54-8c4a-6f9ba8a1c9e3</td>\n",
       "      <td>3b1f7e48-216d-47de-ba79-db350f52bff0</td>\n",
       "      <td>WXS</td>\n",
       "      <td>Annotated Somatic Mutation</td>\n",
       "      <td>VCF</td>\n",
       "      <td>[{'sample_type': 'Blood Derived Normal', 'tissue_type': 'Normal'}, {'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>776f51f8-6f23-42d5-a96b-d7c934dcaa12</td>\n",
       "      <td>3b1f7e48-216d-47de-ba79-db350f52bff0</td>\n",
       "      <td>WXS</td>\n",
       "      <td>Aligned Reads</td>\n",
       "      <td>BAM</td>\n",
       "      <td>[{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>b1680590-4e97-4a3d-b41b-d9e84290de17</td>\n",
       "      <td>3b1f7e48-216d-47de-ba79-db350f52bff0</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>Aligned Reads</td>\n",
       "      <td>BAM</td>\n",
       "      <td>[{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>65cab127-616b-45c7-98b4-fdf8e01881fb</td>\n",
       "      <td>3b1f7e48-216d-47de-ba79-db350f52bff0</td>\n",
       "      <td>WGS</td>\n",
       "      <td>Aligned Reads</td>\n",
       "      <td>BAM</td>\n",
       "      <td>[{'sample_type': 'Blood Derived Normal', 'tissue_type': 'Normal'}]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>a0409565-1baf-46f4-bae9-85a1dbe34160</td>\n",
       "      <td>3b1f7e48-216d-47de-ba79-db350f52bff0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Biospecimen Supplement</td>\n",
       "      <td>BCR XML</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89800</th>\n",
       "      <td>94468afc-faea-4091-af0f-d2df8aec37da</td>\n",
       "      <td>00807dae-9f4a-4fd1-aac2-82eb11bf2afb</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>Gene Expression Quantification</td>\n",
       "      <td>TSV</td>\n",
       "      <td>[{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89801</th>\n",
       "      <td>9bc60601-f424-492f-acbc-38751264d066</td>\n",
       "      <td>00807dae-9f4a-4fd1-aac2-82eb11bf2afb</td>\n",
       "      <td>WGS</td>\n",
       "      <td>Raw Simple Somatic Mutation</td>\n",
       "      <td>VCF</td>\n",
       "      <td>[{'sample_type': 'Blood Derived Normal', 'tissue_type': 'Normal'}, {'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89802</th>\n",
       "      <td>84b47afd-7fb1-4a52-8328-c257a7676176</td>\n",
       "      <td>00807dae-9f4a-4fd1-aac2-82eb11bf2afb</td>\n",
       "      <td>WXS</td>\n",
       "      <td>Annotated Somatic Mutation</td>\n",
       "      <td>VCF</td>\n",
       "      <td>[{'sample_type': 'Blood Derived Normal', 'tissue_type': 'Normal'}, {'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89803</th>\n",
       "      <td>6d870c48-0176-4137-96b9-7f5bb02100d2</td>\n",
       "      <td>00807dae-9f4a-4fd1-aac2-82eb11bf2afb</td>\n",
       "      <td>miRNA-Seq</td>\n",
       "      <td>Aligned Reads</td>\n",
       "      <td>BAM</td>\n",
       "      <td>[{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89804</th>\n",
       "      <td>cc6ca81e-5bb7-4c87-9602-ec5919b443a8</td>\n",
       "      <td>00807dae-9f4a-4fd1-aac2-82eb11bf2afb</td>\n",
       "      <td>WGS</td>\n",
       "      <td>Aligned Reads</td>\n",
       "      <td>BAM</td>\n",
       "      <td>[{'sample_type': 'Blood Derived Normal', 'tissue_type': 'Normal'}]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>89805 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                    file_id  \\\n",
       "0      27016c1c-f696-4a54-8c4a-6f9ba8a1c9e3   \n",
       "1      776f51f8-6f23-42d5-a96b-d7c934dcaa12   \n",
       "2      b1680590-4e97-4a3d-b41b-d9e84290de17   \n",
       "3      65cab127-616b-45c7-98b4-fdf8e01881fb   \n",
       "4      a0409565-1baf-46f4-bae9-85a1dbe34160   \n",
       "...                                     ...   \n",
       "89800  94468afc-faea-4091-af0f-d2df8aec37da   \n",
       "89801  9bc60601-f424-492f-acbc-38751264d066   \n",
       "89802  84b47afd-7fb1-4a52-8328-c257a7676176   \n",
       "89803  6d870c48-0176-4137-96b9-7f5bb02100d2   \n",
       "89804  cc6ca81e-5bb7-4c87-9602-ec5919b443a8   \n",
       "\n",
       "                                    case_id experimental_strategy  \\\n",
       "0      3b1f7e48-216d-47de-ba79-db350f52bff0                   WXS   \n",
       "1      3b1f7e48-216d-47de-ba79-db350f52bff0                   WXS   \n",
       "2      3b1f7e48-216d-47de-ba79-db350f52bff0               RNA-Seq   \n",
       "3      3b1f7e48-216d-47de-ba79-db350f52bff0                   WGS   \n",
       "4      3b1f7e48-216d-47de-ba79-db350f52bff0                   NaN   \n",
       "...                                     ...                   ...   \n",
       "89800  00807dae-9f4a-4fd1-aac2-82eb11bf2afb               RNA-Seq   \n",
       "89801  00807dae-9f4a-4fd1-aac2-82eb11bf2afb                   WGS   \n",
       "89802  00807dae-9f4a-4fd1-aac2-82eb11bf2afb                   WXS   \n",
       "89803  00807dae-9f4a-4fd1-aac2-82eb11bf2afb             miRNA-Seq   \n",
       "89804  00807dae-9f4a-4fd1-aac2-82eb11bf2afb                   WGS   \n",
       "\n",
       "                            data_type data_format  \\\n",
       "0          Annotated Somatic Mutation         VCF   \n",
       "1                       Aligned Reads         BAM   \n",
       "2                       Aligned Reads         BAM   \n",
       "3                       Aligned Reads         BAM   \n",
       "4              Biospecimen Supplement     BCR XML   \n",
       "...                               ...         ...   \n",
       "89800  Gene Expression Quantification         TSV   \n",
       "89801     Raw Simple Somatic Mutation         VCF   \n",
       "89802      Annotated Somatic Mutation         VCF   \n",
       "89803                   Aligned Reads         BAM   \n",
       "89804                   Aligned Reads         BAM   \n",
       "\n",
       "                                                                                                                            samples  \\\n",
       "0      [{'sample_type': 'Blood Derived Normal', 'tissue_type': 'Normal'}, {'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]   \n",
       "1                                                                        [{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]   \n",
       "2                                                                        [{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]   \n",
       "3                                                                [{'sample_type': 'Blood Derived Normal', 'tissue_type': 'Normal'}]   \n",
       "4                                                                                                                               NaN   \n",
       "...                                                                                                                             ...   \n",
       "89800                                                                    [{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]   \n",
       "89801  [{'sample_type': 'Blood Derived Normal', 'tissue_type': 'Normal'}, {'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]   \n",
       "89802  [{'sample_type': 'Blood Derived Normal', 'tissue_type': 'Normal'}, {'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]   \n",
       "89803                                                                    [{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]   \n",
       "89804                                                            [{'sample_type': 'Blood Derived Normal', 'tissue_type': 'Normal'}]   \n",
       "\n",
       "      sample_type tissue_type  is_file_of_interest  \n",
       "0             NaN         NaN                    0  \n",
       "1             NaN         NaN                    0  \n",
       "2             NaN         NaN                    0  \n",
       "3             NaN         NaN                    0  \n",
       "4             NaN         NaN                    0  \n",
       "...           ...         ...                  ...  \n",
       "89800         NaN         NaN                    0  \n",
       "89801         NaN         NaN                    0  \n",
       "89802         NaN         NaN                    0  \n",
       "89803         NaN         NaN                    0  \n",
       "89804         NaN         NaN                    0  \n",
       "\n",
       "[89805 rows x 9 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the DataFrame of flagged files\n",
    "df_files"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Counting Files per Case"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Filter the potential files of interest\n",
    "df_files_of_interest = df_files \\\n",
    "    .query('is_file_of_interest == 1') \\\n",
    "    [['file_id', 'case_id', 'experimental_strategy', 'tissue_type']]\n",
    "\n",
    "# Create an acronym for file type, i.e, experimental strategy and tissue type\n",
    "df_files_of_interest['file_type'] = (\n",
    "    df_files_of_interest['experimental_strategy'] \n",
    "    + '_' + df_files_of_interest['tissue_type']\n",
    ")\n",
    "\n",
    "# Copy some columns of the DataFrame\n",
    "df_file_count = df_files_of_interest[['case_id', 'file_id', 'file_type']].copy()\n",
    "\n",
    "# Create flags for each file types\n",
    "df_file_count['tt_mir'] = np.where(\n",
    "    df_file_count['file_type'] == 'miRNA-Seq_Tumor', 1, 0\n",
    ")\n",
    "df_file_count['tt_mrna'] = np.where(\n",
    "    df_file_count['file_type'] == 'RNA-Seq_Tumor', 1, 0\n",
    ")\n",
    "df_file_count['nt_mir'] = np.where(\n",
    "    df_file_count['file_type'] == 'miRNA-Seq_Normal', 1, 0\n",
    ")\n",
    "df_file_count['nt_mrna'] = np.where(\n",
    "    df_file_count['file_type'] == 'RNA-Seq_Normal', 1, 0\n",
    ")\n",
    "\n",
    "# Initialize the file count per row\n",
    "df_file_count['total_files'] = 1\n",
    "\n",
    "# Count the total number of file types per case\n",
    "df_file_count_agg = df_file_count \\\n",
    "    .groupby('case_id') \\\n",
    "    .agg(\n",
    "        tumor_mir_files=pd.NamedAgg(column='tt_mir', aggfunc='sum'),\n",
    "        tumor_mrna_files=pd.NamedAgg(column='tt_mrna', aggfunc='sum'),\n",
    "        normal_mir_files=pd.NamedAgg(column='nt_mir', aggfunc='sum'),\n",
    "        normal_mrna_files=pd.NamedAgg(column='nt_mrna', aggfunc='sum'),\n",
    "        total_files=pd.NamedAgg(column='total_files', aggfunc='sum'),\n",
    "    ) \\\n",
    "    .sort_values(by=['total_files'], ascending=False) \\\n",
    "    .reset_index()\n",
    "    \n",
    "# Add the file count to the cases DataFrame\n",
    "df_cases = df_cases \\\n",
    "    .merge(\n",
    "        right=df_file_count_agg,\n",
    "        left_on='case_id',\n",
    "        right_on='case_id',\n",
    "        how='left'\n",
    "    )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>case_id</th>\n",
       "      <th>submitter_id</th>\n",
       "      <th>disease_type</th>\n",
       "      <th>pam50_mrna</th>\n",
       "      <th>is_case_of_interest</th>\n",
       "      <th>tumor_mir_files</th>\n",
       "      <th>tumor_mrna_files</th>\n",
       "      <th>normal_mir_files</th>\n",
       "      <th>normal_mrna_files</th>\n",
       "      <th>total_files</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2d29a4ac-98e7-4663-9dd6-5681bc32ac2e</td>\n",
       "      <td>TCGA-A2-A3KC</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>de531604-bd1a-49df-96c2-6de3ae703f1d</td>\n",
       "      <td>TCGA-D8-A1XG</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>de646f6a-6d50-4be2-b3cd-f17554b234df</td>\n",
       "      <td>TCGA-AR-A24L</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>de75d0b9-0f47-4732-8df5-05c350cfcd32</td>\n",
       "      <td>TCGA-AN-A0FY</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal B</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>deba32e4-0e68-4711-941b-3b63bd965afb</td>\n",
       "      <td>TCGA-A2-A3XY</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1093</th>\n",
       "      <td>35bd694d-1dd2-466f-ab27-03320614b40e</td>\n",
       "      <td>TCGA-OL-A66I</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1094</th>\n",
       "      <td>2c90d5e3-ac8c-4c86-a06f-42ccc064eb49</td>\n",
       "      <td>TCGA-D8-A1XZ</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1095</th>\n",
       "      <td>360d1d6d-cf3e-4c78-b36f-ef5e3160aa6b</td>\n",
       "      <td>TCGA-BH-A0AU</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal B</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1096</th>\n",
       "      <td>2cf68894-168b-458b-af4f-53cad72989a8</td>\n",
       "      <td>TCGA-A7-A13F</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal B</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1097</th>\n",
       "      <td>36d77496-99f8-4911-ba7c-89fae32662aa</td>\n",
       "      <td>TCGA-A2-A0T5</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1098 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   case_id  submitter_id  \\\n",
       "0     2d29a4ac-98e7-4663-9dd6-5681bc32ac2e  TCGA-A2-A3KC   \n",
       "1     de531604-bd1a-49df-96c2-6de3ae703f1d  TCGA-D8-A1XG   \n",
       "2     de646f6a-6d50-4be2-b3cd-f17554b234df  TCGA-AR-A24L   \n",
       "3     de75d0b9-0f47-4732-8df5-05c350cfcd32  TCGA-AN-A0FY   \n",
       "4     deba32e4-0e68-4711-941b-3b63bd965afb  TCGA-A2-A3XY   \n",
       "...                                    ...           ...   \n",
       "1093  35bd694d-1dd2-466f-ab27-03320614b40e  TCGA-OL-A66I   \n",
       "1094  2c90d5e3-ac8c-4c86-a06f-42ccc064eb49  TCGA-D8-A1XZ   \n",
       "1095  360d1d6d-cf3e-4c78-b36f-ef5e3160aa6b  TCGA-BH-A0AU   \n",
       "1096  2cf68894-168b-458b-af4f-53cad72989a8  TCGA-A7-A13F   \n",
       "1097  36d77496-99f8-4911-ba7c-89fae32662aa  TCGA-A2-A0T5   \n",
       "\n",
       "                      disease_type pam50_mrna  is_case_of_interest  \\\n",
       "0     Ductal and Lobular Neoplasms        NaN                    0   \n",
       "1     Ductal and Lobular Neoplasms        NaN                    0   \n",
       "2     Ductal and Lobular Neoplasms        NaN                    0   \n",
       "3     Ductal and Lobular Neoplasms  Luminal B                    1   \n",
       "4     Ductal and Lobular Neoplasms        NaN                    0   \n",
       "...                            ...        ...                  ...   \n",
       "1093  Ductal and Lobular Neoplasms        NaN                    0   \n",
       "1094  Ductal and Lobular Neoplasms        NaN                    0   \n",
       "1095  Ductal and Lobular Neoplasms  Luminal B                    1   \n",
       "1096  Ductal and Lobular Neoplasms  Luminal B                    1   \n",
       "1097  Ductal and Lobular Neoplasms  Luminal A                    1   \n",
       "\n",
       "      tumor_mir_files  tumor_mrna_files  normal_mir_files  normal_mrna_files  \\\n",
       "0                 NaN               NaN               NaN                NaN   \n",
       "1                 NaN               NaN               NaN                NaN   \n",
       "2                 NaN               NaN               NaN                NaN   \n",
       "3                 1.0               1.0               0.0                0.0   \n",
       "4                 NaN               NaN               NaN                NaN   \n",
       "...               ...               ...               ...                ...   \n",
       "1093              NaN               NaN               NaN                NaN   \n",
       "1094              NaN               NaN               NaN                NaN   \n",
       "1095              1.0               1.0               1.0                1.0   \n",
       "1096              1.0               1.0               1.0                1.0   \n",
       "1097              1.0               1.0               0.0                0.0   \n",
       "\n",
       "      total_files  \n",
       "0             NaN  \n",
       "1             NaN  \n",
       "2             NaN  \n",
       "3             2.0  \n",
       "4             NaN  \n",
       "...           ...  \n",
       "1093          NaN  \n",
       "1094          NaN  \n",
       "1095          4.0  \n",
       "1096          4.0  \n",
       "1097          2.0  \n",
       "\n",
       "[1098 rows x 10 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the DataFrame of flagged cases with file count\n",
    "df_cases"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Defining the Cases of Interest"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Flagged Cases DataFrame"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Flag cases with files related to tumor tissue analysis\n",
    "df_cases['has_tumor_files_of_interest'] = np.where(\n",
    "    ((df_cases['is_case_of_interest'] == 1)\n",
    "     & (df_cases['tumor_mir_files'] == 1)\n",
    "     & (df_cases['tumor_mrna_files'] == 1)), 1, 0\n",
    ")\n",
    "\n",
    "# Flag cases with files related to normal tissue analysis\n",
    "df_cases['has_normal_files_of_interest'] = np.where(\n",
    "    ((df_cases['is_case_of_interest'] == 1)\n",
    "     & (df_cases['normal_mir_files'] == 1)\n",
    "     & (df_cases['normal_mrna_files'] == 1)), 1, 0\n",
    ")\n",
    "\n",
    "# Rearrange the DataFrame columns\n",
    "df_cases = df_cases.drop(columns=[\n",
    "    'tumor_mir_files',\n",
    "    'tumor_mrna_files',\n",
    "    'normal_mir_files',\n",
    "    'normal_mrna_files',\n",
    "    'total_files',\n",
    "    'is_case_of_interest',\n",
    "])\n",
    "\n",
    "# Update the cases of interest flag\n",
    "df_cases['is_case_of_interest'] = np.where(\n",
    "    ((df_cases['has_tumor_files_of_interest'] == 1)\n",
    "     | (df_cases['has_normal_files_of_interest'] == 1)), 1, 0\n",
    ")\n",
    "\n",
    "# Store the cases metadata DataFrame in a CSV file\n",
    "df_cases.to_csv(BRCA_INTERIM_FILE_PATHS['cases'], index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>case_id</th>\n",
       "      <th>submitter_id</th>\n",
       "      <th>disease_type</th>\n",
       "      <th>pam50_mrna</th>\n",
       "      <th>has_tumor_files_of_interest</th>\n",
       "      <th>has_normal_files_of_interest</th>\n",
       "      <th>is_case_of_interest</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2d29a4ac-98e7-4663-9dd6-5681bc32ac2e</td>\n",
       "      <td>TCGA-A2-A3KC</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>de531604-bd1a-49df-96c2-6de3ae703f1d</td>\n",
       "      <td>TCGA-D8-A1XG</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>de646f6a-6d50-4be2-b3cd-f17554b234df</td>\n",
       "      <td>TCGA-AR-A24L</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>de75d0b9-0f47-4732-8df5-05c350cfcd32</td>\n",
       "      <td>TCGA-AN-A0FY</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal B</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>deba32e4-0e68-4711-941b-3b63bd965afb</td>\n",
       "      <td>TCGA-A2-A3XY</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1093</th>\n",
       "      <td>35bd694d-1dd2-466f-ab27-03320614b40e</td>\n",
       "      <td>TCGA-OL-A66I</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1094</th>\n",
       "      <td>2c90d5e3-ac8c-4c86-a06f-42ccc064eb49</td>\n",
       "      <td>TCGA-D8-A1XZ</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1095</th>\n",
       "      <td>360d1d6d-cf3e-4c78-b36f-ef5e3160aa6b</td>\n",
       "      <td>TCGA-BH-A0AU</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal B</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1096</th>\n",
       "      <td>2cf68894-168b-458b-af4f-53cad72989a8</td>\n",
       "      <td>TCGA-A7-A13F</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal B</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1097</th>\n",
       "      <td>36d77496-99f8-4911-ba7c-89fae32662aa</td>\n",
       "      <td>TCGA-A2-A0T5</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1098 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   case_id  submitter_id  \\\n",
       "0     2d29a4ac-98e7-4663-9dd6-5681bc32ac2e  TCGA-A2-A3KC   \n",
       "1     de531604-bd1a-49df-96c2-6de3ae703f1d  TCGA-D8-A1XG   \n",
       "2     de646f6a-6d50-4be2-b3cd-f17554b234df  TCGA-AR-A24L   \n",
       "3     de75d0b9-0f47-4732-8df5-05c350cfcd32  TCGA-AN-A0FY   \n",
       "4     deba32e4-0e68-4711-941b-3b63bd965afb  TCGA-A2-A3XY   \n",
       "...                                    ...           ...   \n",
       "1093  35bd694d-1dd2-466f-ab27-03320614b40e  TCGA-OL-A66I   \n",
       "1094  2c90d5e3-ac8c-4c86-a06f-42ccc064eb49  TCGA-D8-A1XZ   \n",
       "1095  360d1d6d-cf3e-4c78-b36f-ef5e3160aa6b  TCGA-BH-A0AU   \n",
       "1096  2cf68894-168b-458b-af4f-53cad72989a8  TCGA-A7-A13F   \n",
       "1097  36d77496-99f8-4911-ba7c-89fae32662aa  TCGA-A2-A0T5   \n",
       "\n",
       "                      disease_type pam50_mrna  has_tumor_files_of_interest  \\\n",
       "0     Ductal and Lobular Neoplasms        NaN                            0   \n",
       "1     Ductal and Lobular Neoplasms        NaN                            0   \n",
       "2     Ductal and Lobular Neoplasms        NaN                            0   \n",
       "3     Ductal and Lobular Neoplasms  Luminal B                            1   \n",
       "4     Ductal and Lobular Neoplasms        NaN                            0   \n",
       "...                            ...        ...                          ...   \n",
       "1093  Ductal and Lobular Neoplasms        NaN                            0   \n",
       "1094  Ductal and Lobular Neoplasms        NaN                            0   \n",
       "1095  Ductal and Lobular Neoplasms  Luminal B                            1   \n",
       "1096  Ductal and Lobular Neoplasms  Luminal B                            1   \n",
       "1097  Ductal and Lobular Neoplasms  Luminal A                            1   \n",
       "\n",
       "      has_normal_files_of_interest  is_case_of_interest  \n",
       "0                                0                    0  \n",
       "1                                0                    0  \n",
       "2                                0                    0  \n",
       "3                                0                    1  \n",
       "4                                0                    0  \n",
       "...                            ...                  ...  \n",
       "1093                             0                    0  \n",
       "1094                             0                    0  \n",
       "1095                             1                    1  \n",
       "1096                             1                    1  \n",
       "1097                             0                    1  \n",
       "\n",
       "[1098 rows x 7 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the cases metadata DataFrame in a CSV file\n",
    "df_cases"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Cases of Interest DataFrame"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create the cases of interest metadata DataFrame\n",
    "df_cases_of_interest = df_cases \\\n",
    "    .query('is_case_of_interest == 1') \\\n",
    "    .drop(columns=['is_case_of_interest']) \\\n",
    "    .reset_index(drop=True)\n",
    "    \n",
    "# Store the cases of interest metadata DataFrame in a CSV file\n",
    "df_cases_of_interest.to_csv(\n",
    "    BRCA_INTERIM_FILE_PATHS['cases-of-interest'], index=False\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>case_id</th>\n",
       "      <th>submitter_id</th>\n",
       "      <th>disease_type</th>\n",
       "      <th>pam50_mrna</th>\n",
       "      <th>has_tumor_files_of_interest</th>\n",
       "      <th>has_normal_files_of_interest</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>de75d0b9-0f47-4732-8df5-05c350cfcd32</td>\n",
       "      <td>TCGA-AN-A0FY</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal B</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1fef9454-b3e8-4d9d-a9aa-aa1f1a32b80c</td>\n",
       "      <td>TCGA-E2-A15J</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2021ed1f-dc75-4701-b8b8-1386466e4802</td>\n",
       "      <td>TCGA-BH-A0H7</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>20e8106b-1290-4735-abe4-7621e08e3dc8</td>\n",
       "      <td>TCGA-BH-A0E0</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Basal-like</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>214a4507-d974-4b3e-8525-7408fccc6a0f</td>\n",
       "      <td>TCGA-BH-A0B7</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>HER2-enriched</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>483</th>\n",
       "      <td>33919e92-4ea9-47e5-a6f4-36e51845d50f</td>\n",
       "      <td>TCGA-A8-A09N</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal B</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>484</th>\n",
       "      <td>34ab0cb8-bf19-4cbb-b4eb-3c45ee5aa60b</td>\n",
       "      <td>TCGA-B6-A0IG</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>485</th>\n",
       "      <td>360d1d6d-cf3e-4c78-b36f-ef5e3160aa6b</td>\n",
       "      <td>TCGA-BH-A0AU</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal B</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>486</th>\n",
       "      <td>2cf68894-168b-458b-af4f-53cad72989a8</td>\n",
       "      <td>TCGA-A7-A13F</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal B</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>487</th>\n",
       "      <td>36d77496-99f8-4911-ba7c-89fae32662aa</td>\n",
       "      <td>TCGA-A2-A0T5</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>488 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                  case_id  submitter_id  \\\n",
       "0    de75d0b9-0f47-4732-8df5-05c350cfcd32  TCGA-AN-A0FY   \n",
       "1    1fef9454-b3e8-4d9d-a9aa-aa1f1a32b80c  TCGA-E2-A15J   \n",
       "2    2021ed1f-dc75-4701-b8b8-1386466e4802  TCGA-BH-A0H7   \n",
       "3    20e8106b-1290-4735-abe4-7621e08e3dc8  TCGA-BH-A0E0   \n",
       "4    214a4507-d974-4b3e-8525-7408fccc6a0f  TCGA-BH-A0B7   \n",
       "..                                    ...           ...   \n",
       "483  33919e92-4ea9-47e5-a6f4-36e51845d50f  TCGA-A8-A09N   \n",
       "484  34ab0cb8-bf19-4cbb-b4eb-3c45ee5aa60b  TCGA-B6-A0IG   \n",
       "485  360d1d6d-cf3e-4c78-b36f-ef5e3160aa6b  TCGA-BH-A0AU   \n",
       "486  2cf68894-168b-458b-af4f-53cad72989a8  TCGA-A7-A13F   \n",
       "487  36d77496-99f8-4911-ba7c-89fae32662aa  TCGA-A2-A0T5   \n",
       "\n",
       "                     disease_type     pam50_mrna  has_tumor_files_of_interest  \\\n",
       "0    Ductal and Lobular Neoplasms      Luminal B                            1   \n",
       "1    Ductal and Lobular Neoplasms      Luminal A                            1   \n",
       "2    Ductal and Lobular Neoplasms      Luminal A                            1   \n",
       "3    Ductal and Lobular Neoplasms     Basal-like                            1   \n",
       "4    Ductal and Lobular Neoplasms  HER2-enriched                            1   \n",
       "..                            ...            ...                          ...   \n",
       "483  Ductal and Lobular Neoplasms      Luminal B                            1   \n",
       "484  Ductal and Lobular Neoplasms      Luminal A                            1   \n",
       "485  Ductal and Lobular Neoplasms      Luminal B                            1   \n",
       "486  Ductal and Lobular Neoplasms      Luminal B                            1   \n",
       "487  Ductal and Lobular Neoplasms      Luminal A                            1   \n",
       "\n",
       "     has_normal_files_of_interest  \n",
       "0                               0  \n",
       "1                               0  \n",
       "2                               1  \n",
       "3                               0  \n",
       "4                               1  \n",
       "..                            ...  \n",
       "483                             0  \n",
       "484                             0  \n",
       "485                             1  \n",
       "486                             1  \n",
       "487                             0  \n",
       "\n",
       "[488 rows x 6 columns]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the cases of interest metadata DataFrame\n",
    "df_cases_of_interest"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Defining the Files of Interest"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Flagged Files DataFrame"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Retrieve the flag of interest of the cases    \n",
    "df_files = df_files \\\n",
    "    .merge(\n",
    "        right=df_cases,\n",
    "        left_on='case_id',\n",
    "        right_on='case_id',\n",
    "        how='inner'\n",
    "    ) \\\n",
    "    .drop(columns=['submitter_id', 'disease_type', 'pam50_mrna',])\n",
    "\n",
    "# Flag the files related to tumor tissue analysis\n",
    "df_files['is_tumor_file_of_interest'] = np.where(\n",
    "    ((df_files['is_file_of_interest'] == 1)\n",
    "     & (df_files['has_tumor_files_of_interest'] == 1)\n",
    "     & (df_files['tissue_type'] == 'Tumor')), 1, 0\n",
    ")\n",
    "\n",
    "# Flag the files related to normal tissue analysis\n",
    "df_files['is_normal_file_of_interest'] = np.where(\n",
    "    ((df_files['is_file_of_interest'] == 1)\n",
    "     & (df_files['has_normal_files_of_interest'] == 1)\n",
    "     & (df_files['tissue_type'] == 'Normal')), 1, 0\n",
    ")\n",
    "\n",
    "# Rearrange the DataFrame columns\n",
    "df_files = df_files.drop(columns=[\n",
    "    'sample_type',\n",
    "    'tissue_type',\n",
    "    'is_file_of_interest',\n",
    "    'has_tumor_files_of_interest',\n",
    "    'has_normal_files_of_interest',\n",
    "    'is_case_of_interest',\n",
    "])\n",
    "\n",
    "# Update the files of interest flag\n",
    "df_files['is_file_of_interest'] = np.where(\n",
    "    ((df_files['is_tumor_file_of_interest'] == 1)\n",
    "     | (df_files['is_normal_file_of_interest'] == 1)), 1, 0\n",
    ")\n",
    "\n",
    "# Store the files metadata DataFrame in a CSV file\n",
    "df_files.to_csv(BRCA_INTERIM_FILE_PATHS['files'], index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>file_id</th>\n",
       "      <th>case_id</th>\n",
       "      <th>experimental_strategy</th>\n",
       "      <th>data_type</th>\n",
       "      <th>data_format</th>\n",
       "      <th>samples</th>\n",
       "      <th>is_tumor_file_of_interest</th>\n",
       "      <th>is_normal_file_of_interest</th>\n",
       "      <th>is_file_of_interest</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>27016c1c-f696-4a54-8c4a-6f9ba8a1c9e3</td>\n",
       "      <td>3b1f7e48-216d-47de-ba79-db350f52bff0</td>\n",
       "      <td>WXS</td>\n",
       "      <td>Annotated Somatic Mutation</td>\n",
       "      <td>VCF</td>\n",
       "      <td>[{'sample_type': 'Blood Derived Normal', 'tissue_type': 'Normal'}, {'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>776f51f8-6f23-42d5-a96b-d7c934dcaa12</td>\n",
       "      <td>3b1f7e48-216d-47de-ba79-db350f52bff0</td>\n",
       "      <td>WXS</td>\n",
       "      <td>Aligned Reads</td>\n",
       "      <td>BAM</td>\n",
       "      <td>[{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>b1680590-4e97-4a3d-b41b-d9e84290de17</td>\n",
       "      <td>3b1f7e48-216d-47de-ba79-db350f52bff0</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>Aligned Reads</td>\n",
       "      <td>BAM</td>\n",
       "      <td>[{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>65cab127-616b-45c7-98b4-fdf8e01881fb</td>\n",
       "      <td>3b1f7e48-216d-47de-ba79-db350f52bff0</td>\n",
       "      <td>WGS</td>\n",
       "      <td>Aligned Reads</td>\n",
       "      <td>BAM</td>\n",
       "      <td>[{'sample_type': 'Blood Derived Normal', 'tissue_type': 'Normal'}]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>a0409565-1baf-46f4-bae9-85a1dbe34160</td>\n",
       "      <td>3b1f7e48-216d-47de-ba79-db350f52bff0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Biospecimen Supplement</td>\n",
       "      <td>BCR XML</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89800</th>\n",
       "      <td>94468afc-faea-4091-af0f-d2df8aec37da</td>\n",
       "      <td>00807dae-9f4a-4fd1-aac2-82eb11bf2afb</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>Gene Expression Quantification</td>\n",
       "      <td>TSV</td>\n",
       "      <td>[{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89801</th>\n",
       "      <td>9bc60601-f424-492f-acbc-38751264d066</td>\n",
       "      <td>00807dae-9f4a-4fd1-aac2-82eb11bf2afb</td>\n",
       "      <td>WGS</td>\n",
       "      <td>Raw Simple Somatic Mutation</td>\n",
       "      <td>VCF</td>\n",
       "      <td>[{'sample_type': 'Blood Derived Normal', 'tissue_type': 'Normal'}, {'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89802</th>\n",
       "      <td>84b47afd-7fb1-4a52-8328-c257a7676176</td>\n",
       "      <td>00807dae-9f4a-4fd1-aac2-82eb11bf2afb</td>\n",
       "      <td>WXS</td>\n",
       "      <td>Annotated Somatic Mutation</td>\n",
       "      <td>VCF</td>\n",
       "      <td>[{'sample_type': 'Blood Derived Normal', 'tissue_type': 'Normal'}, {'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89803</th>\n",
       "      <td>6d870c48-0176-4137-96b9-7f5bb02100d2</td>\n",
       "      <td>00807dae-9f4a-4fd1-aac2-82eb11bf2afb</td>\n",
       "      <td>miRNA-Seq</td>\n",
       "      <td>Aligned Reads</td>\n",
       "      <td>BAM</td>\n",
       "      <td>[{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89804</th>\n",
       "      <td>cc6ca81e-5bb7-4c87-9602-ec5919b443a8</td>\n",
       "      <td>00807dae-9f4a-4fd1-aac2-82eb11bf2afb</td>\n",
       "      <td>WGS</td>\n",
       "      <td>Aligned Reads</td>\n",
       "      <td>BAM</td>\n",
       "      <td>[{'sample_type': 'Blood Derived Normal', 'tissue_type': 'Normal'}]</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>89805 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                    file_id  \\\n",
       "0      27016c1c-f696-4a54-8c4a-6f9ba8a1c9e3   \n",
       "1      776f51f8-6f23-42d5-a96b-d7c934dcaa12   \n",
       "2      b1680590-4e97-4a3d-b41b-d9e84290de17   \n",
       "3      65cab127-616b-45c7-98b4-fdf8e01881fb   \n",
       "4      a0409565-1baf-46f4-bae9-85a1dbe34160   \n",
       "...                                     ...   \n",
       "89800  94468afc-faea-4091-af0f-d2df8aec37da   \n",
       "89801  9bc60601-f424-492f-acbc-38751264d066   \n",
       "89802  84b47afd-7fb1-4a52-8328-c257a7676176   \n",
       "89803  6d870c48-0176-4137-96b9-7f5bb02100d2   \n",
       "89804  cc6ca81e-5bb7-4c87-9602-ec5919b443a8   \n",
       "\n",
       "                                    case_id experimental_strategy  \\\n",
       "0      3b1f7e48-216d-47de-ba79-db350f52bff0                   WXS   \n",
       "1      3b1f7e48-216d-47de-ba79-db350f52bff0                   WXS   \n",
       "2      3b1f7e48-216d-47de-ba79-db350f52bff0               RNA-Seq   \n",
       "3      3b1f7e48-216d-47de-ba79-db350f52bff0                   WGS   \n",
       "4      3b1f7e48-216d-47de-ba79-db350f52bff0                   NaN   \n",
       "...                                     ...                   ...   \n",
       "89800  00807dae-9f4a-4fd1-aac2-82eb11bf2afb               RNA-Seq   \n",
       "89801  00807dae-9f4a-4fd1-aac2-82eb11bf2afb                   WGS   \n",
       "89802  00807dae-9f4a-4fd1-aac2-82eb11bf2afb                   WXS   \n",
       "89803  00807dae-9f4a-4fd1-aac2-82eb11bf2afb             miRNA-Seq   \n",
       "89804  00807dae-9f4a-4fd1-aac2-82eb11bf2afb                   WGS   \n",
       "\n",
       "                            data_type data_format  \\\n",
       "0          Annotated Somatic Mutation         VCF   \n",
       "1                       Aligned Reads         BAM   \n",
       "2                       Aligned Reads         BAM   \n",
       "3                       Aligned Reads         BAM   \n",
       "4              Biospecimen Supplement     BCR XML   \n",
       "...                               ...         ...   \n",
       "89800  Gene Expression Quantification         TSV   \n",
       "89801     Raw Simple Somatic Mutation         VCF   \n",
       "89802      Annotated Somatic Mutation         VCF   \n",
       "89803                   Aligned Reads         BAM   \n",
       "89804                   Aligned Reads         BAM   \n",
       "\n",
       "                                                                                                                            samples  \\\n",
       "0      [{'sample_type': 'Blood Derived Normal', 'tissue_type': 'Normal'}, {'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]   \n",
       "1                                                                        [{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]   \n",
       "2                                                                        [{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]   \n",
       "3                                                                [{'sample_type': 'Blood Derived Normal', 'tissue_type': 'Normal'}]   \n",
       "4                                                                                                                               NaN   \n",
       "...                                                                                                                             ...   \n",
       "89800                                                                    [{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]   \n",
       "89801  [{'sample_type': 'Blood Derived Normal', 'tissue_type': 'Normal'}, {'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]   \n",
       "89802  [{'sample_type': 'Blood Derived Normal', 'tissue_type': 'Normal'}, {'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]   \n",
       "89803                                                                    [{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]   \n",
       "89804                                                            [{'sample_type': 'Blood Derived Normal', 'tissue_type': 'Normal'}]   \n",
       "\n",
       "       is_tumor_file_of_interest  is_normal_file_of_interest  \\\n",
       "0                              0                           0   \n",
       "1                              0                           0   \n",
       "2                              0                           0   \n",
       "3                              0                           0   \n",
       "4                              0                           0   \n",
       "...                          ...                         ...   \n",
       "89800                          0                           0   \n",
       "89801                          0                           0   \n",
       "89802                          0                           0   \n",
       "89803                          0                           0   \n",
       "89804                          0                           0   \n",
       "\n",
       "       is_file_of_interest  \n",
       "0                        0  \n",
       "1                        0  \n",
       "2                        0  \n",
       "3                        0  \n",
       "4                        0  \n",
       "...                    ...  \n",
       "89800                    0  \n",
       "89801                    0  \n",
       "89802                    0  \n",
       "89803                    0  \n",
       "89804                    0  \n",
       "\n",
       "[89805 rows x 9 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the files metadata DataFrame in a CSV file\n",
    "df_files"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Files of Interest DataFrame"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create the files of interest metadata DataFrame\n",
    "df_files_of_interest = df_files \\\n",
    "    .query('is_file_of_interest == 1') \\\n",
    "    .drop(columns=['is_file_of_interest']) \\\n",
    "    .reset_index(drop=True)\n",
    "    \n",
    "# Store the files of interest metadata DataFrame in a CSV file\n",
    "df_files_of_interest.to_csv(\n",
    "    BRCA_INTERIM_FILE_PATHS['files-of-interest'], index=False\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>file_id</th>\n",
       "      <th>case_id</th>\n",
       "      <th>experimental_strategy</th>\n",
       "      <th>data_type</th>\n",
       "      <th>data_format</th>\n",
       "      <th>samples</th>\n",
       "      <th>is_tumor_file_of_interest</th>\n",
       "      <th>is_normal_file_of_interest</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>dd0e32d6-4704-4abd-82e4-2041fbc19a79</td>\n",
       "      <td>3c08aabd-d5b5-4bbe-857c-38a7527b2163</td>\n",
       "      <td>miRNA-Seq</td>\n",
       "      <td>Isoform Expression Quantification</td>\n",
       "      <td>TXT</td>\n",
       "      <td>[{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>8644d07c-979f-4ec6-928a-afe9229fb9f1</td>\n",
       "      <td>3c08aabd-d5b5-4bbe-857c-38a7527b2163</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>Gene Expression Quantification</td>\n",
       "      <td>TSV</td>\n",
       "      <td>[{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>09a491b5-1bfb-47d9-963f-69f6c7e6c1e5</td>\n",
       "      <td>abdc76db-f85e-4337-a57e-6d098789da03</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>Gene Expression Quantification</td>\n",
       "      <td>TSV</td>\n",
       "      <td>[{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>02fc7dd7-7e17-4930-8424-52173c33b29b</td>\n",
       "      <td>abdc76db-f85e-4337-a57e-6d098789da03</td>\n",
       "      <td>miRNA-Seq</td>\n",
       "      <td>Isoform Expression Quantification</td>\n",
       "      <td>TXT</td>\n",
       "      <td>[{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>f8e5691a-e24d-4dba-873e-9ffa3ca12bb6</td>\n",
       "      <td>3c612e12-6de8-44fa-a095-805c45474821</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>Gene Expression Quantification</td>\n",
       "      <td>TSV</td>\n",
       "      <td>[{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1079</th>\n",
       "      <td>863c79a7-57d7-4a02-89ab-4abe5db7e4ff</td>\n",
       "      <td>6cdc0d53-f813-4101-81e5-9bee68270536</td>\n",
       "      <td>miRNA-Seq</td>\n",
       "      <td>Isoform Expression Quantification</td>\n",
       "      <td>TXT</td>\n",
       "      <td>[{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1080</th>\n",
       "      <td>24d0d1a9-ce92-4db6-8224-e3dd67733712</td>\n",
       "      <td>10c9be1e-2cc0-45e4-8d2c-c013ea63ef05</td>\n",
       "      <td>miRNA-Seq</td>\n",
       "      <td>Isoform Expression Quantification</td>\n",
       "      <td>TXT</td>\n",
       "      <td>[{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1081</th>\n",
       "      <td>15d7e135-acb7-46ef-a8b1-f424e40a9e4d</td>\n",
       "      <td>10c9be1e-2cc0-45e4-8d2c-c013ea63ef05</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>Gene Expression Quantification</td>\n",
       "      <td>TSV</td>\n",
       "      <td>[{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1082</th>\n",
       "      <td>8ecef704-ed0f-4700-a34e-46a174033332</td>\n",
       "      <td>1174f6e4-ffbe-4e59-a000-8d861c968369</td>\n",
       "      <td>miRNA-Seq</td>\n",
       "      <td>Isoform Expression Quantification</td>\n",
       "      <td>TXT</td>\n",
       "      <td>[{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1083</th>\n",
       "      <td>79f97f73-b650-49cf-8654-53038a6c0de7</td>\n",
       "      <td>1174f6e4-ffbe-4e59-a000-8d861c968369</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>Gene Expression Quantification</td>\n",
       "      <td>TSV</td>\n",
       "      <td>[{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1084 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   file_id  \\\n",
       "0     dd0e32d6-4704-4abd-82e4-2041fbc19a79   \n",
       "1     8644d07c-979f-4ec6-928a-afe9229fb9f1   \n",
       "2     09a491b5-1bfb-47d9-963f-69f6c7e6c1e5   \n",
       "3     02fc7dd7-7e17-4930-8424-52173c33b29b   \n",
       "4     f8e5691a-e24d-4dba-873e-9ffa3ca12bb6   \n",
       "...                                    ...   \n",
       "1079  863c79a7-57d7-4a02-89ab-4abe5db7e4ff   \n",
       "1080  24d0d1a9-ce92-4db6-8224-e3dd67733712   \n",
       "1081  15d7e135-acb7-46ef-a8b1-f424e40a9e4d   \n",
       "1082  8ecef704-ed0f-4700-a34e-46a174033332   \n",
       "1083  79f97f73-b650-49cf-8654-53038a6c0de7   \n",
       "\n",
       "                                   case_id experimental_strategy  \\\n",
       "0     3c08aabd-d5b5-4bbe-857c-38a7527b2163             miRNA-Seq   \n",
       "1     3c08aabd-d5b5-4bbe-857c-38a7527b2163               RNA-Seq   \n",
       "2     abdc76db-f85e-4337-a57e-6d098789da03               RNA-Seq   \n",
       "3     abdc76db-f85e-4337-a57e-6d098789da03             miRNA-Seq   \n",
       "4     3c612e12-6de8-44fa-a095-805c45474821               RNA-Seq   \n",
       "...                                    ...                   ...   \n",
       "1079  6cdc0d53-f813-4101-81e5-9bee68270536             miRNA-Seq   \n",
       "1080  10c9be1e-2cc0-45e4-8d2c-c013ea63ef05             miRNA-Seq   \n",
       "1081  10c9be1e-2cc0-45e4-8d2c-c013ea63ef05               RNA-Seq   \n",
       "1082  1174f6e4-ffbe-4e59-a000-8d861c968369             miRNA-Seq   \n",
       "1083  1174f6e4-ffbe-4e59-a000-8d861c968369               RNA-Seq   \n",
       "\n",
       "                              data_type data_format  \\\n",
       "0     Isoform Expression Quantification         TXT   \n",
       "1        Gene Expression Quantification         TSV   \n",
       "2        Gene Expression Quantification         TSV   \n",
       "3     Isoform Expression Quantification         TXT   \n",
       "4        Gene Expression Quantification         TSV   \n",
       "...                                 ...         ...   \n",
       "1079  Isoform Expression Quantification         TXT   \n",
       "1080  Isoform Expression Quantification         TXT   \n",
       "1081     Gene Expression Quantification         TSV   \n",
       "1082  Isoform Expression Quantification         TXT   \n",
       "1083     Gene Expression Quantification         TSV   \n",
       "\n",
       "                                                         samples  \\\n",
       "0     [{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]   \n",
       "1     [{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]   \n",
       "2     [{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]   \n",
       "3     [{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]   \n",
       "4     [{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]   \n",
       "...                                                          ...   \n",
       "1079  [{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]   \n",
       "1080  [{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]   \n",
       "1081  [{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]   \n",
       "1082  [{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]   \n",
       "1083  [{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]   \n",
       "\n",
       "      is_tumor_file_of_interest  is_normal_file_of_interest  \n",
       "0                             1                           0  \n",
       "1                             1                           0  \n",
       "2                             1                           0  \n",
       "3                             1                           0  \n",
       "4                             1                           0  \n",
       "...                         ...                         ...  \n",
       "1079                          1                           0  \n",
       "1080                          1                           0  \n",
       "1081                          1                           0  \n",
       "1082                          1                           0  \n",
       "1083                          1                           0  \n",
       "\n",
       "[1084 rows x 8 columns]"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the files of interest metadata DataFrame \n",
    "df_files_of_interest"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
