{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# **Preprocessing Metadata from TCGA-BRCA Project**\n",
    "TCGA: The Cancer Genome Atlas  \n",
    "BRCA: Breast Invasive Carcinoma"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Importing Libraries and Configurations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "import os\n",
    "import sys\n",
    "\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import requests\n",
    "\n",
    "# Get the project root (two levels above)\n",
    "sys.path.append(os.path.abspath(os.path.join('..', '..')))\n",
    "\n",
    "from config import (\n",
    "    BRCA_INTERIM_DATA_DIR,\n",
    "    BRCA_PAPER_FILE_PATH,\n",
    "    GDC_API_ENDPOINTS,\n",
    "    GDC_API_REQUESTS_CONSTANTS,\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Function"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "def gdc_api_request(endpoint, fields, filters={}, size=1):\n",
    "    \"\"\"\n",
    "    Send a request to a Genomic Data Commons (GDC) API endpoint with the specified parameters.\n",
    "\n",
    "    Parameters:\n",
    "    -----------\n",
    "    endpoint : str\n",
    "        The URL of the endpoint to send the request to.\n",
    "    fields : list of str\n",
    "        A list of fields to include in the response.\n",
    "    filters : dict, optional\n",
    "        A dictionary of filters to apply to the request. Default is an empty dictionary.\n",
    "    size : int, optional\n",
    "        The number of results to retrieve. Default is 1.\n",
    "\n",
    "    Returns:\n",
    "    --------\n",
    "    list\n",
    "        A list of hits (data items) retrieved from the endpoint response.\n",
    "    \"\"\"\n",
    "    # Parameters used in the endpoint request\n",
    "    params = {\n",
    "        'fields': ','.join(fields),\n",
    "        'filters': filters,\n",
    "        'size': str(size)\n",
    "    }\n",
    "    \n",
    "    # Request the objects of interest to the endpoint\n",
    "    response = requests.post(\n",
    "        url=endpoint,\n",
    "        headers={'Content-Type': 'application/json'},\n",
    "        json=params\n",
    "    )\n",
    "    response.raise_for_status()\n",
    "\n",
    "    # Convert the response content to a JSON\n",
    "    json_response = json.loads(response.content.decode('utf-8'))\n",
    "\n",
    "    return json_response['data']['hits']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# TCGA-BRCA Paper - Supplementary Table 1\n",
    "The Cancer Genome Atlas Network. Comprehensive molecular portraits of human breast tumours. Nature 490, 61–70 (2012). https://doi.org/10.1038/nature11412."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Loading and Filtering Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create the DataFrame of 'Supplementary Table 1' from the TCGA-BRCA paper\n",
    "df_paper_data = pd.read_excel(\n",
    "    BRCA_PAPER_FILE_PATH, sheet_name='SuppTable1', skiprows=1\n",
    ")\n",
    "\n",
    "# Rename DataFrame columns\n",
    "columns = dict()\n",
    "for name in list(df_paper_data.columns):\n",
    "    columns[name] = name.replace(' ', '_').lower()\n",
    "df_paper_data = df_paper_data.rename(columns=columns)\n",
    "\n",
    "# Filter cases of molecular subtypes of interest\n",
    "df_paper_cases = df_paper_data \\\n",
    "    .query(\n",
    "        '(pam50_mrna == \"Basal-like\") or (pam50_mrna == \"HER2-enriched\") ' +\n",
    "        'or (pam50_mrna == \"Luminal A\") or (pam50_mrna == \"Luminal B\")'\n",
    "    )\n",
    "    \n",
    "# # Flags\n",
    "# df_paper_data['is_case_of_interest'] = np.where(\n",
    "#     ((df_paper_data['pam50_mrna'] == 'Basal-like') \n",
    "#      | (df_paper_data['pam50_mrna'] == 'HER2-enriched') \n",
    "#      | (df_paper_data['pam50_mrna'] == 'Luminal A') \n",
    "#      | (df_paper_data['pam50_mrna'] == 'Luminal B')), 1, 0\n",
    "# )"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Paper Cases DataFrame"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Store the DataFrame in a CSV file\n",
    "file_name = 'brca-paper-cases.csv'\n",
    "df_paper_cases.to_csv(f'{BRCA_INTERIM_DATA_DIR}/{file_name}', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>complete_tcga_id</th>\n",
       "      <th>gender</th>\n",
       "      <th>age_at_initial_pathologic_diagnosis</th>\n",
       "      <th>er_status</th>\n",
       "      <th>pr_status</th>\n",
       "      <th>her2_final_status</th>\n",
       "      <th>tumor</th>\n",
       "      <th>tumor--t1_coded</th>\n",
       "      <th>node</th>\n",
       "      <th>node-coded</th>\n",
       "      <th>...</th>\n",
       "      <th>pam50_mrna</th>\n",
       "      <th>sigclust_unsupervised_mrna</th>\n",
       "      <th>sigclust_intrinsic_mrna</th>\n",
       "      <th>mirna_clusters</th>\n",
       "      <th>methylation_clusters</th>\n",
       "      <th>rppa_clusters</th>\n",
       "      <th>cn_clusters</th>\n",
       "      <th>integrated_clusters_(with_pam50)</th>\n",
       "      <th>integrated_clusters_(no_exp)</th>\n",
       "      <th>integrated_clusters_(unsup_exp)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TCGA-A2-A0T2</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>66.0</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>T3</td>\n",
       "      <td>T_Other</td>\n",
       "      <td>N3</td>\n",
       "      <td>Positive</td>\n",
       "      <td>...</td>\n",
       "      <td>Basal-like</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-13.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Basal</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TCGA-A2-A04P</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>36.0</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>T2</td>\n",
       "      <td>T_Other</td>\n",
       "      <td>N3</td>\n",
       "      <td>Positive</td>\n",
       "      <td>...</td>\n",
       "      <td>Basal-like</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-13.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Basal</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TCGA-A1-A0SK</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>54.0</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>T2</td>\n",
       "      <td>T_Other</td>\n",
       "      <td>N0</td>\n",
       "      <td>Negative</td>\n",
       "      <td>...</td>\n",
       "      <td>Basal-like</td>\n",
       "      <td>-6.0</td>\n",
       "      <td>-13.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Basal</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TCGA-A2-A0CM</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>40.0</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>T2</td>\n",
       "      <td>T_Other</td>\n",
       "      <td>N0</td>\n",
       "      <td>Negative</td>\n",
       "      <td>...</td>\n",
       "      <td>Basal-like</td>\n",
       "      <td>-12.0</td>\n",
       "      <td>-13.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>Basal</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TCGA-AR-A1AR</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>50.0</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>T1</td>\n",
       "      <td>T1</td>\n",
       "      <td>N2</td>\n",
       "      <td>Positive</td>\n",
       "      <td>...</td>\n",
       "      <td>Basal-like</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-13.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>509</th>\n",
       "      <td>TCGA-E2-A15K</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>58.0</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>T2</td>\n",
       "      <td>T_Other</td>\n",
       "      <td>N1</td>\n",
       "      <td>Positive</td>\n",
       "      <td>...</td>\n",
       "      <td>Luminal B</td>\n",
       "      <td>-3.0</td>\n",
       "      <td>-12.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>LumA</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>510</th>\n",
       "      <td>TCGA-E2-A15L</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>65.0</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>T2</td>\n",
       "      <td>T_Other</td>\n",
       "      <td>N0</td>\n",
       "      <td>Negative</td>\n",
       "      <td>...</td>\n",
       "      <td>Luminal B</td>\n",
       "      <td>-3.0</td>\n",
       "      <td>-12.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>ReacI</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>511</th>\n",
       "      <td>TCGA-E2-A15M</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>66.0</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>T2</td>\n",
       "      <td>T_Other</td>\n",
       "      <td>N0</td>\n",
       "      <td>Negative</td>\n",
       "      <td>...</td>\n",
       "      <td>Luminal B</td>\n",
       "      <td>-4.0</td>\n",
       "      <td>-4.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ReacI</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>512</th>\n",
       "      <td>TCGA-E2-A15S</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>34.0</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>T2</td>\n",
       "      <td>T_Other</td>\n",
       "      <td>N1</td>\n",
       "      <td>Positive</td>\n",
       "      <td>...</td>\n",
       "      <td>Luminal B</td>\n",
       "      <td>-3.0</td>\n",
       "      <td>-10.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>LumA/B</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>513</th>\n",
       "      <td>TCGA-E2-A15T</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>65.0</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>T2</td>\n",
       "      <td>T_Other</td>\n",
       "      <td>N0</td>\n",
       "      <td>Negative</td>\n",
       "      <td>...</td>\n",
       "      <td>Luminal B</td>\n",
       "      <td>-3.0</td>\n",
       "      <td>-10.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>LumA/B</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>514 rows × 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    complete_tcga_id  gender  age_at_initial_pathologic_diagnosis er_status  \\\n",
       "0       TCGA-A2-A0T2  FEMALE                                 66.0  Negative   \n",
       "1       TCGA-A2-A04P  FEMALE                                 36.0  Negative   \n",
       "2       TCGA-A1-A0SK  FEMALE                                 54.0  Negative   \n",
       "3       TCGA-A2-A0CM  FEMALE                                 40.0  Negative   \n",
       "4       TCGA-AR-A1AR  FEMALE                                 50.0  Negative   \n",
       "..               ...     ...                                  ...       ...   \n",
       "509     TCGA-E2-A15K  FEMALE                                 58.0  Positive   \n",
       "510     TCGA-E2-A15L  FEMALE                                 65.0  Positive   \n",
       "511     TCGA-E2-A15M  FEMALE                                 66.0  Positive   \n",
       "512     TCGA-E2-A15S  FEMALE                                 34.0  Positive   \n",
       "513     TCGA-E2-A15T  FEMALE                                 65.0  Positive   \n",
       "\n",
       "    pr_status her2_final_status tumor tumor--t1_coded node node-coded  ...  \\\n",
       "0    Negative          Negative    T3         T_Other   N3   Positive  ...   \n",
       "1    Negative          Negative    T2         T_Other   N3   Positive  ...   \n",
       "2    Negative          Negative    T2         T_Other   N0   Negative  ...   \n",
       "3    Negative          Negative    T2         T_Other   N0   Negative  ...   \n",
       "4    Negative          Negative    T1              T1   N2   Positive  ...   \n",
       "..        ...               ...   ...             ...  ...        ...  ...   \n",
       "509  Positive          Negative    T2         T_Other   N1   Positive  ...   \n",
       "510  Positive          Negative    T2         T_Other   N0   Negative  ...   \n",
       "511  Positive          Negative    T2         T_Other   N0   Negative  ...   \n",
       "512  Negative          Negative    T2         T_Other   N1   Positive  ...   \n",
       "513  Positive          Negative    T2         T_Other   N0   Negative  ...   \n",
       "\n",
       "     pam50_mrna sigclust_unsupervised_mrna sigclust_intrinsic_mrna  \\\n",
       "0    Basal-like                        0.0                   -13.0   \n",
       "1    Basal-like                        0.0                   -13.0   \n",
       "2    Basal-like                       -6.0                   -13.0   \n",
       "3    Basal-like                      -12.0                   -13.0   \n",
       "4    Basal-like                        0.0                   -13.0   \n",
       "..          ...                        ...                     ...   \n",
       "509   Luminal B                       -3.0                   -12.0   \n",
       "510   Luminal B                       -3.0                   -12.0   \n",
       "511   Luminal B                       -4.0                    -4.0   \n",
       "512   Luminal B                       -3.0                   -10.0   \n",
       "513   Luminal B                       -3.0                   -10.0   \n",
       "\n",
       "    mirna_clusters methylation_clusters rppa_clusters cn_clusters  \\\n",
       "0              3.0                  5.0         Basal         3.0   \n",
       "1              5.0                  5.0         Basal         1.0   \n",
       "2              5.0                  5.0         Basal         1.0   \n",
       "3              4.0                  4.0         Basal         4.0   \n",
       "4              5.0                  5.0           NaN         1.0   \n",
       "..             ...                  ...           ...         ...   \n",
       "509            4.0                  2.0          LumA         2.0   \n",
       "510            NaN                  2.0         ReacI         2.0   \n",
       "511            6.0                  1.0         ReacI         3.0   \n",
       "512            6.0                  4.0        LumA/B         NaN   \n",
       "513            2.0                  4.0        LumA/B         NaN   \n",
       "\n",
       "     integrated_clusters_(with_pam50)  integrated_clusters_(no_exp)  \\\n",
       "0                                 2.0                           2.0   \n",
       "1                                 2.0                           2.0   \n",
       "2                                 2.0                           2.0   \n",
       "3                                 2.0                           1.0   \n",
       "4                                 NaN                           NaN   \n",
       "..                                ...                           ...   \n",
       "509                               4.0                           3.0   \n",
       "510                               NaN                           NaN   \n",
       "511                               3.0                           3.0   \n",
       "512                               NaN                           NaN   \n",
       "513                               NaN                           NaN   \n",
       "\n",
       "     integrated_clusters_(unsup_exp)  \n",
       "0                                2.0  \n",
       "1                                2.0  \n",
       "2                                2.0  \n",
       "3                                1.0  \n",
       "4                                NaN  \n",
       "..                               ...  \n",
       "509                              3.0  \n",
       "510                              NaN  \n",
       "511                              4.0  \n",
       "512                              NaN  \n",
       "513                              NaN  \n",
       "\n",
       "[514 rows x 30 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the DataFrame of cases of interest from the TCGA-BRCA paper\n",
    "df_paper_cases"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# GDC API Metadata\n",
    "GDC: Genomic Data Commons"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Status"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{\n",
      "    \"commit\": \"4bb408881e6dc67eca93ff9fd913629a8f2d11c2\",\n",
      "    \"data_release\": \"Data Release 42.0 - January 30, 2025\",\n",
      "    \"data_release_version\": {\n",
      "        \"major\": 42,\n",
      "        \"minor\": 0,\n",
      "        \"release_date\": \"2025-01-30\"\n",
      "    },\n",
      "    \"status\": \"OK\",\n",
      "    \"tag\": \"7.8.5\",\n",
      "    \"version\": 1\n",
      "}\n"
     ]
    }
   ],
   "source": [
    "# Request the API version and status to the endpoint\n",
    "response = requests.get(GDC_API_ENDPOINTS['status'])\n",
    "\n",
    "# Print the response content\n",
    "try:\n",
    "    parsed = json.loads(response.content.decode('utf-8'))\n",
    "    print(json.dumps(parsed, indent=4))\n",
    "except json.JSONDecodeError:\n",
    "    print('Response is not valid JSON:')\n",
    "    print(response.content.decode('utf-8'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Projects"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Endpoint Response"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Fields available in the endpoint\n",
    "fields = [\n",
    "    'disease_type',\n",
    "    'name',\n",
    "    'primary_site',\n",
    "    'summary.case_count',\n",
    "    'summary.data_categories.case_count',\n",
    "    'summary.data_categories.data_category',\n",
    "    'summary.data_categories.file_count',\n",
    "    'summary.experimental_strategies.case_count',\n",
    "    'summary.experimental_strategies.experimental_strategy',\n",
    "    'summary.experimental_strategies.file_count',\n",
    "    'summary.file_count',\n",
    "]\n",
    "\n",
    "# Filter to be used in the projects endpoint request\n",
    "filters = {\n",
    "    'op': 'in',\n",
    "    'content': {\n",
    "        'field': 'projects.project_id',\n",
    "        'value': GDC_API_REQUESTS_CONSTANTS['projects']\n",
    "    }\n",
    "}\n",
    "\n",
    "# Get the response from the projects endpoint request\n",
    "projects_response = gdc_api_request(\n",
    "    endpoint=GDC_API_ENDPOINTS['projects'],\n",
    "    fields=fields,\n",
    "    filters=filters,\n",
    "    size=len(GDC_API_REQUESTS_CONSTANTS['projects'])\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Project DataFrame"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create the TCGA-BRCA project DataFrame\n",
    "df_project = pd.json_normalize(projects_response) \\\n",
    "    .rename(\n",
    "        columns={\n",
    "            'id': 'project_id',\n",
    "            'name': 'project_name',\n",
    "            'summary.case_count': 'case_count',\n",
    "            'summary.data_categories': 'data_categories',\n",
    "            'summary.experimental_strategies': 'experimental_strategies',\n",
    "            'summary.file_count': 'file_count',\n",
    "        }\n",
    "    ) \\\n",
    "    [[\n",
    "        'project_id',\n",
    "        'project_name',\n",
    "        'primary_site',\n",
    "        'disease_type',\n",
    "        'case_count',\n",
    "        'file_count',\n",
    "        'experimental_strategies',\n",
    "        'data_categories',\n",
    "    ]]\n",
    "\n",
    "# Store the DataFrame in a CSV file\n",
    "file_name = 'brca-api-project.csv'\n",
    "df_project.to_csv(f'{BRCA_INTERIM_DATA_DIR}/{file_name}', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>project_id</th>\n",
       "      <th>project_name</th>\n",
       "      <th>primary_site</th>\n",
       "      <th>disease_type</th>\n",
       "      <th>case_count</th>\n",
       "      <th>file_count</th>\n",
       "      <th>experimental_strategies</th>\n",
       "      <th>data_categories</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TCGA-BRCA</td>\n",
       "      <td>Breast Invasive Carcinoma</td>\n",
       "      <td>[Breast]</td>\n",
       "      <td>[Epithelial Neoplasms, NOS, Adnexal and Skin Appendage Neoplasms, Squamous Cell Neoplasms, Adenomas and Adenocarcinomas, Complex Epithelial Neoplasms, Fibroepithelial Neoplasms, Cystic, Mucinous and Serous Neoplasms, Basal Cell Neoplasms, Ductal and Lobular Neoplasms]</td>\n",
       "      <td>1098</td>\n",
       "      <td>68962</td>\n",
       "      <td>[{'file_count': 11079, 'case_count': 1095, 'experimental_strategy': 'RNA-Seq'}, {'file_count': 17049, 'case_count': 1072, 'experimental_strategy': 'WXS'}, {'file_count': 3621, 'case_count': 1079, 'experimental_strategy': 'miRNA-Seq'}, {'file_count': 10572, 'case_count': 952, 'experimental_strategy': 'WGS'}, {'file_count': 75, 'case_count': 74, 'experimental_strategy': 'ATAC-Seq'}, {'file_count': 14329, 'case_count': 1098, 'experimental_strategy': 'Genotyping Array'}, {'file_count': 3714, 'case_count': 1097, 'experimental_strategy': 'Methylation Array'}, {'file_count': 919, 'case_count': 881, 'experimental_strategy': 'Reverse Phase Protein Array'}, {'file_count': 1133, 'case_count': 1062, 'experimental_strategy': 'Diagnostic Slide'}, {'file_count': 1978, 'case_count': 1093, 'experimental_strategy': 'Tissue Slide'}]</td>\n",
       "      <td>[{'file_count': 19753, 'case_count': 1098, 'data_category': 'Simple Nucleotide Variation'}, {'file_count': 9282, 'case_count': 1098, 'data_category': 'Sequencing Reads'}, {'file_count': 5316, 'case_count': 1098, 'data_category': 'Biospecimen'}, {'file_count': 2288, 'case_count': 1098, 'data_category': 'Clinical'}, {'file_count': 14346, 'case_count': 1098, 'data_category': 'Copy Number Variation'}, {'file_count': 4876, 'case_count': 1097, 'data_category': 'Transcriptome Profiling'}, {'file_count': 3714, 'case_count': 1097, 'data_category': 'DNA Methylation'}, {'file_count': 919, 'case_count': 881, 'data_category': 'Proteome Profiling'}, {'file_count': 2696, 'case_count': 784, 'data_category': 'Somatic Structural Variation'}, {'file_count': 5772, 'case_count': 1098, 'data_category': 'Structural Variation'}]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  project_id               project_name primary_site  \\\n",
       "0  TCGA-BRCA  Breast Invasive Carcinoma     [Breast]   \n",
       "\n",
       "                                                                                                                                                                                                                                                                   disease_type  \\\n",
       "0  [Epithelial Neoplasms, NOS, Adnexal and Skin Appendage Neoplasms, Squamous Cell Neoplasms, Adenomas and Adenocarcinomas, Complex Epithelial Neoplasms, Fibroepithelial Neoplasms, Cystic, Mucinous and Serous Neoplasms, Basal Cell Neoplasms, Ductal and Lobular Neoplasms]   \n",
       "\n",
       "   case_count  file_count  \\\n",
       "0        1098       68962   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     experimental_strategies  \\\n",
       "0  [{'file_count': 11079, 'case_count': 1095, 'experimental_strategy': 'RNA-Seq'}, {'file_count': 17049, 'case_count': 1072, 'experimental_strategy': 'WXS'}, {'file_count': 3621, 'case_count': 1079, 'experimental_strategy': 'miRNA-Seq'}, {'file_count': 10572, 'case_count': 952, 'experimental_strategy': 'WGS'}, {'file_count': 75, 'case_count': 74, 'experimental_strategy': 'ATAC-Seq'}, {'file_count': 14329, 'case_count': 1098, 'experimental_strategy': 'Genotyping Array'}, {'file_count': 3714, 'case_count': 1097, 'experimental_strategy': 'Methylation Array'}, {'file_count': 919, 'case_count': 881, 'experimental_strategy': 'Reverse Phase Protein Array'}, {'file_count': 1133, 'case_count': 1062, 'experimental_strategy': 'Diagnostic Slide'}, {'file_count': 1978, 'case_count': 1093, 'experimental_strategy': 'Tissue Slide'}]   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    data_categories  \n",
       "0  [{'file_count': 19753, 'case_count': 1098, 'data_category': 'Simple Nucleotide Variation'}, {'file_count': 9282, 'case_count': 1098, 'data_category': 'Sequencing Reads'}, {'file_count': 5316, 'case_count': 1098, 'data_category': 'Biospecimen'}, {'file_count': 2288, 'case_count': 1098, 'data_category': 'Clinical'}, {'file_count': 14346, 'case_count': 1098, 'data_category': 'Copy Number Variation'}, {'file_count': 4876, 'case_count': 1097, 'data_category': 'Transcriptome Profiling'}, {'file_count': 3714, 'case_count': 1097, 'data_category': 'DNA Methylation'}, {'file_count': 919, 'case_count': 881, 'data_category': 'Proteome Profiling'}, {'file_count': 2696, 'case_count': 784, 'data_category': 'Somatic Structural Variation'}, {'file_count': 5772, 'case_count': 1098, 'data_category': 'Structural Variation'}]  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the TCGA-BRCA project DataFrame\n",
    "pd.set_option('display.max_colwidth', 900)\n",
    "df_project"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Cases"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Endpoint Response"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# List with the submitter ids of the cases of interest\n",
    "submitter_ids = df_paper_cases['complete_tcga_id'].to_list()\n",
    "\n",
    "# Fields of interest for the cases endpoint request\n",
    "fields = [\n",
    "    'disease_type',\n",
    "    'files.data_format',\n",
    "    'files.data_type',\n",
    "    'files.file_id',\n",
    "    'files.experimental_strategy',\n",
    "    'submitter_id',\n",
    "]\n",
    "\n",
    "# Filters to be used in the cases endpoint request\n",
    "filters = {\n",
    "    'op': 'and',\n",
    "    'content': [\n",
    "        {\n",
    "            'op': 'in',\n",
    "            'content': {\n",
    "                'field': 'project.project_id',\n",
    "                'value': GDC_API_REQUESTS_CONSTANTS['projects']\n",
    "            }\n",
    "        },\n",
    "        {\n",
    "            'op': 'in',\n",
    "            'content': {\n",
    "                'field': 'disease_type',\n",
    "                'value': GDC_API_REQUESTS_CONSTANTS['disease_types']\n",
    "            }  \n",
    "        },\n",
    "        {\n",
    "            'op': 'in',\n",
    "            'content': {\n",
    "                'field': 'submitter_id',\n",
    "                'value': submitter_ids\n",
    "            }\n",
    "        }\n",
    "    ]\n",
    "}\n",
    "\n",
    "# Get the response from the cases endpoint request\n",
    "cases_response = gdc_api_request(\n",
    "    endpoint=GDC_API_ENDPOINTS['cases'],\n",
    "    fields=fields,\n",
    "    filters=filters,\n",
    "    size=len(submitter_ids)\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Cases of Interest DataFrame"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create the DataFrame of TCGA-BRCA cases of interest with associated files\n",
    "df_api_cases = pd.json_normalize(cases_response) \\\n",
    "    .rename(columns={'id': 'case_id'}) \\\n",
    "    [['case_id', 'submitter_id', 'disease_type', 'files']]\n",
    "\n",
    "# Retrieve the molecular subtype classification of cases\n",
    "df_api_cases = df_api_cases \\\n",
    "    .merge(\n",
    "        right=df_paper_cases,\n",
    "        left_on='submitter_id',\n",
    "        right_on='complete_tcga_id',\n",
    "        how='inner'\n",
    "    ) \\\n",
    "    [['case_id', 'submitter_id', 'disease_type', 'pam50_mrna', 'files']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>case_id</th>\n",
       "      <th>submitter_id</th>\n",
       "      <th>disease_type</th>\n",
       "      <th>pam50_mrna</th>\n",
       "      <th>files</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>de75d0b9-0f47-4732-8df5-05c350cfcd32</td>\n",
       "      <td>TCGA-AN-A0FY</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal B</td>\n",
       "      <td>[{'data_format': 'VCF', 'file_id': '9fe6ffdd-7490-452e-9a68-accf26dda31d', 'data_type': 'Annotated Somatic Mutation', 'experimental_strategy': 'WXS'}, {'data_format': 'BEDPE', 'file_id': 'da517742...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1fef9454-b3e8-4d9d-a9aa-aa1f1a32b80c</td>\n",
       "      <td>TCGA-E2-A15J</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>[{'data_format': 'BCR Biotab', 'file_id': '0a40467f-9495-4c5b-b56e-3347a3ee0572', 'data_type': 'Clinical Supplement'}, {'data_format': 'VCF', 'file_id': '9e8b82c0-9bac-4489-b32e-faf591e01f86', 'da...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2021ed1f-dc75-4701-b8b8-1386466e4802</td>\n",
       "      <td>TCGA-BH-A0H7</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>[{'data_format': 'BAM', 'file_id': 'aba03be3-9b5a-44b3-97a7-d89c37c8b721', 'data_type': 'Aligned Reads', 'experimental_strategy': 'RNA-Seq'}, {'data_format': 'TXT', 'file_id': '70a0fb72-e7ca-4fe5-...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>20e8106b-1290-4735-abe4-7621e08e3dc8</td>\n",
       "      <td>TCGA-BH-A0E0</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Basal-like</td>\n",
       "      <td>[{'data_format': 'VCF', 'file_id': '7dc9bd4d-3ee6-42af-8b44-fdd29978b769', 'data_type': 'Annotated Somatic Mutation', 'experimental_strategy': 'WGS'}, {'data_format': 'TSV', 'file_id': 'ba1d64b0-3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>214a4507-d974-4b3e-8525-7408fccc6a0f</td>\n",
       "      <td>TCGA-BH-A0B7</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>HER2-enriched</td>\n",
       "      <td>[{'data_format': 'BEDPE', 'file_id': 'a1a6f518-7460-42cb-b32e-350b0398cfdc', 'data_type': 'Structural Rearrangement', 'experimental_strategy': 'WGS'}, {'data_format': 'TSV', 'file_id': 'f50fdcd8-7...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>500</th>\n",
       "      <td>33919e92-4ea9-47e5-a6f4-36e51845d50f</td>\n",
       "      <td>TCGA-A8-A09N</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal B</td>\n",
       "      <td>[{'data_format': 'TAR', 'file_id': '15fc9a4b-e7d4-4426-b1db-76c714a0ed5e', 'data_type': 'Intermediate Analysis Archive', 'experimental_strategy': 'WGS'}, {'data_format': 'VCF', 'file_id': '9c9b561...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>501</th>\n",
       "      <td>34ab0cb8-bf19-4cbb-b4eb-3c45ee5aa60b</td>\n",
       "      <td>TCGA-B6-A0IG</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>[{'data_format': 'TXT', 'file_id': '54d189b0-90e9-4770-8983-b9ed9ccb9d7d', 'data_type': 'Allele-specific Copy Number Segment', 'experimental_strategy': 'Genotyping Array'}, {'data_format': 'TAR', ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502</th>\n",
       "      <td>360d1d6d-cf3e-4c78-b36f-ef5e3160aa6b</td>\n",
       "      <td>TCGA-BH-A0AU</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal B</td>\n",
       "      <td>[{'data_format': 'VCF', 'file_id': 'ec73e799-e6d8-44b4-b47c-abfe573c8f39', 'data_type': 'Structural Rearrangement', 'experimental_strategy': 'WGS'}, {'data_format': 'PDF', 'file_id': 'fb2fcbdf-cc6...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>503</th>\n",
       "      <td>2cf68894-168b-458b-af4f-53cad72989a8</td>\n",
       "      <td>TCGA-A7-A13F</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal B</td>\n",
       "      <td>[{'data_format': 'BEDPE', 'file_id': '48327159-a47a-4c5f-8434-3a51edeac2cc', 'data_type': 'Structural Rearrangement', 'experimental_strategy': 'WGS'}, {'data_format': 'TXT', 'file_id': '3acfdb09-8...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>504</th>\n",
       "      <td>36d77496-99f8-4911-ba7c-89fae32662aa</td>\n",
       "      <td>TCGA-A2-A0T5</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>[{'data_format': 'BCR Biotab', 'file_id': '0a40467f-9495-4c5b-b56e-3347a3ee0572', 'data_type': 'Clinical Supplement'}, {'data_format': 'TXT', 'file_id': '40d069b7-738c-48f0-bd13-619ea9a3ff29', 'da...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>505 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                  case_id  submitter_id  \\\n",
       "0    de75d0b9-0f47-4732-8df5-05c350cfcd32  TCGA-AN-A0FY   \n",
       "1    1fef9454-b3e8-4d9d-a9aa-aa1f1a32b80c  TCGA-E2-A15J   \n",
       "2    2021ed1f-dc75-4701-b8b8-1386466e4802  TCGA-BH-A0H7   \n",
       "3    20e8106b-1290-4735-abe4-7621e08e3dc8  TCGA-BH-A0E0   \n",
       "4    214a4507-d974-4b3e-8525-7408fccc6a0f  TCGA-BH-A0B7   \n",
       "..                                    ...           ...   \n",
       "500  33919e92-4ea9-47e5-a6f4-36e51845d50f  TCGA-A8-A09N   \n",
       "501  34ab0cb8-bf19-4cbb-b4eb-3c45ee5aa60b  TCGA-B6-A0IG   \n",
       "502  360d1d6d-cf3e-4c78-b36f-ef5e3160aa6b  TCGA-BH-A0AU   \n",
       "503  2cf68894-168b-458b-af4f-53cad72989a8  TCGA-A7-A13F   \n",
       "504  36d77496-99f8-4911-ba7c-89fae32662aa  TCGA-A2-A0T5   \n",
       "\n",
       "                     disease_type     pam50_mrna  \\\n",
       "0    Ductal and Lobular Neoplasms      Luminal B   \n",
       "1    Ductal and Lobular Neoplasms      Luminal A   \n",
       "2    Ductal and Lobular Neoplasms      Luminal A   \n",
       "3    Ductal and Lobular Neoplasms     Basal-like   \n",
       "4    Ductal and Lobular Neoplasms  HER2-enriched   \n",
       "..                            ...            ...   \n",
       "500  Ductal and Lobular Neoplasms      Luminal B   \n",
       "501  Ductal and Lobular Neoplasms      Luminal A   \n",
       "502  Ductal and Lobular Neoplasms      Luminal B   \n",
       "503  Ductal and Lobular Neoplasms      Luminal B   \n",
       "504  Ductal and Lobular Neoplasms      Luminal A   \n",
       "\n",
       "                                                                                                                                                                                                       files  \n",
       "0    [{'data_format': 'VCF', 'file_id': '9fe6ffdd-7490-452e-9a68-accf26dda31d', 'data_type': 'Annotated Somatic Mutation', 'experimental_strategy': 'WXS'}, {'data_format': 'BEDPE', 'file_id': 'da517742...  \n",
       "1    [{'data_format': 'BCR Biotab', 'file_id': '0a40467f-9495-4c5b-b56e-3347a3ee0572', 'data_type': 'Clinical Supplement'}, {'data_format': 'VCF', 'file_id': '9e8b82c0-9bac-4489-b32e-faf591e01f86', 'da...  \n",
       "2    [{'data_format': 'BAM', 'file_id': 'aba03be3-9b5a-44b3-97a7-d89c37c8b721', 'data_type': 'Aligned Reads', 'experimental_strategy': 'RNA-Seq'}, {'data_format': 'TXT', 'file_id': '70a0fb72-e7ca-4fe5-...  \n",
       "3    [{'data_format': 'VCF', 'file_id': '7dc9bd4d-3ee6-42af-8b44-fdd29978b769', 'data_type': 'Annotated Somatic Mutation', 'experimental_strategy': 'WGS'}, {'data_format': 'TSV', 'file_id': 'ba1d64b0-3...  \n",
       "4    [{'data_format': 'BEDPE', 'file_id': 'a1a6f518-7460-42cb-b32e-350b0398cfdc', 'data_type': 'Structural Rearrangement', 'experimental_strategy': 'WGS'}, {'data_format': 'TSV', 'file_id': 'f50fdcd8-7...  \n",
       "..                                                                                                                                                                                                       ...  \n",
       "500  [{'data_format': 'TAR', 'file_id': '15fc9a4b-e7d4-4426-b1db-76c714a0ed5e', 'data_type': 'Intermediate Analysis Archive', 'experimental_strategy': 'WGS'}, {'data_format': 'VCF', 'file_id': '9c9b561...  \n",
       "501  [{'data_format': 'TXT', 'file_id': '54d189b0-90e9-4770-8983-b9ed9ccb9d7d', 'data_type': 'Allele-specific Copy Number Segment', 'experimental_strategy': 'Genotyping Array'}, {'data_format': 'TAR', ...  \n",
       "502  [{'data_format': 'VCF', 'file_id': 'ec73e799-e6d8-44b4-b47c-abfe573c8f39', 'data_type': 'Structural Rearrangement', 'experimental_strategy': 'WGS'}, {'data_format': 'PDF', 'file_id': 'fb2fcbdf-cc6...  \n",
       "503  [{'data_format': 'BEDPE', 'file_id': '48327159-a47a-4c5f-8434-3a51edeac2cc', 'data_type': 'Structural Rearrangement', 'experimental_strategy': 'WGS'}, {'data_format': 'TXT', 'file_id': '3acfdb09-8...  \n",
       "504  [{'data_format': 'BCR Biotab', 'file_id': '0a40467f-9495-4c5b-b56e-3347a3ee0572', 'data_type': 'Clinical Supplement'}, {'data_format': 'TXT', 'file_id': '40d069b7-738c-48f0-bd13-619ea9a3ff29', 'da...  \n",
       "\n",
       "[505 rows x 5 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the DataFrame of TCGA-BRCA cases of interest with associated files\n",
    "pd.set_option('display.max_colwidth', 200)\n",
    "df_api_cases"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Files"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Expression Quantification Files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Explode lists with dictionaries containing case files metadata \n",
    "df_eq_files = df_api_cases.explode('files')\n",
    "\n",
    "# Filter files related to miRNA-Seq or RNA-Seq experimental strategies\n",
    "key = 'experimental_strategy'\n",
    "df_eq_files = df_eq_files[\n",
    "    df_eq_files['files'].apply(\n",
    "        lambda x: (\n",
    "            key in x and (x[key] == 'miRNA-Seq' or x[key] == 'RNA-Seq')\n",
    "        )\n",
    "    )\n",
    "]\n",
    "\n",
    "# Explode dictionary contents with file metadata into columns\n",
    "df_eq_files = pd.concat(\n",
    "    objs=[\n",
    "        df_eq_files.reset_index(drop=True),\n",
    "        pd.json_normalize(df_eq_files['files'])\n",
    "    ],\n",
    "    axis='columns'\n",
    ")\n",
    "\n",
    "# Filter isoform (miRNA-Seq) and gene (RNA-Seq) expression quantification files\n",
    "df_eq_files = df_eq_files \\\n",
    "    .query(\n",
    "        '(data_type == \"Isoform Expression Quantification\") ' +\n",
    "        'or (data_type == \"Gene Expression Quantification\")'\n",
    "    ) \\\n",
    "    .reset_index(drop=True) \\\n",
    "    [['file_id', 'case_id', 'experimental_strategy', 'data_type', 'data_format']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>file_id</th>\n",
       "      <th>case_id</th>\n",
       "      <th>experimental_strategy</th>\n",
       "      <th>data_type</th>\n",
       "      <th>data_format</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>e9fe2319-2624-4ac1-9c41-d0ac44608f02</td>\n",
       "      <td>de75d0b9-0f47-4732-8df5-05c350cfcd32</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>Gene Expression Quantification</td>\n",
       "      <td>TSV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0a8016e1-0321-4421-a5ad-db2fd9caf7ab</td>\n",
       "      <td>de75d0b9-0f47-4732-8df5-05c350cfcd32</td>\n",
       "      <td>miRNA-Seq</td>\n",
       "      <td>Isoform Expression Quantification</td>\n",
       "      <td>TXT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>8437bf97-2354-40cf-8dc4-631c5d7ae63a</td>\n",
       "      <td>1fef9454-b3e8-4d9d-a9aa-aa1f1a32b80c</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>Gene Expression Quantification</td>\n",
       "      <td>TSV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>7a4675e7-4174-4561-ae9c-b7636fbea718</td>\n",
       "      <td>1fef9454-b3e8-4d9d-a9aa-aa1f1a32b80c</td>\n",
       "      <td>miRNA-Seq</td>\n",
       "      <td>Isoform Expression Quantification</td>\n",
       "      <td>TXT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ea1fadc2-1cdc-4658-9619-eeb26ae09da8</td>\n",
       "      <td>2021ed1f-dc75-4701-b8b8-1386466e4802</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>Gene Expression Quantification</td>\n",
       "      <td>TSV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1130</th>\n",
       "      <td>7c275803-bb6d-4fbc-b851-5fcb393f8496</td>\n",
       "      <td>2cf68894-168b-458b-af4f-53cad72989a8</td>\n",
       "      <td>miRNA-Seq</td>\n",
       "      <td>Isoform Expression Quantification</td>\n",
       "      <td>TXT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1131</th>\n",
       "      <td>9c87ef50-5854-4f69-820b-9e0b4d46c982</td>\n",
       "      <td>2cf68894-168b-458b-af4f-53cad72989a8</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>Gene Expression Quantification</td>\n",
       "      <td>TSV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1132</th>\n",
       "      <td>8ca0e4c1-92ab-4653-87e0-cdcc630d8db6</td>\n",
       "      <td>2cf68894-168b-458b-af4f-53cad72989a8</td>\n",
       "      <td>miRNA-Seq</td>\n",
       "      <td>Isoform Expression Quantification</td>\n",
       "      <td>TXT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1133</th>\n",
       "      <td>d5d23d6a-c141-442f-ac4f-714e46b8d9ef</td>\n",
       "      <td>36d77496-99f8-4911-ba7c-89fae32662aa</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>Gene Expression Quantification</td>\n",
       "      <td>TSV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1134</th>\n",
       "      <td>1e4fc580-3846-4ece-a88d-55d0d5e2ac2f</td>\n",
       "      <td>36d77496-99f8-4911-ba7c-89fae32662aa</td>\n",
       "      <td>miRNA-Seq</td>\n",
       "      <td>Isoform Expression Quantification</td>\n",
       "      <td>TXT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1135 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   file_id  \\\n",
       "0     e9fe2319-2624-4ac1-9c41-d0ac44608f02   \n",
       "1     0a8016e1-0321-4421-a5ad-db2fd9caf7ab   \n",
       "2     8437bf97-2354-40cf-8dc4-631c5d7ae63a   \n",
       "3     7a4675e7-4174-4561-ae9c-b7636fbea718   \n",
       "4     ea1fadc2-1cdc-4658-9619-eeb26ae09da8   \n",
       "...                                    ...   \n",
       "1130  7c275803-bb6d-4fbc-b851-5fcb393f8496   \n",
       "1131  9c87ef50-5854-4f69-820b-9e0b4d46c982   \n",
       "1132  8ca0e4c1-92ab-4653-87e0-cdcc630d8db6   \n",
       "1133  d5d23d6a-c141-442f-ac4f-714e46b8d9ef   \n",
       "1134  1e4fc580-3846-4ece-a88d-55d0d5e2ac2f   \n",
       "\n",
       "                                   case_id experimental_strategy  \\\n",
       "0     de75d0b9-0f47-4732-8df5-05c350cfcd32               RNA-Seq   \n",
       "1     de75d0b9-0f47-4732-8df5-05c350cfcd32             miRNA-Seq   \n",
       "2     1fef9454-b3e8-4d9d-a9aa-aa1f1a32b80c               RNA-Seq   \n",
       "3     1fef9454-b3e8-4d9d-a9aa-aa1f1a32b80c             miRNA-Seq   \n",
       "4     2021ed1f-dc75-4701-b8b8-1386466e4802               RNA-Seq   \n",
       "...                                    ...                   ...   \n",
       "1130  2cf68894-168b-458b-af4f-53cad72989a8             miRNA-Seq   \n",
       "1131  2cf68894-168b-458b-af4f-53cad72989a8               RNA-Seq   \n",
       "1132  2cf68894-168b-458b-af4f-53cad72989a8             miRNA-Seq   \n",
       "1133  36d77496-99f8-4911-ba7c-89fae32662aa               RNA-Seq   \n",
       "1134  36d77496-99f8-4911-ba7c-89fae32662aa             miRNA-Seq   \n",
       "\n",
       "                              data_type data_format  \n",
       "0        Gene Expression Quantification         TSV  \n",
       "1     Isoform Expression Quantification         TXT  \n",
       "2        Gene Expression Quantification         TSV  \n",
       "3     Isoform Expression Quantification         TXT  \n",
       "4        Gene Expression Quantification         TSV  \n",
       "...                                 ...         ...  \n",
       "1130  Isoform Expression Quantification         TXT  \n",
       "1131     Gene Expression Quantification         TSV  \n",
       "1132  Isoform Expression Quantification         TXT  \n",
       "1133     Gene Expression Quantification         TSV  \n",
       "1134  Isoform Expression Quantification         TXT  \n",
       "\n",
       "[1135 rows x 5 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the DataFrame of TCGA-BRCA expression quantification files\n",
    "df_eq_files"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Samples Related to Files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "# List with the UUIDs of the files of interest\n",
    "file_ids = df_eq_files['file_id'].to_list()\n",
    "\n",
    "# Fields of interest for the files endpoint request\n",
    "fields = ['cases.samples.tissue_type', 'cases.samples.sample_type']\n",
    "\n",
    "# Filters to be used in the files endpoint request\n",
    "filters = {\n",
    "    'op': 'and',\n",
    "    'content': [\n",
    "        {\n",
    "            'op': 'in',\n",
    "            'content': {\n",
    "                'field': 'cases.samples.sample_type',\n",
    "                'value': GDC_API_REQUESTS_CONSTANTS['sample_types']\n",
    "            }\n",
    "        },\n",
    "        {\n",
    "            'op': 'in',\n",
    "            'content': {\n",
    "                'field': 'file_id',\n",
    "                'value': file_ids\n",
    "            }\n",
    "        }\n",
    "    ]\n",
    "}\n",
    "\n",
    "# Get the response from the files endpoint request\n",
    "files_response = gdc_api_request(\n",
    "    endpoint=GDC_API_ENDPOINTS['files'],\n",
    "    fields=fields,\n",
    "    filters=filters,\n",
    "    size=len(file_ids)\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create the DataFrame of samples related to files\n",
    "df_files_samples = pd.json_normalize(files_response)\n",
    "\n",
    "# Explode the lists of dictionaries with samples metadata\n",
    "df_samples = pd.json_normalize(\n",
    "    pd.json_normalize(\n",
    "        pd.json_normalize(\n",
    "            df_files_samples.explode('cases')['cases']\n",
    "        )['samples']\n",
    "    )[0]\n",
    ")\n",
    "\n",
    "# Concatenate exploded metadata from samples to the files UUIDs\n",
    "df_files_samples = pd.concat(\n",
    "    objs=[df_files_samples, df_samples],\n",
    "    axis='columns'\n",
    ")\n",
    "\n",
    "# Rearrange the columns of the DataFrame\n",
    "df_files_samples = df_files_samples \\\n",
    "    .rename(columns={'id': 'file_id'}) \\\n",
    "    [['file_id', 'tissue_type', 'sample_type']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>file_id</th>\n",
       "      <th>tissue_type</th>\n",
       "      <th>sample_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>dd0e32d6-4704-4abd-82e4-2041fbc19a79</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Primary Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>09a491b5-1bfb-47d9-963f-69f6c7e6c1e5</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Primary Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>f8e5691a-e24d-4dba-873e-9ffa3ca12bb6</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Primary Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>65654739-bed4-4ea7-b75c-ca1f278adcc9</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Primary Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4af55932-ab7a-41e5-aa7f-fb16ab18c629</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Primary Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1120</th>\n",
       "      <td>ed716c0b-4b09-4902-86b8-6955488d60c3</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Primary Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1121</th>\n",
       "      <td>0bc59ba8-158b-4309-b4d5-bb541f892fbf</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Primary Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1122</th>\n",
       "      <td>bc23716d-70e8-486e-b512-8c45654d5d83</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Primary Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1123</th>\n",
       "      <td>8ecef704-ed0f-4700-a34e-46a174033332</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Primary Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1124</th>\n",
       "      <td>79f97f73-b650-49cf-8654-53038a6c0de7</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Primary Tumor</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1125 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   file_id tissue_type    sample_type\n",
       "0     dd0e32d6-4704-4abd-82e4-2041fbc19a79       Tumor  Primary Tumor\n",
       "1     09a491b5-1bfb-47d9-963f-69f6c7e6c1e5       Tumor  Primary Tumor\n",
       "2     f8e5691a-e24d-4dba-873e-9ffa3ca12bb6       Tumor  Primary Tumor\n",
       "3     65654739-bed4-4ea7-b75c-ca1f278adcc9       Tumor  Primary Tumor\n",
       "4     4af55932-ab7a-41e5-aa7f-fb16ab18c629       Tumor  Primary Tumor\n",
       "...                                    ...         ...            ...\n",
       "1120  ed716c0b-4b09-4902-86b8-6955488d60c3       Tumor  Primary Tumor\n",
       "1121  0bc59ba8-158b-4309-b4d5-bb541f892fbf       Tumor  Primary Tumor\n",
       "1122  bc23716d-70e8-486e-b512-8c45654d5d83       Tumor  Primary Tumor\n",
       "1123  8ecef704-ed0f-4700-a34e-46a174033332       Tumor  Primary Tumor\n",
       "1124  79f97f73-b650-49cf-8654-53038a6c0de7       Tumor  Primary Tumor\n",
       "\n",
       "[1125 rows x 3 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the DataFrame of samples related to files\n",
    "df_files_samples"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Files of Interest DataFrame"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create the DataFrame of TCGA-BRCA files of interest with associated samples\n",
    "df_files = df_eq_files \\\n",
    "    .merge(\n",
    "        right=df_files_samples,\n",
    "        left_on='file_id',\n",
    "        right_on='file_id',\n",
    "        how='inner'\n",
    "    ) \\\n",
    "    [[\n",
    "        'file_id',\n",
    "        'case_id',\n",
    "        'tissue_type',\n",
    "        'sample_type',\n",
    "        'experimental_strategy',\n",
    "        'data_type',\n",
    "        'data_format',\n",
    "    ]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>file_id</th>\n",
       "      <th>case_id</th>\n",
       "      <th>tissue_type</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>experimental_strategy</th>\n",
       "      <th>data_type</th>\n",
       "      <th>data_format</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>e9fe2319-2624-4ac1-9c41-d0ac44608f02</td>\n",
       "      <td>de75d0b9-0f47-4732-8df5-05c350cfcd32</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Primary Tumor</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>Gene Expression Quantification</td>\n",
       "      <td>TSV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0a8016e1-0321-4421-a5ad-db2fd9caf7ab</td>\n",
       "      <td>de75d0b9-0f47-4732-8df5-05c350cfcd32</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Primary Tumor</td>\n",
       "      <td>miRNA-Seq</td>\n",
       "      <td>Isoform Expression Quantification</td>\n",
       "      <td>TXT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>8437bf97-2354-40cf-8dc4-631c5d7ae63a</td>\n",
       "      <td>1fef9454-b3e8-4d9d-a9aa-aa1f1a32b80c</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Primary Tumor</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>Gene Expression Quantification</td>\n",
       "      <td>TSV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>7a4675e7-4174-4561-ae9c-b7636fbea718</td>\n",
       "      <td>1fef9454-b3e8-4d9d-a9aa-aa1f1a32b80c</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Primary Tumor</td>\n",
       "      <td>miRNA-Seq</td>\n",
       "      <td>Isoform Expression Quantification</td>\n",
       "      <td>TXT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ea1fadc2-1cdc-4658-9619-eeb26ae09da8</td>\n",
       "      <td>2021ed1f-dc75-4701-b8b8-1386466e4802</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Solid Tissue Normal</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>Gene Expression Quantification</td>\n",
       "      <td>TSV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1120</th>\n",
       "      <td>7c275803-bb6d-4fbc-b851-5fcb393f8496</td>\n",
       "      <td>2cf68894-168b-458b-af4f-53cad72989a8</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Primary Tumor</td>\n",
       "      <td>miRNA-Seq</td>\n",
       "      <td>Isoform Expression Quantification</td>\n",
       "      <td>TXT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1121</th>\n",
       "      <td>9c87ef50-5854-4f69-820b-9e0b4d46c982</td>\n",
       "      <td>2cf68894-168b-458b-af4f-53cad72989a8</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Primary Tumor</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>Gene Expression Quantification</td>\n",
       "      <td>TSV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1122</th>\n",
       "      <td>8ca0e4c1-92ab-4653-87e0-cdcc630d8db6</td>\n",
       "      <td>2cf68894-168b-458b-af4f-53cad72989a8</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Solid Tissue Normal</td>\n",
       "      <td>miRNA-Seq</td>\n",
       "      <td>Isoform Expression Quantification</td>\n",
       "      <td>TXT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1123</th>\n",
       "      <td>d5d23d6a-c141-442f-ac4f-714e46b8d9ef</td>\n",
       "      <td>36d77496-99f8-4911-ba7c-89fae32662aa</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Primary Tumor</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>Gene Expression Quantification</td>\n",
       "      <td>TSV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1124</th>\n",
       "      <td>1e4fc580-3846-4ece-a88d-55d0d5e2ac2f</td>\n",
       "      <td>36d77496-99f8-4911-ba7c-89fae32662aa</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Primary Tumor</td>\n",
       "      <td>miRNA-Seq</td>\n",
       "      <td>Isoform Expression Quantification</td>\n",
       "      <td>TXT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1125 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   file_id  \\\n",
       "0     e9fe2319-2624-4ac1-9c41-d0ac44608f02   \n",
       "1     0a8016e1-0321-4421-a5ad-db2fd9caf7ab   \n",
       "2     8437bf97-2354-40cf-8dc4-631c5d7ae63a   \n",
       "3     7a4675e7-4174-4561-ae9c-b7636fbea718   \n",
       "4     ea1fadc2-1cdc-4658-9619-eeb26ae09da8   \n",
       "...                                    ...   \n",
       "1120  7c275803-bb6d-4fbc-b851-5fcb393f8496   \n",
       "1121  9c87ef50-5854-4f69-820b-9e0b4d46c982   \n",
       "1122  8ca0e4c1-92ab-4653-87e0-cdcc630d8db6   \n",
       "1123  d5d23d6a-c141-442f-ac4f-714e46b8d9ef   \n",
       "1124  1e4fc580-3846-4ece-a88d-55d0d5e2ac2f   \n",
       "\n",
       "                                   case_id tissue_type          sample_type  \\\n",
       "0     de75d0b9-0f47-4732-8df5-05c350cfcd32       Tumor        Primary Tumor   \n",
       "1     de75d0b9-0f47-4732-8df5-05c350cfcd32       Tumor        Primary Tumor   \n",
       "2     1fef9454-b3e8-4d9d-a9aa-aa1f1a32b80c       Tumor        Primary Tumor   \n",
       "3     1fef9454-b3e8-4d9d-a9aa-aa1f1a32b80c       Tumor        Primary Tumor   \n",
       "4     2021ed1f-dc75-4701-b8b8-1386466e4802      Normal  Solid Tissue Normal   \n",
       "...                                    ...         ...                  ...   \n",
       "1120  2cf68894-168b-458b-af4f-53cad72989a8       Tumor        Primary Tumor   \n",
       "1121  2cf68894-168b-458b-af4f-53cad72989a8       Tumor        Primary Tumor   \n",
       "1122  2cf68894-168b-458b-af4f-53cad72989a8      Normal  Solid Tissue Normal   \n",
       "1123  36d77496-99f8-4911-ba7c-89fae32662aa       Tumor        Primary Tumor   \n",
       "1124  36d77496-99f8-4911-ba7c-89fae32662aa       Tumor        Primary Tumor   \n",
       "\n",
       "     experimental_strategy                          data_type data_format  \n",
       "0                  RNA-Seq     Gene Expression Quantification         TSV  \n",
       "1                miRNA-Seq  Isoform Expression Quantification         TXT  \n",
       "2                  RNA-Seq     Gene Expression Quantification         TSV  \n",
       "3                miRNA-Seq  Isoform Expression Quantification         TXT  \n",
       "4                  RNA-Seq     Gene Expression Quantification         TSV  \n",
       "...                    ...                                ...         ...  \n",
       "1120             miRNA-Seq  Isoform Expression Quantification         TXT  \n",
       "1121               RNA-Seq     Gene Expression Quantification         TSV  \n",
       "1122             miRNA-Seq  Isoform Expression Quantification         TXT  \n",
       "1123               RNA-Seq     Gene Expression Quantification         TSV  \n",
       "1124             miRNA-Seq  Isoform Expression Quantification         TXT  \n",
       "\n",
       "[1125 rows x 7 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the DataFrame of TCGA-BRCA files of interest with associated samples\n",
    "df_files"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Filtering Cases and Files"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Counting Files per Cases"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create an acronym for file types\n",
    "df_files['file_type'] = (\n",
    "    df_files['experimental_strategy'] + '_' + df_files['tissue_type']\n",
    ")\n",
    "\n",
    "# Copy some columns of the files DataFrame\n",
    "df_file_count = df_files[['case_id', 'file_id', 'file_type']].copy()\n",
    "\n",
    "# Create numeric flags for the file types\n",
    "df_file_count['TTmiR'] = np.where(\n",
    "    df_file_count['file_type'] == 'miRNA-Seq_Tumor', 1, 0\n",
    ")\n",
    "df_file_count['TTmRNA'] = np.where(\n",
    "    df_file_count['file_type'] == 'RNA-Seq_Tumor', 1, 0\n",
    ")\n",
    "df_file_count['NTmiR'] = np.where(\n",
    "    df_file_count['file_type'] == 'miRNA-Seq_Normal', 1, 0\n",
    ")\n",
    "df_file_count['NTmRNA'] = np.where(\n",
    "    df_file_count['file_type'] == 'RNA-Seq_Normal', 1, 0\n",
    ")\n",
    "\n",
    "# Sum all file types count columns\n",
    "df_file_count['file_count'] = (\n",
    "    df_file_count['TTmiR'] + df_file_count['TTmRNA'] \n",
    "    + df_file_count['NTmiR'] + df_file_count['NTmRNA']\n",
    ")\n",
    "\n",
    "# Count the total number of file types per case\n",
    "df_file_count_agg = df_file_count \\\n",
    "    .groupby('case_id') \\\n",
    "    .agg(\n",
    "        TTmiR_count=pd.NamedAgg(column='TTmiR', aggfunc='sum'),\n",
    "        TTmRNA_count=pd.NamedAgg(column='TTmRNA', aggfunc='sum'),\n",
    "        NTmiR_count=pd.NamedAgg(column='NTmiR', aggfunc='sum'),\n",
    "        NTmRNA_count=pd.NamedAgg(column='NTmRNA', aggfunc='sum'),\n",
    "        file_count=pd.NamedAgg(column='file_count', aggfunc='sum')\n",
    "    ) \\\n",
    "    .sort_values(\n",
    "        by=['file_count', 'TTmiR_count', 'TTmRNA_count', 'NTmiR_count', 'NTmRNA_count'],\n",
    "        ascending=False\n",
    "    )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>case_id</th>\n",
       "      <th>TTmiR_count</th>\n",
       "      <th>TTmRNA_count</th>\n",
       "      <th>NTmiR_count</th>\n",
       "      <th>NTmRNA_count</th>\n",
       "      <th>file_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>8c7e74e0-71ef-49b8-9217-94b8ef740ef9</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>f130f376-5801-40f9-975d-a7e2f7b5670d</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>8785012f-f73e-4d68-87cf-1d804af32782</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0bc5744c-5fa3-45bb-87d0-70a02068b392</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>14267783-5624-4fe5-ba81-9d67f1017474</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499</th>\n",
       "      <td>cc348a26-ee11-47a4-8b51-de922967e175</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>500</th>\n",
       "      <td>dfe6db17-cf45-488e-bd3c-b8433d7343ca</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>501</th>\n",
       "      <td>e3c336f5-c32f-4c5d-81fb-e2408ae145b2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502</th>\n",
       "      <td>ec6930e5-2d83-435a-9a57-a2c5c22caf02</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>503</th>\n",
       "      <td>f947d67f-4aa2-4eb9-977e-21a5ae563613</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>504 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                  case_id  TTmiR_count  TTmRNA_count  \\\n",
       "0    8c7e74e0-71ef-49b8-9217-94b8ef740ef9            3             3   \n",
       "1    f130f376-5801-40f9-975d-a7e2f7b5670d            3             3   \n",
       "2    8785012f-f73e-4d68-87cf-1d804af32782            3             3   \n",
       "3    0bc5744c-5fa3-45bb-87d0-70a02068b392            1             1   \n",
       "4    14267783-5624-4fe5-ba81-9d67f1017474            1             1   \n",
       "..                                    ...          ...           ...   \n",
       "499  cc348a26-ee11-47a4-8b51-de922967e175            0             1   \n",
       "500  dfe6db17-cf45-488e-bd3c-b8433d7343ca            0             1   \n",
       "501  e3c336f5-c32f-4c5d-81fb-e2408ae145b2            0             1   \n",
       "502  ec6930e5-2d83-435a-9a57-a2c5c22caf02            0             1   \n",
       "503  f947d67f-4aa2-4eb9-977e-21a5ae563613            0             1   \n",
       "\n",
       "     NTmiR_count  NTmRNA_count  file_count  \n",
       "0              1             1           8  \n",
       "1              1             1           8  \n",
       "2              0             0           6  \n",
       "3              1             1           4  \n",
       "4              1             1           4  \n",
       "..           ...           ...         ...  \n",
       "499            0             0           1  \n",
       "500            0             0           1  \n",
       "501            0             0           1  \n",
       "502            0             0           1  \n",
       "503            0             0           1  \n",
       "\n",
       "[504 rows x 6 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the DataFrame that aggregates the file types per case\n",
    "df_file_count_agg.reset_index()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Cases of Interest DataFrames"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Cases with Tumor Tissue Analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Filter cases by files associated with tumor tissue\n",
    "df_tumor_cases = df_file_count_agg \\\n",
    "    .query('(TTmiR_count == 1) and (TTmRNA_count == 1)') \\\n",
    "    .reset_index()\n",
    "\n",
    "# Create the DataFrame of cases of interest with tumor tissue analysis\n",
    "df_tumor_cases = df_tumor_cases \\\n",
    "    .merge(\n",
    "        right=df_api_cases,\n",
    "        left_on='case_id',\n",
    "        right_on='case_id',\n",
    "        how='inner'\n",
    "    ) \\\n",
    "    [['case_id', 'submitter_id', 'disease_type', 'pam50_mrna', 'files']]\n",
    "\n",
    "# Store the DataFrame in a CSV file\n",
    "file_name = 'brca-api-cases-tumor-analysis.csv'\n",
    "df_tumor_cases.to_csv(f'{BRCA_INTERIM_DATA_DIR}/{file_name}', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>case_id</th>\n",
       "      <th>submitter_id</th>\n",
       "      <th>disease_type</th>\n",
       "      <th>pam50_mrna</th>\n",
       "      <th>files</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0bc5744c-5fa3-45bb-87d0-70a02068b392</td>\n",
       "      <td>TCGA-BH-A0DG</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>[{'data_format': 'BEDPE', 'file_id': 'f1c01ec0-cf13-4009-a4a1-23665580ab6e', 'data_type': 'Structural Rearrangement', 'experimental_strategy': 'WGS'}, {'data_format': 'BEDPE', 'file_id': '50e7aa21...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14267783-5624-4fe5-ba81-9d67f1017474</td>\n",
       "      <td>TCGA-BH-A0DP</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>[{'data_format': 'BCR Biotab', 'file_id': '0a40467f-9495-4c5b-b56e-3347a3ee0572', 'data_type': 'Clinical Supplement'}, {'data_format': 'BAM', 'file_id': '15a12710-1f2b-41aa-8cf7-054b67f29fbe', 'da...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>17baef7c-d97d-4b98-ab53-503ef856523d</td>\n",
       "      <td>TCGA-A7-A0D9</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>[{'data_format': 'BAM', 'file_id': '8c151dbe-9b21-4fdc-8664-a610ef423606', 'data_type': 'Aligned Reads', 'experimental_strategy': 'WGS'}, {'data_format': 'BEDPE', 'file_id': 'cc14571e-62b5-4f0a-84...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>17c1d42c-cb84-4655-a4cd-b54bae17ecaf</td>\n",
       "      <td>TCGA-E2-A15K</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal B</td>\n",
       "      <td>[{'data_format': 'TXT', 'file_id': 'db86d78c-d6e7-4c62-8942-91b839270b2b', 'data_type': 'Copy Number Segment', 'experimental_strategy': 'Genotyping Array'}, {'data_format': 'SVS', 'file_id': '96ce...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>18eb4dfc-556f-4bf3-a411-4780209ed1e0</td>\n",
       "      <td>TCGA-BH-A0B3</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Basal-like</td>\n",
       "      <td>[{'data_format': 'VCF', 'file_id': '9c9e9481-617b-4ecb-8416-07d62453fa8b', 'data_type': 'Structural Rearrangement', 'experimental_strategy': 'WGS'}, {'data_format': 'TXT', 'file_id': '19259452-1a7...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>481</th>\n",
       "      <td>fb356957-f620-4d99-b675-1355a1d9f932</td>\n",
       "      <td>TCGA-E2-A14O</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal B</td>\n",
       "      <td>[{'data_format': 'SVS', 'file_id': '7846cf1d-aab1-4e14-9f18-f5f7ca4a52bc', 'data_type': 'Slide Image', 'experimental_strategy': 'Tissue Slide'}, {'data_format': 'TXT', 'file_id': '6b9417c3-399d-48...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>482</th>\n",
       "      <td>fb57c598-b45b-478a-9d34-ca1f050d5b7c</td>\n",
       "      <td>TCGA-BH-A0DS</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>[{'data_format': 'SVS', 'file_id': 'f8f16824-4f10-4b71-a8fc-0a2c3176c16a', 'data_type': 'Slide Image', 'experimental_strategy': 'Tissue Slide'}, {'data_format': 'TSV', 'file_id': '0dc31470-2ad5-4e...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>483</th>\n",
       "      <td>fc18d029-9be2-4fa0-9aef-6d647dc55f0b</td>\n",
       "      <td>TCGA-D8-A147</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Basal-like</td>\n",
       "      <td>[{'data_format': 'TXT', 'file_id': '7da02dc3-caf0-4809-99f0-8f5cd0ebff0b', 'data_type': 'Allele-specific Copy Number Segment', 'experimental_strategy': 'Genotyping Array'}, {'data_format': 'TSV', ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>484</th>\n",
       "      <td>fcef8cb5-fb2c-4bfb-82cd-6b9f3145182c</td>\n",
       "      <td>TCGA-BH-A0HP</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>[{'data_format': 'VCF', 'file_id': '64c1118e-5ac4-46e1-9475-7499d19a894b', 'data_type': 'Annotated Somatic Mutation', 'experimental_strategy': 'WXS'}, {'data_format': 'SVS', 'file_id': '3fd44a43-5...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>485</th>\n",
       "      <td>fe15c48b-116d-461e-9b40-68514730fd18</td>\n",
       "      <td>TCGA-A2-A0CP</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>[{'data_format': 'TXT', 'file_id': '0cb8ce0b-9920-4b0f-962b-68b102f9e6a7', 'data_type': 'Methylation Beta Value', 'experimental_strategy': 'Methylation Array'}, {'data_format': 'TXT', 'file_id': '...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>486 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                  case_id  submitter_id  \\\n",
       "0    0bc5744c-5fa3-45bb-87d0-70a02068b392  TCGA-BH-A0DG   \n",
       "1    14267783-5624-4fe5-ba81-9d67f1017474  TCGA-BH-A0DP   \n",
       "2    17baef7c-d97d-4b98-ab53-503ef856523d  TCGA-A7-A0D9   \n",
       "3    17c1d42c-cb84-4655-a4cd-b54bae17ecaf  TCGA-E2-A15K   \n",
       "4    18eb4dfc-556f-4bf3-a411-4780209ed1e0  TCGA-BH-A0B3   \n",
       "..                                    ...           ...   \n",
       "481  fb356957-f620-4d99-b675-1355a1d9f932  TCGA-E2-A14O   \n",
       "482  fb57c598-b45b-478a-9d34-ca1f050d5b7c  TCGA-BH-A0DS   \n",
       "483  fc18d029-9be2-4fa0-9aef-6d647dc55f0b  TCGA-D8-A147   \n",
       "484  fcef8cb5-fb2c-4bfb-82cd-6b9f3145182c  TCGA-BH-A0HP   \n",
       "485  fe15c48b-116d-461e-9b40-68514730fd18  TCGA-A2-A0CP   \n",
       "\n",
       "                     disease_type  pam50_mrna  \\\n",
       "0    Ductal and Lobular Neoplasms   Luminal A   \n",
       "1    Ductal and Lobular Neoplasms   Luminal A   \n",
       "2    Ductal and Lobular Neoplasms   Luminal A   \n",
       "3    Ductal and Lobular Neoplasms   Luminal B   \n",
       "4    Ductal and Lobular Neoplasms  Basal-like   \n",
       "..                            ...         ...   \n",
       "481  Ductal and Lobular Neoplasms   Luminal B   \n",
       "482  Ductal and Lobular Neoplasms   Luminal A   \n",
       "483  Ductal and Lobular Neoplasms  Basal-like   \n",
       "484  Ductal and Lobular Neoplasms   Luminal A   \n",
       "485  Ductal and Lobular Neoplasms   Luminal A   \n",
       "\n",
       "                                                                                                                                                                                                       files  \n",
       "0    [{'data_format': 'BEDPE', 'file_id': 'f1c01ec0-cf13-4009-a4a1-23665580ab6e', 'data_type': 'Structural Rearrangement', 'experimental_strategy': 'WGS'}, {'data_format': 'BEDPE', 'file_id': '50e7aa21...  \n",
       "1    [{'data_format': 'BCR Biotab', 'file_id': '0a40467f-9495-4c5b-b56e-3347a3ee0572', 'data_type': 'Clinical Supplement'}, {'data_format': 'BAM', 'file_id': '15a12710-1f2b-41aa-8cf7-054b67f29fbe', 'da...  \n",
       "2    [{'data_format': 'BAM', 'file_id': '8c151dbe-9b21-4fdc-8664-a610ef423606', 'data_type': 'Aligned Reads', 'experimental_strategy': 'WGS'}, {'data_format': 'BEDPE', 'file_id': 'cc14571e-62b5-4f0a-84...  \n",
       "3    [{'data_format': 'TXT', 'file_id': 'db86d78c-d6e7-4c62-8942-91b839270b2b', 'data_type': 'Copy Number Segment', 'experimental_strategy': 'Genotyping Array'}, {'data_format': 'SVS', 'file_id': '96ce...  \n",
       "4    [{'data_format': 'VCF', 'file_id': '9c9e9481-617b-4ecb-8416-07d62453fa8b', 'data_type': 'Structural Rearrangement', 'experimental_strategy': 'WGS'}, {'data_format': 'TXT', 'file_id': '19259452-1a7...  \n",
       "..                                                                                                                                                                                                       ...  \n",
       "481  [{'data_format': 'SVS', 'file_id': '7846cf1d-aab1-4e14-9f18-f5f7ca4a52bc', 'data_type': 'Slide Image', 'experimental_strategy': 'Tissue Slide'}, {'data_format': 'TXT', 'file_id': '6b9417c3-399d-48...  \n",
       "482  [{'data_format': 'SVS', 'file_id': 'f8f16824-4f10-4b71-a8fc-0a2c3176c16a', 'data_type': 'Slide Image', 'experimental_strategy': 'Tissue Slide'}, {'data_format': 'TSV', 'file_id': '0dc31470-2ad5-4e...  \n",
       "483  [{'data_format': 'TXT', 'file_id': '7da02dc3-caf0-4809-99f0-8f5cd0ebff0b', 'data_type': 'Allele-specific Copy Number Segment', 'experimental_strategy': 'Genotyping Array'}, {'data_format': 'TSV', ...  \n",
       "484  [{'data_format': 'VCF', 'file_id': '64c1118e-5ac4-46e1-9475-7499d19a894b', 'data_type': 'Annotated Somatic Mutation', 'experimental_strategy': 'WXS'}, {'data_format': 'SVS', 'file_id': '3fd44a43-5...  \n",
       "485  [{'data_format': 'TXT', 'file_id': '0cb8ce0b-9920-4b0f-962b-68b102f9e6a7', 'data_type': 'Methylation Beta Value', 'experimental_strategy': 'Methylation Array'}, {'data_format': 'TXT', 'file_id': '...  \n",
       "\n",
       "[486 rows x 5 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the DataFrame of cases of interest with tumor tissue analysis\n",
    "df_tumor_cases"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Cases with Normal Tissue Analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Filter cases by files associated with normal tissue\n",
    "df_normal_cases = df_file_count_agg \\\n",
    "    .query('(NTmiR_count == 1) and (NTmRNA_count == 1)') \\\n",
    "    .reset_index()\n",
    "\n",
    "# Create the DataFrame of cases of interest with normal tissue analysis\n",
    "df_normal_cases = df_normal_cases \\\n",
    "    .merge(\n",
    "        right=df_api_cases,\n",
    "        left_on='case_id',\n",
    "        right_on='case_id',\n",
    "        how='inner'\n",
    "    ) \\\n",
    "    [['case_id', 'submitter_id', 'disease_type', 'pam50_mrna', 'files']]\n",
    "\n",
    "# Store the DataFrame in a CSV file\n",
    "file_name = 'brca-api-cases-normal-analysis.csv'\n",
    "df_normal_cases.to_csv(f'{BRCA_INTERIM_DATA_DIR}/{file_name}', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>case_id</th>\n",
       "      <th>submitter_id</th>\n",
       "      <th>disease_type</th>\n",
       "      <th>pam50_mrna</th>\n",
       "      <th>files</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>8c7e74e0-71ef-49b8-9217-94b8ef740ef9</td>\n",
       "      <td>TCGA-A7-A13E</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Basal-like</td>\n",
       "      <td>[{'data_format': 'TSV', 'file_id': '767bba1d-e80f-49fa-82b1-075a882d0a10', 'data_type': 'Simple Germline Variation', 'experimental_strategy': 'Genotyping Array'}, {'data_format': 'MAF', 'file_id':...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>f130f376-5801-40f9-975d-a7e2f7b5670d</td>\n",
       "      <td>TCGA-A7-A0DB</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>[{'data_format': 'VCF', 'file_id': '886bfc02-b727-47dd-8806-5a34714f8857', 'data_type': 'Annotated Somatic Mutation', 'experimental_strategy': 'WXS'}, {'data_format': 'VCF', 'file_id': 'ef67d03e-b...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0bc5744c-5fa3-45bb-87d0-70a02068b392</td>\n",
       "      <td>TCGA-BH-A0DG</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>[{'data_format': 'BEDPE', 'file_id': 'f1c01ec0-cf13-4009-a4a1-23665580ab6e', 'data_type': 'Structural Rearrangement', 'experimental_strategy': 'WGS'}, {'data_format': 'BEDPE', 'file_id': '50e7aa21...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>14267783-5624-4fe5-ba81-9d67f1017474</td>\n",
       "      <td>TCGA-BH-A0DP</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>[{'data_format': 'BCR Biotab', 'file_id': '0a40467f-9495-4c5b-b56e-3347a3ee0572', 'data_type': 'Clinical Supplement'}, {'data_format': 'BAM', 'file_id': '15a12710-1f2b-41aa-8cf7-054b67f29fbe', 'da...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>17baef7c-d97d-4b98-ab53-503ef856523d</td>\n",
       "      <td>TCGA-A7-A0D9</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>[{'data_format': 'BAM', 'file_id': '8c151dbe-9b21-4fdc-8664-a610ef423606', 'data_type': 'Aligned Reads', 'experimental_strategy': 'WGS'}, {'data_format': 'BEDPE', 'file_id': 'cc14571e-62b5-4f0a-84...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>17c1d42c-cb84-4655-a4cd-b54bae17ecaf</td>\n",
       "      <td>TCGA-E2-A15K</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal B</td>\n",
       "      <td>[{'data_format': 'TXT', 'file_id': 'db86d78c-d6e7-4c62-8942-91b839270b2b', 'data_type': 'Copy Number Segment', 'experimental_strategy': 'Genotyping Array'}, {'data_format': 'SVS', 'file_id': '96ce...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>18eb4dfc-556f-4bf3-a411-4780209ed1e0</td>\n",
       "      <td>TCGA-BH-A0B3</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Basal-like</td>\n",
       "      <td>[{'data_format': 'VCF', 'file_id': '9c9e9481-617b-4ecb-8416-07d62453fa8b', 'data_type': 'Structural Rearrangement', 'experimental_strategy': 'WGS'}, {'data_format': 'TXT', 'file_id': '19259452-1a7...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>2021ed1f-dc75-4701-b8b8-1386466e4802</td>\n",
       "      <td>TCGA-BH-A0H7</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>[{'data_format': 'BAM', 'file_id': 'aba03be3-9b5a-44b3-97a7-d89c37c8b721', 'data_type': 'Aligned Reads', 'experimental_strategy': 'RNA-Seq'}, {'data_format': 'TXT', 'file_id': '70a0fb72-e7ca-4fe5-...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>214a4507-d974-4b3e-8525-7408fccc6a0f</td>\n",
       "      <td>TCGA-BH-A0B7</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>HER2-enriched</td>\n",
       "      <td>[{'data_format': 'BEDPE', 'file_id': 'a1a6f518-7460-42cb-b32e-350b0398cfdc', 'data_type': 'Structural Rearrangement', 'experimental_strategy': 'WGS'}, {'data_format': 'TSV', 'file_id': 'f50fdcd8-7...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>21ef1730-e5a7-47ce-b419-d000bb59ae15</td>\n",
       "      <td>TCGA-BH-A1F0</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Basal-like</td>\n",
       "      <td>[{'data_format': 'SVS', 'file_id': 'db7e4dc3-e9e1-47bd-ab03-7b48e0b1beaa', 'data_type': 'Slide Image', 'experimental_strategy': 'Tissue Slide'}, {'data_format': 'BCR Biotab', 'file_id': '0a40467f-...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>233b02f3-c4f0-4a67-9db5-e68d5cdaccb6</td>\n",
       "      <td>TCGA-BH-A18U</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal B</td>\n",
       "      <td>[{'data_format': 'MAF', 'file_id': '3a53ca7e-b47b-4f3e-b786-4c24b2e8be1a', 'data_type': 'Annotated Somatic Mutation', 'experimental_strategy': 'WXS'}, {'data_format': 'BAM', 'file_id': 'af8c9c0c-3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>24334ac1-28a3-4b61-b590-3f51e375b247</td>\n",
       "      <td>TCGA-BH-A0HK</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>[{'data_format': 'BCR Biotab', 'file_id': '0a40467f-9495-4c5b-b56e-3347a3ee0572', 'data_type': 'Clinical Supplement'}, {'data_format': 'TSV', 'file_id': '027f1116-0dbe-4bb8-b231-11d9089ef551', 'da...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>2b22db1d-54a1-4b9e-a86e-a174cf51d95c</td>\n",
       "      <td>TCGA-BH-A0BS</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>[{'data_format': 'MAF', 'file_id': '6ee85029-62ee-4493-bf9d-5907d2d20a66', 'data_type': 'Masked Somatic Mutation', 'experimental_strategy': 'WXS'}, {'data_format': 'VCF', 'file_id': '2757c162-2525...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>2cf68894-168b-458b-af4f-53cad72989a8</td>\n",
       "      <td>TCGA-A7-A13F</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal B</td>\n",
       "      <td>[{'data_format': 'BEDPE', 'file_id': '48327159-a47a-4c5f-8434-3a51edeac2cc', 'data_type': 'Structural Rearrangement', 'experimental_strategy': 'WGS'}, {'data_format': 'TXT', 'file_id': '3acfdb09-8...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>327a2395-85c8-42e1-9638-8e037a1723ed</td>\n",
       "      <td>TCGA-BH-A0BC</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>[{'data_format': 'MAF', 'file_id': '3e414df5-6a0d-4658-83a6-cc52f8f47a0c', 'data_type': 'Masked Somatic Mutation', 'experimental_strategy': 'WXS'}, {'data_format': 'TXT', 'file_id': '17ecfcdf-37a6...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>360d1d6d-cf3e-4c78-b36f-ef5e3160aa6b</td>\n",
       "      <td>TCGA-BH-A0AU</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal B</td>\n",
       "      <td>[{'data_format': 'VCF', 'file_id': 'ec73e799-e6d8-44b4-b47c-abfe573c8f39', 'data_type': 'Structural Rearrangement', 'experimental_strategy': 'WGS'}, {'data_format': 'PDF', 'file_id': 'fb2fcbdf-cc6...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>417dea5f-f68e-4dab-940e-43ae8c67e5e6</td>\n",
       "      <td>TCGA-BH-A1EV</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>HER2-enriched</td>\n",
       "      <td>[{'data_format': 'BCR SSF XML', 'file_id': 'f4bcf109-3db4-424e-9280-9ffb06fed2ed', 'data_type': 'Biospecimen Supplement'}, {'data_format': 'TSV', 'file_id': '18151c98-ace8-46e4-87d4-9f53b7e0268f',...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>433427a1-bacf-4381-91ba-5fec8a0953f9</td>\n",
       "      <td>TCGA-BH-A18S</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>[{'data_format': 'VCF', 'file_id': 'ea58448f-f513-4b93-ad47-9716354e1b84', 'data_type': 'Annotated Somatic Mutation', 'experimental_strategy': 'WXS'}, {'data_format': 'TXT', 'file_id': '0ebbab0f-e...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>4510295e-8aa7-4ef1-b2b7-91cc902f8200</td>\n",
       "      <td>TCGA-BH-A1EO</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>[{'data_format': 'BCR Biotab', 'file_id': '0a40467f-9495-4c5b-b56e-3347a3ee0572', 'data_type': 'Clinical Supplement'}, {'data_format': 'SVS', 'file_id': 'f42f1f5e-8022-4fce-a9e2-af2702bac381', 'da...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>478eb8aa-50b8-4ce5-98e4-d518ede4b174</td>\n",
       "      <td>TCGA-BH-A0DZ</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>HER2-enriched</td>\n",
       "      <td>[{'data_format': 'BAM', 'file_id': '93c944da-7dc7-480f-a46f-4c8464c21b4a', 'data_type': 'Aligned Reads', 'experimental_strategy': 'WGS'}, {'data_format': 'TSV', 'file_id': 'be6f8b24-e668-41c6-89f7...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>4a032bad-e726-48f2-8f39-e3acc109cc91</td>\n",
       "      <td>TCGA-BH-A0BZ</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal B</td>\n",
       "      <td>[{'data_format': 'BAM', 'file_id': '3e6d86f2-1456-40f6-9d81-a3995f6d810a', 'data_type': 'Aligned Reads', 'experimental_strategy': 'RNA-Seq'}, {'data_format': 'BAM', 'file_id': '0e13e4a0-59ee-423f-...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>50619f8c-10aa-464a-a227-90a7aa6ffd43</td>\n",
       "      <td>TCGA-BH-A18K</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Basal-like</td>\n",
       "      <td>[{'data_format': 'TXT', 'file_id': 'c5b7ba42-889d-4b7a-a810-298ae179e9c2', 'data_type': 'Isoform Expression Quantification', 'experimental_strategy': 'miRNA-Seq'}, {'data_format': 'BEDPE', 'file_i...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>518f231c-f5f9-4a39-a97b-7fd5666d3e50</td>\n",
       "      <td>TCGA-BH-A0AZ</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>[{'data_format': 'VCF', 'file_id': 'cd402e3a-92ba-4ea4-a2c1-172201525fbc', 'data_type': 'Annotated Somatic Mutation', 'experimental_strategy': 'WXS'}, {'data_format': 'TAR', 'file_id': '2b0655d7-4...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>557cb0a3-6f1b-4faf-8296-0a15c24158f9</td>\n",
       "      <td>TCGA-BH-A18P</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>HER2-enriched</td>\n",
       "      <td>[{'data_format': 'VCF', 'file_id': '40e9ca92-253b-44a8-b505-409e415e2843', 'data_type': 'Raw Simple Somatic Mutation', 'experimental_strategy': 'WXS'}, {'data_format': 'BCR Biotab', 'file_id': '0a...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>59be60a3-e0f6-4d63-b98c-574ef92428c4</td>\n",
       "      <td>TCGA-E2-A15M</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal B</td>\n",
       "      <td>[{'data_format': 'TSV', 'file_id': 'bdf8478d-e020-4a4e-bf9f-968ca9263536', 'data_type': 'Transcript Fusion', 'experimental_strategy': 'RNA-Seq'}, {'data_format': 'MAF', 'file_id': 'e5a36fca-6152-4...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>5d1d00c6-fcae-479e-ae1e-de76efd41d98</td>\n",
       "      <td>TCGA-BH-A0BW</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Basal-like</td>\n",
       "      <td>[{'data_format': 'PDF', 'file_id': '530460f1-12da-4420-8116-64ae9e28caaa', 'data_type': 'Pathology Report'}, {'data_format': 'TXT', 'file_id': 'd0178301-200f-4a66-933b-2c6e21628e25', 'data_type': ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>606fbc6a-b41b-441d-9401-51e54912bf5e</td>\n",
       "      <td>TCGA-BH-A0E1</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>[{'data_format': 'TXT', 'file_id': 'f52a43ab-8cf4-4611-a13a-195778748214', 'data_type': 'Isoform Expression Quantification', 'experimental_strategy': 'miRNA-Seq'}, {'data_format': 'BEDPE', 'file_i...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>64089c20-939f-48b4-ae8b-904ad0597145</td>\n",
       "      <td>TCGA-E2-A15I</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>[{'data_format': 'TXT', 'file_id': 'e95cf838-e963-4270-b2ac-0c189b6d496e', 'data_type': 'Isoform Expression Quantification', 'experimental_strategy': 'miRNA-Seq'}, {'data_format': 'TSV', 'file_id'...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>665dd3d3-779c-4abd-b5d7-13342340451d</td>\n",
       "      <td>TCGA-BH-A18N</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>[{'data_format': 'TSV', 'file_id': '78ecb795-9903-44bf-b018-764079b35c4d', 'data_type': 'Protein Expression Quantification', 'experimental_strategy': 'Reverse Phase Protein Array'}, {'data_format'...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>74f31744-0aed-4633-beec-4e203315f0b7</td>\n",
       "      <td>TCGA-BH-A0DT</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>[{'data_format': 'TXT', 'file_id': 'cd64d031-39e6-4e5a-a058-cc8723be8e99', 'data_type': 'Allele-specific Copy Number Segment', 'experimental_strategy': 'Genotyping Array'}, {'data_format': 'BCR SS...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>7875018b-91dc-4ab1-83b0-ee3883e2816e</td>\n",
       "      <td>TCGA-BH-A0C0</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal B</td>\n",
       "      <td>[{'data_format': 'BAM', 'file_id': 'c7dda1c9-6745-4645-9751-ab22f17b30f8', 'data_type': 'Aligned Reads', 'experimental_strategy': 'miRNA-Seq'}, {'data_format': 'MAF', 'file_id': '8340167b-18ef-4c6...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>7bcf6d42-eb8f-4aa2-825b-38671d4f9e3a</td>\n",
       "      <td>TCGA-BH-A0C3</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal B</td>\n",
       "      <td>[{'data_format': 'BCR Biotab', 'file_id': '0a40467f-9495-4c5b-b56e-3347a3ee0572', 'data_type': 'Clinical Supplement'}, {'data_format': 'TXT', 'file_id': 'f8d84812-2fae-449f-9c2a-d38b1c172471', 'da...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>87f03d04-e651-4a6e-aea6-1c946aa46202</td>\n",
       "      <td>TCGA-BH-A0BV</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>[{'data_format': 'VCF', 'file_id': '3d7c7b87-f375-447a-b505-c881649e2f2c', 'data_type': 'Annotated Somatic Mutation', 'experimental_strategy': 'WXS'}, {'data_format': 'TSV', 'file_id': 'ea99e4b0-8...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>8986a141-eae7-4157-b695-02cc6fc3b071</td>\n",
       "      <td>TCGA-BH-A1ET</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>[{'data_format': 'TXT', 'file_id': '490e2975-d4e5-470d-a8d9-9b17940bd3f0', 'data_type': 'Methylation Beta Value', 'experimental_strategy': 'Methylation Array'}, {'data_format': 'BCR Biotab', 'file...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>93ed6066-b567-4e1c-ab81-23370f9d3452</td>\n",
       "      <td>TCGA-BH-A0AY</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal B</td>\n",
       "      <td>[{'data_format': 'CEL', 'file_id': '4688be11-ed2c-4e5b-9cf5-3e93eb27999c', 'data_type': 'Raw Intensities', 'experimental_strategy': 'Genotyping Array'}, {'data_format': 'TSV', 'file_id': '4e747f65...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>9d166970-07c8-4ca3-9cfa-ed0049df9ecc</td>\n",
       "      <td>TCGA-BH-A1EW</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal B</td>\n",
       "      <td>[{'data_format': 'TXT', 'file_id': '5a28ab87-b8af-4db3-a8f1-b7c8a3ecf0ca', 'data_type': 'Methylation Beta Value', 'experimental_strategy': 'Methylation Array'}, {'data_format': 'BCR Biotab', 'file...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>9d95a65b-e41d-4f93-92d4-99dce29ff40d</td>\n",
       "      <td>TCGA-A7-A0CE</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Basal-like</td>\n",
       "      <td>[{'data_format': 'BAM', 'file_id': '7c0cb406-5de2-4e75-855a-907dc692205c', 'data_type': 'Aligned Reads', 'experimental_strategy': 'WGS'}, {'data_format': 'TSV', 'file_id': 'fc670e63-94de-4930-b3ea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>9da462b0-93c2-4305-89f6-7199a30399a7</td>\n",
       "      <td>TCGA-BH-A0B5</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal B</td>\n",
       "      <td>[{'data_format': 'TXT', 'file_id': '56e27f97-4797-4e5a-a84b-ed5709558df2', 'data_type': 'Allele-specific Copy Number Segment', 'experimental_strategy': 'Genotyping Array'}, {'data_format': 'BAM', ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>9ddc3e7b-8b54-4a83-8335-8053940f56c1</td>\n",
       "      <td>TCGA-BH-A18L</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal B</td>\n",
       "      <td>[{'data_format': 'TXT', 'file_id': '2f02bb84-bc0e-449d-9cd0-4b7cf0e4ba54', 'data_type': 'Copy Number Segment', 'experimental_strategy': 'Genotyping Array'}, {'data_format': 'SVS', 'file_id': '17c0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>a1093598-d3a8-4ffe-83fc-bc7d1faff7e5</td>\n",
       "      <td>TCGA-BH-A1EU</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>[{'data_format': 'BAM', 'file_id': '4dc2af8b-95f4-4a54-8a82-8da9fce23b74', 'data_type': 'Aligned Reads', 'experimental_strategy': 'RNA-Seq'}, {'data_format': 'BEDPE', 'file_id': 'c2cfb9b0-25df-44c...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>a32cb96a-78bf-456e-a6ab-2d47b2c67ad4</td>\n",
       "      <td>TCGA-BH-A0BT</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>[{'data_format': 'TSV', 'file_id': '68cc7ff0-92b1-4e29-b729-ca582776301b', 'data_type': 'Transcript Fusion', 'experimental_strategy': 'RNA-Seq'}, {'data_format': 'SVS', 'file_id': '2c83bc8f-ac3c-4...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>a6edb6ca-ae9f-4da7-8ebe-92d83d2987fb</td>\n",
       "      <td>TCGA-E2-A158</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Basal-like</td>\n",
       "      <td>[{'data_format': 'TSV', 'file_id': '62a52136-b629-4049-b3c0-ea77cef0ee0c', 'data_type': 'Simple Germline Variation', 'experimental_strategy': 'Genotyping Array'}, {'data_format': 'BCR Biotab', 'fi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>a851ca14-7bbf-455b-a054-ec03979a4247</td>\n",
       "      <td>TCGA-BH-A0DQ</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>[{'data_format': 'TSV', 'file_id': 'c642f518-6ba5-4c90-9f0c-0bad0ed80199', 'data_type': 'Gene Expression Quantification', 'experimental_strategy': 'RNA-Seq'}, {'data_format': 'BAM', 'file_id': '64...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>b3bb8fd6-ee54-4813-b042-2bbe70115382</td>\n",
       "      <td>TCGA-BH-A0BQ</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>[{'data_format': 'BCR Biotab', 'file_id': '0a40467f-9495-4c5b-b56e-3347a3ee0572', 'data_type': 'Clinical Supplement'}, {'data_format': 'BAM', 'file_id': 'cfa351d1-297c-40ae-a6d4-1ddc93f20af9', 'da...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>be9bf36a-12e1-4386-addf-1686dc8b49de</td>\n",
       "      <td>TCGA-E2-A153</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>[{'data_format': 'TSV', 'file_id': 'e4add373-d8c4-447b-85b0-f1e8922e2901', 'data_type': 'Protein Expression Quantification', 'experimental_strategy': 'Reverse Phase Protein Array'}, {'data_format'...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>cab38ea8-4555-413c-b81e-4621aac4ef85</td>\n",
       "      <td>TCGA-BH-A0H5</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>[{'data_format': 'MAF', 'file_id': '5847a246-65dc-456a-a0af-a5fdb6f4dabc', 'data_type': 'Annotated Somatic Mutation', 'experimental_strategy': 'WXS'}, {'data_format': 'TXT', 'file_id': '1171593e-7...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>cc074b7f-d3b2-4880-902e-bf10e667b665</td>\n",
       "      <td>TCGA-BH-A18J</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal B</td>\n",
       "      <td>[{'data_format': 'SVS', 'file_id': '3984bf8d-e842-46b7-a639-46b7dea4149a', 'data_type': 'Slide Image', 'experimental_strategy': 'Tissue Slide'}, {'data_format': 'BAM', 'file_id': '77d40f65-fca4-49...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>d3d4d709-ab34-4617-aa8c-51a3d0a3efe6</td>\n",
       "      <td>TCGA-BH-A0DO</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>[{'data_format': 'BCR XML', 'file_id': '425014ce-d400-4086-9687-c91c5f1a05bc', 'data_type': 'Clinical Supplement'}, {'data_format': 'VCF', 'file_id': 'ed4bcdde-a76b-4f08-a8ea-84d8b6ae1a90', 'data_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>d6f911b5-e895-43f8-8f86-0ac2f1bc6fae</td>\n",
       "      <td>TCGA-BH-A0BJ</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>[{'data_format': 'TSV', 'file_id': '1f1ad68d-27d9-40a4-a4cd-2e4810d472cf', 'data_type': 'Gene Level Copy Number', 'experimental_strategy': 'Genotyping Array'}, {'data_format': 'BCR Biotab', 'file_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>d8492ebd-3d94-4a4b-8de7-386526691cc2</td>\n",
       "      <td>TCGA-E2-A1BC</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>[{'data_format': 'TAR', 'file_id': '9719ffe7-5154-44fb-8709-eb7dba9e6e34', 'data_type': 'Intermediate Analysis Archive', 'experimental_strategy': 'WGS'}, {'data_format': 'TSV', 'file_id': '42fd957...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>db4bc6aa-2e7d-4bcb-8519-a455f624d33b</td>\n",
       "      <td>TCGA-BH-A18Q</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Basal-like</td>\n",
       "      <td>[{'data_format': 'TSV', 'file_id': '1b9a86e0-b97f-49ba-b227-88022c3f7537', 'data_type': 'Transcript Fusion', 'experimental_strategy': 'RNA-Seq'}, {'data_format': 'BCR XML', 'file_id': '00a6f39d-76...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>dba9beec-5e51-48c0-9229-0ee286a02a1a</td>\n",
       "      <td>TCGA-BH-A0DD</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal B</td>\n",
       "      <td>[{'data_format': 'BEDPE', 'file_id': 'c83c4c6c-0427-46fd-8096-93fd4fa9b17f', 'data_type': 'Transcript Fusion', 'experimental_strategy': 'RNA-Seq'}, {'data_format': 'MAF', 'file_id': '5fc15367-ac66...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>e666081b-caaf-4e8b-9446-807be71201a5</td>\n",
       "      <td>TCGA-BH-A0HA</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>[{'data_format': 'BAM', 'file_id': 'f44ad7c7-ed0b-4082-9a11-5e2b479cb061', 'data_type': 'Aligned Reads', 'experimental_strategy': 'RNA-Seq'}, {'data_format': 'TXT', 'file_id': '7d1d9d40-75b4-4f55-...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>f2bbfa9d-9a9d-4f46-9fde-378e4c44e2ad</td>\n",
       "      <td>TCGA-BH-A0DK</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>[{'data_format': 'TXT', 'file_id': '71b9896d-73fb-4f32-8e6e-e5d839095961', 'data_type': 'miRNA Expression Quantification', 'experimental_strategy': 'miRNA-Seq'}, {'data_format': 'BCR Biotab', 'fil...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>f63722c1-a0bb-4bf9-8a10-1d76a8e1a0ba</td>\n",
       "      <td>TCGA-BH-A18M</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>[{'data_format': 'TXT', 'file_id': '0d8cf5ce-92ba-40ab-a817-a33ff1b5262b', 'data_type': 'Methylation Beta Value', 'experimental_strategy': 'Methylation Array'}, {'data_format': 'TSV', 'file_id': '...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>f6e1ec78-5ad9-4879-9b7e-262d17b166ad</td>\n",
       "      <td>TCGA-BH-A18R</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>HER2-enriched</td>\n",
       "      <td>[{'data_format': 'TSV', 'file_id': 'bd239a8e-56e2-45ca-bc44-bff98b72c1d6', 'data_type': 'Gene Expression Quantification', 'experimental_strategy': 'RNA-Seq'}, {'data_format': 'TSV', 'file_id': 'db...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 case_id  submitter_id  \\\n",
       "0   8c7e74e0-71ef-49b8-9217-94b8ef740ef9  TCGA-A7-A13E   \n",
       "1   f130f376-5801-40f9-975d-a7e2f7b5670d  TCGA-A7-A0DB   \n",
       "2   0bc5744c-5fa3-45bb-87d0-70a02068b392  TCGA-BH-A0DG   \n",
       "3   14267783-5624-4fe5-ba81-9d67f1017474  TCGA-BH-A0DP   \n",
       "4   17baef7c-d97d-4b98-ab53-503ef856523d  TCGA-A7-A0D9   \n",
       "5   17c1d42c-cb84-4655-a4cd-b54bae17ecaf  TCGA-E2-A15K   \n",
       "6   18eb4dfc-556f-4bf3-a411-4780209ed1e0  TCGA-BH-A0B3   \n",
       "7   2021ed1f-dc75-4701-b8b8-1386466e4802  TCGA-BH-A0H7   \n",
       "8   214a4507-d974-4b3e-8525-7408fccc6a0f  TCGA-BH-A0B7   \n",
       "9   21ef1730-e5a7-47ce-b419-d000bb59ae15  TCGA-BH-A1F0   \n",
       "10  233b02f3-c4f0-4a67-9db5-e68d5cdaccb6  TCGA-BH-A18U   \n",
       "11  24334ac1-28a3-4b61-b590-3f51e375b247  TCGA-BH-A0HK   \n",
       "12  2b22db1d-54a1-4b9e-a86e-a174cf51d95c  TCGA-BH-A0BS   \n",
       "13  2cf68894-168b-458b-af4f-53cad72989a8  TCGA-A7-A13F   \n",
       "14  327a2395-85c8-42e1-9638-8e037a1723ed  TCGA-BH-A0BC   \n",
       "15  360d1d6d-cf3e-4c78-b36f-ef5e3160aa6b  TCGA-BH-A0AU   \n",
       "16  417dea5f-f68e-4dab-940e-43ae8c67e5e6  TCGA-BH-A1EV   \n",
       "17  433427a1-bacf-4381-91ba-5fec8a0953f9  TCGA-BH-A18S   \n",
       "18  4510295e-8aa7-4ef1-b2b7-91cc902f8200  TCGA-BH-A1EO   \n",
       "19  478eb8aa-50b8-4ce5-98e4-d518ede4b174  TCGA-BH-A0DZ   \n",
       "20  4a032bad-e726-48f2-8f39-e3acc109cc91  TCGA-BH-A0BZ   \n",
       "21  50619f8c-10aa-464a-a227-90a7aa6ffd43  TCGA-BH-A18K   \n",
       "22  518f231c-f5f9-4a39-a97b-7fd5666d3e50  TCGA-BH-A0AZ   \n",
       "23  557cb0a3-6f1b-4faf-8296-0a15c24158f9  TCGA-BH-A18P   \n",
       "24  59be60a3-e0f6-4d63-b98c-574ef92428c4  TCGA-E2-A15M   \n",
       "25  5d1d00c6-fcae-479e-ae1e-de76efd41d98  TCGA-BH-A0BW   \n",
       "26  606fbc6a-b41b-441d-9401-51e54912bf5e  TCGA-BH-A0E1   \n",
       "27  64089c20-939f-48b4-ae8b-904ad0597145  TCGA-E2-A15I   \n",
       "28  665dd3d3-779c-4abd-b5d7-13342340451d  TCGA-BH-A18N   \n",
       "29  74f31744-0aed-4633-beec-4e203315f0b7  TCGA-BH-A0DT   \n",
       "30  7875018b-91dc-4ab1-83b0-ee3883e2816e  TCGA-BH-A0C0   \n",
       "31  7bcf6d42-eb8f-4aa2-825b-38671d4f9e3a  TCGA-BH-A0C3   \n",
       "32  87f03d04-e651-4a6e-aea6-1c946aa46202  TCGA-BH-A0BV   \n",
       "33  8986a141-eae7-4157-b695-02cc6fc3b071  TCGA-BH-A1ET   \n",
       "34  93ed6066-b567-4e1c-ab81-23370f9d3452  TCGA-BH-A0AY   \n",
       "35  9d166970-07c8-4ca3-9cfa-ed0049df9ecc  TCGA-BH-A1EW   \n",
       "36  9d95a65b-e41d-4f93-92d4-99dce29ff40d  TCGA-A7-A0CE   \n",
       "37  9da462b0-93c2-4305-89f6-7199a30399a7  TCGA-BH-A0B5   \n",
       "38  9ddc3e7b-8b54-4a83-8335-8053940f56c1  TCGA-BH-A18L   \n",
       "39  a1093598-d3a8-4ffe-83fc-bc7d1faff7e5  TCGA-BH-A1EU   \n",
       "40  a32cb96a-78bf-456e-a6ab-2d47b2c67ad4  TCGA-BH-A0BT   \n",
       "41  a6edb6ca-ae9f-4da7-8ebe-92d83d2987fb  TCGA-E2-A158   \n",
       "42  a851ca14-7bbf-455b-a054-ec03979a4247  TCGA-BH-A0DQ   \n",
       "43  b3bb8fd6-ee54-4813-b042-2bbe70115382  TCGA-BH-A0BQ   \n",
       "44  be9bf36a-12e1-4386-addf-1686dc8b49de  TCGA-E2-A153   \n",
       "45  cab38ea8-4555-413c-b81e-4621aac4ef85  TCGA-BH-A0H5   \n",
       "46  cc074b7f-d3b2-4880-902e-bf10e667b665  TCGA-BH-A18J   \n",
       "47  d3d4d709-ab34-4617-aa8c-51a3d0a3efe6  TCGA-BH-A0DO   \n",
       "48  d6f911b5-e895-43f8-8f86-0ac2f1bc6fae  TCGA-BH-A0BJ   \n",
       "49  d8492ebd-3d94-4a4b-8de7-386526691cc2  TCGA-E2-A1BC   \n",
       "50  db4bc6aa-2e7d-4bcb-8519-a455f624d33b  TCGA-BH-A18Q   \n",
       "51  dba9beec-5e51-48c0-9229-0ee286a02a1a  TCGA-BH-A0DD   \n",
       "52  e666081b-caaf-4e8b-9446-807be71201a5  TCGA-BH-A0HA   \n",
       "53  f2bbfa9d-9a9d-4f46-9fde-378e4c44e2ad  TCGA-BH-A0DK   \n",
       "54  f63722c1-a0bb-4bf9-8a10-1d76a8e1a0ba  TCGA-BH-A18M   \n",
       "55  f6e1ec78-5ad9-4879-9b7e-262d17b166ad  TCGA-BH-A18R   \n",
       "\n",
       "                    disease_type     pam50_mrna  \\\n",
       "0   Ductal and Lobular Neoplasms     Basal-like   \n",
       "1   Ductal and Lobular Neoplasms      Luminal A   \n",
       "2   Ductal and Lobular Neoplasms      Luminal A   \n",
       "3   Ductal and Lobular Neoplasms      Luminal A   \n",
       "4   Ductal and Lobular Neoplasms      Luminal A   \n",
       "5   Ductal and Lobular Neoplasms      Luminal B   \n",
       "6   Ductal and Lobular Neoplasms     Basal-like   \n",
       "7   Ductal and Lobular Neoplasms      Luminal A   \n",
       "8   Ductal and Lobular Neoplasms  HER2-enriched   \n",
       "9   Ductal and Lobular Neoplasms     Basal-like   \n",
       "10  Ductal and Lobular Neoplasms      Luminal B   \n",
       "11  Ductal and Lobular Neoplasms      Luminal A   \n",
       "12  Ductal and Lobular Neoplasms      Luminal A   \n",
       "13  Ductal and Lobular Neoplasms      Luminal B   \n",
       "14  Ductal and Lobular Neoplasms      Luminal A   \n",
       "15  Ductal and Lobular Neoplasms      Luminal B   \n",
       "16  Ductal and Lobular Neoplasms  HER2-enriched   \n",
       "17  Ductal and Lobular Neoplasms      Luminal A   \n",
       "18  Ductal and Lobular Neoplasms      Luminal A   \n",
       "19  Ductal and Lobular Neoplasms  HER2-enriched   \n",
       "20  Ductal and Lobular Neoplasms      Luminal B   \n",
       "21  Ductal and Lobular Neoplasms     Basal-like   \n",
       "22  Ductal and Lobular Neoplasms      Luminal A   \n",
       "23  Ductal and Lobular Neoplasms  HER2-enriched   \n",
       "24  Ductal and Lobular Neoplasms      Luminal B   \n",
       "25  Ductal and Lobular Neoplasms     Basal-like   \n",
       "26  Ductal and Lobular Neoplasms      Luminal A   \n",
       "27  Ductal and Lobular Neoplasms      Luminal A   \n",
       "28  Ductal and Lobular Neoplasms      Luminal A   \n",
       "29  Ductal and Lobular Neoplasms      Luminal A   \n",
       "30  Ductal and Lobular Neoplasms      Luminal B   \n",
       "31  Ductal and Lobular Neoplasms      Luminal B   \n",
       "32  Ductal and Lobular Neoplasms      Luminal A   \n",
       "33  Ductal and Lobular Neoplasms      Luminal A   \n",
       "34  Ductal and Lobular Neoplasms      Luminal B   \n",
       "35  Ductal and Lobular Neoplasms      Luminal B   \n",
       "36  Ductal and Lobular Neoplasms     Basal-like   \n",
       "37  Ductal and Lobular Neoplasms      Luminal B   \n",
       "38  Ductal and Lobular Neoplasms      Luminal B   \n",
       "39  Ductal and Lobular Neoplasms      Luminal A   \n",
       "40  Ductal and Lobular Neoplasms      Luminal A   \n",
       "41  Ductal and Lobular Neoplasms     Basal-like   \n",
       "42  Ductal and Lobular Neoplasms      Luminal A   \n",
       "43  Ductal and Lobular Neoplasms      Luminal A   \n",
       "44  Ductal and Lobular Neoplasms      Luminal A   \n",
       "45  Ductal and Lobular Neoplasms      Luminal A   \n",
       "46  Ductal and Lobular Neoplasms      Luminal B   \n",
       "47  Ductal and Lobular Neoplasms      Luminal A   \n",
       "48  Ductal and Lobular Neoplasms      Luminal A   \n",
       "49  Ductal and Lobular Neoplasms      Luminal A   \n",
       "50  Ductal and Lobular Neoplasms     Basal-like   \n",
       "51  Ductal and Lobular Neoplasms      Luminal B   \n",
       "52  Ductal and Lobular Neoplasms      Luminal A   \n",
       "53  Ductal and Lobular Neoplasms      Luminal A   \n",
       "54  Ductal and Lobular Neoplasms      Luminal A   \n",
       "55  Ductal and Lobular Neoplasms  HER2-enriched   \n",
       "\n",
       "                                                                                                                                                                                                      files  \n",
       "0   [{'data_format': 'TSV', 'file_id': '767bba1d-e80f-49fa-82b1-075a882d0a10', 'data_type': 'Simple Germline Variation', 'experimental_strategy': 'Genotyping Array'}, {'data_format': 'MAF', 'file_id':...  \n",
       "1   [{'data_format': 'VCF', 'file_id': '886bfc02-b727-47dd-8806-5a34714f8857', 'data_type': 'Annotated Somatic Mutation', 'experimental_strategy': 'WXS'}, {'data_format': 'VCF', 'file_id': 'ef67d03e-b...  \n",
       "2   [{'data_format': 'BEDPE', 'file_id': 'f1c01ec0-cf13-4009-a4a1-23665580ab6e', 'data_type': 'Structural Rearrangement', 'experimental_strategy': 'WGS'}, {'data_format': 'BEDPE', 'file_id': '50e7aa21...  \n",
       "3   [{'data_format': 'BCR Biotab', 'file_id': '0a40467f-9495-4c5b-b56e-3347a3ee0572', 'data_type': 'Clinical Supplement'}, {'data_format': 'BAM', 'file_id': '15a12710-1f2b-41aa-8cf7-054b67f29fbe', 'da...  \n",
       "4   [{'data_format': 'BAM', 'file_id': '8c151dbe-9b21-4fdc-8664-a610ef423606', 'data_type': 'Aligned Reads', 'experimental_strategy': 'WGS'}, {'data_format': 'BEDPE', 'file_id': 'cc14571e-62b5-4f0a-84...  \n",
       "5   [{'data_format': 'TXT', 'file_id': 'db86d78c-d6e7-4c62-8942-91b839270b2b', 'data_type': 'Copy Number Segment', 'experimental_strategy': 'Genotyping Array'}, {'data_format': 'SVS', 'file_id': '96ce...  \n",
       "6   [{'data_format': 'VCF', 'file_id': '9c9e9481-617b-4ecb-8416-07d62453fa8b', 'data_type': 'Structural Rearrangement', 'experimental_strategy': 'WGS'}, {'data_format': 'TXT', 'file_id': '19259452-1a7...  \n",
       "7   [{'data_format': 'BAM', 'file_id': 'aba03be3-9b5a-44b3-97a7-d89c37c8b721', 'data_type': 'Aligned Reads', 'experimental_strategy': 'RNA-Seq'}, {'data_format': 'TXT', 'file_id': '70a0fb72-e7ca-4fe5-...  \n",
       "8   [{'data_format': 'BEDPE', 'file_id': 'a1a6f518-7460-42cb-b32e-350b0398cfdc', 'data_type': 'Structural Rearrangement', 'experimental_strategy': 'WGS'}, {'data_format': 'TSV', 'file_id': 'f50fdcd8-7...  \n",
       "9   [{'data_format': 'SVS', 'file_id': 'db7e4dc3-e9e1-47bd-ab03-7b48e0b1beaa', 'data_type': 'Slide Image', 'experimental_strategy': 'Tissue Slide'}, {'data_format': 'BCR Biotab', 'file_id': '0a40467f-...  \n",
       "10  [{'data_format': 'MAF', 'file_id': '3a53ca7e-b47b-4f3e-b786-4c24b2e8be1a', 'data_type': 'Annotated Somatic Mutation', 'experimental_strategy': 'WXS'}, {'data_format': 'BAM', 'file_id': 'af8c9c0c-3...  \n",
       "11  [{'data_format': 'BCR Biotab', 'file_id': '0a40467f-9495-4c5b-b56e-3347a3ee0572', 'data_type': 'Clinical Supplement'}, {'data_format': 'TSV', 'file_id': '027f1116-0dbe-4bb8-b231-11d9089ef551', 'da...  \n",
       "12  [{'data_format': 'MAF', 'file_id': '6ee85029-62ee-4493-bf9d-5907d2d20a66', 'data_type': 'Masked Somatic Mutation', 'experimental_strategy': 'WXS'}, {'data_format': 'VCF', 'file_id': '2757c162-2525...  \n",
       "13  [{'data_format': 'BEDPE', 'file_id': '48327159-a47a-4c5f-8434-3a51edeac2cc', 'data_type': 'Structural Rearrangement', 'experimental_strategy': 'WGS'}, {'data_format': 'TXT', 'file_id': '3acfdb09-8...  \n",
       "14  [{'data_format': 'MAF', 'file_id': '3e414df5-6a0d-4658-83a6-cc52f8f47a0c', 'data_type': 'Masked Somatic Mutation', 'experimental_strategy': 'WXS'}, {'data_format': 'TXT', 'file_id': '17ecfcdf-37a6...  \n",
       "15  [{'data_format': 'VCF', 'file_id': 'ec73e799-e6d8-44b4-b47c-abfe573c8f39', 'data_type': 'Structural Rearrangement', 'experimental_strategy': 'WGS'}, {'data_format': 'PDF', 'file_id': 'fb2fcbdf-cc6...  \n",
       "16  [{'data_format': 'BCR SSF XML', 'file_id': 'f4bcf109-3db4-424e-9280-9ffb06fed2ed', 'data_type': 'Biospecimen Supplement'}, {'data_format': 'TSV', 'file_id': '18151c98-ace8-46e4-87d4-9f53b7e0268f',...  \n",
       "17  [{'data_format': 'VCF', 'file_id': 'ea58448f-f513-4b93-ad47-9716354e1b84', 'data_type': 'Annotated Somatic Mutation', 'experimental_strategy': 'WXS'}, {'data_format': 'TXT', 'file_id': '0ebbab0f-e...  \n",
       "18  [{'data_format': 'BCR Biotab', 'file_id': '0a40467f-9495-4c5b-b56e-3347a3ee0572', 'data_type': 'Clinical Supplement'}, {'data_format': 'SVS', 'file_id': 'f42f1f5e-8022-4fce-a9e2-af2702bac381', 'da...  \n",
       "19  [{'data_format': 'BAM', 'file_id': '93c944da-7dc7-480f-a46f-4c8464c21b4a', 'data_type': 'Aligned Reads', 'experimental_strategy': 'WGS'}, {'data_format': 'TSV', 'file_id': 'be6f8b24-e668-41c6-89f7...  \n",
       "20  [{'data_format': 'BAM', 'file_id': '3e6d86f2-1456-40f6-9d81-a3995f6d810a', 'data_type': 'Aligned Reads', 'experimental_strategy': 'RNA-Seq'}, {'data_format': 'BAM', 'file_id': '0e13e4a0-59ee-423f-...  \n",
       "21  [{'data_format': 'TXT', 'file_id': 'c5b7ba42-889d-4b7a-a810-298ae179e9c2', 'data_type': 'Isoform Expression Quantification', 'experimental_strategy': 'miRNA-Seq'}, {'data_format': 'BEDPE', 'file_i...  \n",
       "22  [{'data_format': 'VCF', 'file_id': 'cd402e3a-92ba-4ea4-a2c1-172201525fbc', 'data_type': 'Annotated Somatic Mutation', 'experimental_strategy': 'WXS'}, {'data_format': 'TAR', 'file_id': '2b0655d7-4...  \n",
       "23  [{'data_format': 'VCF', 'file_id': '40e9ca92-253b-44a8-b505-409e415e2843', 'data_type': 'Raw Simple Somatic Mutation', 'experimental_strategy': 'WXS'}, {'data_format': 'BCR Biotab', 'file_id': '0a...  \n",
       "24  [{'data_format': 'TSV', 'file_id': 'bdf8478d-e020-4a4e-bf9f-968ca9263536', 'data_type': 'Transcript Fusion', 'experimental_strategy': 'RNA-Seq'}, {'data_format': 'MAF', 'file_id': 'e5a36fca-6152-4...  \n",
       "25  [{'data_format': 'PDF', 'file_id': '530460f1-12da-4420-8116-64ae9e28caaa', 'data_type': 'Pathology Report'}, {'data_format': 'TXT', 'file_id': 'd0178301-200f-4a66-933b-2c6e21628e25', 'data_type': ...  \n",
       "26  [{'data_format': 'TXT', 'file_id': 'f52a43ab-8cf4-4611-a13a-195778748214', 'data_type': 'Isoform Expression Quantification', 'experimental_strategy': 'miRNA-Seq'}, {'data_format': 'BEDPE', 'file_i...  \n",
       "27  [{'data_format': 'TXT', 'file_id': 'e95cf838-e963-4270-b2ac-0c189b6d496e', 'data_type': 'Isoform Expression Quantification', 'experimental_strategy': 'miRNA-Seq'}, {'data_format': 'TSV', 'file_id'...  \n",
       "28  [{'data_format': 'TSV', 'file_id': '78ecb795-9903-44bf-b018-764079b35c4d', 'data_type': 'Protein Expression Quantification', 'experimental_strategy': 'Reverse Phase Protein Array'}, {'data_format'...  \n",
       "29  [{'data_format': 'TXT', 'file_id': 'cd64d031-39e6-4e5a-a058-cc8723be8e99', 'data_type': 'Allele-specific Copy Number Segment', 'experimental_strategy': 'Genotyping Array'}, {'data_format': 'BCR SS...  \n",
       "30  [{'data_format': 'BAM', 'file_id': 'c7dda1c9-6745-4645-9751-ab22f17b30f8', 'data_type': 'Aligned Reads', 'experimental_strategy': 'miRNA-Seq'}, {'data_format': 'MAF', 'file_id': '8340167b-18ef-4c6...  \n",
       "31  [{'data_format': 'BCR Biotab', 'file_id': '0a40467f-9495-4c5b-b56e-3347a3ee0572', 'data_type': 'Clinical Supplement'}, {'data_format': 'TXT', 'file_id': 'f8d84812-2fae-449f-9c2a-d38b1c172471', 'da...  \n",
       "32  [{'data_format': 'VCF', 'file_id': '3d7c7b87-f375-447a-b505-c881649e2f2c', 'data_type': 'Annotated Somatic Mutation', 'experimental_strategy': 'WXS'}, {'data_format': 'TSV', 'file_id': 'ea99e4b0-8...  \n",
       "33  [{'data_format': 'TXT', 'file_id': '490e2975-d4e5-470d-a8d9-9b17940bd3f0', 'data_type': 'Methylation Beta Value', 'experimental_strategy': 'Methylation Array'}, {'data_format': 'BCR Biotab', 'file...  \n",
       "34  [{'data_format': 'CEL', 'file_id': '4688be11-ed2c-4e5b-9cf5-3e93eb27999c', 'data_type': 'Raw Intensities', 'experimental_strategy': 'Genotyping Array'}, {'data_format': 'TSV', 'file_id': '4e747f65...  \n",
       "35  [{'data_format': 'TXT', 'file_id': '5a28ab87-b8af-4db3-a8f1-b7c8a3ecf0ca', 'data_type': 'Methylation Beta Value', 'experimental_strategy': 'Methylation Array'}, {'data_format': 'BCR Biotab', 'file...  \n",
       "36  [{'data_format': 'BAM', 'file_id': '7c0cb406-5de2-4e75-855a-907dc692205c', 'data_type': 'Aligned Reads', 'experimental_strategy': 'WGS'}, {'data_format': 'TSV', 'file_id': 'fc670e63-94de-4930-b3ea...  \n",
       "37  [{'data_format': 'TXT', 'file_id': '56e27f97-4797-4e5a-a84b-ed5709558df2', 'data_type': 'Allele-specific Copy Number Segment', 'experimental_strategy': 'Genotyping Array'}, {'data_format': 'BAM', ...  \n",
       "38  [{'data_format': 'TXT', 'file_id': '2f02bb84-bc0e-449d-9cd0-4b7cf0e4ba54', 'data_type': 'Copy Number Segment', 'experimental_strategy': 'Genotyping Array'}, {'data_format': 'SVS', 'file_id': '17c0...  \n",
       "39  [{'data_format': 'BAM', 'file_id': '4dc2af8b-95f4-4a54-8a82-8da9fce23b74', 'data_type': 'Aligned Reads', 'experimental_strategy': 'RNA-Seq'}, {'data_format': 'BEDPE', 'file_id': 'c2cfb9b0-25df-44c...  \n",
       "40  [{'data_format': 'TSV', 'file_id': '68cc7ff0-92b1-4e29-b729-ca582776301b', 'data_type': 'Transcript Fusion', 'experimental_strategy': 'RNA-Seq'}, {'data_format': 'SVS', 'file_id': '2c83bc8f-ac3c-4...  \n",
       "41  [{'data_format': 'TSV', 'file_id': '62a52136-b629-4049-b3c0-ea77cef0ee0c', 'data_type': 'Simple Germline Variation', 'experimental_strategy': 'Genotyping Array'}, {'data_format': 'BCR Biotab', 'fi...  \n",
       "42  [{'data_format': 'TSV', 'file_id': 'c642f518-6ba5-4c90-9f0c-0bad0ed80199', 'data_type': 'Gene Expression Quantification', 'experimental_strategy': 'RNA-Seq'}, {'data_format': 'BAM', 'file_id': '64...  \n",
       "43  [{'data_format': 'BCR Biotab', 'file_id': '0a40467f-9495-4c5b-b56e-3347a3ee0572', 'data_type': 'Clinical Supplement'}, {'data_format': 'BAM', 'file_id': 'cfa351d1-297c-40ae-a6d4-1ddc93f20af9', 'da...  \n",
       "44  [{'data_format': 'TSV', 'file_id': 'e4add373-d8c4-447b-85b0-f1e8922e2901', 'data_type': 'Protein Expression Quantification', 'experimental_strategy': 'Reverse Phase Protein Array'}, {'data_format'...  \n",
       "45  [{'data_format': 'MAF', 'file_id': '5847a246-65dc-456a-a0af-a5fdb6f4dabc', 'data_type': 'Annotated Somatic Mutation', 'experimental_strategy': 'WXS'}, {'data_format': 'TXT', 'file_id': '1171593e-7...  \n",
       "46  [{'data_format': 'SVS', 'file_id': '3984bf8d-e842-46b7-a639-46b7dea4149a', 'data_type': 'Slide Image', 'experimental_strategy': 'Tissue Slide'}, {'data_format': 'BAM', 'file_id': '77d40f65-fca4-49...  \n",
       "47  [{'data_format': 'BCR XML', 'file_id': '425014ce-d400-4086-9687-c91c5f1a05bc', 'data_type': 'Clinical Supplement'}, {'data_format': 'VCF', 'file_id': 'ed4bcdde-a76b-4f08-a8ea-84d8b6ae1a90', 'data_...  \n",
       "48  [{'data_format': 'TSV', 'file_id': '1f1ad68d-27d9-40a4-a4cd-2e4810d472cf', 'data_type': 'Gene Level Copy Number', 'experimental_strategy': 'Genotyping Array'}, {'data_format': 'BCR Biotab', 'file_...  \n",
       "49  [{'data_format': 'TAR', 'file_id': '9719ffe7-5154-44fb-8709-eb7dba9e6e34', 'data_type': 'Intermediate Analysis Archive', 'experimental_strategy': 'WGS'}, {'data_format': 'TSV', 'file_id': '42fd957...  \n",
       "50  [{'data_format': 'TSV', 'file_id': '1b9a86e0-b97f-49ba-b227-88022c3f7537', 'data_type': 'Transcript Fusion', 'experimental_strategy': 'RNA-Seq'}, {'data_format': 'BCR XML', 'file_id': '00a6f39d-76...  \n",
       "51  [{'data_format': 'BEDPE', 'file_id': 'c83c4c6c-0427-46fd-8096-93fd4fa9b17f', 'data_type': 'Transcript Fusion', 'experimental_strategy': 'RNA-Seq'}, {'data_format': 'MAF', 'file_id': '5fc15367-ac66...  \n",
       "52  [{'data_format': 'BAM', 'file_id': 'f44ad7c7-ed0b-4082-9a11-5e2b479cb061', 'data_type': 'Aligned Reads', 'experimental_strategy': 'RNA-Seq'}, {'data_format': 'TXT', 'file_id': '7d1d9d40-75b4-4f55-...  \n",
       "53  [{'data_format': 'TXT', 'file_id': '71b9896d-73fb-4f32-8e6e-e5d839095961', 'data_type': 'miRNA Expression Quantification', 'experimental_strategy': 'miRNA-Seq'}, {'data_format': 'BCR Biotab', 'fil...  \n",
       "54  [{'data_format': 'TXT', 'file_id': '0d8cf5ce-92ba-40ab-a817-a33ff1b5262b', 'data_type': 'Methylation Beta Value', 'experimental_strategy': 'Methylation Array'}, {'data_format': 'TSV', 'file_id': '...  \n",
       "55  [{'data_format': 'TSV', 'file_id': 'bd239a8e-56e2-45ca-bc44-bff98b72c1d6', 'data_type': 'Gene Expression Quantification', 'experimental_strategy': 'RNA-Seq'}, {'data_format': 'TSV', 'file_id': 'db...  "
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the DataFrame of cases of interest with normal tissue analysis\n",
    "df_normal_cases"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Files of Interest DataFrames"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Files Associated with Tumor Tissue Analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "# List with the UUIDs of the cases of interest with tumor tissue analysis\n",
    "tumor_cases = df_tumor_cases['case_id'].tolist()\n",
    "\n",
    "# Create the DataFrame of files of interest associated with tumor tissue analysis\n",
    "df_tumor_files = df_files \\\n",
    "    .query('(tissue_type == \"Tumor\") and (case_id == @tumor_cases)') \\\n",
    "    .drop(columns='file_type') \\\n",
    "    .reset_index(drop=True)\n",
    "\n",
    "# Store the DataFrame in a CSV file\n",
    "file_name = 'brca-api-files-tumor-analysis.csv'\n",
    "df_tumor_files.to_csv(f'{BRCA_INTERIM_DATA_DIR}/{file_name}', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>file_id</th>\n",
       "      <th>case_id</th>\n",
       "      <th>tissue_type</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>experimental_strategy</th>\n",
       "      <th>data_type</th>\n",
       "      <th>data_format</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>e9fe2319-2624-4ac1-9c41-d0ac44608f02</td>\n",
       "      <td>de75d0b9-0f47-4732-8df5-05c350cfcd32</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Primary Tumor</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>Gene Expression Quantification</td>\n",
       "      <td>TSV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0a8016e1-0321-4421-a5ad-db2fd9caf7ab</td>\n",
       "      <td>de75d0b9-0f47-4732-8df5-05c350cfcd32</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Primary Tumor</td>\n",
       "      <td>miRNA-Seq</td>\n",
       "      <td>Isoform Expression Quantification</td>\n",
       "      <td>TXT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>8437bf97-2354-40cf-8dc4-631c5d7ae63a</td>\n",
       "      <td>1fef9454-b3e8-4d9d-a9aa-aa1f1a32b80c</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Primary Tumor</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>Gene Expression Quantification</td>\n",
       "      <td>TSV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>7a4675e7-4174-4561-ae9c-b7636fbea718</td>\n",
       "      <td>1fef9454-b3e8-4d9d-a9aa-aa1f1a32b80c</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Primary Tumor</td>\n",
       "      <td>miRNA-Seq</td>\n",
       "      <td>Isoform Expression Quantification</td>\n",
       "      <td>TXT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>15f5f60a-13f0-41a2-8711-171d19c2903a</td>\n",
       "      <td>2021ed1f-dc75-4701-b8b8-1386466e4802</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Primary Tumor</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>Gene Expression Quantification</td>\n",
       "      <td>TSV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>967</th>\n",
       "      <td>8c884351-1c8b-47ff-8bef-1a4f25f645fb</td>\n",
       "      <td>360d1d6d-cf3e-4c78-b36f-ef5e3160aa6b</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Primary Tumor</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>Gene Expression Quantification</td>\n",
       "      <td>TSV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>968</th>\n",
       "      <td>7c275803-bb6d-4fbc-b851-5fcb393f8496</td>\n",
       "      <td>2cf68894-168b-458b-af4f-53cad72989a8</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Primary Tumor</td>\n",
       "      <td>miRNA-Seq</td>\n",
       "      <td>Isoform Expression Quantification</td>\n",
       "      <td>TXT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>969</th>\n",
       "      <td>9c87ef50-5854-4f69-820b-9e0b4d46c982</td>\n",
       "      <td>2cf68894-168b-458b-af4f-53cad72989a8</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Primary Tumor</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>Gene Expression Quantification</td>\n",
       "      <td>TSV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>970</th>\n",
       "      <td>d5d23d6a-c141-442f-ac4f-714e46b8d9ef</td>\n",
       "      <td>36d77496-99f8-4911-ba7c-89fae32662aa</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Primary Tumor</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>Gene Expression Quantification</td>\n",
       "      <td>TSV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>971</th>\n",
       "      <td>1e4fc580-3846-4ece-a88d-55d0d5e2ac2f</td>\n",
       "      <td>36d77496-99f8-4911-ba7c-89fae32662aa</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Primary Tumor</td>\n",
       "      <td>miRNA-Seq</td>\n",
       "      <td>Isoform Expression Quantification</td>\n",
       "      <td>TXT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>972 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                  file_id  \\\n",
       "0    e9fe2319-2624-4ac1-9c41-d0ac44608f02   \n",
       "1    0a8016e1-0321-4421-a5ad-db2fd9caf7ab   \n",
       "2    8437bf97-2354-40cf-8dc4-631c5d7ae63a   \n",
       "3    7a4675e7-4174-4561-ae9c-b7636fbea718   \n",
       "4    15f5f60a-13f0-41a2-8711-171d19c2903a   \n",
       "..                                    ...   \n",
       "967  8c884351-1c8b-47ff-8bef-1a4f25f645fb   \n",
       "968  7c275803-bb6d-4fbc-b851-5fcb393f8496   \n",
       "969  9c87ef50-5854-4f69-820b-9e0b4d46c982   \n",
       "970  d5d23d6a-c141-442f-ac4f-714e46b8d9ef   \n",
       "971  1e4fc580-3846-4ece-a88d-55d0d5e2ac2f   \n",
       "\n",
       "                                  case_id tissue_type    sample_type  \\\n",
       "0    de75d0b9-0f47-4732-8df5-05c350cfcd32       Tumor  Primary Tumor   \n",
       "1    de75d0b9-0f47-4732-8df5-05c350cfcd32       Tumor  Primary Tumor   \n",
       "2    1fef9454-b3e8-4d9d-a9aa-aa1f1a32b80c       Tumor  Primary Tumor   \n",
       "3    1fef9454-b3e8-4d9d-a9aa-aa1f1a32b80c       Tumor  Primary Tumor   \n",
       "4    2021ed1f-dc75-4701-b8b8-1386466e4802       Tumor  Primary Tumor   \n",
       "..                                    ...         ...            ...   \n",
       "967  360d1d6d-cf3e-4c78-b36f-ef5e3160aa6b       Tumor  Primary Tumor   \n",
       "968  2cf68894-168b-458b-af4f-53cad72989a8       Tumor  Primary Tumor   \n",
       "969  2cf68894-168b-458b-af4f-53cad72989a8       Tumor  Primary Tumor   \n",
       "970  36d77496-99f8-4911-ba7c-89fae32662aa       Tumor  Primary Tumor   \n",
       "971  36d77496-99f8-4911-ba7c-89fae32662aa       Tumor  Primary Tumor   \n",
       "\n",
       "    experimental_strategy                          data_type data_format  \n",
       "0                 RNA-Seq     Gene Expression Quantification         TSV  \n",
       "1               miRNA-Seq  Isoform Expression Quantification         TXT  \n",
       "2                 RNA-Seq     Gene Expression Quantification         TSV  \n",
       "3               miRNA-Seq  Isoform Expression Quantification         TXT  \n",
       "4                 RNA-Seq     Gene Expression Quantification         TSV  \n",
       "..                    ...                                ...         ...  \n",
       "967               RNA-Seq     Gene Expression Quantification         TSV  \n",
       "968             miRNA-Seq  Isoform Expression Quantification         TXT  \n",
       "969               RNA-Seq     Gene Expression Quantification         TSV  \n",
       "970               RNA-Seq     Gene Expression Quantification         TSV  \n",
       "971             miRNA-Seq  Isoform Expression Quantification         TXT  \n",
       "\n",
       "[972 rows x 7 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the DataFrame of files of interest associated with tumor tissue analysis\n",
    "df_tumor_files"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Files Associated with Normal Tissue Analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "# List with the UUIDs of the cases of interest with normal tissue analysis\n",
    "normal_cases = df_normal_cases['case_id'].tolist()\n",
    "\n",
    "# Create the DataFrame of files of interest associated with normal tissue analysis\n",
    "df_normal_files = df_files \\\n",
    "    .query('(tissue_type == \"Normal\") and (case_id == @normal_cases)') \\\n",
    "    .drop(columns='file_type') \\\n",
    "    .reset_index(drop=True)\n",
    "\n",
    "# Store the DataFrame in a CSV file\n",
    "file_name = 'brca-api-files-normal-analysis.csv'\n",
    "df_normal_files.to_csv(f'{BRCA_INTERIM_DATA_DIR}/{file_name}', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>file_id</th>\n",
       "      <th>case_id</th>\n",
       "      <th>tissue_type</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>experimental_strategy</th>\n",
       "      <th>data_type</th>\n",
       "      <th>data_format</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ea1fadc2-1cdc-4658-9619-eeb26ae09da8</td>\n",
       "      <td>2021ed1f-dc75-4701-b8b8-1386466e4802</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Solid Tissue Normal</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>Gene Expression Quantification</td>\n",
       "      <td>TSV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ef076f15-36a4-4e8a-a624-8cdeb9c4fa2d</td>\n",
       "      <td>2021ed1f-dc75-4701-b8b8-1386466e4802</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Solid Tissue Normal</td>\n",
       "      <td>miRNA-Seq</td>\n",
       "      <td>Isoform Expression Quantification</td>\n",
       "      <td>TXT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3f5ad0e5-ab95-4c1e-8dfe-09cab1d1500c</td>\n",
       "      <td>214a4507-d974-4b3e-8525-7408fccc6a0f</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Solid Tissue Normal</td>\n",
       "      <td>miRNA-Seq</td>\n",
       "      <td>Isoform Expression Quantification</td>\n",
       "      <td>TXT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3071e512-94ea-4820-9573-668235188e34</td>\n",
       "      <td>214a4507-d974-4b3e-8525-7408fccc6a0f</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Solid Tissue Normal</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>Gene Expression Quantification</td>\n",
       "      <td>TSV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>8790fe0b-8d3a-4f2c-8201-1231da243697</td>\n",
       "      <td>21ef1730-e5a7-47ce-b419-d000bb59ae15</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Solid Tissue Normal</td>\n",
       "      <td>miRNA-Seq</td>\n",
       "      <td>Isoform Expression Quantification</td>\n",
       "      <td>TXT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>107</th>\n",
       "      <td>d4a3af62-0505-49f6-9f34-168bec5cb0d0</td>\n",
       "      <td>a851ca14-7bbf-455b-a054-ec03979a4247</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Solid Tissue Normal</td>\n",
       "      <td>miRNA-Seq</td>\n",
       "      <td>Isoform Expression Quantification</td>\n",
       "      <td>TXT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>cd746011-8fd7-473d-84e7-f39ec60b6717</td>\n",
       "      <td>360d1d6d-cf3e-4c78-b36f-ef5e3160aa6b</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Solid Tissue Normal</td>\n",
       "      <td>miRNA-Seq</td>\n",
       "      <td>Isoform Expression Quantification</td>\n",
       "      <td>TXT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109</th>\n",
       "      <td>f24d6e88-634e-4d86-b993-76a40f1190d0</td>\n",
       "      <td>360d1d6d-cf3e-4c78-b36f-ef5e3160aa6b</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Solid Tissue Normal</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>Gene Expression Quantification</td>\n",
       "      <td>TSV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110</th>\n",
       "      <td>4c2bb8e3-2527-466b-a3bb-0f36c3b2a3e7</td>\n",
       "      <td>2cf68894-168b-458b-af4f-53cad72989a8</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Solid Tissue Normal</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>Gene Expression Quantification</td>\n",
       "      <td>TSV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>111</th>\n",
       "      <td>8ca0e4c1-92ab-4653-87e0-cdcc630d8db6</td>\n",
       "      <td>2cf68894-168b-458b-af4f-53cad72989a8</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Solid Tissue Normal</td>\n",
       "      <td>miRNA-Seq</td>\n",
       "      <td>Isoform Expression Quantification</td>\n",
       "      <td>TXT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>112 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                  file_id  \\\n",
       "0    ea1fadc2-1cdc-4658-9619-eeb26ae09da8   \n",
       "1    ef076f15-36a4-4e8a-a624-8cdeb9c4fa2d   \n",
       "2    3f5ad0e5-ab95-4c1e-8dfe-09cab1d1500c   \n",
       "3    3071e512-94ea-4820-9573-668235188e34   \n",
       "4    8790fe0b-8d3a-4f2c-8201-1231da243697   \n",
       "..                                    ...   \n",
       "107  d4a3af62-0505-49f6-9f34-168bec5cb0d0   \n",
       "108  cd746011-8fd7-473d-84e7-f39ec60b6717   \n",
       "109  f24d6e88-634e-4d86-b993-76a40f1190d0   \n",
       "110  4c2bb8e3-2527-466b-a3bb-0f36c3b2a3e7   \n",
       "111  8ca0e4c1-92ab-4653-87e0-cdcc630d8db6   \n",
       "\n",
       "                                  case_id tissue_type          sample_type  \\\n",
       "0    2021ed1f-dc75-4701-b8b8-1386466e4802      Normal  Solid Tissue Normal   \n",
       "1    2021ed1f-dc75-4701-b8b8-1386466e4802      Normal  Solid Tissue Normal   \n",
       "2    214a4507-d974-4b3e-8525-7408fccc6a0f      Normal  Solid Tissue Normal   \n",
       "3    214a4507-d974-4b3e-8525-7408fccc6a0f      Normal  Solid Tissue Normal   \n",
       "4    21ef1730-e5a7-47ce-b419-d000bb59ae15      Normal  Solid Tissue Normal   \n",
       "..                                    ...         ...                  ...   \n",
       "107  a851ca14-7bbf-455b-a054-ec03979a4247      Normal  Solid Tissue Normal   \n",
       "108  360d1d6d-cf3e-4c78-b36f-ef5e3160aa6b      Normal  Solid Tissue Normal   \n",
       "109  360d1d6d-cf3e-4c78-b36f-ef5e3160aa6b      Normal  Solid Tissue Normal   \n",
       "110  2cf68894-168b-458b-af4f-53cad72989a8      Normal  Solid Tissue Normal   \n",
       "111  2cf68894-168b-458b-af4f-53cad72989a8      Normal  Solid Tissue Normal   \n",
       "\n",
       "    experimental_strategy                          data_type data_format  \n",
       "0                 RNA-Seq     Gene Expression Quantification         TSV  \n",
       "1               miRNA-Seq  Isoform Expression Quantification         TXT  \n",
       "2               miRNA-Seq  Isoform Expression Quantification         TXT  \n",
       "3                 RNA-Seq     Gene Expression Quantification         TSV  \n",
       "4               miRNA-Seq  Isoform Expression Quantification         TXT  \n",
       "..                    ...                                ...         ...  \n",
       "107             miRNA-Seq  Isoform Expression Quantification         TXT  \n",
       "108             miRNA-Seq  Isoform Expression Quantification         TXT  \n",
       "109               RNA-Seq     Gene Expression Quantification         TSV  \n",
       "110               RNA-Seq     Gene Expression Quantification         TSV  \n",
       "111             miRNA-Seq  Isoform Expression Quantification         TXT  \n",
       "\n",
       "[112 rows x 7 columns]"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the DataFrame of files of interest associated with normal tissue analysis\n",
    "df_normal_files"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
