{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# **Access to TCGA-BRCA Metadata from the GDC API**\n",
    "\n",
    "This notebook produces the AT_OR family of data artifacts  \n",
    "\n",
    "- TCGA: The Cancer Genome Atlas  \n",
    "- BRCA: Breast Invasive Carcinoma  \n",
    "- GDC: Genomic Data Commons"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Import Libraries and Configurations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "import os\n",
    "import sys\n",
    "\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import requests\n",
    "\n",
    "# Add project root to Python's path\n",
    "sys.path.append(os.path.abspath(os.path.join('..')))\n",
    "\n",
    "from config import (\n",
    "    BRCA_PREPROCESSING_PARAMETERS,\n",
    "    BRCA_RAW_FILES_PATHS,\n",
    "    GDC_API_ENDPOINTS,\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Function"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "def gdc_api_request(endpoint, fields, filters={}, size=1):\n",
    "    \"\"\"\n",
    "    Send a request to a Genomic Data Commons (GDC) API endpoint with the specified parameters.\n",
    "\n",
    "    Parameters:\n",
    "    -----------\n",
    "    endpoint : str\n",
    "        The URL of the endpoint to send the request to.\n",
    "    fields : list of str\n",
    "        A list of fields to include in the response.\n",
    "    filters : dict, optional\n",
    "        A dictionary of filters to apply to the request. Default is an empty dictionary.\n",
    "    size : int, optional\n",
    "        The number of results to retrieve. Default is 1.\n",
    "\n",
    "    Returns:\n",
    "    --------\n",
    "    list\n",
    "        A list of hits (data items) retrieved from the endpoint response.\n",
    "    \"\"\"\n",
    "    # Parameters used in the endpoint request\n",
    "    params = {\n",
    "        'fields': ','.join(fields),\n",
    "        'filters': filters,\n",
    "        'size': str(size)\n",
    "    }\n",
    "    \n",
    "    # Request the objects of interest to the endpoint\n",
    "    response = requests.post(\n",
    "        url=endpoint,\n",
    "        headers={'Content-Type': 'application/json'},\n",
    "        json=params\n",
    "    )\n",
    "    response.raise_for_status()\n",
    "\n",
    "    # Convert the response content to a JSON\n",
    "    json_response = json.loads(response.content.decode('utf-8'))\n",
    "\n",
    "    return json_response['data']['hits']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# GDC API Requests"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Status Endpoint"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{\n",
      "    \"commit\": \"b46d9fe84b7f2803af806ac98e49e8b7199a84c7\",\n",
      "    \"data_release\": \"Data Release 43.0 - May 07, 2025\",\n",
      "    \"data_release_version\": {\n",
      "        \"major\": 43,\n",
      "        \"minor\": 0,\n",
      "        \"release_date\": \"2025-05-07\"\n",
      "    },\n",
      "    \"status\": \"OK\",\n",
      "    \"tag\": \"7.10\",\n",
      "    \"version\": 1\n",
      "}\n"
     ]
    }
   ],
   "source": [
    "# Request the API version and status to the endpoint\n",
    "response = requests.get(GDC_API_ENDPOINTS['status'])\n",
    "\n",
    "# Print the response content\n",
    "try:\n",
    "    parsed = json.loads(response.content.decode('utf-8'))\n",
    "    print(json.dumps(parsed, indent=4))\n",
    "except json.JSONDecodeError:\n",
    "    print('Response is not valid JSON:')\n",
    "    print(response.content.decode('utf-8'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Projects Endpoint"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Endpoint Response"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Fields available in the endpoint\n",
    "fields = [\n",
    "    'disease_type',\n",
    "    'name',\n",
    "    'primary_site',\n",
    "    'summary.case_count',\n",
    "    'summary.data_categories.case_count',\n",
    "    'summary.data_categories.data_category',\n",
    "    'summary.data_categories.file_count',\n",
    "    'summary.experimental_strategies.case_count',\n",
    "    'summary.experimental_strategies.experimental_strategy',\n",
    "    'summary.experimental_strategies.file_count',\n",
    "    'summary.file_count',\n",
    "]\n",
    "\n",
    "# Filter to be used in the projects endpoint request\n",
    "filters = {\n",
    "    'op': 'in',\n",
    "    'content': {\n",
    "        'field': 'projects.project_id',\n",
    "        'value': BRCA_PREPROCESSING_PARAMETERS['project_ids']\n",
    "    }\n",
    "}\n",
    "\n",
    "# Get the response from the projects endpoint request\n",
    "projects_response = gdc_api_request(\n",
    "    endpoint=GDC_API_ENDPOINTS['projects'],\n",
    "    fields=fields,\n",
    "    filters=filters,\n",
    "    size=len(BRCA_PREPROCESSING_PARAMETERS['project_ids'])\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Project DataFrame"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create the TCGA-BRCA project metadata DataFrame\n",
    "df_project = pd.json_normalize(projects_response) \\\n",
    "    .rename(columns={\n",
    "        'id': 'project_id',\n",
    "        'name': 'project_name',\n",
    "        'summary.case_count': 'case_count',\n",
    "        'summary.data_categories': 'data_categories',\n",
    "        'summary.experimental_strategies': 'experimental_strategies',\n",
    "        'summary.file_count': 'file_count',\n",
    "    }) \\\n",
    "    [[\n",
    "        'project_id',\n",
    "        'project_name',\n",
    "        'primary_site',\n",
    "        'disease_type',\n",
    "        'experimental_strategies',\n",
    "        'data_categories',\n",
    "        'case_count',\n",
    "        'file_count',\n",
    "    ]]\n",
    "\n",
    "# Store the project metadata DataFrame in a CSV file\n",
    "df_project.to_csv(BRCA_RAW_FILES_PATHS['project'], index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>project_id</th>\n",
       "      <th>project_name</th>\n",
       "      <th>primary_site</th>\n",
       "      <th>disease_type</th>\n",
       "      <th>experimental_strategies</th>\n",
       "      <th>data_categories</th>\n",
       "      <th>case_count</th>\n",
       "      <th>file_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TCGA-BRCA</td>\n",
       "      <td>Breast Invasive Carcinoma</td>\n",
       "      <td>[Breast]</td>\n",
       "      <td>[Fibroepithelial Neoplasms, Adnexal and Skin Appendage Neoplasms, Adenomas and Adenocarcinomas, Cystic, Mucinous and Serous Neoplasms, Basal Cell Neoplasms, Squamous Cell Neoplasms, Epithelial Neoplasms, NOS, Complex Epithelial Neoplasms, Ductal and Lobular Neoplasms]</td>\n",
       "      <td>[{'file_count': 11079, 'case_count': 1095, 'experimental_strategy': 'RNA-Seq'}, {'file_count': 17049, 'case_count': 1072, 'experimental_strategy': 'WXS'}, {'file_count': 3621, 'case_count': 1079, 'experimental_strategy': 'miRNA-Seq'}, {'file_count': 12688, 'case_count': 952, 'experimental_strategy': 'WGS'}, {'file_count': 75, 'case_count': 74, 'experimental_strategy': 'ATAC-Seq'}, {'file_count': 14329, 'case_count': 1098, 'experimental_strategy': 'Genotyping Array'}, {'file_count': 3714, 'case_count': 1097, 'experimental_strategy': 'Methylation Array'}, {'file_count': 919, 'case_count': 881, 'experimental_strategy': 'Reverse Phase Protein Array'}, {'file_count': 1133, 'case_count': 1062, 'experimental_strategy': 'Diagnostic Slide'}, {'file_count': 1979, 'case_count': 1093, 'experimental_strategy': 'Tissue Slide'}]</td>\n",
       "      <td>[{'file_count': 21437, 'case_count': 1098, 'data_category': 'Simple Nucleotide Variation'}, {'file_count': 9282, 'case_count': 1098, 'data_category': 'Sequencing Reads'}, {'file_count': 5317, 'case_count': 1098, 'data_category': 'Biospecimen'}, {'file_count': 2288, 'case_count': 1098, 'data_category': 'Clinical'}, {'file_count': 14346, 'case_count': 1098, 'data_category': 'Copy Number Variation'}, {'file_count': 4876, 'case_count': 1097, 'data_category': 'Transcriptome Profiling'}, {'file_count': 3714, 'case_count': 1097, 'data_category': 'DNA Methylation'}, {'file_count': 919, 'case_count': 881, 'data_category': 'Proteome Profiling'}, {'file_count': 3128, 'case_count': 927, 'data_category': 'Somatic Structural Variation'}, {'file_count': 5772, 'case_count': 1098, 'data_category': 'Structural Variation'}]</td>\n",
       "      <td>1098</td>\n",
       "      <td>71079</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  project_id               project_name primary_site  \\\n",
       "0  TCGA-BRCA  Breast Invasive Carcinoma     [Breast]   \n",
       "\n",
       "                                                                                                                                                                                                                                                                   disease_type  \\\n",
       "0  [Fibroepithelial Neoplasms, Adnexal and Skin Appendage Neoplasms, Adenomas and Adenocarcinomas, Cystic, Mucinous and Serous Neoplasms, Basal Cell Neoplasms, Squamous Cell Neoplasms, Epithelial Neoplasms, NOS, Complex Epithelial Neoplasms, Ductal and Lobular Neoplasms]   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     experimental_strategies  \\\n",
       "0  [{'file_count': 11079, 'case_count': 1095, 'experimental_strategy': 'RNA-Seq'}, {'file_count': 17049, 'case_count': 1072, 'experimental_strategy': 'WXS'}, {'file_count': 3621, 'case_count': 1079, 'experimental_strategy': 'miRNA-Seq'}, {'file_count': 12688, 'case_count': 952, 'experimental_strategy': 'WGS'}, {'file_count': 75, 'case_count': 74, 'experimental_strategy': 'ATAC-Seq'}, {'file_count': 14329, 'case_count': 1098, 'experimental_strategy': 'Genotyping Array'}, {'file_count': 3714, 'case_count': 1097, 'experimental_strategy': 'Methylation Array'}, {'file_count': 919, 'case_count': 881, 'experimental_strategy': 'Reverse Phase Protein Array'}, {'file_count': 1133, 'case_count': 1062, 'experimental_strategy': 'Diagnostic Slide'}, {'file_count': 1979, 'case_count': 1093, 'experimental_strategy': 'Tissue Slide'}]   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    data_categories  \\\n",
       "0  [{'file_count': 21437, 'case_count': 1098, 'data_category': 'Simple Nucleotide Variation'}, {'file_count': 9282, 'case_count': 1098, 'data_category': 'Sequencing Reads'}, {'file_count': 5317, 'case_count': 1098, 'data_category': 'Biospecimen'}, {'file_count': 2288, 'case_count': 1098, 'data_category': 'Clinical'}, {'file_count': 14346, 'case_count': 1098, 'data_category': 'Copy Number Variation'}, {'file_count': 4876, 'case_count': 1097, 'data_category': 'Transcriptome Profiling'}, {'file_count': 3714, 'case_count': 1097, 'data_category': 'DNA Methylation'}, {'file_count': 919, 'case_count': 881, 'data_category': 'Proteome Profiling'}, {'file_count': 3128, 'case_count': 927, 'data_category': 'Somatic Structural Variation'}, {'file_count': 5772, 'case_count': 1098, 'data_category': 'Structural Variation'}]   \n",
       "\n",
       "   case_count  file_count  \n",
       "0        1098       71079  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the TCGA-BRCA project metadata DataFrame\n",
    "pd.set_option('display.max_colwidth', 900)\n",
    "df_project"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Cases Endpoint"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Endpoint Response"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Fields of interest for the cases endpoint request\n",
    "fields = ['disease_type', 'submitter_id',]\n",
    "\n",
    "# Filter to be used in the cases endpoint request\n",
    "filters = {\n",
    "    'op': 'in',\n",
    "    'content': {\n",
    "        'field': 'project.project_id',\n",
    "        'value': BRCA_PREPROCESSING_PARAMETERS['project_ids']\n",
    "    }\n",
    "}\n",
    "\n",
    "# Get the response from the cases endpoint request\n",
    "cases_response = gdc_api_request(\n",
    "    endpoint=GDC_API_ENDPOINTS['cases'],\n",
    "    fields=fields,\n",
    "    filters=filters,\n",
    "    size=sum(df_project['case_count'])\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Cases DataFrame"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create the TCGA-BRCA cases metadata DataFrame\n",
    "df_cases = pd.json_normalize(cases_response) \\\n",
    "    .rename(columns={'id': 'case_id'}) \\\n",
    "    [['case_id', 'submitter_id', 'disease_type']]\n",
    "    \n",
    "# Store the cases metadata DataFrame in a CSV file\n",
    "df_cases.to_csv(BRCA_RAW_FILES_PATHS['cases'], index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>case_id</th>\n",
       "      <th>submitter_id</th>\n",
       "      <th>disease_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>b1d44c81-747d-471f-9093-aeb262a17975</td>\n",
       "      <td>TCGA-Z7-A8R6</td>\n",
       "      <td>Epithelial Neoplasms, NOS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>b205bba0-1870-4458-9088-8817e20389fe</td>\n",
       "      <td>TCGA-A8-A09A</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>b205c89f-af62-4186-acad-ed23d243fa98</td>\n",
       "      <td>TCGA-A2-A0YL</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>b26d41cd-393b-4cd4-8925-a9488f7de576</td>\n",
       "      <td>TCGA-C8-A1HE</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>b2aac45b-2073-4c7a-adb9-769a4fdcc111</td>\n",
       "      <td>TCGA-E9-A1NH</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1093</th>\n",
       "      <td>af5453a9-cf1f-40de-aec4-0e0710908fb7</td>\n",
       "      <td>TCGA-A8-A0AD</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1094</th>\n",
       "      <td>af97e043-88cb-4f99-8f8b-9bcbcccdf842</td>\n",
       "      <td>TCGA-EW-A2FR</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1095</th>\n",
       "      <td>b0700958-5f90-4546-b35f-635cd506889b</td>\n",
       "      <td>TCGA-PE-A5DD</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1096</th>\n",
       "      <td>b0bcb829-cee0-4247-bb75-093a5bea89ee</td>\n",
       "      <td>TCGA-E2-A105</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1097</th>\n",
       "      <td>b0f8d698-a30e-4d8d-b0a2-a5a01fac8406</td>\n",
       "      <td>TCGA-A2-A0T4</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1098 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   case_id  submitter_id  \\\n",
       "0     b1d44c81-747d-471f-9093-aeb262a17975  TCGA-Z7-A8R6   \n",
       "1     b205bba0-1870-4458-9088-8817e20389fe  TCGA-A8-A09A   \n",
       "2     b205c89f-af62-4186-acad-ed23d243fa98  TCGA-A2-A0YL   \n",
       "3     b26d41cd-393b-4cd4-8925-a9488f7de576  TCGA-C8-A1HE   \n",
       "4     b2aac45b-2073-4c7a-adb9-769a4fdcc111  TCGA-E9-A1NH   \n",
       "...                                    ...           ...   \n",
       "1093  af5453a9-cf1f-40de-aec4-0e0710908fb7  TCGA-A8-A0AD   \n",
       "1094  af97e043-88cb-4f99-8f8b-9bcbcccdf842  TCGA-EW-A2FR   \n",
       "1095  b0700958-5f90-4546-b35f-635cd506889b  TCGA-PE-A5DD   \n",
       "1096  b0bcb829-cee0-4247-bb75-093a5bea89ee  TCGA-E2-A105   \n",
       "1097  b0f8d698-a30e-4d8d-b0a2-a5a01fac8406  TCGA-A2-A0T4   \n",
       "\n",
       "                      disease_type  \n",
       "0        Epithelial Neoplasms, NOS  \n",
       "1     Ductal and Lobular Neoplasms  \n",
       "2     Ductal and Lobular Neoplasms  \n",
       "3     Ductal and Lobular Neoplasms  \n",
       "4     Ductal and Lobular Neoplasms  \n",
       "...                            ...  \n",
       "1093  Ductal and Lobular Neoplasms  \n",
       "1094  Ductal and Lobular Neoplasms  \n",
       "1095  Ductal and Lobular Neoplasms  \n",
       "1096  Ductal and Lobular Neoplasms  \n",
       "1097  Ductal and Lobular Neoplasms  \n",
       "\n",
       "[1098 rows x 3 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the TCGA-BRCA cases metadata DataFrame\n",
    "df_cases"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Files Endpoint"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Endpoint Response"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Fields of interest for the files endpoint request\n",
    "fields = [\n",
    "    'access',\n",
    "    'cases.case_id',\n",
    "    'cases.samples.sample_type',\n",
    "    'cases.samples.tissue_type',\n",
    "    'data_format',\n",
    "    'data_type',\n",
    "    'experimental_strategy',\n",
    "]\n",
    "\n",
    "# Filter to be used in the files endpoint request\n",
    "filters = {\n",
    "    'op': 'in',\n",
    "    'content': {\n",
    "        'field': 'cases.project.project_id',\n",
    "        'value': BRCA_PREPROCESSING_PARAMETERS['project_ids']\n",
    "    }\n",
    "}\n",
    "\n",
    "# Get the response from the files endpoint request\n",
    "files_response = gdc_api_request(\n",
    "    endpoint=GDC_API_ENDPOINTS['files'],\n",
    "    fields=fields,\n",
    "    filters=filters,\n",
    "    size=sum(df_project['file_count'])\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Files DataFrame"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create the TCGA-BRCA files metadata DataFrame\n",
    "df_files = pd.json_normalize(files_response) \\\n",
    "    .rename(columns={'id': 'file_id'})\n",
    "\n",
    "# Explode lists with dictionaries containing cases metadata\n",
    "df_files = df_files.explode('cases')\n",
    "\n",
    "# Explode dictionary contents with cases metadata into columns\n",
    "df_files = pd.concat(\n",
    "    objs=[\n",
    "        df_files.reset_index(drop=True),\n",
    "        pd.json_normalize(df_files['cases'])\n",
    "    ],\n",
    "    axis='columns'\n",
    ")\n",
    "\n",
    "# Reorganize the DataFrame columns\n",
    "df_files = df_files \\\n",
    "    .drop(columns=['cases']) \\\n",
    "    [[\n",
    "        'file_id',\n",
    "        'case_id',\n",
    "        'access',\n",
    "        'experimental_strategy',\n",
    "        'data_type',\n",
    "        'data_format',\n",
    "        'samples',\n",
    "    ]]\n",
    "\n",
    "# Store the files metadata DataFrame in a CSV file\n",
    "df_files.to_csv(BRCA_RAW_FILES_PATHS['files'], index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>file_id</th>\n",
       "      <th>case_id</th>\n",
       "      <th>access</th>\n",
       "      <th>experimental_strategy</th>\n",
       "      <th>data_type</th>\n",
       "      <th>data_format</th>\n",
       "      <th>samples</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>75a8669a-692c-468c-8469-d2e61f6d37d8</td>\n",
       "      <td>cc348a26-ee11-47a4-8b51-de922967e175</td>\n",
       "      <td>open</td>\n",
       "      <td>Methylation Array</td>\n",
       "      <td>Masked Intensities</td>\n",
       "      <td>IDAT</td>\n",
       "      <td>[{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>7abd94ba-570e-43d4-9360-ad2d8399c960</td>\n",
       "      <td>f2bbfa9d-9a9d-4f46-9fde-378e4c44e2ad</td>\n",
       "      <td>controlled</td>\n",
       "      <td>WGS</td>\n",
       "      <td>Structural Rearrangement</td>\n",
       "      <td>BEDPE</td>\n",
       "      <td>[{'sample_type': 'Blood Derived Normal', 'tissue_type': 'Normal'}, {'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>97ef35dc-9c03-49ce-8103-5e8c55be8805</td>\n",
       "      <td>f2bbfa9d-9a9d-4f46-9fde-378e4c44e2ad</td>\n",
       "      <td>controlled</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>Transcript Fusion</td>\n",
       "      <td>TSV</td>\n",
       "      <td>[{'sample_type': 'Solid Tissue Normal', 'tissue_type': 'Normal'}]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0518551d-4df2-4124-b68d-494200c5586b</td>\n",
       "      <td>a6edb6ca-ae9f-4da7-8ebe-92d83d2987fb</td>\n",
       "      <td>open</td>\n",
       "      <td>WXS</td>\n",
       "      <td>Masked Somatic Mutation</td>\n",
       "      <td>MAF</td>\n",
       "      <td>[{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}, {'sample_type': 'Blood Derived Normal', 'tissue_type': 'Normal'}]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2044bb65-05cf-4ef7-9543-3b7c5c2ff4d5</td>\n",
       "      <td>ec0ab947-9341-4fff-bda4-fdfb9434d508</td>\n",
       "      <td>open</td>\n",
       "      <td>Methylation Array</td>\n",
       "      <td>Masked Intensities</td>\n",
       "      <td>IDAT</td>\n",
       "      <td>[{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91917</th>\n",
       "      <td>c680a261-9fdf-48dd-9e87-274881d700c8</td>\n",
       "      <td>da70cf7e-0e61-4c72-b4c5-c408569d11b8</td>\n",
       "      <td>open</td>\n",
       "      <td>miRNA-Seq</td>\n",
       "      <td>miRNA Expression Quantification</td>\n",
       "      <td>TXT</td>\n",
       "      <td>[{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91918</th>\n",
       "      <td>ad76f12b-1710-4e83-8cf2-21c1b6863d8e</td>\n",
       "      <td>d9627184-b972-4e00-8c11-b0b946ac357a</td>\n",
       "      <td>controlled</td>\n",
       "      <td>WGS</td>\n",
       "      <td>Aligned Reads</td>\n",
       "      <td>BAM</td>\n",
       "      <td>[{'sample_type': 'Blood Derived Normal', 'tissue_type': 'Normal'}]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91919</th>\n",
       "      <td>42eacf87-e962-4b33-859e-29944271e335</td>\n",
       "      <td>d9627184-b972-4e00-8c11-b0b946ac357a</td>\n",
       "      <td>open</td>\n",
       "      <td>Genotyping Array</td>\n",
       "      <td>Masked Copy Number Segment</td>\n",
       "      <td>TXT</td>\n",
       "      <td>[{'sample_type': 'Blood Derived Normal', 'tissue_type': 'Normal'}]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91920</th>\n",
       "      <td>b49a041c-2f46-412f-b10d-b0dcb40737cb</td>\n",
       "      <td>d9627184-b972-4e00-8c11-b0b946ac357a</td>\n",
       "      <td>open</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Biospecimen Supplement</td>\n",
       "      <td>BCR SSF XML</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91921</th>\n",
       "      <td>84bbf11d-b9ad-4376-b8da-2a21b7734801</td>\n",
       "      <td>db8d38ab-cc61-49ee-863d-40060112f2af</td>\n",
       "      <td>open</td>\n",
       "      <td>Methylation Array</td>\n",
       "      <td>Masked Intensities</td>\n",
       "      <td>IDAT</td>\n",
       "      <td>[{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>91922 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                    file_id  \\\n",
       "0      75a8669a-692c-468c-8469-d2e61f6d37d8   \n",
       "1      7abd94ba-570e-43d4-9360-ad2d8399c960   \n",
       "2      97ef35dc-9c03-49ce-8103-5e8c55be8805   \n",
       "3      0518551d-4df2-4124-b68d-494200c5586b   \n",
       "4      2044bb65-05cf-4ef7-9543-3b7c5c2ff4d5   \n",
       "...                                     ...   \n",
       "91917  c680a261-9fdf-48dd-9e87-274881d700c8   \n",
       "91918  ad76f12b-1710-4e83-8cf2-21c1b6863d8e   \n",
       "91919  42eacf87-e962-4b33-859e-29944271e335   \n",
       "91920  b49a041c-2f46-412f-b10d-b0dcb40737cb   \n",
       "91921  84bbf11d-b9ad-4376-b8da-2a21b7734801   \n",
       "\n",
       "                                    case_id      access experimental_strategy  \\\n",
       "0      cc348a26-ee11-47a4-8b51-de922967e175        open     Methylation Array   \n",
       "1      f2bbfa9d-9a9d-4f46-9fde-378e4c44e2ad  controlled                   WGS   \n",
       "2      f2bbfa9d-9a9d-4f46-9fde-378e4c44e2ad  controlled               RNA-Seq   \n",
       "3      a6edb6ca-ae9f-4da7-8ebe-92d83d2987fb        open                   WXS   \n",
       "4      ec0ab947-9341-4fff-bda4-fdfb9434d508        open     Methylation Array   \n",
       "...                                     ...         ...                   ...   \n",
       "91917  da70cf7e-0e61-4c72-b4c5-c408569d11b8        open             miRNA-Seq   \n",
       "91918  d9627184-b972-4e00-8c11-b0b946ac357a  controlled                   WGS   \n",
       "91919  d9627184-b972-4e00-8c11-b0b946ac357a        open      Genotyping Array   \n",
       "91920  d9627184-b972-4e00-8c11-b0b946ac357a        open                   NaN   \n",
       "91921  db8d38ab-cc61-49ee-863d-40060112f2af        open     Methylation Array   \n",
       "\n",
       "                             data_type  data_format  \\\n",
       "0                   Masked Intensities         IDAT   \n",
       "1             Structural Rearrangement        BEDPE   \n",
       "2                    Transcript Fusion          TSV   \n",
       "3              Masked Somatic Mutation          MAF   \n",
       "4                   Masked Intensities         IDAT   \n",
       "...                                ...          ...   \n",
       "91917  miRNA Expression Quantification          TXT   \n",
       "91918                    Aligned Reads          BAM   \n",
       "91919       Masked Copy Number Segment          TXT   \n",
       "91920           Biospecimen Supplement  BCR SSF XML   \n",
       "91921               Masked Intensities         IDAT   \n",
       "\n",
       "                                                                                                                            samples  \n",
       "0                                                                        [{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]  \n",
       "1      [{'sample_type': 'Blood Derived Normal', 'tissue_type': 'Normal'}, {'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]  \n",
       "2                                                                 [{'sample_type': 'Solid Tissue Normal', 'tissue_type': 'Normal'}]  \n",
       "3      [{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}, {'sample_type': 'Blood Derived Normal', 'tissue_type': 'Normal'}]  \n",
       "4                                                                        [{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]  \n",
       "...                                                                                                                             ...  \n",
       "91917                                                                    [{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]  \n",
       "91918                                                            [{'sample_type': 'Blood Derived Normal', 'tissue_type': 'Normal'}]  \n",
       "91919                                                            [{'sample_type': 'Blood Derived Normal', 'tissue_type': 'Normal'}]  \n",
       "91920                                                                                                                           NaN  \n",
       "91921                                                                    [{'sample_type': 'Primary Tumor', 'tissue_type': 'Tumor'}]  \n",
       "\n",
       "[91922 rows x 7 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the TCGA-BRCA files metadata DataFrame\n",
    "pd.set_option('display.max_colwidth', 300)\n",
    "df_files"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "venv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
