{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# **Filtering TCGA-BRCA Metadata from the GDC API**\n",
    "\n",
    "This notebook produces the AT_FOR family of data artifacts  \n",
    "\n",
    "- TCGA: The Cancer Genome Atlas  \n",
    "- BRCA: Breast Invasive Carcinoma  \n",
    "- GDC: Genomic Data Commons"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Import Libraries and Configurations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import sys\n",
    "from ast import literal_eval\n",
    "\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "\n",
    "# Add project root to Python's path\n",
    "sys.path.append(os.path.abspath(os.path.join('..')))\n",
    "\n",
    "from config import (\n",
    "    BRCA_PREPROCESSING_PARAMETERS,\n",
    "    BRCA_PROCESSED_FILES_PATHS,\n",
    "    BRCA_RAW_FILES_PATHS,\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Load the Metadata from the GDC API"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create a DataFrame for the TCGA-BRCA cases metadata\n",
    "df_api_cases = pd.read_csv(BRCA_RAW_FILES_PATHS['cases'])\n",
    "\n",
    "# Create a DataFrame for the TCGA-BRCA files metadata\n",
    "df_files = pd.read_csv(BRCA_RAW_FILES_PATHS['files'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>case_id</th>\n",
       "      <th>submitter_id</th>\n",
       "      <th>disease_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>b1d44c81-747d-471f-9093-aeb262a17975</td>\n",
       "      <td>TCGA-Z7-A8R6</td>\n",
       "      <td>Epithelial Neoplasms, NOS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>b205bba0-1870-4458-9088-8817e20389fe</td>\n",
       "      <td>TCGA-A8-A09A</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>b205c89f-af62-4186-acad-ed23d243fa98</td>\n",
       "      <td>TCGA-A2-A0YL</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>b26d41cd-393b-4cd4-8925-a9488f7de576</td>\n",
       "      <td>TCGA-C8-A1HE</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>b2aac45b-2073-4c7a-adb9-769a4fdcc111</td>\n",
       "      <td>TCGA-E9-A1NH</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1093</th>\n",
       "      <td>af5453a9-cf1f-40de-aec4-0e0710908fb7</td>\n",
       "      <td>TCGA-A8-A0AD</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1094</th>\n",
       "      <td>af97e043-88cb-4f99-8f8b-9bcbcccdf842</td>\n",
       "      <td>TCGA-EW-A2FR</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1095</th>\n",
       "      <td>b0700958-5f90-4546-b35f-635cd506889b</td>\n",
       "      <td>TCGA-PE-A5DD</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1096</th>\n",
       "      <td>b0bcb829-cee0-4247-bb75-093a5bea89ee</td>\n",
       "      <td>TCGA-E2-A105</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1097</th>\n",
       "      <td>b0f8d698-a30e-4d8d-b0a2-a5a01fac8406</td>\n",
       "      <td>TCGA-A2-A0T4</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1098 rows Ã— 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   case_id  submitter_id  \\\n",
       "0     b1d44c81-747d-471f-9093-aeb262a17975  TCGA-Z7-A8R6   \n",
       "1     b205bba0-1870-4458-9088-8817e20389fe  TCGA-A8-A09A   \n",
       "2     b205c89f-af62-4186-acad-ed23d243fa98  TCGA-A2-A0YL   \n",
       "3     b26d41cd-393b-4cd4-8925-a9488f7de576  TCGA-C8-A1HE   \n",
       "4     b2aac45b-2073-4c7a-adb9-769a4fdcc111  TCGA-E9-A1NH   \n",
       "...                                    ...           ...   \n",
       "1093  af5453a9-cf1f-40de-aec4-0e0710908fb7  TCGA-A8-A0AD   \n",
       "1094  af97e043-88cb-4f99-8f8b-9bcbcccdf842  TCGA-EW-A2FR   \n",
       "1095  b0700958-5f90-4546-b35f-635cd506889b  TCGA-PE-A5DD   \n",
       "1096  b0bcb829-cee0-4247-bb75-093a5bea89ee  TCGA-E2-A105   \n",
       "1097  b0f8d698-a30e-4d8d-b0a2-a5a01fac8406  TCGA-A2-A0T4   \n",
       "\n",
       "                      disease_type  \n",
       "0        Epithelial Neoplasms, NOS  \n",
       "1     Ductal and Lobular Neoplasms  \n",
       "2     Ductal and Lobular Neoplasms  \n",
       "3     Ductal and Lobular Neoplasms  \n",
       "4     Ductal and Lobular Neoplasms  \n",
       "...                            ...  \n",
       "1093  Ductal and Lobular Neoplasms  \n",
       "1094  Ductal and Lobular Neoplasms  \n",
       "1095  Ductal and Lobular Neoplasms  \n",
       "1096  Ductal and Lobular Neoplasms  \n",
       "1097  Ductal and Lobular Neoplasms  \n",
       "\n",
       "[1098 rows x 3 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the DataFrame of TCGA-BRCA cases metadata\n",
    "df_api_cases"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>file_id</th>\n",
       "      <th>case_id</th>\n",
       "      <th>access</th>\n",
       "      <th>experimental_strategy</th>\n",
       "      <th>data_type</th>\n",
       "      <th>data_format</th>\n",
       "      <th>samples</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>75a8669a-692c-468c-8469-d2e61f6d37d8</td>\n",
       "      <td>cc348a26-ee11-47a4-8b51-de922967e175</td>\n",
       "      <td>open</td>\n",
       "      <td>Methylation Array</td>\n",
       "      <td>Masked Intensities</td>\n",
       "      <td>IDAT</td>\n",
       "      <td>[{'sample_type': 'Primary Tumor', 'tissue_type...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>7abd94ba-570e-43d4-9360-ad2d8399c960</td>\n",
       "      <td>f2bbfa9d-9a9d-4f46-9fde-378e4c44e2ad</td>\n",
       "      <td>controlled</td>\n",
       "      <td>WGS</td>\n",
       "      <td>Structural Rearrangement</td>\n",
       "      <td>BEDPE</td>\n",
       "      <td>[{'sample_type': 'Blood Derived Normal', 'tiss...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>97ef35dc-9c03-49ce-8103-5e8c55be8805</td>\n",
       "      <td>f2bbfa9d-9a9d-4f46-9fde-378e4c44e2ad</td>\n",
       "      <td>controlled</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>Transcript Fusion</td>\n",
       "      <td>TSV</td>\n",
       "      <td>[{'sample_type': 'Solid Tissue Normal', 'tissu...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0518551d-4df2-4124-b68d-494200c5586b</td>\n",
       "      <td>a6edb6ca-ae9f-4da7-8ebe-92d83d2987fb</td>\n",
       "      <td>open</td>\n",
       "      <td>WXS</td>\n",
       "      <td>Masked Somatic Mutation</td>\n",
       "      <td>MAF</td>\n",
       "      <td>[{'sample_type': 'Primary Tumor', 'tissue_type...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2044bb65-05cf-4ef7-9543-3b7c5c2ff4d5</td>\n",
       "      <td>ec0ab947-9341-4fff-bda4-fdfb9434d508</td>\n",
       "      <td>open</td>\n",
       "      <td>Methylation Array</td>\n",
       "      <td>Masked Intensities</td>\n",
       "      <td>IDAT</td>\n",
       "      <td>[{'sample_type': 'Primary Tumor', 'tissue_type...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91917</th>\n",
       "      <td>c680a261-9fdf-48dd-9e87-274881d700c8</td>\n",
       "      <td>da70cf7e-0e61-4c72-b4c5-c408569d11b8</td>\n",
       "      <td>open</td>\n",
       "      <td>miRNA-Seq</td>\n",
       "      <td>miRNA Expression Quantification</td>\n",
       "      <td>TXT</td>\n",
       "      <td>[{'sample_type': 'Primary Tumor', 'tissue_type...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91918</th>\n",
       "      <td>ad76f12b-1710-4e83-8cf2-21c1b6863d8e</td>\n",
       "      <td>d9627184-b972-4e00-8c11-b0b946ac357a</td>\n",
       "      <td>controlled</td>\n",
       "      <td>WGS</td>\n",
       "      <td>Aligned Reads</td>\n",
       "      <td>BAM</td>\n",
       "      <td>[{'sample_type': 'Blood Derived Normal', 'tiss...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91919</th>\n",
       "      <td>42eacf87-e962-4b33-859e-29944271e335</td>\n",
       "      <td>d9627184-b972-4e00-8c11-b0b946ac357a</td>\n",
       "      <td>open</td>\n",
       "      <td>Genotyping Array</td>\n",
       "      <td>Masked Copy Number Segment</td>\n",
       "      <td>TXT</td>\n",
       "      <td>[{'sample_type': 'Blood Derived Normal', 'tiss...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91920</th>\n",
       "      <td>b49a041c-2f46-412f-b10d-b0dcb40737cb</td>\n",
       "      <td>d9627184-b972-4e00-8c11-b0b946ac357a</td>\n",
       "      <td>open</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Biospecimen Supplement</td>\n",
       "      <td>BCR SSF XML</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91921</th>\n",
       "      <td>84bbf11d-b9ad-4376-b8da-2a21b7734801</td>\n",
       "      <td>db8d38ab-cc61-49ee-863d-40060112f2af</td>\n",
       "      <td>open</td>\n",
       "      <td>Methylation Array</td>\n",
       "      <td>Masked Intensities</td>\n",
       "      <td>IDAT</td>\n",
       "      <td>[{'sample_type': 'Primary Tumor', 'tissue_type...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>91922 rows Ã— 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                    file_id  \\\n",
       "0      75a8669a-692c-468c-8469-d2e61f6d37d8   \n",
       "1      7abd94ba-570e-43d4-9360-ad2d8399c960   \n",
       "2      97ef35dc-9c03-49ce-8103-5e8c55be8805   \n",
       "3      0518551d-4df2-4124-b68d-494200c5586b   \n",
       "4      2044bb65-05cf-4ef7-9543-3b7c5c2ff4d5   \n",
       "...                                     ...   \n",
       "91917  c680a261-9fdf-48dd-9e87-274881d700c8   \n",
       "91918  ad76f12b-1710-4e83-8cf2-21c1b6863d8e   \n",
       "91919  42eacf87-e962-4b33-859e-29944271e335   \n",
       "91920  b49a041c-2f46-412f-b10d-b0dcb40737cb   \n",
       "91921  84bbf11d-b9ad-4376-b8da-2a21b7734801   \n",
       "\n",
       "                                    case_id      access experimental_strategy  \\\n",
       "0      cc348a26-ee11-47a4-8b51-de922967e175        open     Methylation Array   \n",
       "1      f2bbfa9d-9a9d-4f46-9fde-378e4c44e2ad  controlled                   WGS   \n",
       "2      f2bbfa9d-9a9d-4f46-9fde-378e4c44e2ad  controlled               RNA-Seq   \n",
       "3      a6edb6ca-ae9f-4da7-8ebe-92d83d2987fb        open                   WXS   \n",
       "4      ec0ab947-9341-4fff-bda4-fdfb9434d508        open     Methylation Array   \n",
       "...                                     ...         ...                   ...   \n",
       "91917  da70cf7e-0e61-4c72-b4c5-c408569d11b8        open             miRNA-Seq   \n",
       "91918  d9627184-b972-4e00-8c11-b0b946ac357a  controlled                   WGS   \n",
       "91919  d9627184-b972-4e00-8c11-b0b946ac357a        open      Genotyping Array   \n",
       "91920  d9627184-b972-4e00-8c11-b0b946ac357a        open                   NaN   \n",
       "91921  db8d38ab-cc61-49ee-863d-40060112f2af        open     Methylation Array   \n",
       "\n",
       "                             data_type  data_format  \\\n",
       "0                   Masked Intensities         IDAT   \n",
       "1             Structural Rearrangement        BEDPE   \n",
       "2                    Transcript Fusion          TSV   \n",
       "3              Masked Somatic Mutation          MAF   \n",
       "4                   Masked Intensities         IDAT   \n",
       "...                                ...          ...   \n",
       "91917  miRNA Expression Quantification          TXT   \n",
       "91918                    Aligned Reads          BAM   \n",
       "91919       Masked Copy Number Segment          TXT   \n",
       "91920           Biospecimen Supplement  BCR SSF XML   \n",
       "91921               Masked Intensities         IDAT   \n",
       "\n",
       "                                                 samples  \n",
       "0      [{'sample_type': 'Primary Tumor', 'tissue_type...  \n",
       "1      [{'sample_type': 'Blood Derived Normal', 'tiss...  \n",
       "2      [{'sample_type': 'Solid Tissue Normal', 'tissu...  \n",
       "3      [{'sample_type': 'Primary Tumor', 'tissue_type...  \n",
       "4      [{'sample_type': 'Primary Tumor', 'tissue_type...  \n",
       "...                                                  ...  \n",
       "91917  [{'sample_type': 'Primary Tumor', 'tissue_type...  \n",
       "91918  [{'sample_type': 'Blood Derived Normal', 'tiss...  \n",
       "91919  [{'sample_type': 'Blood Derived Normal', 'tiss...  \n",
       "91920                                                NaN  \n",
       "91921  [{'sample_type': 'Primary Tumor', 'tissue_type...  \n",
       "\n",
       "[91922 rows x 7 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the DataFrame of TCGA-BRCA files metadata\n",
    "df_files"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Load and Flag the TCGA-BRCA Paper Data\n",
    "The Cancer Genome Atlas Network. Comprehensive molecular portraits of human breast tumours. Nature 490, 61â€“70 (2012). https://doi.org/10.1038/nature11412."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create the DataFrame of 'Supplementary Table 1' from the TCGA-BRCA paper\n",
    "df_paper_cases = pd.read_excel(\n",
    "    BRCA_RAW_FILES_PATHS['paper'], sheet_name='SuppTable1', skiprows=1\n",
    ")\n",
    "\n",
    "# Standardize DataFrame column names\n",
    "columns = dict()\n",
    "for name in list(df_paper_cases.columns):\n",
    "    columns[name] = name.replace(' ', '_').lower()\n",
    "df_paper_cases = df_paper_cases.rename(columns=columns)\n",
    "\n",
    "# Flag the cases of interest, i.e., those classified into specific molecular\n",
    "# subtypes of breast cancer by the PAM50 signature\n",
    "molecular_subtypes = BRCA_PREPROCESSING_PARAMETERS['molecular_subtypes']\n",
    "df_paper_cases['is_case_of_interest'] = np.where(\n",
    "    df_paper_cases['pam50_mrna'].isin(molecular_subtypes), 1, 0\n",
    ")\n",
    "\n",
    "# Store the DataFrame of flagged cases in a CSV file\n",
    "df_paper_cases.to_csv(BRCA_PROCESSED_FILES_PATHS['paper'], index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>complete_tcga_id</th>\n",
       "      <th>gender</th>\n",
       "      <th>age_at_initial_pathologic_diagnosis</th>\n",
       "      <th>er_status</th>\n",
       "      <th>pr_status</th>\n",
       "      <th>her2_final_status</th>\n",
       "      <th>tumor</th>\n",
       "      <th>tumor--t1_coded</th>\n",
       "      <th>node</th>\n",
       "      <th>node-coded</th>\n",
       "      <th>...</th>\n",
       "      <th>sigclust_unsupervised_mrna</th>\n",
       "      <th>sigclust_intrinsic_mrna</th>\n",
       "      <th>mirna_clusters</th>\n",
       "      <th>methylation_clusters</th>\n",
       "      <th>rppa_clusters</th>\n",
       "      <th>cn_clusters</th>\n",
       "      <th>integrated_clusters_(with_pam50)</th>\n",
       "      <th>integrated_clusters_(no_exp)</th>\n",
       "      <th>integrated_clusters_(unsup_exp)</th>\n",
       "      <th>is_case_of_interest</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TCGA-A2-A0T2</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>66.0</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>T3</td>\n",
       "      <td>T_Other</td>\n",
       "      <td>N3</td>\n",
       "      <td>Positive</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-13.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Basal</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TCGA-A2-A04P</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>36.0</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>T2</td>\n",
       "      <td>T_Other</td>\n",
       "      <td>N3</td>\n",
       "      <td>Positive</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-13.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Basal</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TCGA-A1-A0SK</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>54.0</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>T2</td>\n",
       "      <td>T_Other</td>\n",
       "      <td>N0</td>\n",
       "      <td>Negative</td>\n",
       "      <td>...</td>\n",
       "      <td>-6.0</td>\n",
       "      <td>-13.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Basal</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TCGA-A2-A0CM</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>40.0</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>T2</td>\n",
       "      <td>T_Other</td>\n",
       "      <td>N0</td>\n",
       "      <td>Negative</td>\n",
       "      <td>...</td>\n",
       "      <td>-12.0</td>\n",
       "      <td>-13.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>Basal</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TCGA-AR-A1AR</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>50.0</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>T1</td>\n",
       "      <td>T1</td>\n",
       "      <td>N2</td>\n",
       "      <td>Positive</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-13.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>820</th>\n",
       "      <td>TCGA-AC-A2FF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>821</th>\n",
       "      <td>TCGA-AC-A2FB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>822</th>\n",
       "      <td>TCGA-AC-A2FG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>823</th>\n",
       "      <td>TCGA-GI-A2C8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>824</th>\n",
       "      <td>TCGA-E9-A295</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>825 rows Ã— 31 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    complete_tcga_id  gender  age_at_initial_pathologic_diagnosis er_status  \\\n",
       "0       TCGA-A2-A0T2  FEMALE                                 66.0  Negative   \n",
       "1       TCGA-A2-A04P  FEMALE                                 36.0  Negative   \n",
       "2       TCGA-A1-A0SK  FEMALE                                 54.0  Negative   \n",
       "3       TCGA-A2-A0CM  FEMALE                                 40.0  Negative   \n",
       "4       TCGA-AR-A1AR  FEMALE                                 50.0  Negative   \n",
       "..               ...     ...                                  ...       ...   \n",
       "820     TCGA-AC-A2FF     NaN                                  NaN       NaN   \n",
       "821     TCGA-AC-A2FB     NaN                                  NaN       NaN   \n",
       "822     TCGA-AC-A2FG     NaN                                  NaN       NaN   \n",
       "823     TCGA-GI-A2C8     NaN                                  NaN       NaN   \n",
       "824     TCGA-E9-A295     NaN                                  NaN       NaN   \n",
       "\n",
       "    pr_status her2_final_status tumor tumor--t1_coded node node-coded  ...  \\\n",
       "0    Negative          Negative    T3         T_Other   N3   Positive  ...   \n",
       "1    Negative          Negative    T2         T_Other   N3   Positive  ...   \n",
       "2    Negative          Negative    T2         T_Other   N0   Negative  ...   \n",
       "3    Negative          Negative    T2         T_Other   N0   Negative  ...   \n",
       "4    Negative          Negative    T1              T1   N2   Positive  ...   \n",
       "..        ...               ...   ...             ...  ...        ...  ...   \n",
       "820       NaN               NaN   NaN             NaN  NaN        NaN  ...   \n",
       "821       NaN               NaN   NaN             NaN  NaN        NaN  ...   \n",
       "822       NaN               NaN   NaN             NaN  NaN        NaN  ...   \n",
       "823       NaN               NaN   NaN             NaN  NaN        NaN  ...   \n",
       "824       NaN               NaN   NaN             NaN  NaN        NaN  ...   \n",
       "\n",
       "    sigclust_unsupervised_mrna sigclust_intrinsic_mrna mirna_clusters  \\\n",
       "0                          0.0                   -13.0            3.0   \n",
       "1                          0.0                   -13.0            5.0   \n",
       "2                         -6.0                   -13.0            5.0   \n",
       "3                        -12.0                   -13.0            4.0   \n",
       "4                          0.0                   -13.0            5.0   \n",
       "..                         ...                     ...            ...   \n",
       "820                        NaN                     NaN            NaN   \n",
       "821                        NaN                     NaN            NaN   \n",
       "822                        NaN                     NaN            NaN   \n",
       "823                        NaN                     NaN            NaN   \n",
       "824                        NaN                     NaN            NaN   \n",
       "\n",
       "    methylation_clusters rppa_clusters cn_clusters  \\\n",
       "0                    5.0         Basal         3.0   \n",
       "1                    5.0         Basal         1.0   \n",
       "2                    5.0         Basal         1.0   \n",
       "3                    4.0         Basal         4.0   \n",
       "4                    5.0           NaN         1.0   \n",
       "..                   ...           ...         ...   \n",
       "820                  1.0           NaN         1.0   \n",
       "821                  1.0           NaN         3.0   \n",
       "822                  1.0           NaN         2.0   \n",
       "823                  3.0           NaN         3.0   \n",
       "824                  4.0           NaN         3.0   \n",
       "\n",
       "    integrated_clusters_(with_pam50)  integrated_clusters_(no_exp)  \\\n",
       "0                                2.0                           2.0   \n",
       "1                                2.0                           2.0   \n",
       "2                                2.0                           2.0   \n",
       "3                                2.0                           1.0   \n",
       "4                                NaN                           NaN   \n",
       "..                               ...                           ...   \n",
       "820                              NaN                           NaN   \n",
       "821                              NaN                           NaN   \n",
       "822                              NaN                           NaN   \n",
       "823                              NaN                           NaN   \n",
       "824                              NaN                           NaN   \n",
       "\n",
       "     integrated_clusters_(unsup_exp)  is_case_of_interest  \n",
       "0                                2.0                    1  \n",
       "1                                2.0                    1  \n",
       "2                                2.0                    1  \n",
       "3                                1.0                    1  \n",
       "4                                NaN                    1  \n",
       "..                               ...                  ...  \n",
       "820                              NaN                    0  \n",
       "821                              NaN                    0  \n",
       "822                              NaN                    0  \n",
       "823                              NaN                    0  \n",
       "824                              NaN                    0  \n",
       "\n",
       "[825 rows x 31 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the DataFrame of flagged cases data from the TCGA-BRCA paper\n",
    "df_paper_cases"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Filter the Cases and Files of Interest"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Potential Cases of Interest"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Filter the cases of interest from the paper\n",
    "df_paper_cases_of_interest = df_paper_cases \\\n",
    "    .query('is_case_of_interest == 1') \\\n",
    "    [['complete_tcga_id', 'pam50_mrna']]\n",
    "\n",
    "# Retrieve the molecular subtype classification of the cases\n",
    "df_cases = df_api_cases \\\n",
    "    .merge(\n",
    "        right=df_paper_cases_of_interest,\n",
    "        left_on='submitter_id',\n",
    "        right_on='complete_tcga_id',\n",
    "        how='left'\n",
    "    ) \\\n",
    "    [['case_id', 'submitter_id', 'disease_type', 'pam50_mrna']] \\\n",
    "    .fillna(value={'pam50_mrna': ''})\n",
    "    \n",
    "# Flag the cases of interest, i.e., those classified into a molecular\n",
    "# subtype of breast cancer and a specific type of disease\n",
    "disease_types = BRCA_PREPROCESSING_PARAMETERS['disease_types']\n",
    "molecular_subtypes = BRCA_PREPROCESSING_PARAMETERS['molecular_subtypes']\n",
    "df_cases['is_case_of_interest'] = np.where(\n",
    "    (df_cases['disease_type'].isin(disease_types) \n",
    "     & df_cases['pam50_mrna'].isin(molecular_subtypes)), 1, 0\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>case_id</th>\n",
       "      <th>submitter_id</th>\n",
       "      <th>disease_type</th>\n",
       "      <th>pam50_mrna</th>\n",
       "      <th>is_case_of_interest</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>b1d44c81-747d-471f-9093-aeb262a17975</td>\n",
       "      <td>TCGA-Z7-A8R6</td>\n",
       "      <td>Epithelial Neoplasms, NOS</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>b205bba0-1870-4458-9088-8817e20389fe</td>\n",
       "      <td>TCGA-A8-A09A</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>b205c89f-af62-4186-acad-ed23d243fa98</td>\n",
       "      <td>TCGA-A2-A0YL</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>b26d41cd-393b-4cd4-8925-a9488f7de576</td>\n",
       "      <td>TCGA-C8-A1HE</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>b2aac45b-2073-4c7a-adb9-769a4fdcc111</td>\n",
       "      <td>TCGA-E9-A1NH</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1093</th>\n",
       "      <td>af5453a9-cf1f-40de-aec4-0e0710908fb7</td>\n",
       "      <td>TCGA-A8-A0AD</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1094</th>\n",
       "      <td>af97e043-88cb-4f99-8f8b-9bcbcccdf842</td>\n",
       "      <td>TCGA-EW-A2FR</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1095</th>\n",
       "      <td>b0700958-5f90-4546-b35f-635cd506889b</td>\n",
       "      <td>TCGA-PE-A5DD</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1096</th>\n",
       "      <td>b0bcb829-cee0-4247-bb75-093a5bea89ee</td>\n",
       "      <td>TCGA-E2-A105</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1097</th>\n",
       "      <td>b0f8d698-a30e-4d8d-b0a2-a5a01fac8406</td>\n",
       "      <td>TCGA-A2-A0T4</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal B</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1098 rows Ã— 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   case_id  submitter_id  \\\n",
       "0     b1d44c81-747d-471f-9093-aeb262a17975  TCGA-Z7-A8R6   \n",
       "1     b205bba0-1870-4458-9088-8817e20389fe  TCGA-A8-A09A   \n",
       "2     b205c89f-af62-4186-acad-ed23d243fa98  TCGA-A2-A0YL   \n",
       "3     b26d41cd-393b-4cd4-8925-a9488f7de576  TCGA-C8-A1HE   \n",
       "4     b2aac45b-2073-4c7a-adb9-769a4fdcc111  TCGA-E9-A1NH   \n",
       "...                                    ...           ...   \n",
       "1093  af5453a9-cf1f-40de-aec4-0e0710908fb7  TCGA-A8-A0AD   \n",
       "1094  af97e043-88cb-4f99-8f8b-9bcbcccdf842  TCGA-EW-A2FR   \n",
       "1095  b0700958-5f90-4546-b35f-635cd506889b  TCGA-PE-A5DD   \n",
       "1096  b0bcb829-cee0-4247-bb75-093a5bea89ee  TCGA-E2-A105   \n",
       "1097  b0f8d698-a30e-4d8d-b0a2-a5a01fac8406  TCGA-A2-A0T4   \n",
       "\n",
       "                      disease_type pam50_mrna  is_case_of_interest  \n",
       "0        Epithelial Neoplasms, NOS                               0  \n",
       "1     Ductal and Lobular Neoplasms  Luminal A                    1  \n",
       "2     Ductal and Lobular Neoplasms  Luminal A                    1  \n",
       "3     Ductal and Lobular Neoplasms                               0  \n",
       "4     Ductal and Lobular Neoplasms                               0  \n",
       "...                            ...        ...                  ...  \n",
       "1093  Ductal and Lobular Neoplasms  Luminal A                    1  \n",
       "1094  Ductal and Lobular Neoplasms                               0  \n",
       "1095  Ductal and Lobular Neoplasms                               0  \n",
       "1096  Ductal and Lobular Neoplasms  Luminal A                    1  \n",
       "1097  Ductal and Lobular Neoplasms  Luminal B                    1  \n",
       "\n",
       "[1098 rows x 5 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the DataFrame of potential flagged cases\n",
    "df_cases"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Potential Files of Interest"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Retrieve the flag of interest of the cases\n",
    "df_files = df_files \\\n",
    "    .merge(\n",
    "        right=df_cases,\n",
    "        left_on='case_id',\n",
    "        right_on='case_id',\n",
    "        how='inner'\n",
    "    ) \\\n",
    "    .drop(columns=['submitter_id', 'disease_type', 'pam50_mrna'])\n",
    "\n",
    "# Filter the files based on data type and case\n",
    "data_types = BRCA_PREPROCESSING_PARAMETERS['data_types']\n",
    "df_files_of_interest = df_files \\\n",
    "    .query('(data_type == @data_types) and (is_case_of_interest == 1)')\n",
    "\n",
    "# Explode the lists with dictionaries containing samples metadata\n",
    "df_files_of_interest.loc[:, 'samples'] = \\\n",
    "    df_files_of_interest['samples'].apply(literal_eval)\n",
    "df_files_of_interest = df_files_of_interest.explode('samples')\n",
    "df_files_of_interest = pd.concat(\n",
    "    objs=[\n",
    "        df_files_of_interest.reset_index(drop=True),\n",
    "        pd.json_normalize(df_files_of_interest['samples'])\n",
    "    ],\n",
    "    axis='columns'\n",
    ")\n",
    "\n",
    "# Rearrange the DataFrame columns\n",
    "df_files_of_interest = df_files_of_interest \\\n",
    "    .drop(columns=[\n",
    "        'access',\n",
    "        'experimental_strategy',\n",
    "        'data_type',\n",
    "        'data_format',\n",
    "        'samples',\n",
    "        'is_case_of_interest',\n",
    "    ])\n",
    "    \n",
    "# Flag the files of interest, i.e., those related to a case\n",
    "# of interest, specific data type, and specific sample type\n",
    "sample_types = BRCA_PREPROCESSING_PARAMETERS['sample_types']\n",
    "df_files_of_interest['is_file_of_interest'] = np.where(\n",
    "    df_files_of_interest['sample_type'].isin(sample_types), 1, 0\n",
    ")\n",
    "\n",
    "# Extend the flag column to all files\n",
    "df_files = df_files \\\n",
    "    .merge(\n",
    "        right=df_files_of_interest,\n",
    "        left_on=['file_id', 'case_id'],\n",
    "        right_on=['file_id', 'case_id'],\n",
    "        how='left'\n",
    "    ) \\\n",
    "    .drop(columns=['is_case_of_interest']) \\\n",
    "    .fillna(value={'is_file_of_interest': 0}) \\\n",
    "    .astype({'is_file_of_interest': 'int'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>file_id</th>\n",
       "      <th>case_id</th>\n",
       "      <th>access</th>\n",
       "      <th>experimental_strategy</th>\n",
       "      <th>data_type</th>\n",
       "      <th>data_format</th>\n",
       "      <th>samples</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>tissue_type</th>\n",
       "      <th>is_file_of_interest</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>75a8669a-692c-468c-8469-d2e61f6d37d8</td>\n",
       "      <td>cc348a26-ee11-47a4-8b51-de922967e175</td>\n",
       "      <td>open</td>\n",
       "      <td>Methylation Array</td>\n",
       "      <td>Masked Intensities</td>\n",
       "      <td>IDAT</td>\n",
       "      <td>[{'sample_type': 'Primary Tumor', 'tissue_type...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>7abd94ba-570e-43d4-9360-ad2d8399c960</td>\n",
       "      <td>f2bbfa9d-9a9d-4f46-9fde-378e4c44e2ad</td>\n",
       "      <td>controlled</td>\n",
       "      <td>WGS</td>\n",
       "      <td>Structural Rearrangement</td>\n",
       "      <td>BEDPE</td>\n",
       "      <td>[{'sample_type': 'Blood Derived Normal', 'tiss...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>97ef35dc-9c03-49ce-8103-5e8c55be8805</td>\n",
       "      <td>f2bbfa9d-9a9d-4f46-9fde-378e4c44e2ad</td>\n",
       "      <td>controlled</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>Transcript Fusion</td>\n",
       "      <td>TSV</td>\n",
       "      <td>[{'sample_type': 'Solid Tissue Normal', 'tissu...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0518551d-4df2-4124-b68d-494200c5586b</td>\n",
       "      <td>a6edb6ca-ae9f-4da7-8ebe-92d83d2987fb</td>\n",
       "      <td>open</td>\n",
       "      <td>WXS</td>\n",
       "      <td>Masked Somatic Mutation</td>\n",
       "      <td>MAF</td>\n",
       "      <td>[{'sample_type': 'Primary Tumor', 'tissue_type...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2044bb65-05cf-4ef7-9543-3b7c5c2ff4d5</td>\n",
       "      <td>ec0ab947-9341-4fff-bda4-fdfb9434d508</td>\n",
       "      <td>open</td>\n",
       "      <td>Methylation Array</td>\n",
       "      <td>Masked Intensities</td>\n",
       "      <td>IDAT</td>\n",
       "      <td>[{'sample_type': 'Primary Tumor', 'tissue_type...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91917</th>\n",
       "      <td>c680a261-9fdf-48dd-9e87-274881d700c8</td>\n",
       "      <td>da70cf7e-0e61-4c72-b4c5-c408569d11b8</td>\n",
       "      <td>open</td>\n",
       "      <td>miRNA-Seq</td>\n",
       "      <td>miRNA Expression Quantification</td>\n",
       "      <td>TXT</td>\n",
       "      <td>[{'sample_type': 'Primary Tumor', 'tissue_type...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91918</th>\n",
       "      <td>ad76f12b-1710-4e83-8cf2-21c1b6863d8e</td>\n",
       "      <td>d9627184-b972-4e00-8c11-b0b946ac357a</td>\n",
       "      <td>controlled</td>\n",
       "      <td>WGS</td>\n",
       "      <td>Aligned Reads</td>\n",
       "      <td>BAM</td>\n",
       "      <td>[{'sample_type': 'Blood Derived Normal', 'tiss...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91919</th>\n",
       "      <td>42eacf87-e962-4b33-859e-29944271e335</td>\n",
       "      <td>d9627184-b972-4e00-8c11-b0b946ac357a</td>\n",
       "      <td>open</td>\n",
       "      <td>Genotyping Array</td>\n",
       "      <td>Masked Copy Number Segment</td>\n",
       "      <td>TXT</td>\n",
       "      <td>[{'sample_type': 'Blood Derived Normal', 'tiss...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91920</th>\n",
       "      <td>b49a041c-2f46-412f-b10d-b0dcb40737cb</td>\n",
       "      <td>d9627184-b972-4e00-8c11-b0b946ac357a</td>\n",
       "      <td>open</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Biospecimen Supplement</td>\n",
       "      <td>BCR SSF XML</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91921</th>\n",
       "      <td>84bbf11d-b9ad-4376-b8da-2a21b7734801</td>\n",
       "      <td>db8d38ab-cc61-49ee-863d-40060112f2af</td>\n",
       "      <td>open</td>\n",
       "      <td>Methylation Array</td>\n",
       "      <td>Masked Intensities</td>\n",
       "      <td>IDAT</td>\n",
       "      <td>[{'sample_type': 'Primary Tumor', 'tissue_type...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>91922 rows Ã— 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                    file_id  \\\n",
       "0      75a8669a-692c-468c-8469-d2e61f6d37d8   \n",
       "1      7abd94ba-570e-43d4-9360-ad2d8399c960   \n",
       "2      97ef35dc-9c03-49ce-8103-5e8c55be8805   \n",
       "3      0518551d-4df2-4124-b68d-494200c5586b   \n",
       "4      2044bb65-05cf-4ef7-9543-3b7c5c2ff4d5   \n",
       "...                                     ...   \n",
       "91917  c680a261-9fdf-48dd-9e87-274881d700c8   \n",
       "91918  ad76f12b-1710-4e83-8cf2-21c1b6863d8e   \n",
       "91919  42eacf87-e962-4b33-859e-29944271e335   \n",
       "91920  b49a041c-2f46-412f-b10d-b0dcb40737cb   \n",
       "91921  84bbf11d-b9ad-4376-b8da-2a21b7734801   \n",
       "\n",
       "                                    case_id      access experimental_strategy  \\\n",
       "0      cc348a26-ee11-47a4-8b51-de922967e175        open     Methylation Array   \n",
       "1      f2bbfa9d-9a9d-4f46-9fde-378e4c44e2ad  controlled                   WGS   \n",
       "2      f2bbfa9d-9a9d-4f46-9fde-378e4c44e2ad  controlled               RNA-Seq   \n",
       "3      a6edb6ca-ae9f-4da7-8ebe-92d83d2987fb        open                   WXS   \n",
       "4      ec0ab947-9341-4fff-bda4-fdfb9434d508        open     Methylation Array   \n",
       "...                                     ...         ...                   ...   \n",
       "91917  da70cf7e-0e61-4c72-b4c5-c408569d11b8        open             miRNA-Seq   \n",
       "91918  d9627184-b972-4e00-8c11-b0b946ac357a  controlled                   WGS   \n",
       "91919  d9627184-b972-4e00-8c11-b0b946ac357a        open      Genotyping Array   \n",
       "91920  d9627184-b972-4e00-8c11-b0b946ac357a        open                   NaN   \n",
       "91921  db8d38ab-cc61-49ee-863d-40060112f2af        open     Methylation Array   \n",
       "\n",
       "                             data_type  data_format  \\\n",
       "0                   Masked Intensities         IDAT   \n",
       "1             Structural Rearrangement        BEDPE   \n",
       "2                    Transcript Fusion          TSV   \n",
       "3              Masked Somatic Mutation          MAF   \n",
       "4                   Masked Intensities         IDAT   \n",
       "...                                ...          ...   \n",
       "91917  miRNA Expression Quantification          TXT   \n",
       "91918                    Aligned Reads          BAM   \n",
       "91919       Masked Copy Number Segment          TXT   \n",
       "91920           Biospecimen Supplement  BCR SSF XML   \n",
       "91921               Masked Intensities         IDAT   \n",
       "\n",
       "                                                 samples sample_type  \\\n",
       "0      [{'sample_type': 'Primary Tumor', 'tissue_type...         NaN   \n",
       "1      [{'sample_type': 'Blood Derived Normal', 'tiss...         NaN   \n",
       "2      [{'sample_type': 'Solid Tissue Normal', 'tissu...         NaN   \n",
       "3      [{'sample_type': 'Primary Tumor', 'tissue_type...         NaN   \n",
       "4      [{'sample_type': 'Primary Tumor', 'tissue_type...         NaN   \n",
       "...                                                  ...         ...   \n",
       "91917  [{'sample_type': 'Primary Tumor', 'tissue_type...         NaN   \n",
       "91918  [{'sample_type': 'Blood Derived Normal', 'tiss...         NaN   \n",
       "91919  [{'sample_type': 'Blood Derived Normal', 'tiss...         NaN   \n",
       "91920                                                NaN         NaN   \n",
       "91921  [{'sample_type': 'Primary Tumor', 'tissue_type...         NaN   \n",
       "\n",
       "      tissue_type  is_file_of_interest  \n",
       "0             NaN                    0  \n",
       "1             NaN                    0  \n",
       "2             NaN                    0  \n",
       "3             NaN                    0  \n",
       "4             NaN                    0  \n",
       "...           ...                  ...  \n",
       "91917         NaN                    0  \n",
       "91918         NaN                    0  \n",
       "91919         NaN                    0  \n",
       "91920         NaN                    0  \n",
       "91921         NaN                    0  \n",
       "\n",
       "[91922 rows x 10 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the DataFrame of potential flagged files\n",
    "df_files"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Counting Files per Case"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Filter the potential files of interest\n",
    "df_files_of_interest = df_files \\\n",
    "    .query('is_file_of_interest == 1') \\\n",
    "    [['file_id', 'case_id', 'experimental_strategy', 'tissue_type']]\n",
    "\n",
    "# Create an acronym for file type, i.e, experimental strategy and tissue type\n",
    "df_files_of_interest['file_type'] = (\n",
    "    df_files_of_interest['experimental_strategy'] \n",
    "    + '_' + df_files_of_interest['tissue_type']\n",
    ")\n",
    "\n",
    "# Copy some columns of the DataFrame\n",
    "df_file_count = df_files_of_interest[['case_id', 'file_id', 'file_type']].copy()\n",
    "\n",
    "# Create flags for each file types\n",
    "df_file_count['tt_mir'] = np.where(\n",
    "    df_file_count['file_type'] == 'miRNA-Seq_Tumor', 1, 0\n",
    ")\n",
    "df_file_count['tt_mrna'] = np.where(\n",
    "    df_file_count['file_type'] == 'RNA-Seq_Tumor', 1, 0\n",
    ")\n",
    "df_file_count['nt_mir'] = np.where(\n",
    "    df_file_count['file_type'] == 'miRNA-Seq_Normal', 1, 0\n",
    ")\n",
    "df_file_count['nt_mrna'] = np.where(\n",
    "    df_file_count['file_type'] == 'RNA-Seq_Normal', 1, 0\n",
    ")\n",
    "\n",
    "# Initialize the file count per row\n",
    "df_file_count['total_files'] = 1\n",
    "\n",
    "# Count the total number of file types per case\n",
    "df_file_count_agg = df_file_count \\\n",
    "    .groupby('case_id') \\\n",
    "    .agg(\n",
    "        tumor_mir_files=pd.NamedAgg(column='tt_mir', aggfunc='sum'),\n",
    "        tumor_mrna_files=pd.NamedAgg(column='tt_mrna', aggfunc='sum'),\n",
    "        normal_mir_files=pd.NamedAgg(column='nt_mir', aggfunc='sum'),\n",
    "        normal_mrna_files=pd.NamedAgg(column='nt_mrna', aggfunc='sum'),\n",
    "        total_files=pd.NamedAgg(column='total_files', aggfunc='sum'),\n",
    "    ) \\\n",
    "    .sort_values(by=['total_files'], ascending=False) \\\n",
    "    .reset_index()\n",
    "    \n",
    "# Add the file count to the cases DataFrame\n",
    "df_cases = df_cases \\\n",
    "    .merge(\n",
    "        right=df_file_count_agg,\n",
    "        left_on='case_id',\n",
    "        right_on='case_id',\n",
    "        how='left'\n",
    "    )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>case_id</th>\n",
       "      <th>submitter_id</th>\n",
       "      <th>disease_type</th>\n",
       "      <th>pam50_mrna</th>\n",
       "      <th>is_case_of_interest</th>\n",
       "      <th>tumor_mir_files</th>\n",
       "      <th>tumor_mrna_files</th>\n",
       "      <th>normal_mir_files</th>\n",
       "      <th>normal_mrna_files</th>\n",
       "      <th>total_files</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>b1d44c81-747d-471f-9093-aeb262a17975</td>\n",
       "      <td>TCGA-Z7-A8R6</td>\n",
       "      <td>Epithelial Neoplasms, NOS</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>b205bba0-1870-4458-9088-8817e20389fe</td>\n",
       "      <td>TCGA-A8-A09A</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>b205c89f-af62-4186-acad-ed23d243fa98</td>\n",
       "      <td>TCGA-A2-A0YL</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>b26d41cd-393b-4cd4-8925-a9488f7de576</td>\n",
       "      <td>TCGA-C8-A1HE</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>b2aac45b-2073-4c7a-adb9-769a4fdcc111</td>\n",
       "      <td>TCGA-E9-A1NH</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1093</th>\n",
       "      <td>af5453a9-cf1f-40de-aec4-0e0710908fb7</td>\n",
       "      <td>TCGA-A8-A0AD</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1094</th>\n",
       "      <td>af97e043-88cb-4f99-8f8b-9bcbcccdf842</td>\n",
       "      <td>TCGA-EW-A2FR</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1095</th>\n",
       "      <td>b0700958-5f90-4546-b35f-635cd506889b</td>\n",
       "      <td>TCGA-PE-A5DD</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1096</th>\n",
       "      <td>b0bcb829-cee0-4247-bb75-093a5bea89ee</td>\n",
       "      <td>TCGA-E2-A105</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1097</th>\n",
       "      <td>b0f8d698-a30e-4d8d-b0a2-a5a01fac8406</td>\n",
       "      <td>TCGA-A2-A0T4</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal B</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1098 rows Ã— 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   case_id  submitter_id  \\\n",
       "0     b1d44c81-747d-471f-9093-aeb262a17975  TCGA-Z7-A8R6   \n",
       "1     b205bba0-1870-4458-9088-8817e20389fe  TCGA-A8-A09A   \n",
       "2     b205c89f-af62-4186-acad-ed23d243fa98  TCGA-A2-A0YL   \n",
       "3     b26d41cd-393b-4cd4-8925-a9488f7de576  TCGA-C8-A1HE   \n",
       "4     b2aac45b-2073-4c7a-adb9-769a4fdcc111  TCGA-E9-A1NH   \n",
       "...                                    ...           ...   \n",
       "1093  af5453a9-cf1f-40de-aec4-0e0710908fb7  TCGA-A8-A0AD   \n",
       "1094  af97e043-88cb-4f99-8f8b-9bcbcccdf842  TCGA-EW-A2FR   \n",
       "1095  b0700958-5f90-4546-b35f-635cd506889b  TCGA-PE-A5DD   \n",
       "1096  b0bcb829-cee0-4247-bb75-093a5bea89ee  TCGA-E2-A105   \n",
       "1097  b0f8d698-a30e-4d8d-b0a2-a5a01fac8406  TCGA-A2-A0T4   \n",
       "\n",
       "                      disease_type pam50_mrna  is_case_of_interest  \\\n",
       "0        Epithelial Neoplasms, NOS                               0   \n",
       "1     Ductal and Lobular Neoplasms  Luminal A                    1   \n",
       "2     Ductal and Lobular Neoplasms  Luminal A                    1   \n",
       "3     Ductal and Lobular Neoplasms                               0   \n",
       "4     Ductal and Lobular Neoplasms                               0   \n",
       "...                            ...        ...                  ...   \n",
       "1093  Ductal and Lobular Neoplasms  Luminal A                    1   \n",
       "1094  Ductal and Lobular Neoplasms                               0   \n",
       "1095  Ductal and Lobular Neoplasms                               0   \n",
       "1096  Ductal and Lobular Neoplasms  Luminal A                    1   \n",
       "1097  Ductal and Lobular Neoplasms  Luminal B                    1   \n",
       "\n",
       "      tumor_mir_files  tumor_mrna_files  normal_mir_files  normal_mrna_files  \\\n",
       "0                 NaN               NaN               NaN                NaN   \n",
       "1                 1.0               1.0               0.0                0.0   \n",
       "2                 1.0               1.0               0.0                0.0   \n",
       "3                 NaN               NaN               NaN                NaN   \n",
       "4                 NaN               NaN               NaN                NaN   \n",
       "...               ...               ...               ...                ...   \n",
       "1093              1.0               1.0               0.0                0.0   \n",
       "1094              NaN               NaN               NaN                NaN   \n",
       "1095              NaN               NaN               NaN                NaN   \n",
       "1096              1.0               1.0               0.0                0.0   \n",
       "1097              1.0               1.0               0.0                0.0   \n",
       "\n",
       "      total_files  \n",
       "0             NaN  \n",
       "1             2.0  \n",
       "2             2.0  \n",
       "3             NaN  \n",
       "4             NaN  \n",
       "...           ...  \n",
       "1093          2.0  \n",
       "1094          NaN  \n",
       "1095          NaN  \n",
       "1096          2.0  \n",
       "1097          2.0  \n",
       "\n",
       "[1098 rows x 10 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the DataFrame of potential flagged cases with file count\n",
    "df_cases"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Flagged Cases DataFrame"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Flag cases with files related to tumor tissue analysis\n",
    "df_cases['has_tumor_files_of_interest'] = np.where(\n",
    "    ((df_cases['is_case_of_interest'] == 1)\n",
    "     & (df_cases['tumor_mir_files'] == 1)\n",
    "     & (df_cases['tumor_mrna_files'] == 1)), 1, 0\n",
    ")\n",
    "\n",
    "# Flag cases with files related to normal tissue analysis\n",
    "df_cases['has_normal_files_of_interest'] = np.where(\n",
    "    ((df_cases['is_case_of_interest'] == 1)\n",
    "     & (df_cases['normal_mir_files'] == 1)\n",
    "     & (df_cases['normal_mrna_files'] == 1)), 1, 0\n",
    ")\n",
    "\n",
    "# Rearrange the DataFrame columns\n",
    "df_cases = df_cases.drop(columns=[\n",
    "    'tumor_mir_files',\n",
    "    'tumor_mrna_files',\n",
    "    'normal_mir_files',\n",
    "    'normal_mrna_files',\n",
    "    'total_files',\n",
    "    'is_case_of_interest',\n",
    "])\n",
    "\n",
    "# Update the cases of interest flag\n",
    "df_cases['is_case_of_interest'] = np.where(\n",
    "    ((df_cases['has_tumor_files_of_interest'] == 1)\n",
    "     | (df_cases['has_normal_files_of_interest'] == 1)), 1, 0\n",
    ")\n",
    "\n",
    "# Store the cases metadata DataFrame in a CSV file\n",
    "df_cases.to_csv(BRCA_PROCESSED_FILES_PATHS['cases'], index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>case_id</th>\n",
       "      <th>submitter_id</th>\n",
       "      <th>disease_type</th>\n",
       "      <th>pam50_mrna</th>\n",
       "      <th>has_tumor_files_of_interest</th>\n",
       "      <th>has_normal_files_of_interest</th>\n",
       "      <th>is_case_of_interest</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>b1d44c81-747d-471f-9093-aeb262a17975</td>\n",
       "      <td>TCGA-Z7-A8R6</td>\n",
       "      <td>Epithelial Neoplasms, NOS</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>b205bba0-1870-4458-9088-8817e20389fe</td>\n",
       "      <td>TCGA-A8-A09A</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>b205c89f-af62-4186-acad-ed23d243fa98</td>\n",
       "      <td>TCGA-A2-A0YL</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>b26d41cd-393b-4cd4-8925-a9488f7de576</td>\n",
       "      <td>TCGA-C8-A1HE</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>b2aac45b-2073-4c7a-adb9-769a4fdcc111</td>\n",
       "      <td>TCGA-E9-A1NH</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1093</th>\n",
       "      <td>af5453a9-cf1f-40de-aec4-0e0710908fb7</td>\n",
       "      <td>TCGA-A8-A0AD</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1094</th>\n",
       "      <td>af97e043-88cb-4f99-8f8b-9bcbcccdf842</td>\n",
       "      <td>TCGA-EW-A2FR</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1095</th>\n",
       "      <td>b0700958-5f90-4546-b35f-635cd506889b</td>\n",
       "      <td>TCGA-PE-A5DD</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1096</th>\n",
       "      <td>b0bcb829-cee0-4247-bb75-093a5bea89ee</td>\n",
       "      <td>TCGA-E2-A105</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal A</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1097</th>\n",
       "      <td>b0f8d698-a30e-4d8d-b0a2-a5a01fac8406</td>\n",
       "      <td>TCGA-A2-A0T4</td>\n",
       "      <td>Ductal and Lobular Neoplasms</td>\n",
       "      <td>Luminal B</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1098 rows Ã— 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   case_id  submitter_id  \\\n",
       "0     b1d44c81-747d-471f-9093-aeb262a17975  TCGA-Z7-A8R6   \n",
       "1     b205bba0-1870-4458-9088-8817e20389fe  TCGA-A8-A09A   \n",
       "2     b205c89f-af62-4186-acad-ed23d243fa98  TCGA-A2-A0YL   \n",
       "3     b26d41cd-393b-4cd4-8925-a9488f7de576  TCGA-C8-A1HE   \n",
       "4     b2aac45b-2073-4c7a-adb9-769a4fdcc111  TCGA-E9-A1NH   \n",
       "...                                    ...           ...   \n",
       "1093  af5453a9-cf1f-40de-aec4-0e0710908fb7  TCGA-A8-A0AD   \n",
       "1094  af97e043-88cb-4f99-8f8b-9bcbcccdf842  TCGA-EW-A2FR   \n",
       "1095  b0700958-5f90-4546-b35f-635cd506889b  TCGA-PE-A5DD   \n",
       "1096  b0bcb829-cee0-4247-bb75-093a5bea89ee  TCGA-E2-A105   \n",
       "1097  b0f8d698-a30e-4d8d-b0a2-a5a01fac8406  TCGA-A2-A0T4   \n",
       "\n",
       "                      disease_type pam50_mrna  has_tumor_files_of_interest  \\\n",
       "0        Epithelial Neoplasms, NOS                                       0   \n",
       "1     Ductal and Lobular Neoplasms  Luminal A                            1   \n",
       "2     Ductal and Lobular Neoplasms  Luminal A                            1   \n",
       "3     Ductal and Lobular Neoplasms                                       0   \n",
       "4     Ductal and Lobular Neoplasms                                       0   \n",
       "...                            ...        ...                          ...   \n",
       "1093  Ductal and Lobular Neoplasms  Luminal A                            1   \n",
       "1094  Ductal and Lobular Neoplasms                                       0   \n",
       "1095  Ductal and Lobular Neoplasms                                       0   \n",
       "1096  Ductal and Lobular Neoplasms  Luminal A                            1   \n",
       "1097  Ductal and Lobular Neoplasms  Luminal B                            1   \n",
       "\n",
       "      has_normal_files_of_interest  is_case_of_interest  \n",
       "0                                0                    0  \n",
       "1                                0                    1  \n",
       "2                                0                    1  \n",
       "3                                0                    0  \n",
       "4                                0                    0  \n",
       "...                            ...                  ...  \n",
       "1093                             0                    1  \n",
       "1094                             0                    0  \n",
       "1095                             0                    0  \n",
       "1096                             0                    1  \n",
       "1097                             0                    1  \n",
       "\n",
       "[1098 rows x 7 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the DataFrame of flagged cases metadata\n",
    "df_cases"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Flagged Files DataFrame"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Retrieve the flag of interest of the cases    \n",
    "df_files = df_files \\\n",
    "    .merge(\n",
    "        right=df_cases,\n",
    "        left_on='case_id',\n",
    "        right_on='case_id',\n",
    "        how='inner'\n",
    "    ) \\\n",
    "    .drop(columns=['submitter_id', 'disease_type', 'pam50_mrna',])\n",
    "\n",
    "# Flag the files related to tumor tissue analysis\n",
    "df_files['is_tumor_file_of_interest'] = np.where(\n",
    "    ((df_files['is_file_of_interest'] == 1)\n",
    "     & (df_files['has_tumor_files_of_interest'] == 1)\n",
    "     & (df_files['tissue_type'] == 'Tumor')), 1, 0\n",
    ")\n",
    "\n",
    "# Flag the files related to normal tissue analysis\n",
    "df_files['is_normal_file_of_interest'] = np.where(\n",
    "    ((df_files['is_file_of_interest'] == 1)\n",
    "     & (df_files['has_normal_files_of_interest'] == 1)\n",
    "     & (df_files['tissue_type'] == 'Normal')), 1, 0\n",
    ")\n",
    "\n",
    "# Rearrange the DataFrame columns\n",
    "df_files = df_files.drop(columns=[\n",
    "    'sample_type',\n",
    "    'tissue_type',\n",
    "    'is_file_of_interest',\n",
    "    'has_tumor_files_of_interest',\n",
    "    'has_normal_files_of_interest',\n",
    "    'is_case_of_interest',\n",
    "])\n",
    "\n",
    "# Update the files of interest flag\n",
    "df_files['is_file_of_interest'] = np.where(\n",
    "    ((df_files['is_tumor_file_of_interest'] == 1)\n",
    "     | (df_files['is_normal_file_of_interest'] == 1)), 1, 0\n",
    ")\n",
    "\n",
    "# Store the files metadata DataFrame in a CSV file\n",
    "df_files.to_csv(BRCA_PROCESSED_FILES_PATHS['files'], index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>file_id</th>\n",
       "      <th>case_id</th>\n",
       "      <th>access</th>\n",
       "      <th>experimental_strategy</th>\n",
       "      <th>data_type</th>\n",
       "      <th>data_format</th>\n",
       "      <th>samples</th>\n",
       "      <th>is_tumor_file_of_interest</th>\n",
       "      <th>is_normal_file_of_interest</th>\n",
       "      <th>is_file_of_interest</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>75a8669a-692c-468c-8469-d2e61f6d37d8</td>\n",
       "      <td>cc348a26-ee11-47a4-8b51-de922967e175</td>\n",
       "      <td>open</td>\n",
       "      <td>Methylation Array</td>\n",
       "      <td>Masked Intensities</td>\n",
       "      <td>IDAT</td>\n",
       "      <td>[{'sample_type': 'Primary Tumor', 'tissue_type...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>7abd94ba-570e-43d4-9360-ad2d8399c960</td>\n",
       "      <td>f2bbfa9d-9a9d-4f46-9fde-378e4c44e2ad</td>\n",
       "      <td>controlled</td>\n",
       "      <td>WGS</td>\n",
       "      <td>Structural Rearrangement</td>\n",
       "      <td>BEDPE</td>\n",
       "      <td>[{'sample_type': 'Blood Derived Normal', 'tiss...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>97ef35dc-9c03-49ce-8103-5e8c55be8805</td>\n",
       "      <td>f2bbfa9d-9a9d-4f46-9fde-378e4c44e2ad</td>\n",
       "      <td>controlled</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>Transcript Fusion</td>\n",
       "      <td>TSV</td>\n",
       "      <td>[{'sample_type': 'Solid Tissue Normal', 'tissu...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0518551d-4df2-4124-b68d-494200c5586b</td>\n",
       "      <td>a6edb6ca-ae9f-4da7-8ebe-92d83d2987fb</td>\n",
       "      <td>open</td>\n",
       "      <td>WXS</td>\n",
       "      <td>Masked Somatic Mutation</td>\n",
       "      <td>MAF</td>\n",
       "      <td>[{'sample_type': 'Primary Tumor', 'tissue_type...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2044bb65-05cf-4ef7-9543-3b7c5c2ff4d5</td>\n",
       "      <td>ec0ab947-9341-4fff-bda4-fdfb9434d508</td>\n",
       "      <td>open</td>\n",
       "      <td>Methylation Array</td>\n",
       "      <td>Masked Intensities</td>\n",
       "      <td>IDAT</td>\n",
       "      <td>[{'sample_type': 'Primary Tumor', 'tissue_type...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91917</th>\n",
       "      <td>c680a261-9fdf-48dd-9e87-274881d700c8</td>\n",
       "      <td>da70cf7e-0e61-4c72-b4c5-c408569d11b8</td>\n",
       "      <td>open</td>\n",
       "      <td>miRNA-Seq</td>\n",
       "      <td>miRNA Expression Quantification</td>\n",
       "      <td>TXT</td>\n",
       "      <td>[{'sample_type': 'Primary Tumor', 'tissue_type...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91918</th>\n",
       "      <td>ad76f12b-1710-4e83-8cf2-21c1b6863d8e</td>\n",
       "      <td>d9627184-b972-4e00-8c11-b0b946ac357a</td>\n",
       "      <td>controlled</td>\n",
       "      <td>WGS</td>\n",
       "      <td>Aligned Reads</td>\n",
       "      <td>BAM</td>\n",
       "      <td>[{'sample_type': 'Blood Derived Normal', 'tiss...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91919</th>\n",
       "      <td>42eacf87-e962-4b33-859e-29944271e335</td>\n",
       "      <td>d9627184-b972-4e00-8c11-b0b946ac357a</td>\n",
       "      <td>open</td>\n",
       "      <td>Genotyping Array</td>\n",
       "      <td>Masked Copy Number Segment</td>\n",
       "      <td>TXT</td>\n",
       "      <td>[{'sample_type': 'Blood Derived Normal', 'tiss...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91920</th>\n",
       "      <td>b49a041c-2f46-412f-b10d-b0dcb40737cb</td>\n",
       "      <td>d9627184-b972-4e00-8c11-b0b946ac357a</td>\n",
       "      <td>open</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Biospecimen Supplement</td>\n",
       "      <td>BCR SSF XML</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91921</th>\n",
       "      <td>84bbf11d-b9ad-4376-b8da-2a21b7734801</td>\n",
       "      <td>db8d38ab-cc61-49ee-863d-40060112f2af</td>\n",
       "      <td>open</td>\n",
       "      <td>Methylation Array</td>\n",
       "      <td>Masked Intensities</td>\n",
       "      <td>IDAT</td>\n",
       "      <td>[{'sample_type': 'Primary Tumor', 'tissue_type...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>91922 rows Ã— 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                    file_id  \\\n",
       "0      75a8669a-692c-468c-8469-d2e61f6d37d8   \n",
       "1      7abd94ba-570e-43d4-9360-ad2d8399c960   \n",
       "2      97ef35dc-9c03-49ce-8103-5e8c55be8805   \n",
       "3      0518551d-4df2-4124-b68d-494200c5586b   \n",
       "4      2044bb65-05cf-4ef7-9543-3b7c5c2ff4d5   \n",
       "...                                     ...   \n",
       "91917  c680a261-9fdf-48dd-9e87-274881d700c8   \n",
       "91918  ad76f12b-1710-4e83-8cf2-21c1b6863d8e   \n",
       "91919  42eacf87-e962-4b33-859e-29944271e335   \n",
       "91920  b49a041c-2f46-412f-b10d-b0dcb40737cb   \n",
       "91921  84bbf11d-b9ad-4376-b8da-2a21b7734801   \n",
       "\n",
       "                                    case_id      access experimental_strategy  \\\n",
       "0      cc348a26-ee11-47a4-8b51-de922967e175        open     Methylation Array   \n",
       "1      f2bbfa9d-9a9d-4f46-9fde-378e4c44e2ad  controlled                   WGS   \n",
       "2      f2bbfa9d-9a9d-4f46-9fde-378e4c44e2ad  controlled               RNA-Seq   \n",
       "3      a6edb6ca-ae9f-4da7-8ebe-92d83d2987fb        open                   WXS   \n",
       "4      ec0ab947-9341-4fff-bda4-fdfb9434d508        open     Methylation Array   \n",
       "...                                     ...         ...                   ...   \n",
       "91917  da70cf7e-0e61-4c72-b4c5-c408569d11b8        open             miRNA-Seq   \n",
       "91918  d9627184-b972-4e00-8c11-b0b946ac357a  controlled                   WGS   \n",
       "91919  d9627184-b972-4e00-8c11-b0b946ac357a        open      Genotyping Array   \n",
       "91920  d9627184-b972-4e00-8c11-b0b946ac357a        open                   NaN   \n",
       "91921  db8d38ab-cc61-49ee-863d-40060112f2af        open     Methylation Array   \n",
       "\n",
       "                             data_type  data_format  \\\n",
       "0                   Masked Intensities         IDAT   \n",
       "1             Structural Rearrangement        BEDPE   \n",
       "2                    Transcript Fusion          TSV   \n",
       "3              Masked Somatic Mutation          MAF   \n",
       "4                   Masked Intensities         IDAT   \n",
       "...                                ...          ...   \n",
       "91917  miRNA Expression Quantification          TXT   \n",
       "91918                    Aligned Reads          BAM   \n",
       "91919       Masked Copy Number Segment          TXT   \n",
       "91920           Biospecimen Supplement  BCR SSF XML   \n",
       "91921               Masked Intensities         IDAT   \n",
       "\n",
       "                                                 samples  \\\n",
       "0      [{'sample_type': 'Primary Tumor', 'tissue_type...   \n",
       "1      [{'sample_type': 'Blood Derived Normal', 'tiss...   \n",
       "2      [{'sample_type': 'Solid Tissue Normal', 'tissu...   \n",
       "3      [{'sample_type': 'Primary Tumor', 'tissue_type...   \n",
       "4      [{'sample_type': 'Primary Tumor', 'tissue_type...   \n",
       "...                                                  ...   \n",
       "91917  [{'sample_type': 'Primary Tumor', 'tissue_type...   \n",
       "91918  [{'sample_type': 'Blood Derived Normal', 'tiss...   \n",
       "91919  [{'sample_type': 'Blood Derived Normal', 'tiss...   \n",
       "91920                                                NaN   \n",
       "91921  [{'sample_type': 'Primary Tumor', 'tissue_type...   \n",
       "\n",
       "       is_tumor_file_of_interest  is_normal_file_of_interest  \\\n",
       "0                              0                           0   \n",
       "1                              0                           0   \n",
       "2                              0                           0   \n",
       "3                              0                           0   \n",
       "4                              0                           0   \n",
       "...                          ...                         ...   \n",
       "91917                          0                           0   \n",
       "91918                          0                           0   \n",
       "91919                          0                           0   \n",
       "91920                          0                           0   \n",
       "91921                          0                           0   \n",
       "\n",
       "       is_file_of_interest  \n",
       "0                        0  \n",
       "1                        0  \n",
       "2                        0  \n",
       "3                        0  \n",
       "4                        0  \n",
       "...                    ...  \n",
       "91917                    0  \n",
       "91918                    0  \n",
       "91919                    0  \n",
       "91920                    0  \n",
       "91921                    0  \n",
       "\n",
       "[91922 rows x 10 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the DataFrame of flagged files metadata\n",
    "df_files"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "venv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
