{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# **Preprocessing GDC API Data**\n",
    "GDC: Genomic Data Commons"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Importing Libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "\n",
    "import pandas as pd\n",
    "import requests"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Constants and Path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Base URL HTML from GDC API\n",
    "GDC_API_URL = 'https://api.gdc.cancer.gov'\n",
    "\n",
    "# Endpoint to get API status and version\n",
    "STATUS_ENDPOINT = f'{GDC_API_URL}/status'\n",
    "\n",
    "# Endpoint of projects related to programs\n",
    "PROJECTS_ENDPOINT = f'{GDC_API_URL}/projects'\n",
    "\n",
    "# Endpoint of cases related to projects\n",
    "CASES_ENDPOINT = f'{GDC_API_URL}/cases'\n",
    "\n",
    "# Endpoint of files related to cases\n",
    "FILES_ENDPOINT = f'{GDC_API_URL}/files'\n",
    "\n",
    "# Interim data folder path\n",
    "INTERIM_DATA_PATH = '../../data/interim/gdc-tcga'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# API Status"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{\"commit\":\"48add4be7ac46e7db10e0c6f0e3010d5bb2a50aa\",\"data_release\":\"Data Release 42.0 - January 30, 2025\",\"data_release_version\":{\"major\":42,\"minor\":0,\"release_date\":\"2025-01-30\"},\"status\":\"OK\",\"tag\":\"7.7\",\"version\":1}\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Request the API version and status to the endpoint\n",
    "response = requests.get(STATUS_ENDPOINT)\n",
    "\n",
    "# Print the response content\n",
    "print(response.content.decode('utf-8'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Projects"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## All Projects\n",
    "Total projects contained in the GDC API"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Request the information contained in the endpoint\n",
    "response = requests.get(f'{PROJECTS_ENDPOINT}/_mapping')\n",
    "\n",
    "# Convert the response content to JSON\n",
    "json_response = json.loads(response.content.decode('utf-8'))\n",
    "\n",
    "# Store the fields available in the endpoint\n",
    "fields_projects = json.dumps(json_response['fields'])[1:-2]\n",
    "fields_projects = fields_projects.replace('\"', '')\n",
    "fields_projects = fields_projects.replace(' ', '')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of projects: 86\n"
     ]
    }
   ],
   "source": [
    "# Request the default number of objects to the endpoint\n",
    "response = requests.post(\n",
    "    url=PROJECTS_ENDPOINT,\n",
    "    headers={'Content-Type': 'application/json'},\n",
    "    json={'fields': 'name'}\n",
    ")\n",
    "\n",
    "# Convert the response content to JSON\n",
    "json_response = json.loads(response.content.decode('utf-8'))\n",
    "\n",
    "# Store the total number of projects contained in the GDC API\n",
    "total_projects = json.dumps(json_response['data']['pagination']['total'])\n",
    "print(f'Total number of projects: {total_projects}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Parameters for the endpoint request\n",
    "params = {\n",
    "    'fields': fields_projects,\n",
    "    'size': total_projects\n",
    "}\n",
    "\n",
    "# Request all objects from the endpoint\n",
    "response = requests.post(\n",
    "    url=PROJECTS_ENDPOINT,\n",
    "    headers={'Content-Type': 'application/json'},\n",
    "    json=params\n",
    ")\n",
    "\n",
    "# Convert the response content to JSON\n",
    "json_response = json.loads(response.content.decode('utf-8'))['data']['hits']\n",
    "\n",
    "# Convert JSON boolean values to strings\n",
    "for index in range(len(json_response)):\n",
    "    obj = json_response[index]\n",
    "    for key in obj:\n",
    "        if isinstance(obj[key], bool): \n",
    "            obj[key] = str(obj[key])\n",
    "\n",
    "# Create a DataFrame from the projects JSON\n",
    "df_all_projects = pd.json_normalize(json_response) \\\n",
    "    .sort_values(by='id') \\\n",
    "    .reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>primary_site</th>\n",
       "      <th>dbgap_accession_number</th>\n",
       "      <th>project_id</th>\n",
       "      <th>disease_type</th>\n",
       "      <th>name</th>\n",
       "      <th>releasable</th>\n",
       "      <th>state</th>\n",
       "      <th>released</th>\n",
       "      <th>summary.file_count</th>\n",
       "      <th>summary.data_categories</th>\n",
       "      <th>summary.experimental_strategies</th>\n",
       "      <th>summary.case_count</th>\n",
       "      <th>summary.file_size</th>\n",
       "      <th>program.dbgap_accession_number</th>\n",
       "      <th>program.program_id</th>\n",
       "      <th>program.name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>APOLLO-LUAD</td>\n",
       "      <td>[Bronchus and lung]</td>\n",
       "      <td>phs003011</td>\n",
       "      <td>APOLLO-LUAD</td>\n",
       "      <td>[Adenomas and Adenocarcinomas]</td>\n",
       "      <td>APOLLO1: Proteogenomic characterization of lung adenocarcinoma</td>\n",
       "      <td>False</td>\n",
       "      <td>open</td>\n",
       "      <td>True</td>\n",
       "      <td>1679</td>\n",
       "      <td>[{'file_count': 435, 'case_count': 87, 'data_category': 'Sequencing Reads'}, {'file_count': 194,...</td>\n",
       "      <td>[{'file_count': 783, 'case_count': 87, 'experimental_strategy': 'RNA-Seq'}, {'file_count': 896, ...</td>\n",
       "      <td>87</td>\n",
       "      <td>67389870223275</td>\n",
       "      <td>None</td>\n",
       "      <td>bd5bea9d-29ff-52e4-92f5-463837c6f6cf</td>\n",
       "      <td>APOLLO</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BEATAML1.0-COHORT</td>\n",
       "      <td>[Hematopoietic and reticuloendothelial systems]</td>\n",
       "      <td>phs001657</td>\n",
       "      <td>BEATAML1.0-COHORT</td>\n",
       "      <td>[Plasma Cell Tumors, Myelodysplastic Syndromes, Unknown, Chronic Myeloproliferative Disorders, M...</td>\n",
       "      <td>Functional Genomic Landscape of Acute Myeloid Leukemia</td>\n",
       "      <td>True</td>\n",
       "      <td>open</td>\n",
       "      <td>True</td>\n",
       "      <td>16794</td>\n",
       "      <td>[{'file_count': 8777, 'case_count': 759, 'data_category': 'Simple Nucleotide Variation'}, {'file...</td>\n",
       "      <td>[{'file_count': 7478, 'case_count': 534, 'experimental_strategy': 'WXS'}, {'file_count': 2653, '...</td>\n",
       "      <td>826</td>\n",
       "      <td>42355372208215</td>\n",
       "      <td>None</td>\n",
       "      <td>1eaac0a3-91f6-54b3-9a7c-e34468eaf527</td>\n",
       "      <td>BEATAML1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BEATAML1.0-CRENOLANIB</td>\n",
       "      <td>[Hematopoietic and reticuloendothelial systems]</td>\n",
       "      <td>phs001628</td>\n",
       "      <td>BEATAML1.0-CRENOLANIB</td>\n",
       "      <td>[Myeloid Leukemias]</td>\n",
       "      <td>Clinical Resistance to Crenolanib in Acute Myeloid Leukemia Due to Diverse Molecular Mechanisms</td>\n",
       "      <td>False</td>\n",
       "      <td>open</td>\n",
       "      <td>True</td>\n",
       "      <td>547</td>\n",
       "      <td>[{'file_count': 428, 'case_count': 51, 'data_category': 'Simple Nucleotide Variation'}, {'file_c...</td>\n",
       "      <td>[{'file_count': 547, 'case_count': 56, 'experimental_strategy': 'WXS'}]</td>\n",
       "      <td>56</td>\n",
       "      <td>3608907348655</td>\n",
       "      <td>None</td>\n",
       "      <td>1eaac0a3-91f6-54b3-9a7c-e34468eaf527</td>\n",
       "      <td>BEATAML1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      id                                     primary_site  \\\n",
       "0            APOLLO-LUAD                              [Bronchus and lung]   \n",
       "1      BEATAML1.0-COHORT  [Hematopoietic and reticuloendothelial systems]   \n",
       "2  BEATAML1.0-CRENOLANIB  [Hematopoietic and reticuloendothelial systems]   \n",
       "\n",
       "  dbgap_accession_number             project_id  \\\n",
       "0              phs003011            APOLLO-LUAD   \n",
       "1              phs001657      BEATAML1.0-COHORT   \n",
       "2              phs001628  BEATAML1.0-CRENOLANIB   \n",
       "\n",
       "                                                                                          disease_type  \\\n",
       "0                                                                       [Adenomas and Adenocarcinomas]   \n",
       "1  [Plasma Cell Tumors, Myelodysplastic Syndromes, Unknown, Chronic Myeloproliferative Disorders, M...   \n",
       "2                                                                                  [Myeloid Leukemias]   \n",
       "\n",
       "                                                                                              name  \\\n",
       "0                                   APOLLO1: Proteogenomic characterization of lung adenocarcinoma   \n",
       "1                                           Functional Genomic Landscape of Acute Myeloid Leukemia   \n",
       "2  Clinical Resistance to Crenolanib in Acute Myeloid Leukemia Due to Diverse Molecular Mechanisms   \n",
       "\n",
       "  releasable state released  summary.file_count  \\\n",
       "0      False  open     True                1679   \n",
       "1       True  open     True               16794   \n",
       "2      False  open     True                 547   \n",
       "\n",
       "                                                                               summary.data_categories  \\\n",
       "0  [{'file_count': 435, 'case_count': 87, 'data_category': 'Sequencing Reads'}, {'file_count': 194,...   \n",
       "1  [{'file_count': 8777, 'case_count': 759, 'data_category': 'Simple Nucleotide Variation'}, {'file...   \n",
       "2  [{'file_count': 428, 'case_count': 51, 'data_category': 'Simple Nucleotide Variation'}, {'file_c...   \n",
       "\n",
       "                                                                       summary.experimental_strategies  \\\n",
       "0  [{'file_count': 783, 'case_count': 87, 'experimental_strategy': 'RNA-Seq'}, {'file_count': 896, ...   \n",
       "1  [{'file_count': 7478, 'case_count': 534, 'experimental_strategy': 'WXS'}, {'file_count': 2653, '...   \n",
       "2                              [{'file_count': 547, 'case_count': 56, 'experimental_strategy': 'WXS'}]   \n",
       "\n",
       "   summary.case_count  summary.file_size program.dbgap_accession_number  \\\n",
       "0                  87     67389870223275                           None   \n",
       "1                 826     42355372208215                           None   \n",
       "2                  56      3608907348655                           None   \n",
       "\n",
       "                     program.program_id program.name  \n",
       "0  bd5bea9d-29ff-52e4-92f5-463837c6f6cf       APOLLO  \n",
       "1  1eaac0a3-91f6-54b3-9a7c-e34468eaf527   BEATAML1.0  \n",
       "2  1eaac0a3-91f6-54b3-9a7c-e34468eaf527   BEATAML1.0  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Write the DataFrame in a CSV file\n",
    "file_name = 'gdc-projects.csv'\n",
    "df_all_projects.to_csv(f'{INTERIM_DATA_PATH}/{file_name}', index=False)\n",
    "\n",
    "# Print the DataFrame partially\n",
    "pd.set_option('display.max_colwidth', 100)\n",
    "df_all_projects.head(3)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Projects of Interest\n",
    "Projects Released with miRNA-Seq and RNA-Seq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of projects of interest: 47\n"
     ]
    }
   ],
   "source": [
    "# Filter released projects and drop unnecessary columns\n",
    "df_projects = df_all_projects \\\n",
    "    .query('released == \"True\"') \\\n",
    "    .drop(\n",
    "        columns=[\n",
    "            'dbgap_accession_number',\n",
    "            'id',\n",
    "            'releasable',\n",
    "            'released',\n",
    "            'program.dbgap_accession_number',\n",
    "            'program.name',\n",
    "            'program.program_id',\n",
    "            'state',\n",
    "            'summary.data_categories',\n",
    "            'summary.file_count',\n",
    "            'summary.file_size'\n",
    "        ]\n",
    "    )\n",
    "\n",
    "# Initialize lists for counting cases with miRNA-Seq and RNA-Seq in projects\n",
    "miRNA_Seq_case_count = [0] * df_projects.shape[0]\n",
    "RNA_Seq_case_count = [0] * df_projects.shape[0]\n",
    "\n",
    "# Fill the lists with counts of cases with miRNA-Seq and RNA-Seq in each project\n",
    "for index in range(df_projects.shape[0]):\n",
    "    for info in df_projects['summary.experimental_strategies'][index]:\n",
    "        if info['experimental_strategy'] in ['miRNA-Seq', 'RNA-Seq']:\n",
    "            if info['experimental_strategy'] == 'miRNA-Seq':\n",
    "                miRNA_Seq_case_count[index] = info['case_count']\n",
    "            else:\n",
    "                RNA_Seq_case_count[index] = info['case_count']\n",
    "\n",
    "# Add the count of cases with miRNA-Seq and RNA-Seq to the DataFrame\n",
    "df_projects['miRNA-Seq_case_count'] = miRNA_Seq_case_count\n",
    "df_projects['RNA-Seq_case_count'] = RNA_Seq_case_count\n",
    "\n",
    "# Filter projects with miRNA-Seq and RNA-Seq as experimental strategy\n",
    "df_projects = df_projects \\\n",
    "    .query('(`miRNA-Seq_case_count` > 0) & (`RNA-Seq_case_count` > 0)') \\\n",
    "    .drop(columns='summary.experimental_strategies') \\\n",
    "    .reset_index(drop=True)\n",
    "\n",
    "# Rearrange and rename the DataFrame columns\n",
    "columns = [\n",
    "    'project_id',\n",
    "    'name',\n",
    "    'primary_site',\n",
    "    'disease_type',\n",
    "    'summary.case_count',\n",
    "    'miRNA-Seq_case_count',\n",
    "    'RNA-Seq_case_count'\n",
    "]\n",
    "df_projects = df_projects[columns] \\\n",
    "    .rename(\n",
    "        columns={\n",
    "            'name': 'project_name',\n",
    "            'summary.case_count': 'case_count',\n",
    "        }\n",
    "    )\n",
    "\n",
    "# Print the total number of projects with the desired characteristics\n",
    "print(f'Total number of projects of interest: {df_projects.shape[0]}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>project_id</th>\n",
       "      <th>project_name</th>\n",
       "      <th>primary_site</th>\n",
       "      <th>disease_type</th>\n",
       "      <th>case_count</th>\n",
       "      <th>miRNA-Seq_case_count</th>\n",
       "      <th>RNA-Seq_case_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CGCI-BLGSP</td>\n",
       "      <td>Burkitt Lymphoma Genome Sequencing Project</td>\n",
       "      <td>[Hematopoietic and reticuloendothelial systems]</td>\n",
       "      <td>[Mature B-Cell Lymphomas]</td>\n",
       "      <td>324</td>\n",
       "      <td>278</td>\n",
       "      <td>283</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CGCI-HTMCP-CC</td>\n",
       "      <td>HIV+ Tumor Molecular Characterization Project - Cervical Cancer</td>\n",
       "      <td>[Cervix uteri]</td>\n",
       "      <td>[Adenomas and Adenocarcinomas, Squamous Cell Neoplasms, Epithelial Neoplasms, NOS, Complex Epith...</td>\n",
       "      <td>212</td>\n",
       "      <td>118</td>\n",
       "      <td>123</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CGCI-HTMCP-DLBCL</td>\n",
       "      <td>HIV+ Tumor Molecular Characterization Project - Diffuse Large B-Cell Lymphoma</td>\n",
       "      <td>[Hematopoietic and reticuloendothelial systems]</td>\n",
       "      <td>[Mature B-Cell Lymphomas]</td>\n",
       "      <td>70</td>\n",
       "      <td>60</td>\n",
       "      <td>55</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         project_id  \\\n",
       "0        CGCI-BLGSP   \n",
       "1     CGCI-HTMCP-CC   \n",
       "2  CGCI-HTMCP-DLBCL   \n",
       "\n",
       "                                                                    project_name  \\\n",
       "0                                     Burkitt Lymphoma Genome Sequencing Project   \n",
       "1                HIV+ Tumor Molecular Characterization Project - Cervical Cancer   \n",
       "2  HIV+ Tumor Molecular Characterization Project - Diffuse Large B-Cell Lymphoma   \n",
       "\n",
       "                                      primary_site  \\\n",
       "0  [Hematopoietic and reticuloendothelial systems]   \n",
       "1                                   [Cervix uteri]   \n",
       "2  [Hematopoietic and reticuloendothelial systems]   \n",
       "\n",
       "                                                                                          disease_type  \\\n",
       "0                                                                            [Mature B-Cell Lymphomas]   \n",
       "1  [Adenomas and Adenocarcinomas, Squamous Cell Neoplasms, Epithelial Neoplasms, NOS, Complex Epith...   \n",
       "2                                                                            [Mature B-Cell Lymphomas]   \n",
       "\n",
       "   case_count  miRNA-Seq_case_count  RNA-Seq_case_count  \n",
       "0         324                   278                 283  \n",
       "1         212                   118                 123  \n",
       "2          70                    60                  55  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Write the DataFrame in a CSV file\n",
    "file_name = 'gdc-projects-of-interest.csv'\n",
    "df_projects.to_csv(f'{INTERIM_DATA_PATH}/{file_name}', index=False)\n",
    "\n",
    "# Print the DataFrame partially\n",
    "pd.set_option('display.max_colwidth', 100)\n",
    "df_projects.head(3)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Cases and Files"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Cases of Interest \n",
    "Cases with miRNA-Seq and RNA-Seq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Fields of interest for the endpoint request\n",
    "fields = [\n",
    "    'disease_type',\n",
    "    'files.access',\n",
    "    'files.data_category',\n",
    "    'files.data_format',\n",
    "    'files.data_type',\n",
    "    'files.file_id',\n",
    "    'files.experimental_strategy',\n",
    "    'primary_site',\n",
    "    'project.project_id'\n",
    "]\n",
    "fields = ','.join(fields)\n",
    "\n",
    "# Initialize the DataFrame of cases and files\n",
    "df_cases_and_files = pd.DataFrame()\n",
    "\n",
    "# Search for cases with miRNA-Seq and RNA-Seq in each project of interest\n",
    "for index in range(df_projects.shape[0]):\n",
    "    # Filters used in the endpoint request\n",
    "    filters = {\n",
    "        'op': 'and',\n",
    "        'content': [\n",
    "            {\n",
    "                'op': '=',\n",
    "                'content': {\n",
    "                    'field': 'project.project_id',\n",
    "                    'value': df_projects['project_id'][index]\n",
    "                }\n",
    "            },\n",
    "            {\n",
    "                'op': 'in',\n",
    "                'content': {\n",
    "                    'field': 'files.experimental_strategy',\n",
    "                    'value': ['miRNA-Seq', 'RNA-Seq']\n",
    "                }\n",
    "            }\n",
    "        ]\n",
    "    }\n",
    "\n",
    "    # Parameters for the endpoint request\n",
    "    params = {\n",
    "        'fields': fields,\n",
    "        'filters': filters,\n",
    "        'size': str(df_projects['case_count'][index])\n",
    "    }\n",
    "\n",
    "    # Request all project objects to the endpoint\n",
    "    response = requests.post(\n",
    "        url=CASES_ENDPOINT,\n",
    "        headers={'Content-Type': 'application/json'},\n",
    "        json=params\n",
    "    )\n",
    "\n",
    "    # Transform the response content to a DataFrame\n",
    "    json_response = json.loads(response.content.decode('utf-8'))\n",
    "    df_project_cases = pd.json_normalize(json_response['data']['hits'])\n",
    "\n",
    "    # Concatenate the cases of this project with the others\n",
    "    if df_cases_and_files.empty == False:\n",
    "        df_cases_and_files = pd.concat(\n",
    "            [df_cases_and_files, df_project_cases], ignore_index=True\n",
    "        )\n",
    "    else:\n",
    "        df_cases_and_files = df_project_cases.copy()\n",
    "\n",
    "# Rename some columns of the DataFrame\n",
    "df_cases_and_files = df_cases_and_files.rename(\n",
    "    columns={'id': 'case_id', 'project.project_id': 'project_id'}\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>case_id</th>\n",
       "      <th>primary_site</th>\n",
       "      <th>disease_type</th>\n",
       "      <th>project_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>cbb553da-c61c-4734-ac58-ef4580aaef47</td>\n",
       "      <td>Hematopoietic and reticuloendothelial systems</td>\n",
       "      <td>Mature B-Cell Lymphomas</td>\n",
       "      <td>CGCI-BLGSP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>681ff2e8-6073-4dab-a398-1f2203c2d7f5</td>\n",
       "      <td>Hematopoietic and reticuloendothelial systems</td>\n",
       "      <td>Mature B-Cell Lymphomas</td>\n",
       "      <td>CGCI-BLGSP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>886e1609-e89c-4d5e-ab9d-4f9ae18c937a</td>\n",
       "      <td>Hematopoietic and reticuloendothelial systems</td>\n",
       "      <td>Mature B-Cell Lymphomas</td>\n",
       "      <td>CGCI-BLGSP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>fdc9dadf-33c7-4b64-8bad-d6bb56fb2209</td>\n",
       "      <td>Hematopoietic and reticuloendothelial systems</td>\n",
       "      <td>Mature B-Cell Lymphomas</td>\n",
       "      <td>CGCI-BLGSP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>edf2e5ba-5702-4f96-97f7-124d70bb16aa</td>\n",
       "      <td>Hematopoietic and reticuloendothelial systems</td>\n",
       "      <td>Mature B-Cell Lymphomas</td>\n",
       "      <td>CGCI-BLGSP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16785</th>\n",
       "      <td>e21d8018-8fe9-4c92-8b36-28d7d3f7df2b</td>\n",
       "      <td>Eye and adnexa</td>\n",
       "      <td>Nevi and Melanomas</td>\n",
       "      <td>TCGA-UVM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16786</th>\n",
       "      <td>ebdabfcb-1c68-4ca1-8b7f-3ea9cdac3a6a</td>\n",
       "      <td>Eye and adnexa</td>\n",
       "      <td>Nevi and Melanomas</td>\n",
       "      <td>TCGA-UVM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16787</th>\n",
       "      <td>edbf6720-467e-4928-b1cb-0e7336358d18</td>\n",
       "      <td>Eye and adnexa</td>\n",
       "      <td>Nevi and Melanomas</td>\n",
       "      <td>TCGA-UVM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16788</th>\n",
       "      <td>eeab0529-31e6-4edb-bd2c-a21f2a169381</td>\n",
       "      <td>Eye and adnexa</td>\n",
       "      <td>Nevi and Melanomas</td>\n",
       "      <td>TCGA-UVM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16789</th>\n",
       "      <td>f78d8541-b317-4c8a-aa66-f172a8eaf14f</td>\n",
       "      <td>Eye and adnexa</td>\n",
       "      <td>Nevi and Melanomas</td>\n",
       "      <td>TCGA-UVM</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>16790 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                    case_id  \\\n",
       "0      cbb553da-c61c-4734-ac58-ef4580aaef47   \n",
       "1      681ff2e8-6073-4dab-a398-1f2203c2d7f5   \n",
       "2      886e1609-e89c-4d5e-ab9d-4f9ae18c937a   \n",
       "3      fdc9dadf-33c7-4b64-8bad-d6bb56fb2209   \n",
       "4      edf2e5ba-5702-4f96-97f7-124d70bb16aa   \n",
       "...                                     ...   \n",
       "16785  e21d8018-8fe9-4c92-8b36-28d7d3f7df2b   \n",
       "16786  ebdabfcb-1c68-4ca1-8b7f-3ea9cdac3a6a   \n",
       "16787  edbf6720-467e-4928-b1cb-0e7336358d18   \n",
       "16788  eeab0529-31e6-4edb-bd2c-a21f2a169381   \n",
       "16789  f78d8541-b317-4c8a-aa66-f172a8eaf14f   \n",
       "\n",
       "                                        primary_site             disease_type  \\\n",
       "0      Hematopoietic and reticuloendothelial systems  Mature B-Cell Lymphomas   \n",
       "1      Hematopoietic and reticuloendothelial systems  Mature B-Cell Lymphomas   \n",
       "2      Hematopoietic and reticuloendothelial systems  Mature B-Cell Lymphomas   \n",
       "3      Hematopoietic and reticuloendothelial systems  Mature B-Cell Lymphomas   \n",
       "4      Hematopoietic and reticuloendothelial systems  Mature B-Cell Lymphomas   \n",
       "...                                              ...                      ...   \n",
       "16785                                 Eye and adnexa       Nevi and Melanomas   \n",
       "16786                                 Eye and adnexa       Nevi and Melanomas   \n",
       "16787                                 Eye and adnexa       Nevi and Melanomas   \n",
       "16788                                 Eye and adnexa       Nevi and Melanomas   \n",
       "16789                                 Eye and adnexa       Nevi and Melanomas   \n",
       "\n",
       "       project_id  \n",
       "0      CGCI-BLGSP  \n",
       "1      CGCI-BLGSP  \n",
       "2      CGCI-BLGSP  \n",
       "3      CGCI-BLGSP  \n",
       "4      CGCI-BLGSP  \n",
       "...           ...  \n",
       "16785    TCGA-UVM  \n",
       "16786    TCGA-UVM  \n",
       "16787    TCGA-UVM  \n",
       "16788    TCGA-UVM  \n",
       "16789    TCGA-UVM  \n",
       "\n",
       "[16790 rows x 4 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Create the cases DataFrame\n",
    "df_cases = df_cases_and_files.drop(columns='files')\n",
    "\n",
    "# Write the DataFrame in a CSV file\n",
    "file_name = 'gdc-cases-of-interest.csv'\n",
    "df_cases.to_csv(f'{INTERIM_DATA_PATH}/{file_name}', index=False)\n",
    "\n",
    "# Print the DataFrame\n",
    "df_cases"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Files of Interest \n",
    "Files from miRNA-Seq and RNA-Seq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Explode the lists dictionary over case files\n",
    "df_cases_and_files = df_cases_and_files.explode('files')\n",
    "\n",
    "# Filter case files related to miRNA-Seq or RNA-Seq\n",
    "key = 'experimental_strategy'\n",
    "df_cases_and_files = df_cases_and_files[\n",
    "    df_cases_and_files['files'].apply(\n",
    "        lambda x: (\n",
    "            key in x and (x[key] == 'miRNA-Seq' or x[key] == 'RNA-Seq')\n",
    "        )\n",
    "    )\n",
    "]\n",
    "\n",
    "# Create the files DataFrame\n",
    "df_files = pd.concat(\n",
    "    objs=[\n",
    "        df_cases_and_files.reset_index(drop=True),\n",
    "        pd.json_normalize(df_cases_and_files['files'])\n",
    "    ],\n",
    "    axis='columns'\n",
    ")\n",
    "df_files = df_files.drop(\n",
    "    columns=['disease_type', 'files', 'primary_site', 'project_id']\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>case_id</th>\n",
       "      <th>data_format</th>\n",
       "      <th>access</th>\n",
       "      <th>file_id</th>\n",
       "      <th>data_type</th>\n",
       "      <th>data_category</th>\n",
       "      <th>experimental_strategy</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>cbb553da-c61c-4734-ac58-ef4580aaef47</td>\n",
       "      <td>TSV</td>\n",
       "      <td>controlled</td>\n",
       "      <td>7fad3cc2-cc77-4f0f-b63c-a94a274f10e2</td>\n",
       "      <td>Transcript Fusion</td>\n",
       "      <td>Structural Variation</td>\n",
       "      <td>RNA-Seq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>cbb553da-c61c-4734-ac58-ef4580aaef47</td>\n",
       "      <td>TSV</td>\n",
       "      <td>open</td>\n",
       "      <td>030842d9-69b2-4938-9475-31c43cce6f7a</td>\n",
       "      <td>Gene Expression Quantification</td>\n",
       "      <td>Transcriptome Profiling</td>\n",
       "      <td>RNA-Seq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>cbb553da-c61c-4734-ac58-ef4580aaef47</td>\n",
       "      <td>TSV</td>\n",
       "      <td>controlled</td>\n",
       "      <td>862a63ae-d83e-431d-94fb-613570e01477</td>\n",
       "      <td>Splice Junction Quantification</td>\n",
       "      <td>Transcriptome Profiling</td>\n",
       "      <td>RNA-Seq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>cbb553da-c61c-4734-ac58-ef4580aaef47</td>\n",
       "      <td>TSV</td>\n",
       "      <td>controlled</td>\n",
       "      <td>44816e46-1474-4c6c-a4c8-c2cee21329d0</td>\n",
       "      <td>Transcript Fusion</td>\n",
       "      <td>Structural Variation</td>\n",
       "      <td>RNA-Seq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>cbb553da-c61c-4734-ac58-ef4580aaef47</td>\n",
       "      <td>BEDPE</td>\n",
       "      <td>controlled</td>\n",
       "      <td>a93709d9-9a6f-4ceb-aeca-a9afd48c13b1</td>\n",
       "      <td>Transcript Fusion</td>\n",
       "      <td>Structural Variation</td>\n",
       "      <td>RNA-Seq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>232548</th>\n",
       "      <td>f78d8541-b317-4c8a-aa66-f172a8eaf14f</td>\n",
       "      <td>BEDPE</td>\n",
       "      <td>controlled</td>\n",
       "      <td>8a7a812c-cb90-492a-982f-f27e3ce39794</td>\n",
       "      <td>Transcript Fusion</td>\n",
       "      <td>Structural Variation</td>\n",
       "      <td>RNA-Seq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>232549</th>\n",
       "      <td>f78d8541-b317-4c8a-aa66-f172a8eaf14f</td>\n",
       "      <td>TSV</td>\n",
       "      <td>controlled</td>\n",
       "      <td>c948d42f-8eb8-4425-b8dd-27b973ccc6b7</td>\n",
       "      <td>Splice Junction Quantification</td>\n",
       "      <td>Transcriptome Profiling</td>\n",
       "      <td>RNA-Seq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>232550</th>\n",
       "      <td>f78d8541-b317-4c8a-aa66-f172a8eaf14f</td>\n",
       "      <td>BAM</td>\n",
       "      <td>controlled</td>\n",
       "      <td>c5a9c2d9-1971-4eaf-923e-f2d1df90fdcb</td>\n",
       "      <td>Aligned Reads</td>\n",
       "      <td>Sequencing Reads</td>\n",
       "      <td>RNA-Seq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>232551</th>\n",
       "      <td>f78d8541-b317-4c8a-aa66-f172a8eaf14f</td>\n",
       "      <td>TSV</td>\n",
       "      <td>open</td>\n",
       "      <td>e1849623-895a-4981-b8c1-25ad0ec1143b</td>\n",
       "      <td>Gene Expression Quantification</td>\n",
       "      <td>Transcriptome Profiling</td>\n",
       "      <td>RNA-Seq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>232552</th>\n",
       "      <td>f78d8541-b317-4c8a-aa66-f172a8eaf14f</td>\n",
       "      <td>TXT</td>\n",
       "      <td>open</td>\n",
       "      <td>b42c5f8b-f027-46a6-9485-80d65664b3c8</td>\n",
       "      <td>miRNA Expression Quantification</td>\n",
       "      <td>Transcriptome Profiling</td>\n",
       "      <td>miRNA-Seq</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>232553 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                     case_id data_format      access  \\\n",
       "0       cbb553da-c61c-4734-ac58-ef4580aaef47         TSV  controlled   \n",
       "1       cbb553da-c61c-4734-ac58-ef4580aaef47         TSV        open   \n",
       "2       cbb553da-c61c-4734-ac58-ef4580aaef47         TSV  controlled   \n",
       "3       cbb553da-c61c-4734-ac58-ef4580aaef47         TSV  controlled   \n",
       "4       cbb553da-c61c-4734-ac58-ef4580aaef47       BEDPE  controlled   \n",
       "...                                      ...         ...         ...   \n",
       "232548  f78d8541-b317-4c8a-aa66-f172a8eaf14f       BEDPE  controlled   \n",
       "232549  f78d8541-b317-4c8a-aa66-f172a8eaf14f         TSV  controlled   \n",
       "232550  f78d8541-b317-4c8a-aa66-f172a8eaf14f         BAM  controlled   \n",
       "232551  f78d8541-b317-4c8a-aa66-f172a8eaf14f         TSV        open   \n",
       "232552  f78d8541-b317-4c8a-aa66-f172a8eaf14f         TXT        open   \n",
       "\n",
       "                                     file_id                        data_type  \\\n",
       "0       7fad3cc2-cc77-4f0f-b63c-a94a274f10e2                Transcript Fusion   \n",
       "1       030842d9-69b2-4938-9475-31c43cce6f7a   Gene Expression Quantification   \n",
       "2       862a63ae-d83e-431d-94fb-613570e01477   Splice Junction Quantification   \n",
       "3       44816e46-1474-4c6c-a4c8-c2cee21329d0                Transcript Fusion   \n",
       "4       a93709d9-9a6f-4ceb-aeca-a9afd48c13b1                Transcript Fusion   \n",
       "...                                      ...                              ...   \n",
       "232548  8a7a812c-cb90-492a-982f-f27e3ce39794                Transcript Fusion   \n",
       "232549  c948d42f-8eb8-4425-b8dd-27b973ccc6b7   Splice Junction Quantification   \n",
       "232550  c5a9c2d9-1971-4eaf-923e-f2d1df90fdcb                    Aligned Reads   \n",
       "232551  e1849623-895a-4981-b8c1-25ad0ec1143b   Gene Expression Quantification   \n",
       "232552  b42c5f8b-f027-46a6-9485-80d65664b3c8  miRNA Expression Quantification   \n",
       "\n",
       "                  data_category experimental_strategy  \n",
       "0          Structural Variation               RNA-Seq  \n",
       "1       Transcriptome Profiling               RNA-Seq  \n",
       "2       Transcriptome Profiling               RNA-Seq  \n",
       "3          Structural Variation               RNA-Seq  \n",
       "4          Structural Variation               RNA-Seq  \n",
       "...                         ...                   ...  \n",
       "232548     Structural Variation               RNA-Seq  \n",
       "232549  Transcriptome Profiling               RNA-Seq  \n",
       "232550         Sequencing Reads               RNA-Seq  \n",
       "232551  Transcriptome Profiling               RNA-Seq  \n",
       "232552  Transcriptome Profiling             miRNA-Seq  \n",
       "\n",
       "[232553 rows x 7 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Write the DataFrame in a CSV file\n",
    "file_name = 'gdc-files-of-interest.csv'\n",
    "df_files.to_csv(f'{INTERIM_DATA_PATH}/{file_name}', index=False)\n",
    "\n",
    "# Print the DataFrame\n",
    "df_files"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
