{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# **Exploratory Data Analysis of TCGA Data from GDC API**\n",
    "TCGA: The Cancer Genome Atlas  \n",
    "GDC: Genomic Data Commons"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Importing Libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Constants and Paths"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Endpoint for files download\n",
    "DATA_ENDPOINT = 'https://api.gdc.cancer.gov/data'\n",
    "\n",
    "# Data folder path\n",
    "DATA_PATH = '../../data'\n",
    "\n",
    "# External data folder path\n",
    "EXTERNAL_DATA_PATH = f'{DATA_PATH}/external/gdc-tcga'\n",
    "\n",
    "# Interim data folder path\n",
    "INTERIM_DATA_PATH = f'{DATA_PATH}/interim/gdc-tcga'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Data Loading and Preparation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# DataFrame with TCGA projects\n",
    "df_projects = pd.read_csv(f'{INTERIM_DATA_PATH}/tcga-projects.csv')\n",
    "\n",
    "# DataFrame with the cases of interest\n",
    "df_cases = pd.read_csv(f'{INTERIM_DATA_PATH}/tcga-cases-of-interest.csv')\n",
    "\n",
    "# DataFrame with the files of interest\n",
    "df_files = pd.read_csv(f'{INTERIM_DATA_PATH}/tcga-files-of-interest.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Join both cases and files DataFrames for the exploratory data analysis (EDA)\n",
    "df_eda = df_cases \\\n",
    "    .merge(\n",
    "        right=df_files,\n",
    "        left_on='case_id',\n",
    "        right_on='case_id',\n",
    "        how='inner'\n",
    "    )\n",
    "\n",
    "# Create an acronym column for the file data type\n",
    "df_eda['data_acronym'] = np.where(\n",
    "    df_eda['data_type'] == 'Gene Expression Quantification', 'RNA', 'microRNA'\n",
    ")\n",
    "\n",
    "# Create an acronym column that combines file data type and sample tissue type\n",
    "df_eda['data_and_sample'] = (\n",
    "    df_eda['data_acronym'] + '-' + df_eda['tissue_type']\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>case_id</th>\n",
       "      <th>case_primary_site</th>\n",
       "      <th>case_disease_type</th>\n",
       "      <th>project_id</th>\n",
       "      <th>file_id</th>\n",
       "      <th>data_format</th>\n",
       "      <th>data_type</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>tissue_type</th>\n",
       "      <th>created_datetime</th>\n",
       "      <th>updated_datetime</th>\n",
       "      <th>data_acronym</th>\n",
       "      <th>data_and_sample</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4298ccdb-2e6d-4267-822d-75b021364084</td>\n",
       "      <td>Kidney</td>\n",
       "      <td>Adenomas and Adenocarcinomas</td>\n",
       "      <td>TCGA-KIRC</td>\n",
       "      <td>7ebb3c64-2d8b-4388-8ad8-555fb7cea1b4</td>\n",
       "      <td>TXT</td>\n",
       "      <td>Isoform Expression Quantification</td>\n",
       "      <td>Primary Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>2018-03-20T00:16:05.303968-05:00</td>\n",
       "      <td>2024-07-29T21:20:32.946706-05:00</td>\n",
       "      <td>microRNA</td>\n",
       "      <td>microRNA-Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4298ccdb-2e6d-4267-822d-75b021364084</td>\n",
       "      <td>Kidney</td>\n",
       "      <td>Adenomas and Adenocarcinomas</td>\n",
       "      <td>TCGA-KIRC</td>\n",
       "      <td>6d0df9bf-69d6-447f-b2d9-76391e95d61d</td>\n",
       "      <td>TSV</td>\n",
       "      <td>Gene Expression Quantification</td>\n",
       "      <td>Primary Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>2021-12-13T19:31:32.504341-06:00</td>\n",
       "      <td>2024-07-30T11:11:25.450940-05:00</td>\n",
       "      <td>RNA</td>\n",
       "      <td>RNA-Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>a2663a86-a006-4867-9e88-2b523df48303</td>\n",
       "      <td>Kidney</td>\n",
       "      <td>Adenomas and Adenocarcinomas</td>\n",
       "      <td>TCGA-KIRC</td>\n",
       "      <td>9f7e5a27-9a09-474e-a2d6-ed478a124fe3</td>\n",
       "      <td>TXT</td>\n",
       "      <td>Isoform Expression Quantification</td>\n",
       "      <td>Primary Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>2018-03-19T23:43:21.888100-05:00</td>\n",
       "      <td>2024-07-29T21:23:47.515560-05:00</td>\n",
       "      <td>microRNA</td>\n",
       "      <td>microRNA-Tumor</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                case_id case_primary_site  \\\n",
       "0  4298ccdb-2e6d-4267-822d-75b021364084            Kidney   \n",
       "1  4298ccdb-2e6d-4267-822d-75b021364084            Kidney   \n",
       "2  a2663a86-a006-4867-9e88-2b523df48303            Kidney   \n",
       "\n",
       "              case_disease_type project_id  \\\n",
       "0  Adenomas and Adenocarcinomas  TCGA-KIRC   \n",
       "1  Adenomas and Adenocarcinomas  TCGA-KIRC   \n",
       "2  Adenomas and Adenocarcinomas  TCGA-KIRC   \n",
       "\n",
       "                                file_id data_format  \\\n",
       "0  7ebb3c64-2d8b-4388-8ad8-555fb7cea1b4         TXT   \n",
       "1  6d0df9bf-69d6-447f-b2d9-76391e95d61d         TSV   \n",
       "2  9f7e5a27-9a09-474e-a2d6-ed478a124fe3         TXT   \n",
       "\n",
       "                           data_type    sample_type tissue_type  \\\n",
       "0  Isoform Expression Quantification  Primary Tumor       Tumor   \n",
       "1     Gene Expression Quantification  Primary Tumor       Tumor   \n",
       "2  Isoform Expression Quantification  Primary Tumor       Tumor   \n",
       "\n",
       "                   created_datetime                  updated_datetime  \\\n",
       "0  2018-03-20T00:16:05.303968-05:00  2024-07-29T21:20:32.946706-05:00   \n",
       "1  2021-12-13T19:31:32.504341-06:00  2024-07-30T11:11:25.450940-05:00   \n",
       "2  2018-03-19T23:43:21.888100-05:00  2024-07-29T21:23:47.515560-05:00   \n",
       "\n",
       "  data_acronym data_and_sample  \n",
       "0     microRNA  microRNA-Tumor  \n",
       "1          RNA       RNA-Tumor  \n",
       "2     microRNA  microRNA-Tumor  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the EDA DataFrame partially\n",
    "df_eda.head(3)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Objects Quantification"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "32"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Total number of projects of interest\n",
    "df_eda['project_id'].nunique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "10207"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Total number of cases of interest\n",
    "df_eda['case_id'].nunique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "21714"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Total number of files of interest\n",
    "df_eda['file_id'].nunique()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Projects"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Cases and Files per Project"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>project_id</th>\n",
       "      <th>case_count</th>\n",
       "      <th>file_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TCGA-BRCA</td>\n",
       "      <td>1097</td>\n",
       "      <td>2424</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TCGA-UCEC</td>\n",
       "      <td>559</td>\n",
       "      <td>1166</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TCGA-KIRC</td>\n",
       "      <td>534</td>\n",
       "      <td>1228</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TCGA-HNSC</td>\n",
       "      <td>528</td>\n",
       "      <td>1131</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TCGA-OV</td>\n",
       "      <td>522</td>\n",
       "      <td>987</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>TCGA-LUAD</td>\n",
       "      <td>519</td>\n",
       "      <td>1163</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>TCGA-LGG</td>\n",
       "      <td>516</td>\n",
       "      <td>1028</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>TCGA-THCA</td>\n",
       "      <td>507</td>\n",
       "      <td>1129</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>TCGA-LUSC</td>\n",
       "      <td>504</td>\n",
       "      <td>1095</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>TCGA-PRAD</td>\n",
       "      <td>498</td>\n",
       "      <td>1103</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>TCGA-COAD</td>\n",
       "      <td>459</td>\n",
       "      <td>985</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>TCGA-STAD</td>\n",
       "      <td>439</td>\n",
       "      <td>939</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>TCGA-BLCA</td>\n",
       "      <td>412</td>\n",
       "      <td>867</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>TCGA-LIHC</td>\n",
       "      <td>376</td>\n",
       "      <td>843</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>TCGA-GBM</td>\n",
       "      <td>336</td>\n",
       "      <td>654</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>TCGA-CESC</td>\n",
       "      <td>307</td>\n",
       "      <td>617</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>TCGA-KIRP</td>\n",
       "      <td>291</td>\n",
       "      <td>647</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>TCGA-SARC</td>\n",
       "      <td>261</td>\n",
       "      <td>520</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>TCGA-ESCA</td>\n",
       "      <td>185</td>\n",
       "      <td>396</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>TCGA-PCPG</td>\n",
       "      <td>179</td>\n",
       "      <td>364</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>TCGA-PAAD</td>\n",
       "      <td>178</td>\n",
       "      <td>364</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>TCGA-READ</td>\n",
       "      <td>167</td>\n",
       "      <td>340</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>TCGA-TGCT</td>\n",
       "      <td>150</td>\n",
       "      <td>300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>TCGA-THYM</td>\n",
       "      <td>124</td>\n",
       "      <td>248</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>TCGA-SKCM</td>\n",
       "      <td>105</td>\n",
       "      <td>203</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>TCGA-MESO</td>\n",
       "      <td>87</td>\n",
       "      <td>174</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>TCGA-ACC</td>\n",
       "      <td>80</td>\n",
       "      <td>159</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>TCGA-UVM</td>\n",
       "      <td>80</td>\n",
       "      <td>160</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>TCGA-KICH</td>\n",
       "      <td>66</td>\n",
       "      <td>182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>TCGA-UCS</td>\n",
       "      <td>57</td>\n",
       "      <td>114</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>TCGA-DLBC</td>\n",
       "      <td>48</td>\n",
       "      <td>95</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>TCGA-CHOL</td>\n",
       "      <td>36</td>\n",
       "      <td>89</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   project_id  case_count  file_count\n",
       "0   TCGA-BRCA        1097        2424\n",
       "1   TCGA-UCEC         559        1166\n",
       "2   TCGA-KIRC         534        1228\n",
       "3   TCGA-HNSC         528        1131\n",
       "4     TCGA-OV         522         987\n",
       "5   TCGA-LUAD         519        1163\n",
       "6    TCGA-LGG         516        1028\n",
       "7   TCGA-THCA         507        1129\n",
       "8   TCGA-LUSC         504        1095\n",
       "9   TCGA-PRAD         498        1103\n",
       "10  TCGA-COAD         459         985\n",
       "11  TCGA-STAD         439         939\n",
       "12  TCGA-BLCA         412         867\n",
       "13  TCGA-LIHC         376         843\n",
       "14   TCGA-GBM         336         654\n",
       "15  TCGA-CESC         307         617\n",
       "16  TCGA-KIRP         291         647\n",
       "17  TCGA-SARC         261         520\n",
       "18  TCGA-ESCA         185         396\n",
       "19  TCGA-PCPG         179         364\n",
       "20  TCGA-PAAD         178         364\n",
       "21  TCGA-READ         167         340\n",
       "22  TCGA-TGCT         150         300\n",
       "23  TCGA-THYM         124         248\n",
       "24  TCGA-SKCM         105         203\n",
       "25  TCGA-MESO          87         174\n",
       "26   TCGA-ACC          80         159\n",
       "27   TCGA-UVM          80         160\n",
       "28  TCGA-KICH          66         182\n",
       "29   TCGA-UCS          57         114\n",
       "30  TCGA-DLBC          48          95\n",
       "31  TCGA-CHOL          36          89"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Count the total number of cases and files per project \n",
    "df_eda \\\n",
    "    .groupby('project_id') \\\n",
    "    .agg(\n",
    "        case_count=pd.NamedAgg(column='case_id', aggfunc='nunique'),\n",
    "        file_count=pd.NamedAgg(column='file_id', aggfunc='nunique')\n",
    "    ) \\\n",
    "    .sort_values(by='case_count', ascending=False) \\\n",
    "    .reset_index()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Sample Types and Data File Types per Project"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>case_count</th>\n",
       "      <th>file_count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>project_id</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>data_acronym</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">TCGA-ACC</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">Primary Tumor</th>\n",
       "      <th>RNA</th>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>80</td>\n",
       "      <td>80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">TCGA-BLCA</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">Primary Tumor</th>\n",
       "      <th>RNA</th>\n",
       "      <td>406</td>\n",
       "      <td>412</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>409</td>\n",
       "      <td>417</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">Solid Tissue Normal</th>\n",
       "      <th>RNA</th>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">TCGA-BRCA</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">Primary Tumor</th>\n",
       "      <th>RNA</th>\n",
       "      <td>1095</td>\n",
       "      <td>1111</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>1078</td>\n",
       "      <td>1096</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">Solid Tissue Normal</th>\n",
       "      <th>RNA</th>\n",
       "      <td>113</td>\n",
       "      <td>113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>104</td>\n",
       "      <td>104</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">TCGA-CESC</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">Primary Tumor</th>\n",
       "      <th>RNA</th>\n",
       "      <td>304</td>\n",
       "      <td>304</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>307</td>\n",
       "      <td>307</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">Solid Tissue Normal</th>\n",
       "      <th>RNA</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">TCGA-CHOL</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">Primary Tumor</th>\n",
       "      <th>RNA</th>\n",
       "      <td>35</td>\n",
       "      <td>35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>36</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">Solid Tissue Normal</th>\n",
       "      <th>RNA</th>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">TCGA-COAD</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">Primary Tumor</th>\n",
       "      <th>RNA</th>\n",
       "      <td>458</td>\n",
       "      <td>481</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>444</td>\n",
       "      <td>455</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">Solid Tissue Normal</th>\n",
       "      <th>RNA</th>\n",
       "      <td>41</td>\n",
       "      <td>41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">TCGA-DLBC</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">Primary Tumor</th>\n",
       "      <th>RNA</th>\n",
       "      <td>48</td>\n",
       "      <td>48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>47</td>\n",
       "      <td>47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">TCGA-ESCA</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">Primary Tumor</th>\n",
       "      <th>RNA</th>\n",
       "      <td>184</td>\n",
       "      <td>184</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>184</td>\n",
       "      <td>186</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">Solid Tissue Normal</th>\n",
       "      <th>RNA</th>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">TCGA-GBM</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">Primary Tumor</th>\n",
       "      <th>RNA</th>\n",
       "      <td>284</td>\n",
       "      <td>372</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>267</td>\n",
       "      <td>272</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">Solid Tissue Normal</th>\n",
       "      <th>RNA</th>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">TCGA-HNSC</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">Primary Tumor</th>\n",
       "      <th>RNA</th>\n",
       "      <td>520</td>\n",
       "      <td>520</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>523</td>\n",
       "      <td>523</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">Solid Tissue Normal</th>\n",
       "      <th>RNA</th>\n",
       "      <td>44</td>\n",
       "      <td>44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>44</td>\n",
       "      <td>44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">TCGA-KICH</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">Primary Tumor</th>\n",
       "      <th>RNA</th>\n",
       "      <td>66</td>\n",
       "      <td>66</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>66</td>\n",
       "      <td>66</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">Solid Tissue Normal</th>\n",
       "      <th>RNA</th>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">TCGA-KIRC</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">Primary Tumor</th>\n",
       "      <th>RNA</th>\n",
       "      <td>533</td>\n",
       "      <td>541</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>516</td>\n",
       "      <td>544</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">Solid Tissue Normal</th>\n",
       "      <th>RNA</th>\n",
       "      <td>72</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>71</td>\n",
       "      <td>71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">TCGA-KIRP</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">Primary Tumor</th>\n",
       "      <th>RNA</th>\n",
       "      <td>290</td>\n",
       "      <td>290</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>291</td>\n",
       "      <td>291</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">Solid Tissue Normal</th>\n",
       "      <th>RNA</th>\n",
       "      <td>32</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>34</td>\n",
       "      <td>34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">TCGA-LGG</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">Primary Tumor</th>\n",
       "      <th>RNA</th>\n",
       "      <td>516</td>\n",
       "      <td>516</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>512</td>\n",
       "      <td>512</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">TCGA-LIHC</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">Primary Tumor</th>\n",
       "      <th>RNA</th>\n",
       "      <td>371</td>\n",
       "      <td>371</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>372</td>\n",
       "      <td>372</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">Solid Tissue Normal</th>\n",
       "      <th>RNA</th>\n",
       "      <td>50</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>50</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">TCGA-LUAD</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">Primary Tumor</th>\n",
       "      <th>RNA</th>\n",
       "      <td>516</td>\n",
       "      <td>539</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>513</td>\n",
       "      <td>519</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">Solid Tissue Normal</th>\n",
       "      <th>RNA</th>\n",
       "      <td>59</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>46</td>\n",
       "      <td>46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">TCGA-LUSC</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">Primary Tumor</th>\n",
       "      <th>RNA</th>\n",
       "      <td>501</td>\n",
       "      <td>511</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>488</td>\n",
       "      <td>488</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">Solid Tissue Normal</th>\n",
       "      <th>RNA</th>\n",
       "      <td>51</td>\n",
       "      <td>51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>45</td>\n",
       "      <td>45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">TCGA-MESO</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">Primary Tumor</th>\n",
       "      <th>RNA</th>\n",
       "      <td>87</td>\n",
       "      <td>87</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>87</td>\n",
       "      <td>87</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">TCGA-OV</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">Primary Tumor</th>\n",
       "      <th>RNA</th>\n",
       "      <td>421</td>\n",
       "      <td>421</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>522</td>\n",
       "      <td>566</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">TCGA-PAAD</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">Primary Tumor</th>\n",
       "      <th>RNA</th>\n",
       "      <td>178</td>\n",
       "      <td>178</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>178</td>\n",
       "      <td>178</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">Solid Tissue Normal</th>\n",
       "      <th>RNA</th>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">TCGA-PCPG</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">Primary Tumor</th>\n",
       "      <th>RNA</th>\n",
       "      <td>179</td>\n",
       "      <td>179</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>179</td>\n",
       "      <td>179</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">Solid Tissue Normal</th>\n",
       "      <th>RNA</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">TCGA-PRAD</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">Primary Tumor</th>\n",
       "      <th>RNA</th>\n",
       "      <td>497</td>\n",
       "      <td>501</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>494</td>\n",
       "      <td>498</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">Solid Tissue Normal</th>\n",
       "      <th>RNA</th>\n",
       "      <td>52</td>\n",
       "      <td>52</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>52</td>\n",
       "      <td>52</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">TCGA-READ</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">Primary Tumor</th>\n",
       "      <th>RNA</th>\n",
       "      <td>166</td>\n",
       "      <td>166</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>161</td>\n",
       "      <td>161</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">Solid Tissue Normal</th>\n",
       "      <th>RNA</th>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"3\" valign=\"top\">TCGA-SARC</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">Primary Tumor</th>\n",
       "      <th>RNA</th>\n",
       "      <td>259</td>\n",
       "      <td>259</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>259</td>\n",
       "      <td>259</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Solid Tissue Normal</th>\n",
       "      <th>RNA</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">TCGA-SKCM</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">Primary Tumor</th>\n",
       "      <th>RNA</th>\n",
       "      <td>103</td>\n",
       "      <td>103</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>97</td>\n",
       "      <td>97</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">Solid Tissue Normal</th>\n",
       "      <th>RNA</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">TCGA-STAD</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">Primary Tumor</th>\n",
       "      <th>RNA</th>\n",
       "      <td>412</td>\n",
       "      <td>412</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>436</td>\n",
       "      <td>446</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">Solid Tissue Normal</th>\n",
       "      <th>RNA</th>\n",
       "      <td>36</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>41</td>\n",
       "      <td>45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">TCGA-TGCT</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">Primary Tumor</th>\n",
       "      <th>RNA</th>\n",
       "      <td>150</td>\n",
       "      <td>150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>150</td>\n",
       "      <td>150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">TCGA-THCA</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">Primary Tumor</th>\n",
       "      <th>RNA</th>\n",
       "      <td>505</td>\n",
       "      <td>505</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>506</td>\n",
       "      <td>506</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">Solid Tissue Normal</th>\n",
       "      <th>RNA</th>\n",
       "      <td>59</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>59</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">TCGA-THYM</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">Primary Tumor</th>\n",
       "      <th>RNA</th>\n",
       "      <td>120</td>\n",
       "      <td>120</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>124</td>\n",
       "      <td>124</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">Solid Tissue Normal</th>\n",
       "      <th>RNA</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">TCGA-UCEC</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">Primary Tumor</th>\n",
       "      <th>RNA</th>\n",
       "      <td>545</td>\n",
       "      <td>553</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>538</td>\n",
       "      <td>545</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">Solid Tissue Normal</th>\n",
       "      <th>RNA</th>\n",
       "      <td>35</td>\n",
       "      <td>35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">TCGA-UCS</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">Primary Tumor</th>\n",
       "      <th>RNA</th>\n",
       "      <td>57</td>\n",
       "      <td>57</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>57</td>\n",
       "      <td>57</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">TCGA-UVM</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">Primary Tumor</th>\n",
       "      <th>RNA</th>\n",
       "      <td>80</td>\n",
       "      <td>80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>microRNA</th>\n",
       "      <td>80</td>\n",
       "      <td>80</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                             case_count  file_count\n",
       "project_id sample_type         data_acronym                        \n",
       "TCGA-ACC   Primary Tumor       RNA                   79          79\n",
       "                               microRNA              80          80\n",
       "TCGA-BLCA  Primary Tumor       RNA                  406         412\n",
       "                               microRNA             409         417\n",
       "           Solid Tissue Normal RNA                   19          19\n",
       "                               microRNA              19          19\n",
       "TCGA-BRCA  Primary Tumor       RNA                 1095        1111\n",
       "                               microRNA            1078        1096\n",
       "           Solid Tissue Normal RNA                  113         113\n",
       "                               microRNA             104         104\n",
       "TCGA-CESC  Primary Tumor       RNA                  304         304\n",
       "                               microRNA             307         307\n",
       "           Solid Tissue Normal RNA                    3           3\n",
       "                               microRNA               3           3\n",
       "TCGA-CHOL  Primary Tumor       RNA                   35          35\n",
       "                               microRNA              36          36\n",
       "           Solid Tissue Normal RNA                    9           9\n",
       "                               microRNA               9           9\n",
       "TCGA-COAD  Primary Tumor       RNA                  458         481\n",
       "                               microRNA             444         455\n",
       "           Solid Tissue Normal RNA                   41          41\n",
       "                               microRNA               8           8\n",
       "TCGA-DLBC  Primary Tumor       RNA                   48          48\n",
       "                               microRNA              47          47\n",
       "TCGA-ESCA  Primary Tumor       RNA                  184         184\n",
       "                               microRNA             184         186\n",
       "           Solid Tissue Normal RNA                   13          13\n",
       "                               microRNA              13          13\n",
       "TCGA-GBM   Primary Tumor       RNA                  284         372\n",
       "                               microRNA             267         272\n",
       "           Solid Tissue Normal RNA                    5           5\n",
       "                               microRNA               5           5\n",
       "TCGA-HNSC  Primary Tumor       RNA                  520         520\n",
       "                               microRNA             523         523\n",
       "           Solid Tissue Normal RNA                   44          44\n",
       "                               microRNA              44          44\n",
       "TCGA-KICH  Primary Tumor       RNA                   66          66\n",
       "                               microRNA              66          66\n",
       "           Solid Tissue Normal RNA                   25          25\n",
       "                               microRNA              25          25\n",
       "TCGA-KIRC  Primary Tumor       RNA                  533         541\n",
       "                               microRNA             516         544\n",
       "           Solid Tissue Normal RNA                   72          72\n",
       "                               microRNA              71          71\n",
       "TCGA-KIRP  Primary Tumor       RNA                  290         290\n",
       "                               microRNA             291         291\n",
       "           Solid Tissue Normal RNA                   32          32\n",
       "                               microRNA              34          34\n",
       "TCGA-LGG   Primary Tumor       RNA                  516         516\n",
       "                               microRNA             512         512\n",
       "TCGA-LIHC  Primary Tumor       RNA                  371         371\n",
       "                               microRNA             372         372\n",
       "           Solid Tissue Normal RNA                   50          50\n",
       "                               microRNA              50          50\n",
       "TCGA-LUAD  Primary Tumor       RNA                  516         539\n",
       "                               microRNA             513         519\n",
       "           Solid Tissue Normal RNA                   59          59\n",
       "                               microRNA              46          46\n",
       "TCGA-LUSC  Primary Tumor       RNA                  501         511\n",
       "                               microRNA             488         488\n",
       "           Solid Tissue Normal RNA                   51          51\n",
       "                               microRNA              45          45\n",
       "TCGA-MESO  Primary Tumor       RNA                   87          87\n",
       "                               microRNA              87          87\n",
       "TCGA-OV    Primary Tumor       RNA                  421         421\n",
       "                               microRNA             522         566\n",
       "TCGA-PAAD  Primary Tumor       RNA                  178         178\n",
       "                               microRNA             178         178\n",
       "           Solid Tissue Normal RNA                    4           4\n",
       "                               microRNA               4           4\n",
       "TCGA-PCPG  Primary Tumor       RNA                  179         179\n",
       "                               microRNA             179         179\n",
       "           Solid Tissue Normal RNA                    3           3\n",
       "                               microRNA               3           3\n",
       "TCGA-PRAD  Primary Tumor       RNA                  497         501\n",
       "                               microRNA             494         498\n",
       "           Solid Tissue Normal RNA                   52          52\n",
       "                               microRNA              52          52\n",
       "TCGA-READ  Primary Tumor       RNA                  166         166\n",
       "                               microRNA             161         161\n",
       "           Solid Tissue Normal RNA                   10          10\n",
       "                               microRNA               3           3\n",
       "TCGA-SARC  Primary Tumor       RNA                  259         259\n",
       "                               microRNA             259         259\n",
       "           Solid Tissue Normal RNA                    2           2\n",
       "TCGA-SKCM  Primary Tumor       RNA                  103         103\n",
       "                               microRNA              97          97\n",
       "           Solid Tissue Normal RNA                    1           1\n",
       "                               microRNA               2           2\n",
       "TCGA-STAD  Primary Tumor       RNA                  412         412\n",
       "                               microRNA             436         446\n",
       "           Solid Tissue Normal RNA                   36          36\n",
       "                               microRNA              41          45\n",
       "TCGA-TGCT  Primary Tumor       RNA                  150         150\n",
       "                               microRNA             150         150\n",
       "TCGA-THCA  Primary Tumor       RNA                  505         505\n",
       "                               microRNA             506         506\n",
       "           Solid Tissue Normal RNA                   59          59\n",
       "                               microRNA              59          59\n",
       "TCGA-THYM  Primary Tumor       RNA                  120         120\n",
       "                               microRNA             124         124\n",
       "           Solid Tissue Normal RNA                    2           2\n",
       "                               microRNA               2           2\n",
       "TCGA-UCEC  Primary Tumor       RNA                  545         553\n",
       "                               microRNA             538         545\n",
       "           Solid Tissue Normal RNA                   35          35\n",
       "                               microRNA              33          33\n",
       "TCGA-UCS   Primary Tumor       RNA                   57          57\n",
       "                               microRNA              57          57\n",
       "TCGA-UVM   Primary Tumor       RNA                   80          80\n",
       "                               microRNA              80          80"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Count the total number of cases and files per project\n",
    "pd.set_option('display.max_rows', 500)\n",
    "df_eda \\\n",
    "    .groupby(['project_id', 'sample_type', 'data_acronym']) \\\n",
    "    .agg(\n",
    "        case_count=pd.NamedAgg(column='case_id', aggfunc='nunique'),\n",
    "        file_count=pd.NamedAgg(column='file_id', aggfunc='nunique')\n",
    "    ) \\\n",
    "    .sort_values(by=['project_id', 'sample_type'], ascending=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Cases and Files"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Sample Types and Data File Types per Case"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Copy some columns of the EDA DataFrame\n",
    "df_cases_eda = df_eda[['case_id', 'file_id', 'data_and_sample']].copy()\n",
    "\n",
    "# Count the total number of files type\n",
    "df_cases_eda['NTmRNA'] = np.where(\n",
    "    df_cases_eda['data_and_sample'] == 'RNA-Normal', 1, 0\n",
    ")\n",
    "df_cases_eda['NTmicroRNA'] = np.where(\n",
    "    df_cases_eda['data_and_sample'] == 'microRNA-Normal', 1, 0\n",
    ")\n",
    "df_cases_eda['TTmRNA'] = np.where(\n",
    "    df_cases_eda['data_and_sample'] == 'RNA-Tumor', 1, 0\n",
    ")\n",
    "df_cases_eda['TTmicroRNA'] = np.where(\n",
    "    df_cases_eda['data_and_sample'] == 'microRNA-Tumor', 1, 0\n",
    ")\n",
    "\n",
    "# Sum all count columns\n",
    "df_cases_eda['file_count'] = (\n",
    "    df_cases_eda['NTmRNA'] + df_cases_eda['NTmicroRNA'] + \n",
    "    df_cases_eda['TTmRNA'] + df_cases_eda['TTmicroRNA']\n",
    ")\n",
    "\n",
    "# Count the total number of files type per case\n",
    "df_cases_eda = df_cases_eda \\\n",
    "    .groupby('case_id') \\\n",
    "    .agg(\n",
    "        TTmRNA_count=pd.NamedAgg(column='TTmRNA', aggfunc='sum'),\n",
    "        TTmicroRNA_count=pd.NamedAgg(column='TTmicroRNA', aggfunc='sum'),\n",
    "        NTmRNA_count=pd.NamedAgg(column='NTmRNA', aggfunc='sum'),\n",
    "        NTmicroRNA_count=pd.NamedAgg(column='NTmicroRNA', aggfunc='sum'),\n",
    "        file_count=pd.NamedAgg(column='file_count', aggfunc='sum')\n",
    "    ) \\\n",
    "    .sort_values(\n",
    "        by=['file_count', 'TTmRNA_count', 'TTmicroRNA_count', 'NTmRNA_count', 'NTmicroRNA_count'],\n",
    "        ascending=False\n",
    "    )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TTmRNA_count</th>\n",
       "      <th>TTmicroRNA_count</th>\n",
       "      <th>NTmRNA_count</th>\n",
       "      <th>NTmicroRNA_count</th>\n",
       "      <th>file_count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>case_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>556fcbc8-172a-4af1-8822-ae036e8d68e8</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8c7e74e0-71ef-49b8-9217-94b8ef740ef9</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>f130f376-5801-40f9-975d-a7e2f7b5670d</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0c0b610e-fe4c-406d-a5ed-5cc3b11dabf5</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>889aec8e-14ba-48d9-8fe1-f2416e82b333</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>f9824a6e-7a97-445c-8846-df8d8cddedaa</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>f9cf605f-f287-4eba-8e8f-8ba47e14ecc4</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>fdd4adb8-9295-480a-9352-305b5eb51187</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42ebfec9-6e28-4d72-9b9f-23a836800921</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>e5bc45ce-8a14-40b5-b9b5-ce45609fef3a</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10207 rows  5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      TTmRNA_count  TTmicroRNA_count  \\\n",
       "case_id                                                                \n",
       "556fcbc8-172a-4af1-8822-ae036e8d68e8             3                 3   \n",
       "8c7e74e0-71ef-49b8-9217-94b8ef740ef9             3                 3   \n",
       "f130f376-5801-40f9-975d-a7e2f7b5670d             3                 3   \n",
       "0c0b610e-fe4c-406d-a5ed-5cc3b11dabf5             3                 3   \n",
       "889aec8e-14ba-48d9-8fe1-f2416e82b333             3                 3   \n",
       "...                                            ...               ...   \n",
       "f9824a6e-7a97-445c-8846-df8d8cddedaa             0                 1   \n",
       "f9cf605f-f287-4eba-8e8f-8ba47e14ecc4             0                 1   \n",
       "fdd4adb8-9295-480a-9352-305b5eb51187             0                 1   \n",
       "42ebfec9-6e28-4d72-9b9f-23a836800921             0                 0   \n",
       "e5bc45ce-8a14-40b5-b9b5-ce45609fef3a             0                 0   \n",
       "\n",
       "                                      NTmRNA_count  NTmicroRNA_count  \\\n",
       "case_id                                                                \n",
       "556fcbc8-172a-4af1-8822-ae036e8d68e8             1                 1   \n",
       "8c7e74e0-71ef-49b8-9217-94b8ef740ef9             1                 1   \n",
       "f130f376-5801-40f9-975d-a7e2f7b5670d             1                 1   \n",
       "0c0b610e-fe4c-406d-a5ed-5cc3b11dabf5             1                 0   \n",
       "889aec8e-14ba-48d9-8fe1-f2416e82b333             1                 0   \n",
       "...                                            ...               ...   \n",
       "f9824a6e-7a97-445c-8846-df8d8cddedaa             0                 0   \n",
       "f9cf605f-f287-4eba-8e8f-8ba47e14ecc4             0                 0   \n",
       "fdd4adb8-9295-480a-9352-305b5eb51187             0                 0   \n",
       "42ebfec9-6e28-4d72-9b9f-23a836800921             1                 0   \n",
       "e5bc45ce-8a14-40b5-b9b5-ce45609fef3a             0                 1   \n",
       "\n",
       "                                      file_count  \n",
       "case_id                                           \n",
       "556fcbc8-172a-4af1-8822-ae036e8d68e8           8  \n",
       "8c7e74e0-71ef-49b8-9217-94b8ef740ef9           8  \n",
       "f130f376-5801-40f9-975d-a7e2f7b5670d           8  \n",
       "0c0b610e-fe4c-406d-a5ed-5cc3b11dabf5           7  \n",
       "889aec8e-14ba-48d9-8fe1-f2416e82b333           7  \n",
       "...                                          ...  \n",
       "f9824a6e-7a97-445c-8846-df8d8cddedaa           1  \n",
       "f9cf605f-f287-4eba-8e8f-8ba47e14ecc4           1  \n",
       "fdd4adb8-9295-480a-9352-305b5eb51187           1  \n",
       "42ebfec9-6e28-4d72-9b9f-23a836800921           1  \n",
       "e5bc45ce-8a14-40b5-b9b5-ce45609fef3a           1  \n",
       "\n",
       "[10207 rows x 5 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the total number of files type per case\n",
    "df_cases_eda"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Case Types Summarization"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Calculate the normalized count of case types\n",
    "count_abs = df_cases_eda.value_counts(ascending=False)\n",
    "count_norm = df_cases_eda.value_counts(normalize=True, ascending=False)\n",
    "\n",
    "# Summarizes the different types of cases\n",
    "df_case_counts = pd.DataFrame({\n",
    "    'count': count_abs,\n",
    "    'count%': round(count_norm * 100, 2)\n",
    "})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TTmRNA_count</th>\n",
       "      <th>TTmicroRNA_count</th>\n",
       "      <th>NTmRNA_count</th>\n",
       "      <th>NTmicroRNA_count</th>\n",
       "      <th>file_count</th>\n",
       "      <th>count</th>\n",
       "      <th>count%</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>8867</td>\n",
       "      <td>86.87</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>594</td>\n",
       "      <td>5.82</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>177</td>\n",
       "      <td>1.73</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>168</td>\n",
       "      <td>1.65</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>98</td>\n",
       "      <td>0.96</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>88</td>\n",
       "      <td>0.86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>48</td>\n",
       "      <td>0.47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>38</td>\n",
       "      <td>0.37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>35</td>\n",
       "      <td>0.34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>17</td>\n",
       "      <td>0.17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>17</td>\n",
       "      <td>0.17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>10</td>\n",
       "      <td>0.10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>10</td>\n",
       "      <td>0.10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "      <td>0.07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>4</td>\n",
       "      <td>0.04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>0.04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>0.03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>3</td>\n",
       "      <td>0.03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>8</td>\n",
       "      <td>3</td>\n",
       "      <td>0.03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>2</td>\n",
       "      <td>0.02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>0.02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>0.02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>0.01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>0.01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0.01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    TTmRNA_count  TTmicroRNA_count  NTmRNA_count  NTmicroRNA_count  \\\n",
       "0              1                 1             0                 0   \n",
       "1              1                 1             1                 1   \n",
       "2              1                 0             0                 0   \n",
       "3              0                 1             0                 0   \n",
       "4              1                 1             1                 0   \n",
       "5              2                 1             0                 0   \n",
       "6              0                 2             0                 0   \n",
       "7              1                 1             0                 1   \n",
       "8              1                 2             0                 0   \n",
       "9              0                 0             1                 1   \n",
       "10             3                 3             0                 0   \n",
       "11             2                 2             0                 0   \n",
       "12             3                 1             0                 0   \n",
       "13             0                 1             1                 1   \n",
       "14             2                 0             0                 0   \n",
       "15             2                 2             1                 1   \n",
       "16             1                 0             1                 1   \n",
       "17             3                 1             1                 1   \n",
       "18             3                 1             1                 0   \n",
       "19             3                 3             1                 1   \n",
       "20             3                 3             1                 0   \n",
       "21             1                 1             1                 2   \n",
       "22             1                 1             0                 2   \n",
       "23             2                 2             1                 0   \n",
       "24             0                 0             1                 0   \n",
       "25             3                 2             0                 0   \n",
       "26             1                 0             1                 0   \n",
       "27             0                 0             0                 1   \n",
       "\n",
       "    file_count  count  count%  \n",
       "0            2   8867   86.87  \n",
       "1            4    594    5.82  \n",
       "2            1    177    1.73  \n",
       "3            1    168    1.65  \n",
       "4            3     98    0.96  \n",
       "5            3     88    0.86  \n",
       "6            2     48    0.47  \n",
       "7            3     38    0.37  \n",
       "8            3     35    0.34  \n",
       "9            2     17    0.17  \n",
       "10           6     17    0.17  \n",
       "11           4     10    0.10  \n",
       "12           4     10    0.10  \n",
       "13           3      7    0.07  \n",
       "14           2      5    0.05  \n",
       "15           6      4    0.04  \n",
       "16           3      4    0.04  \n",
       "17           6      3    0.03  \n",
       "18           5      3    0.03  \n",
       "19           8      3    0.03  \n",
       "20           7      2    0.02  \n",
       "21           5      2    0.02  \n",
       "22           4      2    0.02  \n",
       "23           5      1    0.01  \n",
       "24           1      1    0.01  \n",
       "25           5      1    0.01  \n",
       "26           2      1    0.01  \n",
       "27           1      1    0.01  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Print the summarization of the different types of cases\n",
    "df_case_counts.reset_index()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# BRCA Project"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Data and Informations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>project_id</th>\n",
       "      <th>project_primary_site</th>\n",
       "      <th>project_disease_type</th>\n",
       "      <th>project_name</th>\n",
       "      <th>project_case_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>TCGA-BRCA</td>\n",
       "      <td>['Breast']</td>\n",
       "      <td>['Epithelial Neoplasms, NOS', 'Adnexal and Skin Appendage Neoplasms', 'Squamous Cell Neoplasms', 'Adenomas and Adenocarcinomas', 'Complex Epithelial Neoplasms', 'Fibroepithelial Neoplasms', 'Cystic, Mucinous and Serous Neoplasms', 'Basal Cell Neoplasms', 'Ductal and Lobular Neoplasms']</td>\n",
       "      <td>Breast Invasive Carcinoma</td>\n",
       "      <td>1098</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   project_id project_primary_site  \\\n",
       "32  TCGA-BRCA           ['Breast']   \n",
       "\n",
       "                                                                                                                                                                                                                                                                              project_disease_type  \\\n",
       "32  ['Epithelial Neoplasms, NOS', 'Adnexal and Skin Appendage Neoplasms', 'Squamous Cell Neoplasms', 'Adenomas and Adenocarcinomas', 'Complex Epithelial Neoplasms', 'Fibroepithelial Neoplasms', 'Cystic, Mucinous and Serous Neoplasms', 'Basal Cell Neoplasms', 'Ductal and Lobular Neoplasms']   \n",
       "\n",
       "                 project_name  project_case_count  \n",
       "32  Breast Invasive Carcinoma                1098  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Query the data related to TCGA-BRCA \n",
    "df_brca_eda = df_eda \\\n",
    "    .query('project_id == \"TCGA-BRCA\"') \\\n",
    "    .merge(\n",
    "        right=df_cases_eda,\n",
    "        left_on='case_id',\n",
    "        right_on='case_id',\n",
    "        how='inner'\n",
    "    )\n",
    "\n",
    "# Print TCGA-BRCA basic informations\n",
    "pd.set_option('display.max_colwidth', 300)\n",
    "df_projects.query('project_id == \"TCGA-BRCA\"')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Cases per Disease Type"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>case_disease_type</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Ductal and Lobular Neoplasms</th>\n",
       "      <td>1053</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Cystic, Mucinous and Serous Neoplasms</th>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Complex Epithelial Neoplasms</th>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Epithelial Neoplasms, NOS</th>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Adenomas and Adenocarcinomas</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Fibroepithelial Neoplasms</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Squamous Cell Neoplasms</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Adnexal and Skin Appendage Neoplasms</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Basal Cell Neoplasms</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       count\n",
       "case_disease_type                           \n",
       "Ductal and Lobular Neoplasms            1053\n",
       "Cystic, Mucinous and Serous Neoplasms     16\n",
       "Complex Epithelial Neoplasms              14\n",
       "Epithelial Neoplasms, NOS                  5\n",
       "Adenomas and Adenocarcinomas               3\n",
       "Fibroepithelial Neoplasms                  2\n",
       "Squamous Cell Neoplasms                    2\n",
       "Adnexal and Skin Appendage Neoplasms       1\n",
       "Basal Cell Neoplasms                       1"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Count the number of cases per disease type\n",
    "df_brca_eda \\\n",
    "    .groupby('case_disease_type') \\\n",
    "    .agg(\n",
    "        count=pd.NamedAgg(column='case_id', aggfunc='nunique'),\n",
    "    ) \\\n",
    "    .sort_values(by='count', ascending=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Case Types Summarization"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TTmRNA_count</th>\n",
       "      <th>TTmicroRNA_count</th>\n",
       "      <th>NTmRNA_count</th>\n",
       "      <th>NTmicroRNA_count</th>\n",
       "      <th>file_count</th>\n",
       "      <th>count</th>\n",
       "      <th>count%</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>955</td>\n",
       "      <td>87.06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>99</td>\n",
       "      <td>9.02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>18</td>\n",
       "      <td>1.64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>9</td>\n",
       "      <td>0.82</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>0.36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>0.27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>0.18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "      <td>0.18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>8</td>\n",
       "      <td>2</td>\n",
       "      <td>0.18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>0.09</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    TTmRNA_count  TTmicroRNA_count  NTmRNA_count  NTmicroRNA_count  \\\n",
       "0              1                 1             0                 0   \n",
       "1              1                 1             1                 1   \n",
       "2              1                 0             0                 0   \n",
       "3              1                 1             1                 0   \n",
       "4              2                 2             0                 0   \n",
       "5              3                 3             0                 0   \n",
       "6              0                 1             0                 0   \n",
       "7              1                 2             0                 0   \n",
       "8              2                 2             1                 1   \n",
       "9              3                 3             1                 1   \n",
       "10             1                 0             1                 1   \n",
       "\n",
       "    file_count  count  count%  \n",
       "0            2    955   87.06  \n",
       "1            4     99    9.02  \n",
       "2            1     18    1.64  \n",
       "3            3      9    0.82  \n",
       "4            4      4    0.36  \n",
       "5            6      3    0.27  \n",
       "6            1      2    0.18  \n",
       "7            3      2    0.18  \n",
       "8            6      2    0.18  \n",
       "9            8      2    0.18  \n",
       "10           3      1    0.09  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Drop the unnecessary columns and duplicated rows\n",
    "columns = [\n",
    "    'TTmRNA_count',\n",
    "    'TTmicroRNA_count',\n",
    "    'NTmRNA_count',\n",
    "    'NTmicroRNA_count',\n",
    "    'file_count'\n",
    "]\n",
    "df_brca_cases_eda = df_brca_eda.drop_duplicates(subset='case_id')[columns]\n",
    "\n",
    "# Calculate the normalized count of case types\n",
    "count_abs = df_brca_cases_eda.value_counts(ascending=False)\n",
    "count_norm = df_brca_cases_eda.value_counts(normalize=True, ascending=False)\n",
    "\n",
    "# Summarizes the different types of cases\n",
    "pd.DataFrame({\n",
    "    'count': count_abs,\n",
    "    'count%': round(count_norm * 100, 2)\n",
    "}).reset_index()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# CESC Project"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Data and Informations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>project_id</th>\n",
       "      <th>project_primary_site</th>\n",
       "      <th>project_disease_type</th>\n",
       "      <th>project_name</th>\n",
       "      <th>project_case_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>TCGA-CESC</td>\n",
       "      <td>['Ovary', 'Cervix uteri']</td>\n",
       "      <td>['Complex Epithelial Neoplasms', 'Adenomas and Adenocarcinomas', 'Squamous Cell Neoplasms', 'Cystic, Mucinous and Serous Neoplasms']</td>\n",
       "      <td>Cervical Squamous Cell Carcinoma and Endocervical Adenocarcinoma</td>\n",
       "      <td>307</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   project_id       project_primary_site  \\\n",
       "16  TCGA-CESC  ['Ovary', 'Cervix uteri']   \n",
       "\n",
       "                                                                                                                    project_disease_type  \\\n",
       "16  ['Complex Epithelial Neoplasms', 'Adenomas and Adenocarcinomas', 'Squamous Cell Neoplasms', 'Cystic, Mucinous and Serous Neoplasms']   \n",
       "\n",
       "                                                        project_name  \\\n",
       "16  Cervical Squamous Cell Carcinoma and Endocervical Adenocarcinoma   \n",
       "\n",
       "    project_case_count  \n",
       "16                 307  "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Query the data related to TCGA-CESC \n",
    "df_cesc_eda = df_eda \\\n",
    "    .query('project_id == \"TCGA-CESC\"') \\\n",
    "    .merge(\n",
    "        right=df_cases_eda,\n",
    "        left_on='case_id',\n",
    "        right_on='case_id',\n",
    "        how='inner'\n",
    "    )\n",
    "\n",
    "# Print TCGA-CESC basic informations\n",
    "pd.set_option('display.max_colwidth', 300)\n",
    "df_projects.query('project_id == \"TCGA-CESC\"')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Cases per Disease Type"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>case_disease_type</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Squamous Cell Neoplasms</th>\n",
       "      <td>254</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Adenomas and Adenocarcinomas</th>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Cystic, Mucinous and Serous Neoplasms</th>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Complex Epithelial Neoplasms</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       count\n",
       "case_disease_type                           \n",
       "Squamous Cell Neoplasms                  254\n",
       "Adenomas and Adenocarcinomas              32\n",
       "Cystic, Mucinous and Serous Neoplasms     17\n",
       "Complex Epithelial Neoplasms               4"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Count the number of cases per disease type\n",
    "df_cesc_eda \\\n",
    "    .groupby('case_disease_type') \\\n",
    "    .agg(\n",
    "        count=pd.NamedAgg(column='case_id', aggfunc='nunique'),\n",
    "    ) \\\n",
    "    .sort_values(by='count', ascending=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Case Types Summarization"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TTmRNA_count</th>\n",
       "      <th>TTmicroRNA_count</th>\n",
       "      <th>NTmRNA_count</th>\n",
       "      <th>NTmicroRNA_count</th>\n",
       "      <th>file_count</th>\n",
       "      <th>count</th>\n",
       "      <th>count%</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>301</td>\n",
       "      <td>98.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>0.98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>0.98</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   TTmRNA_count  TTmicroRNA_count  NTmRNA_count  NTmicroRNA_count  file_count  \\\n",
       "0             1                 1             0                 0           2   \n",
       "1             0                 1             0                 0           1   \n",
       "2             1                 1             1                 1           4   \n",
       "\n",
       "   count  count%  \n",
       "0    301   98.05  \n",
       "1      3    0.98  \n",
       "2      3    0.98  "
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Drop the unnecessary columns and duplicated rows\n",
    "columns = [\n",
    "    'TTmRNA_count',\n",
    "    'TTmicroRNA_count',\n",
    "    'NTmRNA_count',\n",
    "    'NTmicroRNA_count',\n",
    "    'file_count'\n",
    "]\n",
    "df_cesc_cases_eda = df_cesc_eda.drop_duplicates(subset='case_id')[columns]\n",
    "\n",
    "# Calculate the normalized count of case types\n",
    "count_abs = df_cesc_cases_eda.value_counts(ascending=False)\n",
    "count_norm = df_cesc_cases_eda.value_counts(normalize=True, ascending=False)\n",
    "\n",
    "# Summarizes the different types of cases\n",
    "pd.DataFrame({\n",
    "    'count': count_abs,\n",
    "    'count%': round(count_norm * 100, 2)\n",
    "}).reset_index()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# OV Project"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Data and Informations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>project_id</th>\n",
       "      <th>project_primary_site</th>\n",
       "      <th>project_disease_type</th>\n",
       "      <th>project_name</th>\n",
       "      <th>project_case_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>TCGA-OV</td>\n",
       "      <td>['Ovary', 'Retroperitoneum and peritoneum']</td>\n",
       "      <td>['Cystic, Mucinous and Serous Neoplasms', 'Not Reported']</td>\n",
       "      <td>Ovarian Serous Cystadenocarcinoma</td>\n",
       "      <td>608</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   project_id                         project_primary_site  \\\n",
       "30    TCGA-OV  ['Ovary', 'Retroperitoneum and peritoneum']   \n",
       "\n",
       "                                         project_disease_type  \\\n",
       "30  ['Cystic, Mucinous and Serous Neoplasms', 'Not Reported']   \n",
       "\n",
       "                         project_name  project_case_count  \n",
       "30  Ovarian Serous Cystadenocarcinoma                 608  "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Query the data related to TCGA-OV\n",
    "df_ov_eda = df_eda \\\n",
    "    .query('project_id == \"TCGA-OV\"') \\\n",
    "    .merge(\n",
    "        right=df_cases_eda,\n",
    "        left_on='case_id',\n",
    "        right_on='case_id',\n",
    "        how='inner'\n",
    "    )\n",
    "\n",
    "# Print TCGA-OV basic informations\n",
    "pd.set_option('display.max_colwidth', 300)\n",
    "df_projects.query('project_id == \"TCGA-OV\"')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Cases per Disease Type"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>case_disease_type</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Cystic, Mucinous and Serous Neoplasms</th>\n",
       "      <td>522</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       count\n",
       "case_disease_type                           \n",
       "Cystic, Mucinous and Serous Neoplasms    522"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Count the number of cases per disease type\n",
    "df_ov_eda \\\n",
    "    .groupby('case_disease_type') \\\n",
    "    .agg(\n",
    "        count=pd.NamedAgg(column='case_id', aggfunc='nunique'),\n",
    "    ) \\\n",
    "    .sort_values(by='count', ascending=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Case Types Summarization"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TTmRNA_count</th>\n",
       "      <th>TTmicroRNA_count</th>\n",
       "      <th>NTmRNA_count</th>\n",
       "      <th>NTmicroRNA_count</th>\n",
       "      <th>file_count</th>\n",
       "      <th>count</th>\n",
       "      <th>count%</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>420</td>\n",
       "      <td>80.46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>58</td>\n",
       "      <td>11.11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>43</td>\n",
       "      <td>8.24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>0.19</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   TTmRNA_count  TTmicroRNA_count  NTmRNA_count  NTmicroRNA_count  file_count  \\\n",
       "0             1                 1             0                 0           2   \n",
       "1             0                 1             0                 0           1   \n",
       "2             0                 2             0                 0           2   \n",
       "3             1                 2             0                 0           3   \n",
       "\n",
       "   count  count%  \n",
       "0    420   80.46  \n",
       "1     58   11.11  \n",
       "2     43    8.24  \n",
       "3      1    0.19  "
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Drop the unnecessary columns and duplicated rows\n",
    "columns = [\n",
    "    'TTmRNA_count',\n",
    "    'TTmicroRNA_count',\n",
    "    'NTmRNA_count',\n",
    "    'NTmicroRNA_count',\n",
    "    'file_count'\n",
    "]\n",
    "df_ov_cases_eda = df_ov_eda.drop_duplicates(subset='case_id')[columns]\n",
    "\n",
    "# Calculate the normalized count of case types\n",
    "count_abs = df_ov_cases_eda.value_counts(ascending=False)\n",
    "count_norm = df_ov_cases_eda.value_counts(normalize=True, ascending=False)\n",
    "\n",
    "# Summarizes the different types of cases\n",
    "pd.DataFrame({\n",
    "    'count': count_abs,\n",
    "    'count%': round(count_norm * 100, 2)\n",
    "}).reset_index()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# UCEC Project"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Data and Informations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>project_id</th>\n",
       "      <th>project_primary_site</th>\n",
       "      <th>project_disease_type</th>\n",
       "      <th>project_name</th>\n",
       "      <th>project_case_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>TCGA-UCEC</td>\n",
       "      <td>['Corpus uteri', 'Uterus, NOS']</td>\n",
       "      <td>['Epithelial Neoplasms, NOS', 'Adenomas and Adenocarcinomas', 'Cystic, Mucinous and Serous Neoplasms', 'Not Reported']</td>\n",
       "      <td>Uterine Corpus Endometrial Carcinoma</td>\n",
       "      <td>560</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   project_id             project_primary_site  \\\n",
       "28  TCGA-UCEC  ['Corpus uteri', 'Uterus, NOS']   \n",
       "\n",
       "                                                                                                      project_disease_type  \\\n",
       "28  ['Epithelial Neoplasms, NOS', 'Adenomas and Adenocarcinomas', 'Cystic, Mucinous and Serous Neoplasms', 'Not Reported']   \n",
       "\n",
       "                            project_name  project_case_count  \n",
       "28  Uterine Corpus Endometrial Carcinoma                 560  "
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Query the data related to TCGA-UCEC\n",
    "df_ucec_eda = df_eda \\\n",
    "    .query('project_id == \"TCGA-UCEC\"') \\\n",
    "    .merge(\n",
    "        right=df_cases_eda,\n",
    "        left_on='case_id',\n",
    "        right_on='case_id',\n",
    "        how='inner'\n",
    "    )\n",
    "\n",
    "# Print TCGA-UCEC basic informations\n",
    "pd.set_option('display.max_colwidth', 300)\n",
    "df_projects.query('project_id == \"TCGA-UCEC\"')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Cases per Disease Type"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>case_disease_type</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Adenomas and Adenocarcinomas</th>\n",
       "      <td>414</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Cystic, Mucinous and Serous Neoplasms</th>\n",
       "      <td>131</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Not Reported</th>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Epithelial Neoplasms, NOS</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       count\n",
       "case_disease_type                           \n",
       "Adenomas and Adenocarcinomas             414\n",
       "Cystic, Mucinous and Serous Neoplasms    131\n",
       "Not Reported                              12\n",
       "Epithelial Neoplasms, NOS                  2"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Count the number of cases per disease type\n",
    "df_ucec_eda \\\n",
    "    .groupby('case_disease_type') \\\n",
    "    .agg(\n",
    "        count=pd.NamedAgg(column='case_id', aggfunc='nunique'),\n",
    "    ) \\\n",
    "    .sort_values(by='count', ascending=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Case Types Summarization"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TTmRNA_count</th>\n",
       "      <th>TTmicroRNA_count</th>\n",
       "      <th>NTmRNA_count</th>\n",
       "      <th>NTmicroRNA_count</th>\n",
       "      <th>file_count</th>\n",
       "      <th>count</th>\n",
       "      <th>count%</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>510</td>\n",
       "      <td>91.23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>21</td>\n",
       "      <td>3.76</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>11</td>\n",
       "      <td>1.97</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>8</td>\n",
       "      <td>1.43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>0.54</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>0.18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>0.18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>0.18</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   TTmRNA_count  TTmicroRNA_count  NTmRNA_count  NTmicroRNA_count  file_count  \\\n",
       "0             1                 1             0                 0           2   \n",
       "1             1                 1             1                 1           4   \n",
       "2             0                 0             1                 1           2   \n",
       "3             1                 0             0                 0           1   \n",
       "4             3                 3             0                 0           6   \n",
       "5             0                 1             0                 0           1   \n",
       "6             0                 0             1                 0           1   \n",
       "7             1                 0             1                 1           3   \n",
       "8             1                 1             1                 0           3   \n",
       "9             3                 2             0                 0           5   \n",
       "\n",
       "   count  count%  \n",
       "0    510   91.23  \n",
       "1     21    3.76  \n",
       "2     11    1.97  \n",
       "3      8    1.43  \n",
       "4      3    0.54  \n",
       "5      2    0.36  \n",
       "6      1    0.18  \n",
       "7      1    0.18  \n",
       "8      1    0.18  \n",
       "9      1    0.18  "
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Drop the unnecessary columns and duplicated rows\n",
    "columns = [\n",
    "    'TTmRNA_count',\n",
    "    'TTmicroRNA_count',\n",
    "    'NTmRNA_count',\n",
    "    'NTmicroRNA_count',\n",
    "    'file_count'\n",
    "]\n",
    "df_ucec_cases_eda = df_ucec_eda.drop_duplicates(subset='case_id')[columns]\n",
    "\n",
    "# Calculate the normalized count of case types\n",
    "count_abs = df_ucec_cases_eda.value_counts(ascending=False)\n",
    "count_norm = df_ucec_cases_eda.value_counts(normalize=True, ascending=False)\n",
    "\n",
    "# Summarizes the different types of cases\n",
    "pd.DataFrame({\n",
    "    'count': count_abs,\n",
    "    'count%': round(count_norm * 100, 2)\n",
    "}).reset_index()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# UCS Project"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Data and Informations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>project_id</th>\n",
       "      <th>project_primary_site</th>\n",
       "      <th>project_disease_type</th>\n",
       "      <th>project_name</th>\n",
       "      <th>project_case_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TCGA-UCS</td>\n",
       "      <td>['Corpus uteri', 'Uterus, NOS']</td>\n",
       "      <td>['Complex Mixed and Stromal Neoplasms', 'Basal Cell Neoplasms']</td>\n",
       "      <td>Uterine Carcinosarcoma</td>\n",
       "      <td>57</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  project_id             project_primary_site  \\\n",
       "1   TCGA-UCS  ['Corpus uteri', 'Uterus, NOS']   \n",
       "\n",
       "                                              project_disease_type  \\\n",
       "1  ['Complex Mixed and Stromal Neoplasms', 'Basal Cell Neoplasms']   \n",
       "\n",
       "             project_name  project_case_count  \n",
       "1  Uterine Carcinosarcoma                  57  "
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Query the data related to TCGA-UCS\n",
    "df_ucs_eda = df_eda \\\n",
    "    .query('project_id == \"TCGA-UCS\"') \\\n",
    "    .merge(\n",
    "        right=df_cases_eda,\n",
    "        left_on='case_id',\n",
    "        right_on='case_id',\n",
    "        how='inner'\n",
    "    )\n",
    "\n",
    "# Print TCGA-UCS basic informations\n",
    "pd.set_option('display.max_colwidth', 300)\n",
    "df_projects.query('project_id == \"TCGA-UCS\"')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Cases per Disease Type"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>case_disease_type</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Complex Mixed and Stromal Neoplasms</th>\n",
       "      <td>56</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Basal Cell Neoplasms</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                     count\n",
       "case_disease_type                         \n",
       "Complex Mixed and Stromal Neoplasms     56\n",
       "Basal Cell Neoplasms                     1"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Count the number of cases per disease type\n",
    "df_ucs_eda \\\n",
    "    .groupby('case_disease_type') \\\n",
    "    .agg(\n",
    "        count=pd.NamedAgg(column='case_id', aggfunc='nunique'),\n",
    "    ) \\\n",
    "    .sort_values(by='count', ascending=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Case Types Summarization"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TTmRNA_count</th>\n",
       "      <th>TTmicroRNA_count</th>\n",
       "      <th>NTmRNA_count</th>\n",
       "      <th>NTmicroRNA_count</th>\n",
       "      <th>file_count</th>\n",
       "      <th>count</th>\n",
       "      <th>count%</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>57</td>\n",
       "      <td>100.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   TTmRNA_count  TTmicroRNA_count  NTmRNA_count  NTmicroRNA_count  file_count  \\\n",
       "0             1                 1             0                 0           2   \n",
       "\n",
       "   count  count%  \n",
       "0     57   100.0  "
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Drop the unnecessary columns and duplicated rows\n",
    "columns = [\n",
    "    'TTmRNA_count',\n",
    "    'TTmicroRNA_count',\n",
    "    'NTmRNA_count',\n",
    "    'NTmicroRNA_count',\n",
    "    'file_count'\n",
    "]\n",
    "df_ucs_cases_eda = df_ucs_eda.drop_duplicates(subset='case_id')[columns]\n",
    "\n",
    "# Calculate the normalized count of case types\n",
    "count_abs = df_ucs_cases_eda.value_counts(ascending=False)\n",
    "count_norm = df_ucs_cases_eda.value_counts(normalize=True, ascending=False)\n",
    "\n",
    "# Summarizes the different types of cases\n",
    "pd.DataFrame({\n",
    "    'count': count_abs,\n",
    "    'count%': round(count_norm * 100, 2)\n",
    "}).reset_index()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
